{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "\n",
    "import re\n",
    "import os\n",
    "import csv\n",
    "\n",
    "from tqdm.notebook import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded dataset with 457 rows\n"
     ]
    }
   ],
   "source": [
    "DATA_PATH = 'outputs/humanatlas.csv'\n",
    "\n",
    "data = pd.read_csv(DATA_PATH, lineterminator=\"\\n\")\n",
    "\n",
    "print(f'Loaded dataset with {len(data)} rows')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('EDAM/edam_topics.txt', 'r') as f:\n",
    "    edam_topics = [topic.strip() for topic in f.readlines()]\n",
    "\n",
    "quoted_topics = [topic for topic in edam_topics if topic.startswith('\"') and topic.endswith('\"')]\n",
    "\n",
    "# Remove quotes\n",
    "edam_topics = [topic[1:-1] if topic.startswith('\"') and topic.endswith('\"') else topic for topic in edam_topics]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Format Outputs\n",
    "\n",
    "Split outputs on tab, and check for other separators that GPT may have used in error."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].str.replace('\\\\t', '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_topics(topics):\n",
    "    cleaned_topics = [topic.strip() for topic in topics.split('\\t')]\n",
    "    for i in range(len(cleaned_topics)):\n",
    "        for quoted_topic in quoted_topics:\n",
    "            if quoted_topic.replace('\\\"', '').lower() in cleaned_topics[i].lower():\n",
    "                cleaned_topics[i] = cleaned_topics[i].replace(quoted_topic.replace('\\\"', ''), quoted_topic)\n",
    "                break\n",
    "            else:\n",
    "                cleaned_topics[i] = cleaned_topics[i].replace('\\\"', '')\n",
    "    return cleaned_topics\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(split_topics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "75064f31e8a74e06a853a8c036f2ae5b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/457 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "separators = ['    ', '   ', '  ', '\\n', '<TAB>', 'TAB', '<tab>', '(tab)', '<Tab>', '[tab]', '▪️', '(Tab)', '\\xa0\\xa0\\xa0\\xa0', '\\xa0', '\\u2003', '、', '\\x0b', '\\x0c', ';', '.', '--', '-', '–', '_', '\\\\', '\\\\n', '/', '@', '|', '\\r', '+', '<', '>', '·']\n",
    "\n",
    "# Join the separators with the regex OR operator |\n",
    "sep_pattern = '|'.join(map(re.escape, separators))\n",
    "\n",
    "for i in tqdm(range(len(data))):\n",
    "    # Replace commas not enclosed in double quotes with |\n",
    "    data['Predictions'][i] = [re.sub(r',(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)', '|', pred) for pred in data['Predictions'][i]]\n",
    "    # Split on the separators\n",
    "    split_list = [re.split(sep_pattern, pred) for pred in data['Predictions'][i]]\n",
    "    # Flatten the list\n",
    "    data['Predictions'][i] = [item for sublist in split_list for item in sublist]\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(lambda x: [item.strip() for item in x])\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(lambda x: [re.sub(r'Category \\d+:', '', pred) for pred in x])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "is_one_dimensional = all(isinstance(pred, str) for preds in data['Predictions'] for pred in preds)\n",
    "print(is_one_dimensional)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## Capture any weirdly formatted outputs (using the wrong separators)\n",
    "\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('    ') if 0 < len(x) <= 1 and '    ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('   ') if 0 < len(x) <= 1 and '   ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('  ') if 0 < len(x) <= 1 and '  ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\n') if 0 < len(x) <= 1 and '\\n' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<TAB>') if 0 < len(x) <= 1 and '<TAB>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('TAB') if 0 < len(x) <= 1 and 'TAB' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<tab>') if 0 < len(x) <= 1 and '<tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('(tab)') if 0 < len(x) <= 1 and '(tab)' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<Tab>') if 0 < len(x) <= 1 and '<Tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('[tab]') if 0 < len(x) <= 1 and '[tab]' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('▪️') if 0 < len(x) <= 1 and '▪️' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('(Tab)') if 0 < len(x) <= 1 and '<Tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\xa0\\xa0\\xa0\\xa0') if 0 < len(x) <= 1 and '\\xa0\\xa0\\xa0\\xa0' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\xa0') if 0 < len(x) <= 1 and '\\xa0' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\u2003') if 0 < len(x) <= 1 and '\\u2003' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('、') if 0 < len(x) <= 1 and '、' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\x0b') if 0 < len(x) <= 1 and '\\x0b' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\x0c') if 0 < len(x) <= 1 and '\\x0c' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split(';') if 0 < len(x) <= 1 and ';' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('.') if 0 < len(x) <= 1 and '.' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('--') if 0 < len(x) <= 1 and '--' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('-') if 0 < len(x) <= 1 and '-' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('–') if 0 < len(x) <= 1 and '–' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('_') if 0 < len(x) <= 1 and '_' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\\\') if 0 < len(x) <= 1 and '\\\\' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\\\n') if 0 < len(x) <= 1 and '\\\\n' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('/') if 0 < len(x) <= 1 and '/' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('@') if 0 < len(x) <= 1 and '@' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('|') if 0 < len(x) <= 1 and '|' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\r') if 0 < len(x) <= 1 and '\\r' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('+') if 0 < len(x) <= 1 and '+' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<') if 0 < len(x) <= 1 and '<' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('>') if 0 < len(x) <= 1 and '>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: [pred.strip() for pred in csv.reader([list(x)[0]],skipinitialspace=True, delimiter=',', quotechar='\"').__next__()])\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: [re.sub(r'Category \\d+:', '', pred) for pred in x])\n",
    "# # data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split(', ') if len(x) <= 1 and ', ' in list(x)[0] else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if any of the quoted topics, or their equivalent without quotes, \n",
    "# are in any of the prediction sets with length less than or equal to 1. \"\"\n",
    "# If there is, then add the quotes back in if they don't have them,\n",
    "#  and then split on commas while avoiding anything inside quotes\n",
    "def process_predictions(predictions):\n",
    "    processed_predictions = []\n",
    "    for prediction in predictions:\n",
    "        formatted = False\n",
    "        for topic in quoted_topics:\n",
    "            formatted_topic = topic.replace('\\\"', '')\n",
    "            if formatted_topic in prediction:\n",
    "                processed_prediction = prediction.replace(formatted_topic, f'{topic}')\n",
    "                processed_predictions.append(processed_prediction)\n",
    "                formatted = True\n",
    "                break\n",
    "        if not formatted:\n",
    "            processed_predictions.append(prediction)\n",
    "        \n",
    "    final_predictions = []\n",
    "    for prediction in processed_predictions:\n",
    "        if '\\\"' in prediction:\n",
    "            parts = re.findall(r'[^\"]+|\"[^\"]+\"', prediction)\n",
    "            final_predictions.extend(parts)\n",
    "        else:\n",
    "            final_predictions.extend([pred.strip() for pred in prediction.split(',')])\n",
    "    return set(final_predictions)\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(process_predictions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unexpected predictions: 3\n",
      "Original Index: 277, Prediction: Spatial transcriptomic profiling\n",
      "Original Index: 389, Prediction: Single cell RNA sequencing (RNAseq)\n",
      "Original Index: 392, Prediction: Single cell transcriptome sequencing\n"
     ]
    }
   ],
   "source": [
    "filtered_predictions = data[data['Predictions'].apply(len) <= 1]['Predictions']\n",
    "\n",
    "# Filter out any expected predictions. May or may not be hallucinations\n",
    "unexpected_predictions = []\n",
    "for original_index, pred_set in filtered_predictions.items():\n",
    "    lst = list(pred_set)\n",
    "    if len(lst) == 0:\n",
    "        continue\n",
    "    prediction = lst[0]\n",
    "    if '\\\"' not in prediction and ' ' in prediction and prediction not in edam_topics:\n",
    "        unexpected_predictions.append((original_index, prediction))\n",
    "\n",
    "# Print the unexpected predictions and their corresponding original indices\n",
    "count = len(unexpected_predictions)\n",
    "print(f\"Number of unexpected predictions: {count}\")\n",
    "for original_index, prediction in unexpected_predictions:\n",
    "    print(f\"Original Index: {original_index}, Prediction: {prediction}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hallucinations\n",
    "\n",
    "Filter out topics not in the EDAM topics list. The filtered topics may be matched to a topic or synonym->topic in the next section."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Hallucinations'] = data['Predictions'].apply(lambda preds: set([pred.replace('.', '').replace('\\\"', '') for pred in preds if pred.replace('.', '').replace('\\\"', '') not in edam_topics]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].apply(lambda preds: set([pred.replace('.', '').replace('\\\"', '') for pred in preds if pred.replace('.', '').replace('\\\"', '') in edam_topics]))\n",
    "data['Predictions'] = data.apply(lambda row: set([topic for topic in row['Predictions'] if topic not in row['Hallucinations']]), axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Synonym matching\n",
    "\n",
    "Check for mispelled/misformatted topics or synonyms using levenshtein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "edam = pd.read_csv('EDAM/EDAM.csv')\n",
    "\n",
    "edam = edam[edam['Class ID'].str.contains('topic')].reset_index(drop=True)\n",
    "# edam['Preferred Label'].apply(lambda topic: topic.replace('\\\"', ''))\n",
    "edam = edam[edam['Preferred Label'].isin([topic.replace('\\\"', '') for topic in edam_topics])].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "edam['Synonyms'] = edam['Synonyms'].fillna('').apply(lambda x: x.split('|') if x != '' else [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_topics = set([topic.replace('\\\"', '') for topic in edam_topics]) - set(edam['Preferred Label'])\n",
    "missing_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "synonym_dict = {}\n",
    "\n",
    "for index, row in edam.iterrows():\n",
    "    for synonym in row['Synonyms']:\n",
    "        synonym_dict[synonym] = row['Preferred Label']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Molecular docking': 'Molecular modelling',\n",
       " 'Homology modeling': 'Molecular modelling',\n",
       " 'Docking': 'Molecular modelling',\n",
       " 'Comparative modelling': 'Molecular modelling',\n",
       " 'Homology modelling': 'Molecular modelling',\n",
       " 'Evolution': 'Evolutionary biology',\n",
       " 'Freshwater science': 'Freshwater biology',\n",
       " 'Nutrition': 'Nutritional science',\n",
       " 'Nutrition science': 'Nutritional science',\n",
       " 'Dietetics': 'Nutritional science',\n",
       " 'Cardiovascular medicine': 'Cardiology',\n",
       " 'Heart disease': 'Cardiology',\n",
       " 'Cardiovascular disease': 'Cardiology',\n",
       " 'Gene features': 'Gene structure',\n",
       " 'Fusion genes': 'Gene structure',\n",
       " 'Transcriptome': 'Transcriptomics',\n",
       " 'Comparative transcriptomics': 'Transcriptomics',\n",
       " 'Ancestral genomes': 'Paleogenomics',\n",
       " 'Paleogenetics': 'Paleogenomics',\n",
       " 'Panomics': 'Multiomics',\n",
       " 'Pan-omics': 'Multiomics',\n",
       " 'Integrative omics': 'Multiomics',\n",
       " 'Multi-omics': 'Multiomics',\n",
       " 'RNA metabarcoding': 'Metabarcoding',\n",
       " 'DNA metabarcoding': 'Metabarcoding',\n",
       " 'eDNA metabarcoding': 'Metabarcoding',\n",
       " 'Environmental metabarcoding': 'Metabarcoding',\n",
       " 'eRNA metabarcoding': 'Metabarcoding',\n",
       " 'Environmental RNA (eRNA)': 'Metabarcoding',\n",
       " 'Environmental sequencing': 'Metagenomics',\n",
       " 'Taxonomic profiling': 'Metabarcoding',\n",
       " 'Environmental DNA (eDNA)': 'Metagenomics',\n",
       " 'Expression': 'Gene expression',\n",
       " 'Codon usage': 'Gene expression',\n",
       " 'Gene transcription': 'Gene expression',\n",
       " 'Transcription': 'Gene expression',\n",
       " 'Gene expression profiling': 'Gene expression',\n",
       " 'Gene translation': 'Gene expression',\n",
       " 'DNA microarrays': 'Gene expression',\n",
       " 'DNA chips': 'Gene expression',\n",
       " 'Small RNA': 'RNA',\n",
       " 'Electron diffraction experiment': 'Electron microscopy',\n",
       " 'Transmission electron microscopy': 'Electron microscopy',\n",
       " 'SEM': 'Electron microscopy',\n",
       " 'Scanning electron microscopy': 'Electron microscopy',\n",
       " 'TEM': 'Electron microscopy',\n",
       " 'Electron crystallography': 'Electron microscopy',\n",
       " 'Single particle electron microscopy': 'Electron microscopy',\n",
       " 'Immunoproteins': 'Immunoproteins and antigens',\n",
       " 'Antigens': 'Immunoproteins and antigens',\n",
       " 'Immunopeptides': 'Immunoproteins and antigens',\n",
       " 'Therapeutic antibodies': 'Immunoproteins and antigens',\n",
       " 'Translation': 'Protein expression',\n",
       " 'Neuroscience': 'Neurobiology',\n",
       " 'Molecular neuroscience': 'Neurobiology',\n",
       " 'Systemetic neuroscience': 'Neurobiology',\n",
       " 'Neurophysiology': 'Neurobiology',\n",
       " 'Reproductive disorders': 'Reproductive health',\n",
       " 'Andrology': 'Reproductive health',\n",
       " 'Fertility medicine': 'Reproductive health',\n",
       " 'Family planning': 'Reproductive health',\n",
       " 'Findable, accessible, interoperable, reusable data': 'FAIR data',\n",
       " 'Open data': 'FAIR data',\n",
       " 'FAIRification': 'FAIR data',\n",
       " 'FAIR data principles': 'FAIR data',\n",
       " 'Genetic modification': 'Genetic engineering',\n",
       " 'Genetic manipulation': 'Genetic engineering',\n",
       " 'Genome engineering': 'Genetic engineering',\n",
       " 'Genome editing': 'Genetic engineering',\n",
       " 'Audiovestibular medicine': 'Ear, nose and throat medicine',\n",
       " 'Otorhinolaryngology': 'Ear, nose and throat medicine',\n",
       " 'Otolaryngology': 'Ear, nose and throat medicine',\n",
       " 'Head and neck disorders': 'Ear, nose and throat medicine',\n",
       " 'Data integration': 'Data integration and warehousing',\n",
       " 'Data warehousing': 'Data integration and warehousing',\n",
       " 'Neurological disorders': 'Neurology',\n",
       " 'Life sciences': 'Biosciences',\n",
       " 'Primers': 'Probes and primers',\n",
       " 'Primer quality': 'Probes and primers',\n",
       " 'Probes': 'Probes and primers',\n",
       " 'Epistatic interactions': 'Epistasis',\n",
       " 'Epistatic genetic interaction': 'Epistasis',\n",
       " 'Ecological informatics': 'Ecology',\n",
       " 'Computational ecology': 'Ecology',\n",
       " 'Ecosystem science': 'Ecology',\n",
       " 'Ecoinformatics': 'Ecology',\n",
       " 'Pipelines': 'Workflows',\n",
       " 'Tool interoperability': 'Workflows',\n",
       " 'Software integration': 'Workflows',\n",
       " 'Tool integration': 'Workflows',\n",
       " 'Membrane proteins': 'Membrane and lipoproteins',\n",
       " 'Transmembrane proteins': 'Membrane and lipoproteins',\n",
       " 'Lipoproteins': 'Membrane and lipoproteins',\n",
       " 'Biomedical research': 'Medicine',\n",
       " 'Experimental medicine': 'Medicine',\n",
       " 'Clinical medicine': 'Medicine',\n",
       " 'Internal medicine': 'Medicine',\n",
       " 'General medicine': 'Medicine',\n",
       " 'Environmental microbiology': 'Microbial ecology',\n",
       " 'Microbiome': 'Microbial ecology',\n",
       " 'Molecular community analysis': 'Microbial ecology',\n",
       " 'Community analysis': 'Microbial ecology',\n",
       " 'Pharmacodynamics': 'Drug metabolism',\n",
       " 'ADME': 'Drug metabolism',\n",
       " 'Pharmacokinetics and pharmacodynamics': 'Drug metabolism',\n",
       " 'Drug distribution': 'Drug metabolism',\n",
       " 'Pharmacokinetics': 'Drug metabolism',\n",
       " 'Drug absorption': 'Drug metabolism',\n",
       " 'Drug excretion': 'Drug metabolism',\n",
       " 'RNA structure prediction': 'Structure prediction',\n",
       " 'Nucleic acid folding': 'Structure prediction',\n",
       " 'Nucleic acid design': 'Structure prediction',\n",
       " 'DNA structure prediction': 'Structure prediction',\n",
       " 'Protein fold recognition': 'Structure prediction',\n",
       " 'Protein structure prediction': 'Structure prediction',\n",
       " 'Nucleic acid structure prediction': 'Structure prediction',\n",
       " 'Protein 3D motifs': 'Protein structural motifs and surfaces',\n",
       " 'Protein structural motifs': 'Protein structural motifs and surfaces',\n",
       " 'Protein structural features': 'Protein structural motifs and surfaces',\n",
       " 'Protein surfaces': 'Protein structural motifs and surfaces',\n",
       " 'Structural motifs': 'Protein structural motifs and surfaces',\n",
       " 'Data design': 'Data architecture, analysis and design',\n",
       " 'Data architecture': 'Data architecture, analysis and design',\n",
       " 'Data analysis': 'Data architecture, analysis and design',\n",
       " 'Applied ontology': 'Ontology and terminology',\n",
       " 'Ontologies': 'Ontology and terminology',\n",
       " 'Ontology relations': 'Ontology and terminology',\n",
       " 'Upper ontology': 'Ontology and terminology',\n",
       " 'Terminology': 'Ontology and terminology',\n",
       " 'Ontology': 'Ontology and terminology',\n",
       " 'Protein stability': 'Protein folding, stability and design',\n",
       " 'Protein design': 'Protein folding, stability and design',\n",
       " 'Protein residue interactions': 'Protein folding, stability and design',\n",
       " 'Protein folding': 'Protein folding, stability and design',\n",
       " 'Rational protein design': 'Protein folding, stability and design',\n",
       " 'Metadata management': 'Data management',\n",
       " 'Data stewardship': 'Data governance',\n",
       " 'Data privacy': 'Data security',\n",
       " 'Genes, gene family or system': 'Gene and protein families',\n",
       " 'Gene system': 'Gene and protein families',\n",
       " 'Gene family': 'Gene and protein families',\n",
       " 'Protein sequence classification': 'Gene and protein families',\n",
       " 'Protein families': 'Gene and protein families',\n",
       " 'Gene families': 'Gene and protein families',\n",
       " 'Protein post-translational modification': 'Protein modifications',\n",
       " 'Post-translational modifications': 'Protein modifications',\n",
       " 'PTMs': 'Protein modifications',\n",
       " 'Post-translation modifications': 'Protein modifications',\n",
       " 'Protein post-translational modifications': 'Protein modifications',\n",
       " 'Protein chemical modifications': 'Protein modifications',\n",
       " 'GO:0006464': 'Protein modifications',\n",
       " 'MOD:00000': 'Protein modifications',\n",
       " 'Pharmaceutical biotechnology': 'Medical biotechnology',\n",
       " 'Lipidomics': 'Lipids',\n",
       " 'Pain management': 'Pain medicine',\n",
       " 'Algiatry': 'Pain medicine',\n",
       " 'Protein functional sites': 'Protein binding sites',\n",
       " 'Protein cleavage sites': 'Protein binding sites',\n",
       " 'Enzyme active site': 'Protein binding sites',\n",
       " 'Protein-nucleic acid binding sites': 'Protein binding sites',\n",
       " 'Protein key folding sites': 'Protein binding sites',\n",
       " 'EEG': 'Electroencephalography',\n",
       " 'Communicable disease': 'Infectious disease',\n",
       " 'Transmissible disease': 'Infectious disease',\n",
       " 'Exomes': 'Genomics',\n",
       " 'Genomes': 'Genomics',\n",
       " 'Personal genomics': 'Genomics',\n",
       " 'Synthetic genomics': 'Genomics',\n",
       " 'Genome annotation': 'Genomics',\n",
       " 'Viral genomics': 'Genomics',\n",
       " 'Whole genomes': 'Genomics',\n",
       " 'Alternative splicing': 'RNA splicing',\n",
       " 'Splice sites': 'RNA splicing',\n",
       " 'Humans': 'Human biology',\n",
       " 'Biological processes': 'Molecular biology',\n",
       " 'Assembly': 'Sequence assembly',\n",
       " 'Diagnostic markers': 'Biomarkers',\n",
       " 'Clinical immunology': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Allergy': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immune disorders': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immunomodulators': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immunotherapeutics': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Urology': 'Urology and nephrology',\n",
       " 'Nephrology': 'Urology and nephrology',\n",
       " 'Urological disorders': 'Urology and nephrology',\n",
       " 'Kidney disease': 'Urology and nephrology',\n",
       " 'Phylogeny reconstruction': 'Phylogeny',\n",
       " 'Phylogenetic dating': 'Phylogeny',\n",
       " 'Phylogenetic simulation': 'Phylogeny',\n",
       " 'Phylogenetic clocks': 'Phylogeny',\n",
       " 'Phylogenetic stratigraphy': 'Phylogeny',\n",
       " 'Optometry': 'Ophthalmology',\n",
       " 'Eye disoders': 'Ophthalmology',\n",
       " 'Chemoinformatics': 'Cheminformatics',\n",
       " 'Chemical informatics': 'Cheminformatics',\n",
       " 'AMR': 'Antimicrobial Resistance',\n",
       " 'Multidrug resistance': 'Antimicrobial Resistance',\n",
       " 'Multiple drug resistance (MDR)': 'Antimicrobial Resistance',\n",
       " 'Multiresistance': 'Antimicrobial Resistance',\n",
       " 'Antiprotozoal resistance': 'Antimicrobial Resistance',\n",
       " 'Antifungal resistance': 'Antimicrobial Resistance',\n",
       " 'Extensive drug resistance (XDR)': 'Antimicrobial Resistance',\n",
       " 'Pandrug resistance (PDR)': 'Antimicrobial Resistance',\n",
       " 'Total drug resistance (TDR)': 'Antimicrobial Resistance',\n",
       " 'Antiviral resistance': 'Antimicrobial Resistance',\n",
       " 'Biomimeic chemistry': 'Synthetic biology',\n",
       " 'Patient safety': 'Safety sciences',\n",
       " 'Drug safety': 'Safety sciences',\n",
       " 'Functional sites': 'Sequence sites, features and motifs',\n",
       " 'Sequence features': 'Sequence sites, features and motifs',\n",
       " 'Sequence sites': 'Sequence sites, features and motifs',\n",
       " 'Sequence profiles': 'Sequence sites, features and motifs',\n",
       " 'Sequence motifs': 'Sequence sites, features and motifs',\n",
       " 'HMMs': 'Sequence sites, features and motifs',\n",
       " 'Transplantation': 'Surgery',\n",
       " 'Environment': 'Environmental sciences',\n",
       " 'Biological sequences': 'Sequence analysis',\n",
       " 'Sequence databases': 'Sequence analysis',\n",
       " 'Animal experimentation': 'Laboratory animal science',\n",
       " 'Animal research': 'Laboratory animal science',\n",
       " 'Animal testing': 'Laboratory animal science',\n",
       " 'In vivo testing': 'Laboratory animal science',\n",
       " 'Medical physics': 'Biophysics',\n",
       " 'Neutron diffraction experiment': 'Neutron diffraction',\n",
       " 'Elastic neutron scattering': 'Neutron diffraction',\n",
       " 'Neutron microscopy': 'Neutron diffraction',\n",
       " 'NMR-based metabolomics': 'Metabolomics',\n",
       " 'LC-MS-based metabolomics': 'Metabolomics',\n",
       " 'MS-based metabolomics': 'Metabolomics',\n",
       " 'Exometabolomics': 'Metabolomics',\n",
       " 'Metabonomics': 'Metabolomics',\n",
       " 'MS-based targeted metabolomics': 'Metabolomics',\n",
       " 'Metabolome': 'Metabolomics',\n",
       " 'MS-based untargeted metabolomics': 'Metabolomics',\n",
       " 'Metabolites': 'Metabolomics',\n",
       " 'Mass spectrometry-based metabolomics': 'Metabolomics',\n",
       " 'Biological engineering': 'Bioengineering',\n",
       " 'Structure determination': 'Structural biology',\n",
       " 'Structural assignment': 'Structural biology',\n",
       " 'Structural determination': 'Structural biology',\n",
       " 'Rheumatology': 'Musculoskeletal medicine',\n",
       " 'Musculoskeletal disorders': 'Musculoskeletal medicine',\n",
       " 'Orthopaedics': 'Musculoskeletal medicine',\n",
       " 'Health care science': 'Medicines research and development',\n",
       " 'Health care research': 'Medicines research and development',\n",
       " 'Drug discovery and development': 'Medicines research and development',\n",
       " 'Hepatology': 'Hepatic and biliary medicine',\n",
       " 'Liver disorders': 'Hepatic and biliary medicine',\n",
       " 'Transmembrane regions': 'Protein folds and structural domains',\n",
       " 'Protein structural domains': 'Protein folds and structural domains',\n",
       " 'Protein membrane regions': 'Protein folds and structural domains',\n",
       " 'Intramembrane regions': 'Protein folds and structural domains',\n",
       " 'Protein folds': 'Protein folds and structural domains',\n",
       " 'Protein topological domains': 'Protein folds and structural domains',\n",
       " 'Protein domains': 'Protein folds and structural domains',\n",
       " 'Protein transmembrane regions': 'Protein folds and structural domains',\n",
       " 'Pharmacogenetics': 'Pharmacogenomics',\n",
       " 'Disease': 'Pathology',\n",
       " 'Data collection': 'Data acquisition',\n",
       " 'Public health': 'Public health and epidemiology',\n",
       " 'Epidemiology': 'Public health and epidemiology',\n",
       " 'Molecular motions': 'Molecular dynamics',\n",
       " 'Molecular flexibility': 'Molecular dynamics',\n",
       " 'Protein dynamics': 'Molecular dynamics',\n",
       " 'Fluidics': 'Microfluidics',\n",
       " 'Precision medicine': 'Personalised medicine',\n",
       " 'Molecular diagnostics': 'Personalised medicine',\n",
       " 'Computer programming': 'Software engineering',\n",
       " 'Software development': 'Software engineering',\n",
       " 'Data structures': 'Software engineering',\n",
       " 'Algorithms': 'Software engineering',\n",
       " 'Programming languages': 'Software engineering',\n",
       " 'Immunology and genetics': 'Immunogenetics',\n",
       " 'Immungenetics': 'Immunogenetics',\n",
       " 'Immune system genetics': 'Immunogenetics',\n",
       " 'Immunogenes': 'Immunogenetics',\n",
       " 'Methylation profiles': 'Epigenetics',\n",
       " 'DNA methylation': 'Epigenetics',\n",
       " 'Histone modification': 'Epigenetics',\n",
       " 'Clinical trial': 'Preclinical and clinical studies',\n",
       " 'Preclinical studies': 'Preclinical and clinical studies',\n",
       " 'Clinical studies': 'Preclinical and clinical studies',\n",
       " 'Drug trials': 'Preclinical and clinical studies',\n",
       " 'Clinical study': 'Preclinical and clinical studies',\n",
       " 'Preclinical study': 'Preclinical and clinical studies',\n",
       " 'Challenge study': 'Animal study',\n",
       " 'Functional RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Regulatory RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small nucleolar RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'miRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'microRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small interfering RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'siRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'lncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'snoRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'snRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Long non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'piwi-interacting RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'piRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small nuclear RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small and long non-coding RNAs': 'Functional, regulatory and non-coding RNA',\n",
       " 'Long ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'DNA structural variation': 'Structural variation',\n",
       " 'Genomic structural variation': 'Structural variation',\n",
       " 'Insertion': 'Structural variation',\n",
       " 'Translocation': 'Structural variation',\n",
       " 'Deletion': 'Structural variation',\n",
       " 'Inversion': 'Structural variation',\n",
       " 'Duplication': 'Structural variation',\n",
       " 'Metabolic disorders': 'Endocrinology and metabolism',\n",
       " 'Endocrinology': 'Endocrinology and metabolism',\n",
       " 'Metabolism': 'Endocrinology and metabolism',\n",
       " 'Endocrine disorders': 'Endocrinology and metabolism',\n",
       " 'Primer binding sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Sequence tagged sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Nucleic acid sequence features': 'Nucleic acid sites, features and motifs',\n",
       " 'Nucleic acid functional sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Spectroscopy': 'NMR',\n",
       " 'NMR spectroscopy': 'NMR',\n",
       " 'Nuclear magnetic resonance spectroscopy': 'NMR',\n",
       " 'Nuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'Heteronuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'Rotational Frame Nuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'NOESY': 'NMR',\n",
       " 'ROESY': 'NMR',\n",
       " 'HOESY': 'NMR',\n",
       " 'Molecular infection biology': 'Microbiology',\n",
       " 'Microbial physiology': 'Microbiology',\n",
       " 'Microbial genetics': 'Microbiology',\n",
       " 'Antimicrobial stewardship': 'Microbiology',\n",
       " 'Microbiological surveillance': 'Microbiology',\n",
       " 'Microbial surveillance': 'Microbiology',\n",
       " 'Molecular microbiology': 'Microbiology',\n",
       " 'Medical microbiology': 'Microbiology',\n",
       " 'Enzymology': 'Enzymes',\n",
       " 'Quality assurance': 'Quality affairs',\n",
       " 'Good laboratory practice': 'Quality affairs',\n",
       " 'Good clinical practice': 'Quality affairs',\n",
       " 'Good manufacturing practice': 'Quality affairs',\n",
       " 'Synthetic organic chemistry': 'Synthetic chemistry',\n",
       " 'Obstetrics': 'Gynaecology and obstetrics',\n",
       " 'Gynaecology': 'Gynaecology and obstetrics',\n",
       " 'Gynaecological disorders': 'Gynaecology and obstetrics',\n",
       " 'Protein repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Nucleic acid repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Repeat sequences': 'Sequence composition, complexity and repeats',\n",
       " 'Protein sequence repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence complexity': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence composition': 'Sequence composition, complexity and repeats',\n",
       " 'Low complexity sequences': 'Sequence composition, complexity and repeats',\n",
       " 'Community genomics': 'Metagenomics',\n",
       " 'Ecogenomics': 'Metagenomics',\n",
       " 'Biome sequencing': 'Metagenomics',\n",
       " 'Environmental genomics': 'Metagenomics',\n",
       " 'Environmental omics': 'Metagenomics',\n",
       " 'Shotgun metagenomics': 'Metagenomics',\n",
       " 'Computational toxicology': 'Toxicology',\n",
       " 'Toxicoinformatics': 'Toxicology',\n",
       " 'DNA analysis': 'DNA',\n",
       " 'Ancient DNA': 'DNA',\n",
       " 'Chromosomes': 'DNA',\n",
       " 'Protein sequence features': 'Protein sites, features and motifs',\n",
       " 'Signal peptide cleavage sites': 'Protein sites, features and motifs',\n",
       " 'Cellular processes': 'Cell biology',\n",
       " 'Cells': 'Cell biology',\n",
       " 'Protein subcellular localization': 'Cell biology',\n",
       " 'EKG': 'Electrocardiography',\n",
       " 'ECG': 'Electrocardiography',\n",
       " 'Electrophysiology': 'Physiology',\n",
       " 'Artificial Intelligence': 'Machine learning',\n",
       " 'Active learning': 'Machine learning',\n",
       " 'Knowledge representation': 'Machine learning',\n",
       " 'Unsupervised learning': 'Machine learning',\n",
       " 'Ensembl learning': 'Machine learning',\n",
       " 'Neural networks': 'Machine learning',\n",
       " 'Recommender system': 'Machine learning',\n",
       " 'Kernel methods': 'Machine learning',\n",
       " 'Reinforcement learning': 'Machine learning',\n",
       " 'Supervised learning': 'Machine learning',\n",
       " 'Reverse vaccinology': 'Vaccinology',\n",
       " 'Structural vaccinology': 'Vaccinology',\n",
       " 'Rational vaccine design': 'Vaccinology',\n",
       " 'Vaccine design': 'Vaccinology',\n",
       " 'Structure-based immunogen design': 'Vaccinology',\n",
       " 'Pulmonary medicine': 'Respiratory medicine',\n",
       " 'Pulmonology': 'Respiratory medicine',\n",
       " 'Pulmonary disorders': 'Respiratory medicine',\n",
       " 'Respiratory disease': 'Respiratory medicine',\n",
       " 'Biological models': 'Molecular interactions, pathways and networks',\n",
       " 'Genetic information processing pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Interactions': 'Molecular interactions, pathways and networks',\n",
       " 'Networks': 'Molecular interactions, pathways and networks',\n",
       " 'Signal transduction pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Biological pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Signaling pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Gene regulatory networks': 'Molecular interactions, pathways and networks',\n",
       " 'Disease pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Metabolic pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Interactome': 'Molecular interactions, pathways and networks',\n",
       " 'Environmental information processing pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Molecular interactions': 'Molecular interactions, pathways and networks',\n",
       " 'Biological networks': 'Molecular interactions, pathways and networks',\n",
       " 'Cellular process pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Data clean-up': 'Data quality management',\n",
       " 'Data integrity': 'Data quality management',\n",
       " 'Data cleaning': 'Data quality management',\n",
       " 'Data quality': 'Data quality management',\n",
       " 'Biological system modelling': 'Systems biology',\n",
       " 'Systems modelling': 'Systems biology',\n",
       " 'Biological modelling': 'Systems biology',\n",
       " 'Cell culture': 'Cell culture collection',\n",
       " 'Biogeochemical cycle': 'Carbon cycle',\n",
       " 'Anaesthetics': 'Anaesthesiology',\n",
       " 'Gene imprinting': 'Genomic imprinting',\n",
       " 'Computational pharmacology': 'Pharmacology',\n",
       " 'Pharmacoinformatics': 'Pharmacology',\n",
       " 'Mental health': 'Psychiatry',\n",
       " 'Psychiatric disorders': 'Psychiatry',\n",
       " 'Plants': 'Plant biology',\n",
       " 'Plant science': 'Plant biology',\n",
       " 'Botany': 'Plant biology',\n",
       " 'Plant': 'Plant biology',\n",
       " 'Plant genetics': 'Plant biology',\n",
       " 'Plant ecology': 'Plant biology',\n",
       " 'Plant anatomy': 'Plant biology',\n",
       " 'Plant cell biology': 'Plant biology',\n",
       " 'Plant physiology': 'Plant biology',\n",
       " 'Clinical veterinary medicine': 'Veterinary medicine',\n",
       " 'Matrix/scaffold attachment region': 'DNA binding sites',\n",
       " 'Restriction sites': 'DNA binding sites',\n",
       " 'Matrix-attachment region': 'DNA binding sites',\n",
       " 'Scaffold-attachment region': 'DNA binding sites',\n",
       " 'Nucleosome exclusion sequences': 'DNA binding sites',\n",
       " 'Ribosome binding sites': 'DNA binding sites',\n",
       " 'Specimen collections': 'Sample collections',\n",
       " 'biosamples': 'Sample collections',\n",
       " 'samples': 'Sample collections',\n",
       " 'Biomedical informatics': 'Medical informatics',\n",
       " 'Clinical informatics': 'Medical informatics',\n",
       " 'Health informatics': 'Medical informatics',\n",
       " 'Healthcare informatics': 'Medical informatics',\n",
       " 'Health and disease': 'Medical informatics',\n",
       " 'Crystallography': 'X-ray diffraction',\n",
       " 'X-ray microscopy': 'X-ray diffraction',\n",
       " 'X-ray crystallography': 'X-ray diffraction',\n",
       " 'Mathematical biology': 'Computational biology',\n",
       " 'Biomathematics': 'Computational biology',\n",
       " 'Theoretical biology': 'Computational biology',\n",
       " 'Heredity': 'Genetics',\n",
       " 'Genes': 'Genetics',\n",
       " 'Information science': 'Informatics',\n",
       " 'Knowledge management': 'Informatics',\n",
       " 'Information management': 'Informatics',\n",
       " 'Alternative medicine': 'Complementary medicine',\n",
       " 'Integrative medicine': 'Complementary medicine',\n",
       " 'Holistic medicine': 'Complementary medicine',\n",
       " 'Protein sorting': 'Protein targeting and localisation',\n",
       " 'Protein targeting': 'Protein targeting and localisation',\n",
       " 'Protein localisation': 'Protein targeting and localisation',\n",
       " 'Child health': 'Paediatrics',\n",
       " 'KDD': 'Data mining',\n",
       " 'Knowledge discovery in databases': 'Data mining',\n",
       " 'Pattern recognition': 'Data mining',\n",
       " 'Chemical screening': 'Compound libraries and screening',\n",
       " 'Compound library': 'Compound libraries and screening',\n",
       " 'Small chemical compounds libraries': 'Compound libraries and screening',\n",
       " 'Chemical library': 'Compound libraries and screening',\n",
       " 'Small compounds libraries': 'Compound libraries and screening',\n",
       " 'Target identification and validation': 'Compound libraries and screening',\n",
       " 'Database curation': 'Data submission, annotation, and curation',\n",
       " 'Data provenance': 'Data submission, annotation, and curation',\n",
       " 'Data curation': 'Data submission, annotation, and curation',\n",
       " 'Bottom-up proteomics': 'Proteomics',\n",
       " 'Peptide identification': 'Proteomics',\n",
       " 'Quantitative proteomics': 'Proteomics',\n",
       " 'Discovery proteomics': 'Proteomics',\n",
       " 'Targeted proteomics': 'Proteomics',\n",
       " 'MS-based targeted proteomics': 'Proteomics',\n",
       " 'MS-based untargeted proteomics': 'Proteomics',\n",
       " 'Protein and peptide identification': 'Proteomics',\n",
       " 'Metaproteomics': 'Proteomics',\n",
       " 'Top-down proteomics': 'Proteomics',\n",
       " 'Laboratory mouse': 'Mouse clinic',\n",
       " 'Nucleic acid informatics': 'Nucleic acids',\n",
       " 'Nucleic acid bioinformatics': 'Nucleic acids',\n",
       " 'Nucleic acid physicochemistry': 'Nucleic acids',\n",
       " 'Nucleic acid properties': 'Nucleic acids',\n",
       " 'Cloud computing': 'Computer science',\n",
       " 'High performance computing': 'Computer science',\n",
       " 'High-performance computing': 'Computer science',\n",
       " 'HPC': 'Computer science',\n",
       " 'CNV deletion': 'Copy number variation',\n",
       " 'Copy number variant': 'Copy number variation',\n",
       " 'CNV duplication': 'Copy number variation',\n",
       " 'Complex CNV': 'Copy number variation',\n",
       " 'CNV insertion / amplification': 'Copy number variation',\n",
       " 'Animal biology': 'Zoology',\n",
       " 'Metazoa': 'Zoology',\n",
       " 'Animal': 'Zoology',\n",
       " 'Animals': 'Zoology',\n",
       " 'Animal physiology': 'Zoology',\n",
       " 'Animal genetics': 'Zoology',\n",
       " 'Entomology': 'Zoology',\n",
       " 'Drug formulation and delivery': 'Biotherapeutics',\n",
       " 'Drug delivery': 'Biotherapeutics',\n",
       " 'Drug formulation': 'Biotherapeutics',\n",
       " 'Traumatology': 'Trauma medicine',\n",
       " 'Structural bioinformatics': 'Structure analysis',\n",
       " 'Biomolecular structure': 'Structure analysis',\n",
       " 'Structure databases': 'Structure analysis',\n",
       " 'Structures': 'Structure analysis',\n",
       " 'Molecular structure': 'Structure analysis',\n",
       " 'Computational structural biology': 'Structure analysis',\n",
       " 'Structure data resources': 'Structure analysis',\n",
       " 'NLP': 'Natural language processing',\n",
       " 'Literature mining': 'Natural language processing',\n",
       " 'BioNLP': 'Natural language processing',\n",
       " 'Text analytics': 'Natural language processing',\n",
       " 'Text data mining': 'Natural language processing',\n",
       " 'Text mining': 'Natural language processing',\n",
       " 'Gastrointestinal disorders': 'Gastroenterology',\n",
       " 'biobanking': 'Biobank',\n",
       " 'Tissue collection': 'Biobank',\n",
       " 'Databases': 'Database management',\n",
       " 'Content management': 'Database management',\n",
       " 'Record management': 'Database management',\n",
       " 'Document management': 'Database management',\n",
       " 'File management': 'Database management',\n",
       " 'Database administration': 'Database management',\n",
       " 'Information systems': 'Database management',\n",
       " 'Biological science': 'Biology',\n",
       " 'Cryobiology': 'Biology',\n",
       " 'Aerobiology': 'Biology',\n",
       " 'Chronobiology': 'Biology',\n",
       " 'Biological rhythms': 'Biology',\n",
       " 'Reproductive biology': 'Biology',\n",
       " 'Behavioural biology': 'Biology',\n",
       " 'Protein-nucleic acid interactions': 'Protein interactions',\n",
       " 'Protein-DNA interactions': 'Protein interactions',\n",
       " 'Protein-RNA interactions': 'Protein interactions',\n",
       " 'Protein-RNA interaction': 'Protein interactions',\n",
       " 'Protein-ligand interactions': 'Protein interactions',\n",
       " 'Protein-protein interactions': 'Protein interactions',\n",
       " 'Protein interaction map': 'Protein interactions',\n",
       " 'Protein interactome': 'Protein interactions',\n",
       " 'Protein interaction networks': 'Protein interactions',\n",
       " 'Protein-DNA interaction': 'Protein interactions',\n",
       " 'Genetic linkage': 'Mapping',\n",
       " 'Synteny': 'Mapping',\n",
       " 'Linkage': 'Mapping',\n",
       " 'Linkage mapping': 'Mapping',\n",
       " 'Gerontology': 'Geriatric medicine',\n",
       " 'Ageing': 'Geriatric medicine',\n",
       " 'Aging': 'Geriatric medicine',\n",
       " 'Geriatrics': 'Geriatric medicine',\n",
       " 'Applied microbiology': 'Biotechnology',\n",
       " 'RiboSeq': 'Ribosome Profiling',\n",
       " 'ribosomal footprinting': 'Ribosome Profiling',\n",
       " 'ribo-seq': 'Ribosome Profiling',\n",
       " 'RIBO-seq': 'Ribosome Profiling',\n",
       " 'Ribo-Seq': 'Ribosome Profiling',\n",
       " 'translation footprinting': 'Ribosome Profiling',\n",
       " 'Blood disorders': 'Haematology',\n",
       " 'Haematological disorders': 'Haematology',\n",
       " 'Protein features (disordered structure)': 'Protein disordered structure',\n",
       " 'Drug design': 'Medicinal chemistry',\n",
       " 'Nucleic acid structure': 'Nucleic acid structure analysis',\n",
       " 'Nucleic acid denaturation': 'Nucleic acid structure analysis',\n",
       " 'Nucleic acid thermodynamics': 'Nucleic acid structure analysis',\n",
       " 'DNA melting': 'Nucleic acid structure analysis',\n",
       " 'DNA structure': 'Nucleic acid structure analysis',\n",
       " 'RNA structure': 'Nucleic acid structure analysis',\n",
       " 'RNA alignment': 'Nucleic acid structure analysis',\n",
       " 'RNA structure alignment': 'Nucleic acid structure analysis',\n",
       " 'Cancer biology': 'Oncology',\n",
       " 'Cancer': 'Oncology',\n",
       " 'Neoplasm': 'Oncology',\n",
       " 'Neoplasms': 'Oncology',\n",
       " 'Genotype-phenotype analysis': 'Genotype and phenotype',\n",
       " 'Genotype-phenotype': 'Genotype and phenotype',\n",
       " 'Genotype and phenotype resources': 'Genotype and phenotype',\n",
       " 'Genotype': 'Genotype and phenotype',\n",
       " 'Genotyping': 'Genotype and phenotype',\n",
       " 'Phenotype': 'Genotype and phenotype',\n",
       " 'Phenotyping': 'Genotype and phenotype',\n",
       " 'Drug development science': 'Drug development',\n",
       " 'Medicines development': 'Drug development',\n",
       " 'Medicine development': 'Drug development',\n",
       " 'DNA variation': 'Genetic variation',\n",
       " 'Polymorphism': 'Genetic variation',\n",
       " 'Mutation': 'Genetic variation',\n",
       " 'Genomic variation': 'Genetic variation',\n",
       " 'Somatic mutations': 'Genetic variation',\n",
       " 'Protein structure': 'Protein structure analysis',\n",
       " 'Protein tertiary structure': 'Protein structure analysis',\n",
       " 'Organisms': 'Model organisms',\n",
       " 'Stem cell research': 'Regenerative medicine',\n",
       " 'Nucleosome positioning': 'DNA packaging',\n",
       " 'Computational immunology': 'Immunoinformatics',\n",
       " 'Dermatological disorders': 'Dermatology',\n",
       " 'Biological chemistry': 'Biochemistry',\n",
       " 'Glycomics': 'Biochemistry',\n",
       " 'Pathobiochemistry': 'Biochemistry',\n",
       " 'Phytochemistry': 'Biochemistry',\n",
       " 'PolyA signal': 'Gene transcripts',\n",
       " 'PolyA site': 'Gene transcripts',\n",
       " 'Fusion transcripts': 'Gene transcripts',\n",
       " 'Gene transcript features': 'Gene transcripts',\n",
       " 'Coding RNA': 'Gene transcripts',\n",
       " 'Transit peptide coding sequence': 'Gene transcripts',\n",
       " 'Signal peptide coding sequence': 'Gene transcripts',\n",
       " 'mRNA features': 'Gene transcripts',\n",
       " 'Introns': 'Gene transcripts',\n",
       " 'Exons': 'Gene transcripts',\n",
       " 'cDNA': 'Gene transcripts',\n",
       " 'mRNA': 'Gene transcripts',\n",
       " 'EST': 'Gene transcripts',\n",
       " 'Development': 'Developmental biology',\n",
       " 'Transposons': 'Mobile genetic elements',\n",
       " 'Biomedical sciences': 'Biomedical science',\n",
       " 'Health science': 'Biomedical science',\n",
       " 'Protein physicochemistry': 'Protein properties',\n",
       " 'Protein hydropathy': 'Protein properties',\n",
       " 'Functional analysis': 'Function analysis',\n",
       " 'Protein function prediction': 'Function analysis',\n",
       " 'Protein function analysis': 'Function analysis',\n",
       " 'VT 1.7.10 Polymer science': 'Chemistry',\n",
       " 'Polymer science': 'Chemistry',\n",
       " 'Chemical science': 'Chemistry',\n",
       " 'Nuclear chemistry': 'Chemistry',\n",
       " 'Physical chemistry': 'Chemistry',\n",
       " 'Inorganic chemistry': 'Chemistry',\n",
       " 'Organic chemistry': 'Chemistry',\n",
       " 'Mathematical chemistry': 'Chemistry',\n",
       " 'Protein features (secondary structure)': 'Protein secondary structure',\n",
       " 'Protein super-secondary structure': 'Protein secondary structure',\n",
       " 'Agriculture': 'Agricultural science',\n",
       " 'Agronomy': 'Agricultural science',\n",
       " 'Agroecology': 'Agricultural science',\n",
       " 'Animal breeding': 'Agricultural science',\n",
       " 'Food security': 'Agricultural science',\n",
       " 'Phytomedicine': 'Agricultural science',\n",
       " 'Soil science': 'Agricultural science',\n",
       " 'Farming systems research': 'Agricultural science',\n",
       " 'Plant nutrition': 'Agricultural science',\n",
       " 'Animal husbandry': 'Agricultural science',\n",
       " 'Food process engineering': 'Agricultural science',\n",
       " 'Plant cultivation': 'Agricultural science',\n",
       " 'Plant breeding': 'Agricultural science',\n",
       " 'Animal nutrition': 'Agricultural science',\n",
       " 'Horticulture': 'Agricultural science',\n",
       " 'Plant pathology': 'Agricultural science',\n",
       " 'Agricultural economics': 'Agricultural science',\n",
       " 'Agricultural biotechnology': 'Agricultural science',\n",
       " 'Protein informatics': 'Proteins',\n",
       " 'Protein bioinformatics': 'Proteins',\n",
       " 'Protein databases': 'Proteins',\n",
       " 'Healthcare RA': 'Regulatory affairs',\n",
       " 'Tree of life': 'Cladistics',\n",
       " 'Microsatellites': 'DNA polymorphism',\n",
       " 'Variable number of tandem repeat polymorphism': 'DNA polymorphism',\n",
       " 'snps': 'DNA polymorphism',\n",
       " 'RFLP': 'DNA polymorphism',\n",
       " 'VNTR': 'DNA polymorphism',\n",
       " 'SNP': 'DNA polymorphism',\n",
       " 'Single nucleotide polymorphism': 'DNA polymorphism',\n",
       " 'Acute medicine': 'Critical care medicine',\n",
       " 'Emergency medicine': 'Critical care medicine',\n",
       " 'Intensive care medicine': 'Critical care medicine',\n",
       " 'Chemical structures': 'Small molecules',\n",
       " 'Toxins and targets': 'Small molecules',\n",
       " 'Amino acids': 'Small molecules',\n",
       " 'Targets': 'Small molecules',\n",
       " 'Drugs and target structures': 'Small molecules',\n",
       " 'Peptides': 'Small molecules',\n",
       " 'Toxins': 'Small molecules',\n",
       " 'Peptides and amino acids': 'Small molecules',\n",
       " 'Metabolite structures': 'Small molecules',\n",
       " 'Target structures': 'Small molecules',\n",
       " 'Drug structures': 'Small molecules',\n",
       " 'Drug targets': 'Small molecules',\n",
       " 'CHEBI:23367': 'Small molecules',\n",
       " 'Laboratory resources': 'Laboratory information management',\n",
       " 'Regulatory genomics': 'Gene regulation',\n",
       " 'Transcription factors': 'Transcription factors and regulatory sites',\n",
       " '-35 signals': 'Transcription factors and regulatory sites',\n",
       " 'CpG islands': 'Transcription factors and regulatory sites',\n",
       " '-10 signals': 'Transcription factors and regulatory sites',\n",
       " 'Enhancers': 'Transcription factors and regulatory sites',\n",
       " 'TFBS': 'Transcription factors and regulatory sites',\n",
       " 'Transcriptional regulatory sites': 'Transcription factors and regulatory sites',\n",
       " 'GC signals': 'Transcription factors and regulatory sites',\n",
       " 'Attenuators': 'Transcription factors and regulatory sites',\n",
       " 'CAAT signals': 'Transcription factors and regulatory sites',\n",
       " 'TATA signals': 'Transcription factors and regulatory sites',\n",
       " 'Terminators': 'Transcription factors and regulatory sites',\n",
       " 'Isochores': 'Transcription factors and regulatory sites',\n",
       " 'CAT box': 'Transcription factors and regulatory sites',\n",
       " 'Promoters': 'Transcription factors and regulatory sites',\n",
       " 'Transcription factor binding sites': 'Transcription factors and regulatory sites',\n",
       " 'CCAAT box': 'Transcription factors and regulatory sites',\n",
       " 'cryo-EM': 'Cryogenic electron microscopy',\n",
       " 'Computer graphics': 'Data visualisation',\n",
       " 'Data rendering': 'Data visualisation'}"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synonym_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4e3dfca45ff942bb9b92efd5d640f564",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/457 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "{'cell RNA': 'RNA',\n",
       " 'Spatial mapping of developing human brain regions': 'Mapping',\n",
       " 'cell RNA sequencing': 'RNA',\n",
       " 'Transcription factor inhibition': 'Gene expression',\n",
       " 'RNA sequencing': 'RNA',\n",
       " 'Development of light': 'Developmental biology',\n",
       " 'nucleus RNA sequencing': 'RNA',\n",
       " 'Transcriptome analysis': 'Transcriptomics',\n",
       " 'Bulk transcriptome analysis': 'Transcriptomics',\n",
       " 'Single cell RNA sequencing (scRNAseq)': 'RNA',\n",
       " 'Gene ontology functional enrichment analysis': 'Ontology and terminology',\n",
       " 'cell RNA sequencing (scRNA': 'RNA',\n",
       " 'Dense genotyping data integration': 'Genotype and phenotype',\n",
       " 'cell transcriptome studies': 'Transcriptomics',\n",
       " 'clamp electrophysiology': 'Physiology',\n",
       " 'cell RNA sequencing (scRNAseq)': 'RNA',\n",
       " '10X Genomics technology': 'Genomics',\n",
       " 'marker expression analysis': 'Gene expression',\n",
       " 'Single cell RNA sequencing': 'RNA',\n",
       " 'nucleus transcriptome profiling': 'Transcriptomics',\n",
       " 'regulatory element mapping': 'Mapping',\n",
       " '10x Genomics Chromium Instrument': 'Genomics',\n",
       " 'Cell culture': 'Cell culture collection',\n",
       " 'Functional analysis': 'Function analysis',\n",
       " 'cell transcriptomics': 'Transcriptomics',\n",
       " 'Spatial transcriptomics': 'Transcriptomics',\n",
       " 'Viral RNA': 'RNA',\n",
       " 'Gene expression analysis': 'Gene expression',\n",
       " 'Single cell RNA': 'RNA',\n",
       " 'Single nucleus RNA sequencing': 'RNA',\n",
       " '10x Genomics platform': 'Genomics',\n",
       " 'Single cell transcriptome analysis': 'Transcriptomics',\n",
       " 'nucleus RNA': 'RNA',\n",
       " 'Spatially resolved transcriptomics': 'Transcriptomics',\n",
       " 'PMA': 'DNA',\n",
       " 'Peripheral blood mononuclear cells (PBMCs) isolation': 'Cell biology',\n",
       " 'Gene expression meta': 'Gene expression',\n",
       " 'Genome': 'Genomics',\n",
       " 'Cell': 'Cell biology',\n",
       " 'Multiome RNA': 'RNA',\n",
       " 'Plasma proteomics': 'Proteomics',\n",
       " 'Illumina NovaSeq S2 and SP 100 cycle dual lane flow cells': 'Cell biology',\n",
       " 'Bulk RNA': 'RNA',\n",
       " 'scRNA': 'RNA',\n",
       " 'Transcription factor analysis': 'Gene expression',\n",
       " 'cell transcriptome profiling': 'Transcriptomics',\n",
       " 'omics single cell profiling': 'Omics',\n",
       " 'Live Animal Return': 'Zoology',\n",
       " 'In situ RNA hybridization': 'RNA',\n",
       " 'sorting cells': 'Cell biology',\n",
       " 'Single cell transcriptome profiling': 'Transcriptomics',\n",
       " 'Single nuclear RNA sequencing (snRNA': 'RNA',\n",
       " 'Tissue collection': 'Biobank',\n",
       " '10X Visium spatial transcriptomics': 'Transcriptomics',\n",
       " 'cell gene expression profiling': 'Gene expression',\n",
       " 'Sorting of NK cells': 'Cell biology',\n",
       " 'Multimodal mapping': 'Mapping',\n",
       " 'Immune landscape mapping': 'Mapping',\n",
       " 'RNA velocity analysis': 'RNA',\n",
       " '10X Genomics Chromium': 'Genomics',\n",
       " 'mRNA profiling': 'Gene transcripts',\n",
       " 'Bioinformatics analysis': 'Bioinformatics',\n",
       " 'cell transcriptome analysis': 'Transcriptomics',\n",
       " 'Deletion analysis': 'Structural variation',\n",
       " 'Multiomics integration using scGlue': 'Multiomics',\n",
       " 'Single nucleus RNA sequencing (snRNA': 'RNA',\n",
       " 'De novo mutation calling from single': 'Genetic variation',\n",
       " '10x Genomics technology': 'Genomics',\n",
       " 'Transplantation': 'Surgery',\n",
       " 'cell interactions': 'Molecular interactions, pathways and networks',\n",
       " 'High throughput spatial transcriptomics': 'Transcriptomics',\n",
       " 'Fibroblast single cell RNA sequencing': 'RNA',\n",
       " 'Whole transcriptome amplification (WTA)': 'Transcriptomics',\n",
       " 'Expression magnitude deviations analysis': 'Gene expression',\n",
       " 'Single cell transcriptomics': 'Transcriptomics',\n",
       " 'nucleus RNA sequencing (snRNA': 'RNA',\n",
       " 'Differential gene expression analysis': 'Gene expression',\n",
       " 'Viral transcriptome analysis': 'Transcriptomics',\n",
       " 'gene expression program analysis': 'Gene expression',\n",
       " 'Single nucleus RNA': 'RNA',\n",
       " 'Septal myectomy surgery': 'Surgery',\n",
       " 'Bioinformatic pipelines': 'Workflows',\n",
       " 'RNA extraction': 'RNA',\n",
       " 'nuclei RNA sequencing': 'RNA',\n",
       " 'nuclei RNA': 'RNA',\n",
       " 'silico algorithm development': 'Developmental biology',\n",
       " 'expression network analysis (WGCNA)': 'Gene expression',\n",
       " 'Stimulation of Treg cells in vitro': 'Cell biology',\n",
       " 'PCR': 'NMR',\n",
       " 'Single Nuclear RNA Sequencing': 'RNA',\n",
       " 'Serum Aptamer Proteomics': 'Proteomics',\n",
       " 'Transcriptome data analysis': 'Transcriptomics',\n",
       " 'Single nuclear RNA sequencing': 'RNA',\n",
       " 'Single nuclear RNA expression': 'RNA',\n",
       " 'Spatial RNA expression': 'RNA',\n",
       " 'Single cell RNA expression': 'RNA',\n",
       " 'cell epigenomics': 'Epigenomics',\n",
       " 'mapping': 'Mapping',\n",
       " 'cell transcriptomics meta': 'Transcriptomics',\n",
       " 'gene expression analysis': 'Gene expression',\n",
       " '10X Genomics Chromium single': 'Genomics',\n",
       " 'Cell Culture': 'Cell culture collection',\n",
       " 'RNA Sequencing': 'RNA',\n",
       " 'Seq v4 Ultra Low Input RNA Kit for Sequencing': 'RNA',\n",
       " 'Cell RNA Sequencing': 'RNA',\n",
       " 'cell expression profiling': 'Gene expression',\n",
       " 'Primary human nonsmoker airway basal stem cells (ABSCs)': 'Cell biology',\n",
       " 'expression quantitative trait loci (eQTL) analysis': 'Gene expression',\n",
       " 'gene expression studies': 'Gene expression',\n",
       " 'cell transcriptome sequencing': 'Transcriptomics',\n",
       " 'TCR': 'NMR',\n",
       " 'BCR': 'NMR',\n",
       " 'Protein': 'Proteins',\n",
       " 'cell whole transcriptome sequencing (scRNAseq)': 'Transcriptomics',\n",
       " 'Analysis of cell surface proteins': 'Proteins',\n",
       " 'Lineage': 'Mapping',\n",
       " 'specific transcription factor analysis': 'Gene expression',\n",
       " 'Molecular mapping': 'Mapping',\n",
       " 'Single cell mRNA sequencing': 'Gene transcripts',\n",
       " 'TNF': 'DNA',\n",
       " \"10x genomics 3' solution\": 'Genomics',\n",
       " 'Induced pluripotent stem cells (iPSCs)': 'Cell biology',\n",
       " 'Bulk DNA methylome analysis': 'DNA',\n",
       " 'Whole transcriptome analysis': 'Transcriptomics',\n",
       " 'Single Cell RNA': 'RNA',\n",
       " '10X Genomics single': 'Genomics',\n",
       " 'Single cell RNA sequencing (sc': 'RNA',\n",
       " \"10x Genomics 3' single cell RNA sequencing\": 'Genomics',\n",
       " 'Transcriptome profiling': 'Transcriptomics',\n",
       " 'receptor pair gene expression analysis': 'Gene expression',\n",
       " 'cell RNA Sequencing': 'RNA',\n",
       " 'Orthogonal RNA and protein expression assessments': 'RNA',\n",
       " 'RAISIN RNA': 'RNA',\n",
       " '10X genomics Kit': 'Genomics',\n",
       " 'precision transcriptomics': 'Transcriptomics',\n",
       " 'single cell RNA sequencing': 'RNA',\n",
       " 'bulk genomics': 'Genomics',\n",
       " 'cell RNA sequencing (RNA': 'RNA',\n",
       " '10X Genomics ChromiumTM Single Cell 3’ Library and Gel Bead Kit v2': 'Genomics',\n",
       " 'Targeted proteomics': 'Proteomics',\n",
       " 'Network': 'Molecular interactions, pathways and networks',\n",
       " 'Isolation of memory T cells': 'Cell biology',\n",
       " 'Single nucleus RNA sequencing (snRNAseq)': 'RNA',\n",
       " 'cell RNA expression profiling': 'RNA',\n",
       " 'Transcriptomics analysis': 'Transcriptomics',\n",
       " '10X Genomics Single Cell Gene Expression': 'Genomics',\n",
       " 'genetic mapping': 'Mapping',\n",
       " 'Data mapping with CellRanger': 'Mapping',\n",
       " 'driven framework development': 'Developmental biology',\n",
       " 'Single nuclei RNA sequencing': 'RNA',\n",
       " '10x Genomics Visium platform': 'Genomics',\n",
       " 'cell genomics': 'Genomics',\n",
       " '10x Genomic 5 expression platform': 'Gene expression',\n",
       " 'Data analysis': 'Data architecture, analysis and design',\n",
       " 'Bone marrow transplantation model': 'Surgery',\n",
       " 'Bulk RNA sequencing': 'RNA',\n",
       " 'Single cell RNA sequencing (scRNA': 'RNA',\n",
       " 'mRNA analysis': 'Gene transcripts',\n",
       " 'RNA in situ hybridization': 'RNA',\n",
       " 'mRNA sequencing': 'Gene transcripts',\n",
       " 'Transcriptome sequencing': 'Transcriptomics',\n",
       " 'cell Allele Specific Expression (sc': 'Gene expression',\n",
       " 'ASE) mapping': 'Mapping',\n",
       " 'caQTL) mapping': 'Mapping',\n",
       " 'eQTL) mapping': 'Mapping',\n",
       " 'cell multiomics profiling': 'Multiomics',\n",
       " 'cell expression quantitative trait loci (sc': 'Gene expression',\n",
       " 'ASCA) mapping': 'Mapping',\n",
       " 'Sequencing of epithelial cells from tracheal donors': 'Cell biology',\n",
       " 'cell RNA sequencing technology': 'RNA',\n",
       " 'single cell RNA sequencing (scRNA': 'RNA',\n",
       " 'Transcription factor identification': 'Gene expression',\n",
       " 'microfluidics': 'Microfluidics',\n",
       " 'cell mRNA sequencing': 'Gene transcripts',\n",
       " '500 colonic mesenchymal cells': 'Cell biology',\n",
       " 'Connectome mapping': 'Mapping',\n",
       " 'T cells': 'Cell biology',\n",
       " 'RNA (snRNA': 'RNA',\n",
       " 'PBMC samples': 'Sample collections',\n",
       " 'Single cell genomics': 'Genomics',\n",
       " 'DNA methylation profiling': 'DNA',\n",
       " 'Bulk RNA sequencing (RNA': 'RNA',\n",
       " 'like cells': 'Cell biology',\n",
       " 'seq) using 10x Genomics': 'Genomics',\n",
       " 'Human induced pluripotent stem cells (hiPSCs)': 'Cell biology',\n",
       " 'Analysis of gene expression of known RefSeq genes': 'Gene expression',\n",
       " 'Isolation of individual cells': 'Cell biology',\n",
       " 'Expression quantitative trait loci mapping (eQTL mapping)': 'Mapping',\n",
       " 'Algorithm development': 'Developmental biology',\n",
       " 'mRNA transcript profiling': 'Gene transcripts',\n",
       " '10x Genomics single cell isolation system': 'Genomics',\n",
       " 'Differential expression analysis': 'Gene expression',\n",
       " 'transcriptome analysis': 'Transcriptomics',\n",
       " 'X chromosome RNA level analysis': 'RNA',\n",
       " 'specific gene expression analysis': 'Gene expression',\n",
       " 'Spatial mapping techniques': 'Mapping',\n",
       " 'cells': 'Cell biology',\n",
       " 'Expression quantitative trait loci (eQTL) analysis': 'Gene expression',\n",
       " 'Lung transplantation': 'Surgery',\n",
       " 'Re': 'RNA',\n",
       " 'Whole genome RNA sequencing': 'RNA',\n",
       " 'CG': 'Electrocardiography',\n",
       " 'specific deletion': 'Structural variation',\n",
       " 'Animal models of viral exacerbations': 'Zoology',\n",
       " 'Animal models': 'Zoology',\n",
       " 'cell proteomics': 'Proteomics',\n",
       " 'Unbiased proteomics': 'Proteomics',\n",
       " 'scale RNA sequencing': 'RNA',\n",
       " 'Gene expression patterns analysis': 'Gene expression',\n",
       " 'Expression quantitative trait mapping': 'Mapping',\n",
       " 'Transcriptome': 'Transcriptomics',\n",
       " 'Reprogramming fibroblasts into induced pluripotent stem cells (iPSCs)': 'Cell biology',\n",
       " 'Bioprinting organoids composed of a single deposition of cells (R0)': 'Cell biology',\n",
       " 'Differentiation of iPS cells': 'Cell biology',\n",
       " 'Bioprinting organoids composed of a thin line of cells (R40)': 'Cell biology',\n",
       " 'Mapping to a custom reference human genome': 'Mapping',\n",
       " 'cell omics analysis': 'Omics',\n",
       " 'Protein expression data analysis': 'Gene expression',\n",
       " 'RNA expression data analysis': 'RNA',\n",
       " '10X Genomics Chromium platform': 'Genomics',\n",
       " 'multiomics analysis': 'Multiomics',\n",
       " 'trophoblast stem cells': 'Cell biology',\n",
       " 'Identification of marker genes': 'Genetics',\n",
       " 'Bulk genetic and expression analysis': 'Gene expression',\n",
       " 'Mutation analysis': 'Genetic variation',\n",
       " 'Transcriptome noise analysis': 'Transcriptomics',\n",
       " 'cell RNA sequencing analysis': 'RNA',\n",
       " 'Single cell RNA sequencing (RNAseq)': 'RNA',\n",
       " 'transcriptomics profiling': 'Transcriptomics',\n",
       " 'Electrophysiology': 'Physiology',\n",
       " 'Generation of induced pluripotent stem cells': 'Cell biology',\n",
       " 'Single cell transcriptome sequencing': 'Transcriptomics',\n",
       " '10X genomics': 'Genomics',\n",
       " 'peripheral blood mononuclear cells (PBMCs)': 'Cell biology',\n",
       " 'nucleus RNA sequencing (snRNAseq)': 'RNA',\n",
       " 'Organoid transplantation': 'Surgery',\n",
       " 'T cells antigen receptor (TCR) analysis': 'Cell biology',\n",
       " 'cell RNA sequencing (RNAseq)': 'RNA',\n",
       " 'based algorithms': 'Software engineering',\n",
       " 'Spatial proteomics': 'Proteomics',\n",
       " 'Deconvolution of bulk gene expression samples': 'Gene expression',\n",
       " 'Cleavage Under Targets and Release Using Nuclease (CUT&RUN)': 'Small molecules',\n",
       " '10x Chromium genomics platform': 'Genomics',\n",
       " 'nucleus transcriptomics': 'Transcriptomics',\n",
       " 'Silencing of gene expression': 'Gene expression',\n",
       " 'scale gene expression analysis': 'Gene expression',\n",
       " 'islet RNA': 'RNA',\n",
       " 'Massively parallel single nucleus RNA sequencing (snRNA': 'RNA',\n",
       " 'Longitudinal assessment of circulating proteins': 'Proteins',\n",
       " 'Single nucleus gene expression sequencing': 'Gene expression',\n",
       " 'Immune phenotyping': 'Genotype and phenotype',\n",
       " '10x genomics': 'Genomics',\n",
       " 'cell RNA profiling': 'RNA',\n",
       " 'Collection of cells': 'Cell biology',\n",
       " 'omics analysis': 'Omics',\n",
       " 'bulk RNA': 'RNA',\n",
       " 'Isolation of single cells from nasal swabs': 'Cell biology',\n",
       " 'cell transcriptome libraries': 'Transcriptomics',\n",
       " 'RNA sequencing (RNA': 'RNA',\n",
       " 'nucleus RNA sequencing (NucSeq)': 'RNA',\n",
       " 'wide expression quantitative trait loci (eQTL) analysis': 'Gene expression'}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import Levenshtein\n",
    "\n",
    "hallucinations = data['Hallucinations']\n",
    "\n",
    "matched_topics = {}\n",
    "\n",
    "for hallucination_set in tqdm(hallucinations):\n",
    "    for hallucination in hallucination_set:\n",
    "        if hallucination in matched_topics:\n",
    "            continue\n",
    "        matched = False\n",
    "        # First check for a match in the topics list\n",
    "        sorted_topics = sorted(edam_topics, key=lambda topic: Levenshtein.distance(hallucination, topic))\n",
    "        for topic in sorted_topics:\n",
    "            distance = Levenshtein.distance(hallucination, topic)\n",
    "            if  0 < distance <= 2:\n",
    "                matched_topics[hallucination] = topic\n",
    "                matched = True\n",
    "                break\n",
    "        \n",
    "        # If the hallucination has already been matched, skip to the next hallucination\n",
    "        if matched:\n",
    "            continue\n",
    "        \n",
    "        # If no match in the topics list, look through the available synonyms\n",
    "        sorted_synonyms = sorted(synonym_dict.keys(), key=lambda topic: Levenshtein.distance(hallucination, topic))\n",
    "        for topic in sorted_synonyms:\n",
    "            distance = Levenshtein.distance(hallucination, topic)\n",
    "            if 0 <= distance <= 1:\n",
    "                matched_topics[hallucination] = synonym_dict[topic]\n",
    "                matched = True\n",
    "                break     \n",
    "\n",
    "        if matched:\n",
    "            continue\n",
    "\n",
    "        for topic in sorted_topics:\n",
    "            if topic.lower() in hallucination.lower().split():\n",
    "                matched_topics[hallucination] = topic\n",
    "                break\n",
    "        # No break reached\n",
    "        else:\n",
    "            for topic in sorted_synonyms:\n",
    "                if topic.lower() in hallucination.lower().split():\n",
    "                    matched_topics[hallucination] = synonym_dict[topic]\n",
    "                    break\n",
    "\n",
    "matched_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'cell RNA' in row 0 matches topic 'RNA'\n",
      "'Spatial mapping of developing human brain regions' in row 0 matches topic 'Mapping'\n",
      "'cell RNA sequencing' in row 1 matches topic 'RNA'\n",
      "'Transcription factor inhibition' in row 1 matches topic 'Gene expression'\n",
      "'RNA sequencing' in row 2 matches topic 'RNA'\n",
      "'Development of light' in row 2 matches topic 'Developmental biology'\n",
      "'cell RNA sequencing' in row 3 matches topic 'RNA'\n",
      "'nucleus RNA sequencing' in row 4 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 4 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 5 matches topic 'RNA'\n",
      "'Bulk transcriptome analysis' in row 5 matches topic 'Transcriptomics'\n",
      "'Single cell RNA sequencing (scRNAseq)' in row 6 matches topic 'RNA'\n",
      "'Gene ontology functional enrichment analysis' in row 7 matches topic 'Ontology and terminology'\n",
      "'cell RNA sequencing (scRNA' in row 7 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 8 matches topic 'RNA'\n",
      "'Dense genotyping data integration' in row 8 matches topic 'Genotype and phenotype'\n",
      "'cell RNA sequencing' in row 9 matches topic 'RNA'\n",
      "'cell transcriptome studies' in row 9 matches topic 'Transcriptomics'\n",
      "'clamp electrophysiology' in row 9 matches topic 'Physiology'\n",
      "'cell RNA sequencing (scRNA' in row 10 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNAseq)' in row 11 matches topic 'RNA'\n",
      "'10X Genomics technology' in row 11 matches topic 'Genomics'\n",
      "'cell RNA sequencing' in row 12 matches topic 'RNA'\n",
      "'marker expression analysis' in row 13 matches topic 'Gene expression'\n",
      "'Single cell RNA sequencing' in row 14 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 15 matches topic 'RNA'\n",
      "'nucleus transcriptome profiling' in row 16 matches topic 'Transcriptomics'\n",
      "'regulatory element mapping' in row 16 matches topic 'Mapping'\n",
      "'cell RNA sequencing (scRNA' in row 17 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 17 matches topic 'Transcriptomics'\n",
      "'10x Genomics Chromium Instrument' in row 18 matches topic 'Genomics'\n",
      "'Cell culture' in row 18 matches topic 'Cell culture collection'\n",
      "'cell RNA sequencing (scRNA' in row 20 matches topic 'RNA'\n",
      "'Functional analysis' in row 20 matches topic 'Function analysis'\n",
      "'cell RNA sequencing' in row 21 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 22 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 24 matches topic 'Transcriptomics'\n",
      "'Spatial transcriptomics' in row 24 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 25 matches topic 'RNA'\n",
      "'Viral RNA' in row 26 matches topic 'RNA'\n",
      "'Gene expression analysis' in row 26 matches topic 'Gene expression'\n",
      "'Single cell RNA' in row 27 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 28 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 29 matches topic 'RNA'\n",
      "'Single nucleus RNA sequencing' in row 30 matches topic 'RNA'\n",
      "'10x Genomics platform' in row 31 matches topic 'Genomics'\n",
      "'Single cell transcriptome analysis' in row 31 matches topic 'Transcriptomics'\n",
      "'nucleus RNA' in row 32 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 34 matches topic 'RNA'\n",
      "'nucleus RNA' in row 35 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 36 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 38 matches topic 'RNA'\n",
      "'Spatially resolved transcriptomics' in row 40 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 40 matches topic 'Transcriptomics'\n",
      "'PMA' in row 41 matches topic 'DNA'\n",
      "'cell RNA' in row 41 matches topic 'RNA'\n",
      "'Peripheral blood mononuclear cells (PBMCs) isolation' in row 41 matches topic 'Cell biology'\n",
      "'cell RNA sequencing (scRNA' in row 42 matches topic 'RNA'\n",
      "'Cell culture' in row 42 matches topic 'Cell culture collection'\n",
      "'cell RNA sequencing' in row 43 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 44 matches topic 'RNA'\n",
      "'Gene expression meta' in row 44 matches topic 'Gene expression'\n",
      "'Genome' in row 44 matches topic 'Genomics'\n",
      "'cell RNA sequencing (scRNA' in row 45 matches topic 'RNA'\n",
      "'Cell' in row 45 matches topic 'Cell biology'\n",
      "'Multiome RNA' in row 46 matches topic 'RNA'\n",
      "'Plasma proteomics' in row 47 matches topic 'Proteomics'\n",
      "'cell RNA' in row 47 matches topic 'RNA'\n",
      "'Illumina NovaSeq S2 and SP 100 cycle dual lane flow cells' in row 47 matches topic 'Cell biology'\n",
      "'Bulk RNA' in row 47 matches topic 'RNA'\n",
      "'scRNA' in row 47 matches topic 'RNA'\n",
      "'Transcription factor analysis' in row 48 matches topic 'Gene expression'\n",
      "'Single cell RNA' in row 48 matches topic 'RNA'\n",
      "'cell transcriptome profiling' in row 49 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 50 matches topic 'RNA'\n",
      "'RNA sequencing' in row 51 matches topic 'RNA'\n",
      "'omics single cell profiling' in row 51 matches topic 'Omics'\n",
      "'Gene expression analysis' in row 51 matches topic 'Gene expression'\n",
      "'cell RNA' in row 52 matches topic 'RNA'\n",
      "'Live Animal Return' in row 53 matches topic 'Zoology'\n",
      "'cell transcriptomics' in row 54 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 54 matches topic 'RNA'\n",
      "'nucleus RNA' in row 54 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 55 matches topic 'RNA'\n",
      "'Bulk RNA' in row 55 matches topic 'RNA'\n",
      "'In situ RNA hybridization' in row 55 matches topic 'RNA'\n",
      "'sorting cells' in row 55 matches topic 'Cell biology'\n",
      "'Single cell transcriptome profiling' in row 56 matches topic 'Transcriptomics'\n",
      "'Single nuclear RNA sequencing (snRNA' in row 57 matches topic 'RNA'\n",
      "'Tissue collection' in row 58 matches topic 'Biobank'\n",
      "'cell RNA sequencing (scRNA' in row 58 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 58 matches topic 'Transcriptomics'\n",
      "'Gene expression analysis' in row 58 matches topic 'Gene expression'\n",
      "'Transcriptome analysis' in row 59 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 61 matches topic 'RNA'\n",
      "'10X Visium spatial transcriptomics' in row 61 matches topic 'Transcriptomics'\n",
      "'cell gene expression profiling' in row 62 matches topic 'Gene expression'\n",
      "'cell RNA' in row 63 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 65 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 66 matches topic 'RNA'\n",
      "'Sorting of NK cells' in row 66 matches topic 'Cell biology'\n",
      "'Multimodal mapping' in row 67 matches topic 'Mapping'\n",
      "'Immune landscape mapping' in row 67 matches topic 'Mapping'\n",
      "'RNA sequencing' in row 69 matches topic 'RNA'\n",
      "'RNA velocity analysis' in row 69 matches topic 'RNA'\n",
      "'Cell' in row 69 matches topic 'Cell biology'\n",
      "'cell RNA sequencing' in row 70 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 71 matches topic 'Transcriptomics'\n",
      "'10X Genomics Chromium' in row 71 matches topic 'Genomics'\n",
      "'nucleus RNA sequencing' in row 72 matches topic 'RNA'\n",
      "'mRNA profiling' in row 73 matches topic 'Gene transcripts'\n",
      "'cell RNA sequencing (scRNA' in row 73 matches topic 'RNA'\n",
      "'Bioinformatics analysis' in row 73 matches topic 'Bioinformatics'\n",
      "'cell transcriptome analysis' in row 74 matches topic 'Transcriptomics'\n",
      "'Deletion analysis' in row 74 matches topic 'Structural variation'\n",
      "'Multiomics integration using scGlue' in row 76 matches topic 'Multiomics'\n",
      "'Single nucleus RNA sequencing (snRNA' in row 76 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 77 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 78 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 78 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 78 matches topic 'Transcriptomics'\n",
      "'De novo mutation calling from single' in row 78 matches topic 'Genetic variation'\n",
      "'cell RNA sequencing (scRNA' in row 79 matches topic 'RNA'\n",
      "'10x Genomics technology' in row 79 matches topic 'Genomics'\n",
      "'Transplantation' in row 80 matches topic 'Surgery'\n",
      "'Cell culture' in row 80 matches topic 'Cell culture collection'\n",
      "'cell RNA sequencing' in row 81 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 81 matches topic 'Transcriptomics'\n",
      "'cell interactions' in row 81 matches topic 'Molecular interactions, pathways and networks'\n",
      "'cell RNA' in row 82 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 83 matches topic 'RNA'\n",
      "'High throughput spatial transcriptomics' in row 83 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 84 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 85 matches topic 'RNA'\n",
      "'Fibroblast single cell RNA sequencing' in row 85 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 86 matches topic 'RNA'\n",
      "'Whole transcriptome amplification (WTA)' in row 87 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing (scRNA' in row 87 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 88 matches topic 'RNA'\n",
      "'Expression magnitude deviations analysis' in row 88 matches topic 'Gene expression'\n",
      "'Single cell transcriptomics' in row 89 matches topic 'Transcriptomics'\n",
      "'nucleus RNA sequencing (snRNA' in row 90 matches topic 'RNA'\n",
      "'Differential gene expression analysis' in row 90 matches topic 'Gene expression'\n",
      "'Viral transcriptome analysis' in row 90 matches topic 'Transcriptomics'\n",
      "'Single cell RNA' in row 91 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 92 matches topic 'Transcriptomics'\n",
      "'Bioinformatics analysis' in row 92 matches topic 'Bioinformatics'\n",
      "'cell RNA sequencing' in row 95 matches topic 'RNA'\n",
      "'gene expression program analysis' in row 95 matches topic 'Gene expression'\n",
      "'nucleus RNA sequencing' in row 96 matches topic 'RNA'\n",
      "'Single nucleus RNA' in row 97 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 98 matches topic 'RNA'\n",
      "'Septal myectomy surgery' in row 99 matches topic 'Surgery'\n",
      "'cell RNA sequencing (scRNA' in row 99 matches topic 'RNA'\n",
      "'RNA sequencing' in row 101 matches topic 'RNA'\n",
      "'Bioinformatic pipelines' in row 101 matches topic 'Workflows'\n",
      "'RNA extraction' in row 101 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 102 matches topic 'RNA'\n",
      "'nuclei RNA sequencing' in row 103 matches topic 'RNA'\n",
      "'Transplantation' in row 104 matches topic 'Surgery'\n",
      "'cell RNA sequencing (scRNA' in row 104 matches topic 'RNA'\n",
      "'nuclei RNA' in row 106 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 107 matches topic 'RNA'\n",
      "'scRNA' in row 108 matches topic 'RNA'\n",
      "'silico algorithm development' in row 109 matches topic 'Developmental biology'\n",
      "'Single cell RNA' in row 109 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 111 matches topic 'RNA'\n",
      "'cell RNA' in row 112 matches topic 'RNA'\n",
      "'expression network analysis (WGCNA)' in row 112 matches topic 'Gene expression'\n",
      "'Stimulation of Treg cells in vitro' in row 113 matches topic 'Cell biology'\n",
      "'nucleus RNA sequencing (snRNA' in row 114 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 115 matches topic 'RNA'\n",
      "'PCR' in row 116 matches topic 'NMR'\n",
      "'Single Nuclear RNA Sequencing' in row 116 matches topic 'RNA'\n",
      "'Serum Aptamer Proteomics' in row 116 matches topic 'Proteomics'\n",
      "'Transcriptome data analysis' in row 117 matches topic 'Transcriptomics'\n",
      "'Single cell RNA' in row 118 matches topic 'RNA'\n",
      "'cell RNA' in row 119 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 120 matches topic 'Transcriptomics'\n",
      "'Spatial transcriptomics' in row 120 matches topic 'Transcriptomics'\n",
      "'Single nuclear RNA sequencing' in row 121 matches topic 'RNA'\n",
      "'Single nuclear RNA expression' in row 122 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 122 matches topic 'Transcriptomics'\n",
      "'Spatial RNA expression' in row 122 matches topic 'RNA'\n",
      "'Single cell RNA expression' in row 122 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 123 matches topic 'RNA'\n",
      "'nucleus RNA sequencing' in row 124 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 125 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 126 matches topic 'RNA'\n",
      "'Genome' in row 127 matches topic 'Genomics'\n",
      "'cell epigenomics' in row 127 matches topic 'Epigenomics'\n",
      "'mapping' in row 127 matches topic 'Mapping'\n",
      "'cell RNA sequencing' in row 128 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 128 matches topic 'Transcriptomics'\n",
      "'Spatial transcriptomics' in row 129 matches topic 'Transcriptomics'\n",
      "'nucleus RNA sequencing' in row 129 matches topic 'RNA'\n",
      "'cell transcriptomics meta' in row 130 matches topic 'Transcriptomics'\n",
      "'gene expression analysis' in row 130 matches topic 'Gene expression'\n",
      "'10X Genomics Chromium single' in row 131 matches topic 'Genomics'\n",
      "'Single cell RNA sequencing' in row 133 matches topic 'RNA'\n",
      "'Cell Culture' in row 134 matches topic 'Cell culture collection'\n",
      "'RNA Sequencing' in row 134 matches topic 'RNA'\n",
      "'Seq v4 Ultra Low Input RNA Kit for Sequencing' in row 134 matches topic 'RNA'\n",
      "'Cell RNA Sequencing' in row 134 matches topic 'RNA'\n",
      "'Single nucleus RNA' in row 136 matches topic 'RNA'\n",
      "'cell expression profiling' in row 137 matches topic 'Gene expression'\n",
      "'Primary human nonsmoker airway basal stem cells (ABSCs)' in row 138 matches topic 'Cell biology'\n",
      "'cell RNA sequencing (scRNA' in row 139 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 140 matches topic 'RNA'\n",
      "'expression quantitative trait loci (eQTL) analysis' in row 141 matches topic 'Gene expression'\n",
      "'cell epigenomics' in row 141 matches topic 'Epigenomics'\n",
      "'gene expression studies' in row 141 matches topic 'Gene expression'\n",
      "'mapping' in row 141 matches topic 'Mapping'\n",
      "'cell RNA sequencing' in row 142 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 142 matches topic 'Transcriptomics'\n",
      "'Bulk RNA' in row 142 matches topic 'RNA'\n",
      "'cell transcriptome sequencing' in row 143 matches topic 'Transcriptomics'\n",
      "'TCR' in row 143 matches topic 'NMR'\n",
      "'BCR' in row 143 matches topic 'NMR'\n",
      "'cell RNA sequencing' in row 143 matches topic 'RNA'\n",
      "'Protein' in row 143 matches topic 'Proteins'\n",
      "'Single nucleus RNA' in row 144 matches topic 'RNA'\n",
      "'RNA sequencing' in row 145 matches topic 'RNA'\n",
      "'nucleus RNA sequencing (snRNA' in row 146 matches topic 'RNA'\n",
      "'cell transcriptome sequencing' in row 147 matches topic 'Transcriptomics'\n",
      "'Spatial transcriptomics' in row 147 matches topic 'Transcriptomics'\n",
      "'nucleus RNA sequencing' in row 149 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 151 matches topic 'RNA'\n",
      "'RNA sequencing' in row 151 matches topic 'RNA'\n",
      "'cell transcriptome profiling' in row 152 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 153 matches topic 'RNA'\n",
      "'cell whole transcriptome sequencing (scRNAseq)' in row 153 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 154 matches topic 'RNA'\n",
      "'Analysis of cell surface proteins' in row 154 matches topic 'Proteins'\n",
      "'Single cell transcriptomics' in row 155 matches topic 'Transcriptomics'\n",
      "'Single cell transcriptome profiling' in row 155 matches topic 'Transcriptomics'\n",
      "'scRNA' in row 156 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 157 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 159 matches topic 'Transcriptomics'\n",
      "'Lineage' in row 159 matches topic 'Mapping'\n",
      "'specific transcription factor analysis' in row 159 matches topic 'Gene expression'\n",
      "'cell RNA sequencing' in row 160 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 160 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 161 matches topic 'RNA'\n",
      "'Molecular mapping' in row 161 matches topic 'Mapping'\n",
      "'Single cell mRNA sequencing' in row 163 matches topic 'Gene transcripts'\n",
      "'TNF' in row 165 matches topic 'DNA'\n",
      "'cell RNA sequencing' in row 166 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 167 matches topic 'RNA'\n",
      "'Differential gene expression analysis' in row 167 matches topic 'Gene expression'\n",
      "'cell RNA sequencing' in row 168 matches topic 'RNA'\n",
      "'Gene expression analysis' in row 169 matches topic 'Gene expression'\n",
      "'cell RNA' in row 170 matches topic 'RNA'\n",
      "'Genome' in row 170 matches topic 'Genomics'\n",
      "'10x genomics 3' solution' in row 170 matches topic 'Genomics'\n",
      "'cell RNA' in row 171 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 172 matches topic 'RNA'\n",
      "'Induced pluripotent stem cells (iPSCs)' in row 172 matches topic 'Cell biology'\n",
      "'Single cell RNA sequencing' in row 173 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 173 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 174 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 174 matches topic 'Transcriptomics'\n",
      "'cell transcriptome analysis' in row 175 matches topic 'Transcriptomics'\n",
      "'Bulk DNA methylome analysis' in row 175 matches topic 'DNA'\n",
      "'Bulk transcriptome analysis' in row 175 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 177 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 177 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 178 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 180 matches topic 'Transcriptomics'\n",
      "'Whole transcriptome analysis' in row 180 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing (scRNA' in row 181 matches topic 'RNA'\n",
      "'cell RNA' in row 182 matches topic 'RNA'\n",
      "'cell RNA' in row 183 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 184 matches topic 'RNA'\n",
      "'Protein' in row 184 matches topic 'Proteins'\n",
      "'cell RNA sequencing' in row 185 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 186 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 187 matches topic 'RNA'\n",
      "'Single Cell RNA' in row 187 matches topic 'RNA'\n",
      "'10X Genomics single' in row 187 matches topic 'Genomics'\n",
      "'cell RNA' in row 188 matches topic 'RNA'\n",
      "'Single cell RNA sequencing (sc' in row 189 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 190 matches topic 'RNA'\n",
      "'10x Genomics 3' single cell RNA sequencing' in row 191 matches topic 'Genomics'\n",
      "'Single cell RNA sequencing' in row 191 matches topic 'RNA'\n",
      "'Transcriptome profiling' in row 192 matches topic 'Transcriptomics'\n",
      "'Differential gene expression analysis' in row 193 matches topic 'Gene expression'\n",
      "'receptor pair gene expression analysis' in row 193 matches topic 'Gene expression'\n",
      "'Single nucleus RNA' in row 193 matches topic 'RNA'\n",
      "'cell RNA Sequencing' in row 194 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 195 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 196 matches topic 'RNA'\n",
      "'Orthogonal RNA and protein expression assessments' in row 196 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 197 matches topic 'RNA'\n",
      "'RAISIN RNA' in row 198 matches topic 'RNA'\n",
      "'cell RNA' in row 199 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 200 matches topic 'Transcriptomics'\n",
      "'10X genomics Kit' in row 200 matches topic 'Genomics'\n",
      "'precision transcriptomics' in row 200 matches topic 'Transcriptomics'\n",
      "'single cell RNA sequencing' in row 201 matches topic 'RNA'\n",
      "'bulk genomics' in row 201 matches topic 'Genomics'\n",
      "'RNA extraction' in row 202 matches topic 'RNA'\n",
      "'cell RNA sequencing (RNA' in row 202 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 203 matches topic 'Transcriptomics'\n",
      "'10X Genomics ChromiumTM Single Cell 3’ Library and Gel Bead Kit v2' in row 203 matches topic 'Genomics'\n",
      "'Targeted proteomics' in row 203 matches topic 'Proteomics'\n",
      "'cell RNA sequencing' in row 204 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 207 matches topic 'RNA'\n",
      "'Genome' in row 207 matches topic 'Genomics'\n",
      "'Genome' in row 208 matches topic 'Genomics'\n",
      "'cell RNA sequencing (scRNA' in row 208 matches topic 'RNA'\n",
      "'Network' in row 208 matches topic 'Molecular interactions, pathways and networks'\n",
      "'Single cell mRNA sequencing' in row 209 matches topic 'Gene transcripts'\n",
      "'nucleus RNA sequencing' in row 210 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 211 matches topic 'RNA'\n",
      "'Isolation of memory T cells' in row 211 matches topic 'Cell biology'\n",
      "'Single nucleus RNA sequencing (snRNAseq)' in row 212 matches topic 'RNA'\n",
      "'cell RNA expression profiling' in row 213 matches topic 'RNA'\n",
      "'cell RNA' in row 213 matches topic 'RNA'\n",
      "'cell RNA' in row 214 matches topic 'RNA'\n",
      "'Transcriptomics analysis' in row 214 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing (scRNA' in row 215 matches topic 'RNA'\n",
      "'cell RNA' in row 216 matches topic 'RNA'\n",
      "'10X Genomics Single Cell Gene Expression' in row 216 matches topic 'Genomics'\n",
      "'cell RNA sequencing' in row 217 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 220 matches topic 'RNA'\n",
      "'genetic mapping' in row 220 matches topic 'Mapping'\n",
      "'cell RNA sequencing (scRNA' in row 221 matches topic 'RNA'\n",
      "'cell RNA' in row 222 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 223 matches topic 'RNA'\n",
      "'Data mapping with CellRanger' in row 223 matches topic 'Mapping'\n",
      "'driven framework development' in row 225 matches topic 'Developmental biology'\n",
      "'Single nuclei RNA sequencing' in row 225 matches topic 'RNA'\n",
      "'10x Genomics Visium platform' in row 225 matches topic 'Genomics'\n",
      "'Single cell RNA sequencing' in row 226 matches topic 'RNA'\n",
      "'Single nuclei RNA sequencing' in row 226 matches topic 'RNA'\n",
      "'cell RNA' in row 227 matches topic 'RNA'\n",
      "'cell genomics' in row 227 matches topic 'Genomics'\n",
      "'10x Genomic 5 expression platform' in row 227 matches topic 'Gene expression'\n",
      "'cell RNA sequencing (scRNA' in row 228 matches topic 'RNA'\n",
      "'nucleus RNA' in row 228 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 229 matches topic 'RNA'\n",
      "'Transcriptome profiling' in row 229 matches topic 'Transcriptomics'\n",
      "'cell genomics' in row 230 matches topic 'Genomics'\n",
      "'cell RNA sequencing' in row 231 matches topic 'RNA'\n",
      "'Transcriptome profiling' in row 231 matches topic 'Transcriptomics'\n",
      "'Data analysis' in row 231 matches topic 'Data architecture, analysis and design'\n",
      "'cell RNA sequencing' in row 232 matches topic 'RNA'\n",
      "'Single cell RNA' in row 233 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 234 matches topic 'RNA'\n",
      "'Bone marrow transplantation model' in row 234 matches topic 'Surgery'\n",
      "'Bulk RNA sequencing' in row 235 matches topic 'RNA'\n",
      "'Single cell RNA sequencing (scRNA' in row 235 matches topic 'RNA'\n",
      "'mRNA analysis' in row 236 matches topic 'Gene transcripts'\n",
      "'cell RNA sequencing' in row 237 matches topic 'RNA'\n",
      "'RNA in situ hybridization' in row 237 matches topic 'RNA'\n",
      "'Cell' in row 237 matches topic 'Cell biology'\n",
      "'cell RNA sequencing' in row 238 matches topic 'RNA'\n",
      "'Transcriptome profiling' in row 238 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 239 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 240 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 241 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 242 matches topic 'RNA'\n",
      "'mRNA sequencing' in row 242 matches topic 'Gene transcripts'\n",
      "'cell RNA sequencing' in row 243 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 244 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 245 matches topic 'RNA'\n",
      "'Transcriptome sequencing' in row 245 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 246 matches topic 'RNA'\n",
      "'cell Allele Specific Expression (sc' in row 247 matches topic 'Gene expression'\n",
      "'ASE) mapping' in row 247 matches topic 'Mapping'\n",
      "'caQTL) mapping' in row 247 matches topic 'Mapping'\n",
      "'eQTL) mapping' in row 247 matches topic 'Mapping'\n",
      "'cell multiomics profiling' in row 247 matches topic 'Multiomics'\n",
      "'cell expression quantitative trait loci (sc' in row 247 matches topic 'Gene expression'\n",
      "'ASCA) mapping' in row 247 matches topic 'Mapping'\n",
      "'Bulk RNA sequencing' in row 248 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 248 matches topic 'RNA'\n",
      "'Sequencing of epithelial cells from tracheal donors' in row 249 matches topic 'Cell biology'\n",
      "'Single cell RNA' in row 249 matches topic 'RNA'\n",
      "'nucleus RNA sequencing (snRNA' in row 250 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 250 matches topic 'RNA'\n",
      "'Single cell RNA' in row 251 matches topic 'RNA'\n",
      "'cell RNA sequencing technology' in row 253 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 254 matches topic 'RNA'\n",
      "'single cell RNA sequencing (scRNA' in row 255 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 256 matches topic 'RNA'\n",
      "'Single nuclei RNA sequencing' in row 257 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 257 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 259 matches topic 'RNA'\n",
      "'Transcription factor identification' in row 259 matches topic 'Gene expression'\n",
      "'cell RNA sequencing' in row 261 matches topic 'RNA'\n",
      "'microfluidics' in row 262 matches topic 'Microfluidics'\n",
      "'cell mRNA sequencing' in row 262 matches topic 'Gene transcripts'\n",
      "'500 colonic mesenchymal cells' in row 263 matches topic 'Cell biology'\n",
      "'cell transcriptomics' in row 264 matches topic 'Transcriptomics'\n",
      "'cell interactions' in row 264 matches topic 'Molecular interactions, pathways and networks'\n",
      "'Spatial transcriptomics' in row 265 matches topic 'Transcriptomics'\n",
      "'Connectome mapping' in row 265 matches topic 'Mapping'\n",
      "'Single cell transcriptome analysis' in row 266 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 267 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 268 matches topic 'RNA'\n",
      "'T cells' in row 268 matches topic 'Cell biology'\n",
      "'cell RNA' in row 269 matches topic 'RNA'\n",
      "'cell RNA' in row 270 matches topic 'RNA'\n",
      "'nucleus RNA sequencing' in row 271 matches topic 'RNA'\n",
      "'Bulk RNA' in row 272 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 272 matches topic 'RNA'\n",
      "'Single cell RNA' in row 273 matches topic 'RNA'\n",
      "'RNA (snRNA' in row 275 matches topic 'RNA'\n",
      "'Single cell RNA' in row 276 matches topic 'RNA'\n",
      "'Gene expression analysis' in row 278 matches topic 'Gene expression'\n",
      "'PBMC samples' in row 279 matches topic 'Sample collections'\n",
      "'Single cell genomics' in row 279 matches topic 'Genomics'\n",
      "'cell RNA sequencing' in row 280 matches topic 'RNA'\n",
      "'DNA methylation profiling' in row 281 matches topic 'DNA'\n",
      "'cell RNA sequencing (RNA' in row 281 matches topic 'RNA'\n",
      "'Bulk RNA sequencing (RNA' in row 281 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 282 matches topic 'RNA'\n",
      "'like cells' in row 283 matches topic 'Cell biology'\n",
      "'Single cell RNA' in row 283 matches topic 'RNA'\n",
      "'seq) using 10x Genomics' in row 283 matches topic 'Genomics'\n",
      "'cell RNA' in row 284 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 285 matches topic 'RNA'\n",
      "'Human induced pluripotent stem cells (hiPSCs)' in row 285 matches topic 'Cell biology'\n",
      "'cell RNA' in row 286 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 287 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 288 matches topic 'RNA'\n",
      "'Analysis of gene expression of known RefSeq genes' in row 289 matches topic 'Gene expression'\n",
      "'Isolation of individual cells' in row 289 matches topic 'Cell biology'\n",
      "'Single cell RNA' in row 289 matches topic 'RNA'\n",
      "'Bulk RNA sequencing' in row 291 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 291 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 293 matches topic 'Transcriptomics'\n",
      "'Single nucleus RNA sequencing' in row 293 matches topic 'RNA'\n",
      "'RNA sequencing' in row 294 matches topic 'RNA'\n",
      "'Gene expression analysis' in row 295 matches topic 'Gene expression'\n",
      "'cell RNA sequencing (scRNA' in row 296 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 297 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 298 matches topic 'Transcriptomics'\n",
      "'Expression quantitative trait loci mapping (eQTL mapping)' in row 299 matches topic 'Mapping'\n",
      "'cell RNA sequencing (scRNA' in row 299 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 300 matches topic 'Transcriptomics'\n",
      "'Algorithm development' in row 300 matches topic 'Developmental biology'\n",
      "'cell RNA sequencing (scRNA' in row 301 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 303 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 304 matches topic 'RNA'\n",
      "'cell RNA' in row 305 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 306 matches topic 'RNA'\n",
      "'mRNA transcript profiling' in row 307 matches topic 'Gene transcripts'\n",
      "'cell genomics' in row 307 matches topic 'Genomics'\n",
      "'Genome' in row 308 matches topic 'Genomics'\n",
      "'Single cell RNA sequencing' in row 309 matches topic 'RNA'\n",
      "'10x Genomics single cell isolation system' in row 310 matches topic 'Genomics'\n",
      "'cell RNA sequencing' in row 311 matches topic 'RNA'\n",
      "'Data analysis' in row 311 matches topic 'Data architecture, analysis and design'\n",
      "'Genome' in row 312 matches topic 'Genomics'\n",
      "'Bulk RNA' in row 312 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 312 matches topic 'RNA'\n",
      "'10x Genomics platform' in row 312 matches topic 'Genomics'\n",
      "'Differential expression analysis' in row 313 matches topic 'Gene expression'\n",
      "'cell mRNA sequencing' in row 314 matches topic 'Gene transcripts'\n",
      "'Functional analysis' in row 314 matches topic 'Function analysis'\n",
      "'Single cell RNA sequencing (scRNA' in row 316 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 317 matches topic 'RNA'\n",
      "'transcriptome analysis' in row 318 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 319 matches topic 'RNA'\n",
      "'Transcriptome profiling' in row 319 matches topic 'Transcriptomics'\n",
      "'X chromosome RNA level analysis' in row 319 matches topic 'RNA'\n",
      "'Lineage' in row 319 matches topic 'Mapping'\n",
      "'specific gene expression analysis' in row 319 matches topic 'Gene expression'\n",
      "'cell RNA' in row 320 matches topic 'RNA'\n",
      "'cell transcriptome analysis' in row 321 matches topic 'Transcriptomics'\n",
      "'nucleus RNA sequencing (snRNA' in row 322 matches topic 'RNA'\n",
      "'cell RNA' in row 323 matches topic 'RNA'\n",
      "'nucleus RNA' in row 323 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 324 matches topic 'Transcriptomics'\n",
      "'Transcriptome profiling' in row 324 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 325 matches topic 'Transcriptomics'\n",
      "'Spatial mapping techniques' in row 325 matches topic 'Mapping'\n",
      "'cell RNA sequencing' in row 326 matches topic 'RNA'\n",
      "'cells' in row 326 matches topic 'Cell biology'\n",
      "'cell mRNA sequencing' in row 327 matches topic 'Gene transcripts'\n",
      "'cell RNA sequencing' in row 328 matches topic 'RNA'\n",
      "'Expression quantitative trait loci (eQTL) analysis' in row 328 matches topic 'Gene expression'\n",
      "'cell RNA sequencing' in row 329 matches topic 'RNA'\n",
      "'Lung transplantation' in row 329 matches topic 'Surgery'\n",
      "'cell mRNA sequencing' in row 330 matches topic 'Gene transcripts'\n",
      "'nuclei RNA' in row 331 matches topic 'RNA'\n",
      "'Bulk RNA' in row 331 matches topic 'RNA'\n",
      "'nucleus RNA' in row 332 matches topic 'RNA'\n",
      "'Single cell RNA sequencing (scRNA' in row 333 matches topic 'RNA'\n",
      "'Bulk RNA sequencing (RNA' in row 333 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 334 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing (scRNA' in row 334 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 335 matches topic 'RNA'\n",
      "'Re' in row 335 matches topic 'RNA'\n",
      "'cell RNA' in row 336 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 337 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 338 matches topic 'RNA'\n",
      "'cell RNA' in row 339 matches topic 'RNA'\n",
      "'Single cell RNA sequencing (scRNA' in row 340 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 341 matches topic 'RNA'\n",
      "'Whole genome RNA sequencing' in row 341 matches topic 'RNA'\n",
      "'CG' in row 341 matches topic 'Electrocardiography'\n",
      "'cell RNA sequencing (scRNAseq)' in row 342 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNAseq)' in row 343 matches topic 'RNA'\n",
      "'specific deletion' in row 344 matches topic 'Structural variation'\n",
      "'Animal models of viral exacerbations' in row 344 matches topic 'Zoology'\n",
      "'cell RNA sequencing' in row 345 matches topic 'RNA'\n",
      "'Transcriptome analysis' in row 345 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 346 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 347 matches topic 'RNA'\n",
      "'scRNA' in row 348 matches topic 'RNA'\n",
      "'Animal models' in row 349 matches topic 'Zoology'\n",
      "'Cell culture' in row 349 matches topic 'Cell culture collection'\n",
      "'cell RNA sequencing' in row 350 matches topic 'RNA'\n",
      "'nucleus RNA sequencing (snRNA' in row 351 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 351 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing (scRNA' in row 351 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 352 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 353 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNAseq)' in row 354 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 355 matches topic 'RNA'\n",
      "'cell transcriptomics' in row 356 matches topic 'Transcriptomics'\n",
      "'cell proteomics' in row 356 matches topic 'Proteomics'\n",
      "'Unbiased proteomics' in row 356 matches topic 'Proteomics'\n",
      "'cell RNA' in row 357 matches topic 'RNA'\n",
      "'scale RNA sequencing' in row 358 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 359 matches topic 'RNA'\n",
      "'Gene expression patterns analysis' in row 359 matches topic 'Gene expression'\n",
      "'cell RNA' in row 360 matches topic 'RNA'\n",
      "'cell RNA' in row 361 matches topic 'RNA'\n",
      "'Expression quantitative trait mapping' in row 361 matches topic 'Mapping'\n",
      "'Transcriptome' in row 361 matches topic 'Transcriptomics'\n",
      "'Reprogramming fibroblasts into induced pluripotent stem cells (iPSCs)' in row 361 matches topic 'Cell biology'\n",
      "'Single cell RNA' in row 362 matches topic 'RNA'\n",
      "'Bioprinting organoids composed of a single deposition of cells (R0)' in row 363 matches topic 'Cell biology'\n",
      "'Differentiation of iPS cells' in row 363 matches topic 'Cell biology'\n",
      "'Single cell RNA' in row 363 matches topic 'RNA'\n",
      "'Bioprinting organoids composed of a thin line of cells (R40)' in row 363 matches topic 'Cell biology'\n",
      "'cell RNA sequencing' in row 365 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 366 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 367 matches topic 'RNA'\n",
      "'Mapping to a custom reference human genome' in row 367 matches topic 'Mapping'\n",
      "'Cell culture' in row 367 matches topic 'Cell culture collection'\n",
      "'cell omics analysis' in row 368 matches topic 'Omics'\n",
      "'Protein expression data analysis' in row 368 matches topic 'Gene expression'\n",
      "'RNA expression data analysis' in row 368 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 369 matches topic 'RNA'\n",
      "'Single cell RNA' in row 370 matches topic 'RNA'\n",
      "'10X Genomics Chromium platform' in row 371 matches topic 'Genomics'\n",
      "'nucleus RNA' in row 371 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 372 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 372 matches topic 'Transcriptomics'\n",
      "'multiomics analysis' in row 372 matches topic 'Multiomics'\n",
      "'trophoblast stem cells' in row 372 matches topic 'Cell biology'\n",
      "'cell RNA sequencing' in row 373 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 373 matches topic 'RNA'\n",
      "'Single cell RNA' in row 374 matches topic 'RNA'\n",
      "'Identification of marker genes' in row 375 matches topic 'Genetics'\n",
      "'cell RNA sequencing (scRNA' in row 375 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 376 matches topic 'RNA'\n",
      "'Bulk genetic and expression analysis' in row 376 matches topic 'Gene expression'\n",
      "'cell RNA' in row 377 matches topic 'RNA'\n",
      "'cell proteomics' in row 377 matches topic 'Proteomics'\n",
      "'cell RNA sequencing' in row 378 matches topic 'RNA'\n",
      "'nucleus RNA' in row 379 matches topic 'RNA'\n",
      "'Mutation analysis' in row 380 matches topic 'Genetic variation'\n",
      "'cell transcriptome analysis' in row 380 matches topic 'Transcriptomics'\n",
      "'RNA sequencing' in row 380 matches topic 'RNA'\n",
      "'Transcriptome noise analysis' in row 380 matches topic 'Transcriptomics'\n",
      "'cell transcriptomics' in row 381 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 382 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 384 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 385 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 385 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing analysis' in row 386 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 387 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 388 matches topic 'RNA'\n",
      "'Single cell RNA sequencing (RNAseq)' in row 389 matches topic 'RNA'\n",
      "'nucleus RNA sequencing' in row 390 matches topic 'RNA'\n",
      "'transcriptomics profiling' in row 390 matches topic 'Transcriptomics'\n",
      "'Electrophysiology' in row 391 matches topic 'Physiology'\n",
      "'Generation of induced pluripotent stem cells' in row 391 matches topic 'Cell biology'\n",
      "'Single cell transcriptome sequencing' in row 392 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 393 matches topic 'RNA'\n",
      "'Bulk RNA' in row 393 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 394 matches topic 'RNA'\n",
      "'10X genomics' in row 394 matches topic 'Genomics'\n",
      "'peripheral blood mononuclear cells (PBMCs)' in row 395 matches topic 'Cell biology'\n",
      "'cell RNA sequencing (RNA' in row 395 matches topic 'RNA'\n",
      "'Cell' in row 395 matches topic 'Cell biology'\n",
      "'Gene expression analysis' in row 395 matches topic 'Gene expression'\n",
      "'nucleus RNA sequencing (snRNAseq)' in row 396 matches topic 'RNA'\n",
      "'cell RNA' in row 398 matches topic 'RNA'\n",
      "'nucleus RNA' in row 398 matches topic 'RNA'\n",
      "'nucleus RNA sequencing (snRNA' in row 399 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 399 matches topic 'RNA'\n",
      "'BCR' in row 400 matches topic 'NMR'\n",
      "'Single cell RNA' in row 400 matches topic 'RNA'\n",
      "'Single cell RNA' in row 401 matches topic 'RNA'\n",
      "'Organoid transplantation' in row 401 matches topic 'Surgery'\n",
      "'cell RNA' in row 402 matches topic 'RNA'\n",
      "'Differential gene expression analysis' in row 402 matches topic 'Gene expression'\n",
      "'T cells antigen receptor (TCR) analysis' in row 402 matches topic 'Cell biology'\n",
      "'cell transcriptomics' in row 403 matches topic 'Transcriptomics'\n",
      "'Cell' in row 403 matches topic 'Cell biology'\n",
      "'Single cell mRNA sequencing' in row 404 matches topic 'Gene transcripts'\n",
      "'cell RNA sequencing (scRNA' in row 405 matches topic 'RNA'\n",
      "'Single nucleus RNA sequencing' in row 406 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 406 matches topic 'RNA'\n",
      "'Single cell RNA' in row 407 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 408 matches topic 'RNA'\n",
      "'Single Cell RNA' in row 408 matches topic 'RNA'\n",
      "'10X Genomics single' in row 408 matches topic 'Genomics'\n",
      "'cell RNA sequencing (RNAseq)' in row 409 matches topic 'RNA'\n",
      "'based algorithms' in row 410 matches topic 'Software engineering'\n",
      "'Single cell RNA sequencing (scRNA' in row 410 matches topic 'RNA'\n",
      "'Spatial proteomics' in row 411 matches topic 'Proteomics'\n",
      "'Spatial transcriptomics' in row 411 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 413 matches topic 'RNA'\n",
      "'Deconvolution of bulk gene expression samples' in row 413 matches topic 'Gene expression'\n",
      "'nucleus RNA' in row 414 matches topic 'RNA'\n",
      "'Cleavage Under Targets and Release Using Nuclease (CUT&RUN)' in row 415 matches topic 'Small molecules'\n",
      "'Single nucleus RNA sequencing (snRNA' in row 415 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 416 matches topic 'Transcriptomics'\n",
      "'10x Chromium genomics platform' in row 417 matches topic 'Genomics'\n",
      "'Single cell RNA' in row 417 matches topic 'RNA'\n",
      "'nucleus transcriptomics' in row 418 matches topic 'Transcriptomics'\n",
      "'Silencing of gene expression' in row 418 matches topic 'Gene expression'\n",
      "'scale gene expression analysis' in row 418 matches topic 'Gene expression'\n",
      "'cell RNA' in row 419 matches topic 'RNA'\n",
      "'Bioinformatics analysis' in row 419 matches topic 'Bioinformatics'\n",
      "'islet RNA' in row 419 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 420 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 420 matches topic 'Transcriptomics'\n",
      "'nucleus RNA' in row 421 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 422 matches topic 'RNA'\n",
      "'Gene expression analysis' in row 422 matches topic 'Gene expression'\n",
      "'Massively parallel single nucleus RNA sequencing (snRNA' in row 423 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 424 matches topic 'RNA'\n",
      "'Induced pluripotent stem cells (iPSCs)' in row 424 matches topic 'Cell biology'\n",
      "'cell RNA sequencing' in row 425 matches topic 'RNA'\n",
      "'nucleus RNA sequencing' in row 425 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNAseq)' in row 426 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 427 matches topic 'RNA'\n",
      "'cell genomics' in row 428 matches topic 'Genomics'\n",
      "'cell transcriptome analysis' in row 429 matches topic 'Transcriptomics'\n",
      "'Longitudinal assessment of circulating proteins' in row 429 matches topic 'Proteins'\n",
      "'nucleus RNA sequencing (snRNA' in row 430 matches topic 'RNA'\n",
      "'Single nucleus gene expression sequencing' in row 430 matches topic 'Gene expression'\n",
      "'cell RNA sequencing' in row 431 matches topic 'RNA'\n",
      "'RNA sequencing' in row 432 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 433 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 434 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 434 matches topic 'RNA'\n",
      "'cell RNA' in row 435 matches topic 'RNA'\n",
      "'nucleus RNA' in row 435 matches topic 'RNA'\n",
      "'Single cell RNA sequencing' in row 436 matches topic 'RNA'\n",
      "'Immune phenotyping' in row 437 matches topic 'Genotype and phenotype'\n",
      "'cell RNA sequencing (scRNA' in row 438 matches topic 'RNA'\n",
      "'Bulk RNA sequencing (RNA' in row 438 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 439 matches topic 'RNA'\n",
      "'10x genomics' in row 439 matches topic 'Genomics'\n",
      "'nuclei RNA' in row 440 matches topic 'RNA'\n",
      "'cell RNA profiling' in row 441 matches topic 'RNA'\n",
      "'Collection of cells' in row 441 matches topic 'Cell biology'\n",
      "'Gene expression analysis' in row 441 matches topic 'Gene expression'\n",
      "'nucleus RNA sequencing' in row 442 matches topic 'RNA'\n",
      "'omics analysis' in row 443 matches topic 'Omics'\n",
      "'cell RNA sequencing' in row 444 matches topic 'RNA'\n",
      "'cell RNA' in row 445 matches topic 'RNA'\n",
      "'bulk RNA' in row 445 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 446 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 447 matches topic 'RNA'\n",
      "'Spatial transcriptomics' in row 447 matches topic 'Transcriptomics'\n",
      "'cell RNA sequencing' in row 448 matches topic 'RNA'\n",
      "'Isolation of single cells from nasal swabs' in row 448 matches topic 'Cell biology'\n",
      "'cell transcriptome libraries' in row 448 matches topic 'Transcriptomics'\n",
      "'cell RNA' in row 449 matches topic 'RNA'\n",
      "'RNA sequencing (RNA' in row 450 matches topic 'RNA'\n",
      "'Genome' in row 450 matches topic 'Genomics'\n",
      "'nucleus RNA sequencing (NucSeq)' in row 450 matches topic 'RNA'\n",
      "'wide expression quantitative trait loci (eQTL) analysis' in row 450 matches topic 'Gene expression'\n",
      "'Single cell RNA sequencing (scRNA' in row 451 matches topic 'RNA'\n",
      "'Bulk RNA' in row 452 matches topic 'RNA'\n",
      "'Single nucleus RNA' in row 452 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 453 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 454 matches topic 'RNA'\n",
      "'cell RNA sequencing (scRNA' in row 455 matches topic 'RNA'\n",
      "'cell RNA sequencing' in row 456 matches topic 'RNA'\n"
     ]
    }
   ],
   "source": [
    "for index, row in data.iterrows():\n",
    "    if len(row['Hallucinations']) > 0:\n",
    "        for hallucination in list(row['Hallucinations']):\n",
    "            if hallucination in matched_topics:\n",
    "                print(f\"'{hallucination}' in row {index} matches topic '{matched_topics[hallucination]}'\")\n",
    "                data.at[index, 'Predictions'].add(matched_topics[hallucination])\n",
    "                continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_id</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>Model</th>\n",
       "      <th>Predictions</th>\n",
       "      <th>Hallucinations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hca_ef1d9888-fa86-47a4-bb72-0ab0f20f7004</td>\n",
       "      <td>Single-cell atlas of early human brain development highlights heterogeneity of human neuroepithelial cells and early radial glia.</td>\n",
       "      <td>The human cortex comprises diverse cell types that emerge from an initially uniform neuroepithelium that gives rise to radial glia, the neural stem cells of the cortex. To characterize the earliest stages of human brain development, we performed single-cell RNA-sequencing across regions of the developing human brain, including the telencephalon, diencephalon, midbrain, hindbrain and cerebellum. We identify nine progenitor populations physically proximal to the telencephalon, suggesting more heterogeneity than previously described, including a highly prevalent mesenchymal-like population that disappears once neurogenesis begins. Comparison of human and mouse progenitor populations at corresponding stages identifies two progenitor clusters that are enriched in the early stages of human cortical development. We also find that organoid systems display low fidelity to neuroepithelial and early radial glia cell types, but improve as neurogenesis progresses. Overall, we provide a comprehensive molecular and spatial atlas of early stages of human brain and cortical development.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA, Mapping}</td>\n",
       "      <td>{, sequencing, cell RNA, Comparative analysis with mouse progenitor populations, Single, Spatial mapping of developing human brain regions}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hca_923d3231-7295-4184-b3f6-c3082766a8c7</td>\n",
       "      <td>A Single-Cell Transcriptomic Atlas of Human Skin Aging.</td>\n",
       "      <td>Skin undergoes constant self-renewal, and its functional decline is a visible consequence of aging. Understanding human skin aging requires in-depth knowledge of the molecular and functional properties of various skin cell types. We performed single-cell RNA sequencing of human eyelid skin from healthy individuals across different ages and identified eleven canonical cell types, as well as six subpopulations of basal cells. Further analysis revealed progressive accumulation of photoaging-related changes and increased chronic inflammation with age. Transcriptional factors involved in the developmental process underwent early-onset decline during aging. Furthermore, inhibition of key transcription factors HES1 in fibroblasts and KLF6 in keratinocytes not only compromised cell proliferation, but also increased inflammation and cellular senescence during aging. Lastly, we found that genetic activation of HES1 or pharmacological treatment with quercetin alleviated cellular senescence of dermal fibroblasts. These findings provide a single-cell molecular framework of human skin aging, providing a rich resource for developing therapeutic strategies against aging-related skin disorders.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA, Gene expression}</td>\n",
       "      <td>{cell RNA sequencing, 2, Genetic activation, Transcription factor inhibition, 5, 3, Transcriptomic analysis, Pharmacological treatment, 1, 4, Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hca_1dddae6e-3753-48af-b20e-fa22abad125d</td>\n",
       "      <td>Cell Types of the Human Retina and Its Organoids at Single-Cell Resolution</td>\n",
       "      <td>Human organoids recapitulating the cell-type diversity and function of their target organ are valuable for basic and translational research. We developed light-sensitive human retinal organoids with multiple nuclear and synaptic layers and functional synapses. We sequenced the RNA of 285,441 single cells from these organoids at seven developmental time points and from the periphery, fovea, pigment epithelium and choroid of light-responsive adult human retinas, and performed histochemistry. Cell types in organoids matured in vitro to a stable \"developed\" state at a rate similar to human retina development in vivo. Transcriptomes of organoid cell types converged toward the transcriptomes of adult peripheral retinal cell types. Expression of disease-associated genes was cell-type-specific in adult retina, and cell-type specificity was retained in organoids. We implicate unexpected cell types in diseases such as macular degeneration. This resource identifies cellular targets for studying disease mechanisms in organoids and for targeted repair in human retinas.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA, Developmental biology}</td>\n",
       "      <td>{, RNA sequencing, Development of light, Histochemistry, sensitive human retinal organoids}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hca_95f07e6e-6a73-4e1b-a880-c83996b3aa5c</td>\n",
       "      <td>Immunophenotyping of COVID-19 and influenza highlights the role of type I interferons in development of severe COVID-19</td>\n",
       "      <td>Although most severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)–infected individuals experience mild coronavirus disease 2019 (COVID-19), some patients suffer from severe COVID-19, which is accompanied by acute respiratory distress syndrome and systemic inflammation. To identify factors driving severe progression of COVID-19, we performed single-cell RNA sequencing using peripheral blood mononuclear cells (PBMCs) obtained from healthy donors, patients with mild or severe COVID-19, and patients with severe influenza. Patients with COVID-19 exhibited hyperinflammatory signatures across all types of cells among PBMCs, particularly up-regulation of the tumor necrosis factor/interleukin-1β (TNF/IL-1β)–driven inflammatory response as compared with severe influenza. In classical monocytes from patients with severe COVID-19, type I interferon (IFN) response coexisted with the TNF/IL-1β–driven inflammation, and this was not seen in patients with milder COVID-19. We documented type I IFN–driven inflammatory features in patients with severe influenza as well. On the basis of this, we propose that the type I IFN response plays a pivotal role in exacerbating inflammation in severe COVID-19.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{, cell RNA sequencing, Immunophenotyping, Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hca_dcc28fb3-7bab-48ce-bc4b-684c00e133ce</td>\n",
       "      <td>A single-cell atlas of entorhinal cortex from individuals with Alzheimer's disease reveals cell-type-specific gene expression regulation.</td>\n",
       "      <td>There is currently little information available about how individual cell types contribute to Alzheimer's disease. Here we applied single-nucleus RNA sequencing to entorhinal cortex samples from control and Alzheimer's disease brains (n = 6 per group), yielding a total of 13,214 high-quality nuclei. We detail cell-type-specific gene expression patterns, unveiling how transcriptional changes in specific cell subpopulations are associated with Alzheimer's disease. We report that the Alzheimer's disease risk gene APOE is specifically repressed in Alzheimer's disease oligodendrocyte progenitor cells and astrocyte subpopulations and upregulated in an Alzheimer's disease-specific microglial subopulation. Integrating transcription factor regulatory modules with Alzheimer's disease risk loci revealed drivers of cell-type-specific state transitions towards Alzheimer's disease. For example, transcription factor EB, a master regulator of lysosomal function, regulates multiple disease genes in a specific Alzheimer's disease astrocyte subpopulation. These results provide insights into the coordinated control of Alzheimer's disease risk genes and their cell-type-specific contribution to disease susceptibility. These results are available at http://adsn.ddnetbio.com.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA, Transcriptomics}</td>\n",
       "      <td>{Integrative analysis, 2, nucleus RNA sequencing, 3, Regulatory module analysis, 1, 4, Single, Transcriptome analysis}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>hca_bd400331-54b9-4fcc-bff6-6bb8b079ee1f</td>\n",
       "      <td>Sex-Specific Control of Human Heart Maturation by the Progesterone Receptor.</td>\n",
       "      <td>Background: Despite in-depth knowledge of the molecular mechanisms controlling embryonic heart development, little is known about the signals governing postnatal maturation of the human heart. Methods: Single nucleus RNA-sequencing (snRNA-seq) of 54,140 nuclei from 9 human donors was used to profile transcriptional changes in diverse cardiac cell types during maturation from fetal stages to adulthood. Bulk RNA-sequencing and the assay for transposase-accessible chromatin using sequencing (ATAC-seq) were used to further validate transcriptional changes and to profile alterations in the chromatin accessibility landscape in purified cardiomyocyte nuclei from 21 human donors. Functional validation studies of sex steroids implicated in cardiac maturation were performed in human pluripotent stem cell-derived cardiac organoids and mice. Results: Our data identify the progesterone receptor as a key mediator of sex-dependent transcriptional programs during cardiomyocyte maturation. Functional validation studies in human cardiac organoids and mice demonstrate the progesterone receptor drives sex-specific metabolic programs and maturation of cardiac contractile properties. Conclusions: These data provide a blueprint for understanding human heart maturation in both sexes and reveal an important role for the progesterone receptor in human heart development.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{sequencing, 2, 3, Bulk RNA, 1, seq), sequencing (snRNA, accessible chromatin using sequencing (ATAC, Single nucleus RNA, Assay for transposase}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>hca_415eb773-cadb-43d1-ab89-7d160d5cfc7d</td>\n",
       "      <td>Single-cell analysis reveals innate lymphoid cell lineage infidelity in atopic dermatitis</td>\n",
       "      <td>BackgroundAlthough ample knowledge exists about phenotype and function of cutaneous T lymphocytes, much less is known about the lymphocytic components of the skin's innate immune system.ObjectiveTo better understand the biologic role of cutaneous innate lymphoid cells (ILCs), we investigated their phenotypic and molecular features under physiologic (normal human skin [NHS]) and pathologic (lesional skin of patients with atopic dermatitis [AD]) conditions.MethodsSkin punch biopsies and reduction sheets as well as blood specimens were obtained from either patients with AD or healthy individuals. Cell and/or tissue samples were analyzed by flow cytometry, immunohistochemistry, and single-cell RNA sequencing or subjected to in vitro/ex vivo culture.ResultsNotwithstanding substantial quantitative differences between NHS and AD skin, we found that the vast majority of cutaneous ILCs belong to the CRTH2+ subset and reside in the upper skin layers. Single-cell RNA sequencing of cutaneous ILC-enriched cell samples confirmed the predominance of biologically heterogeneous group 2 ILCs and, for the first time, demonstrated considerable ILC lineage infidelity (coexpression of genes typical of either type 2 [GATA3 and IL13] or type 3/17 [RORC, IL22, and IL26] immunity within individual cells) in lesional AD skin, and to a much lesser extent, in NHS. Similar events were demonstrated in ILCs from skin explant cultures and in vitro expanded ILCs from the peripheral blood.ConclusionThese findings support the concept that instead of being a stable entity with well-defined components, the skin immune system consists of a network of highly flexible cellular players that are capable of adjusting their function to the needs and challenges of the environment.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{, cell RNA sequencing, Flow cytometry, In vitro, Immunohistochemistry, Single, ex vivo culture}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>hca_a29952d9-925e-40f4-8a1c-274f118f1f51</td>\n",
       "      <td>Bone marrow plasma cells from hip replacement surgeries</td>\n",
       "      <td>Our aims are to generate a full representation of human hematopoiesis in blood and bone marrow of humans using a multi-tier and iterative collection and analysis of 200,000 cells from ten healthy human bone</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{Bone marrow aspiration, cell RNA sequencing, 2, 3, tier collection and analysis, 1, 4, Single, Iterative analysis, Multi}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>hca_d2111fac-3fc4-4f42-9b6d-32cd6a828267</td>\n",
       "      <td>Human cerebral organoids recapitulate gene expression programs of fetal neocortex development.</td>\n",
       "      <td>Cerebral organoids – three-dimensional cultures of human cerebral tissue derived from pluripotent stem cells – have emerged as models of human cortical development. However, the extent to which in vitro organoid systems recapitulate neural progenitor cell proliferation and neuronal differentiation programs observed in vivo remains unclear. Here we use single-cell RNA sequencing (scRNA-seq) to dissect and compare cell composition and progenitor-to-neuron lineage relationships in human cerebral organoids and fetal neocortex. Covariation network analysis using the fetal neocortex data reveals known and novel interactions among genes central to neural progenitor proliferation and neuronal differentiation. In the organoid, we detect diverse progenitors and differentiated cell types of neuronal and mesenchymal lineages, and identify cells that derived from regions resembling the fetal neocortex. We find that these organoid cortical cells use gene expression programs remarkably similar to those of the fetal tissue in order to organize into cerebral cortex-like regions. Our comparison of in vivo and in vitro cortical single cell transcriptomes illuminates the genetic features underlying human cortical development that can be studied in organoid cultures.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{, Covariation network analysis, cell RNA sequencing (scRNA, seq), Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>hca_b91c623b-1945-4727-b167-0a93027b0d3f</td>\n",
       "      <td>Circuits between infected macrophages and T cells in SARS-CoV-2 pneumonia.</td>\n",
       "      <td>Some patients infected with severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) develop severe pneumonia and acute respiratory distress syndrome1 (ARDS). Distinct clinical features in these patients have led to speculation that the immune response to virus in the SARS-CoV-2-infected alveolus differs from that in other types of pneumonia2. Here we investigate SARS-CoV-2 pathobiology by characterizing the immune response in the alveoli of patients infected with the virus. We collected bronchoalveolar lavage fluid samples from 88 patients with SARS-CoV-2-induced respiratory failure and 211 patients with known or suspected pneumonia from other pathogens, and analysed them using flow cytometry and bulk transcriptomic profiling. We performed single-cell RNA sequencing on 10 bronchoalveolar lavage fluid samples collected from patients with severe coronavirus disease 2019 (COVID-19) within 48 h of intubation. In the majority of patients with SARS-CoV-2 infection, the alveolar space was persistently enriched in T cells and monocytes. Bulk and single-cell transcriptomic profiling suggested that SARS-CoV-2 infects alveolar macrophages, which in turn respond by producing T cell chemoattractants. These T cells produce interferon-γ to induce inflammatory cytokine release from alveolar macrophages and further promote T cell activation. Collectively, our results suggest that SARS-CoV-2 causes a slowly unfolding, spatially limited alveolitis in which alveolar macrophages containing SARS-CoV-2 and T cells form a positive feedback loop that drives persistent alveolar inflammation.</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{RNA}</td>\n",
       "      <td>{, cell RNA sequencing, Bulk transcriptomic profiling, Flow cytometry, Single}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>454 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                          _id  \\\n",
       "0    hca_ef1d9888-fa86-47a4-bb72-0ab0f20f7004   \n",
       "1    hca_923d3231-7295-4184-b3f6-c3082766a8c7   \n",
       "2    hca_1dddae6e-3753-48af-b20e-fa22abad125d   \n",
       "3    hca_95f07e6e-6a73-4e1b-a880-c83996b3aa5c   \n",
       "4    hca_dcc28fb3-7bab-48ce-bc4b-684c00e133ce   \n",
       "..                                        ...   \n",
       "452  hca_bd400331-54b9-4fcc-bff6-6bb8b079ee1f   \n",
       "453  hca_415eb773-cadb-43d1-ab89-7d160d5cfc7d   \n",
       "454  hca_a29952d9-925e-40f4-8a1c-274f118f1f51   \n",
       "455  hca_d2111fac-3fc4-4f42-9b6d-32cd6a828267   \n",
       "456  hca_b91c623b-1945-4727-b167-0a93027b0d3f   \n",
       "\n",
       "                                                                                                                                          name  \\\n",
       "0            Single-cell atlas of early human brain development highlights heterogeneity of human neuroepithelial cells and early radial glia.   \n",
       "1                                                                                      A Single-Cell Transcriptomic Atlas of Human Skin Aging.   \n",
       "2                                                                   Cell Types of the Human Retina and Its Organoids at Single-Cell Resolution   \n",
       "3                      Immunophenotyping of COVID-19 and influenza highlights the role of type I interferons in development of severe COVID-19   \n",
       "4    A single-cell atlas of entorhinal cortex from individuals with Alzheimer's disease reveals cell-type-specific gene expression regulation.   \n",
       "..                                                                                                                                         ...   \n",
       "452                                                               Sex-Specific Control of Human Heart Maturation by the Progesterone Receptor.   \n",
       "453                                                  Single-cell analysis reveals innate lymphoid cell lineage infidelity in atopic dermatitis   \n",
       "454                                                                                    Bone marrow plasma cells from hip replacement surgeries   \n",
       "455                                             Human cerebral organoids recapitulate gene expression programs of fetal neocortex development.   \n",
       "456                                                                 Circuits between infected macrophages and T cells in SARS-CoV-2 pneumonia.   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               description  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           The human cortex comprises diverse cell types that emerge from an initially uniform neuroepithelium that gives rise to radial glia, the neural stem cells of the cortex. To characterize the earliest stages of human brain development, we performed single-cell RNA-sequencing across regions of the developing human brain, including the telencephalon, diencephalon, midbrain, hindbrain and cerebellum. We identify nine progenitor populations physically proximal to the telencephalon, suggesting more heterogeneity than previously described, including a highly prevalent mesenchymal-like population that disappears once neurogenesis begins. Comparison of human and mouse progenitor populations at corresponding stages identifies two progenitor clusters that are enriched in the early stages of human cortical development. We also find that organoid systems display low fidelity to neuroepithelial and early radial glia cell types, but improve as neurogenesis progresses. Overall, we provide a comprehensive molecular and spatial atlas of early stages of human brain and cortical development.   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             Skin undergoes constant self-renewal, and its functional decline is a visible consequence of aging. Understanding human skin aging requires in-depth knowledge of the molecular and functional properties of various skin cell types. We performed single-cell RNA sequencing of human eyelid skin from healthy individuals across different ages and identified eleven canonical cell types, as well as six subpopulations of basal cells. Further analysis revealed progressive accumulation of photoaging-related changes and increased chronic inflammation with age. Transcriptional factors involved in the developmental process underwent early-onset decline during aging. Furthermore, inhibition of key transcription factors HES1 in fibroblasts and KLF6 in keratinocytes not only compromised cell proliferation, but also increased inflammation and cellular senescence during aging. Lastly, we found that genetic activation of HES1 or pharmacological treatment with quercetin alleviated cellular senescence of dermal fibroblasts. These findings provide a single-cell molecular framework of human skin aging, providing a rich resource for developing therapeutic strategies against aging-related skin disorders.   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Human organoids recapitulating the cell-type diversity and function of their target organ are valuable for basic and translational research. We developed light-sensitive human retinal organoids with multiple nuclear and synaptic layers and functional synapses. We sequenced the RNA of 285,441 single cells from these organoids at seven developmental time points and from the periphery, fovea, pigment epithelium and choroid of light-responsive adult human retinas, and performed histochemistry. Cell types in organoids matured in vitro to a stable \"developed\" state at a rate similar to human retina development in vivo. Transcriptomes of organoid cell types converged toward the transcriptomes of adult peripheral retinal cell types. Expression of disease-associated genes was cell-type-specific in adult retina, and cell-type specificity was retained in organoids. We implicate unexpected cell types in diseases such as macular degeneration. This resource identifies cellular targets for studying disease mechanisms in organoids and for targeted repair in human retinas.   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  Although most severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2)–infected individuals experience mild coronavirus disease 2019 (COVID-19), some patients suffer from severe COVID-19, which is accompanied by acute respiratory distress syndrome and systemic inflammation. To identify factors driving severe progression of COVID-19, we performed single-cell RNA sequencing using peripheral blood mononuclear cells (PBMCs) obtained from healthy donors, patients with mild or severe COVID-19, and patients with severe influenza. Patients with COVID-19 exhibited hyperinflammatory signatures across all types of cells among PBMCs, particularly up-regulation of the tumor necrosis factor/interleukin-1β (TNF/IL-1β)–driven inflammatory response as compared with severe influenza. In classical monocytes from patients with severe COVID-19, type I interferon (IFN) response coexisted with the TNF/IL-1β–driven inflammation, and this was not seen in patients with milder COVID-19. We documented type I IFN–driven inflammatory features in patients with severe influenza as well. On the basis of this, we propose that the type I IFN response plays a pivotal role in exacerbating inflammation in severe COVID-19.   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 There is currently little information available about how individual cell types contribute to Alzheimer's disease. Here we applied single-nucleus RNA sequencing to entorhinal cortex samples from control and Alzheimer's disease brains (n = 6 per group), yielding a total of 13,214 high-quality nuclei. We detail cell-type-specific gene expression patterns, unveiling how transcriptional changes in specific cell subpopulations are associated with Alzheimer's disease. We report that the Alzheimer's disease risk gene APOE is specifically repressed in Alzheimer's disease oligodendrocyte progenitor cells and astrocyte subpopulations and upregulated in an Alzheimer's disease-specific microglial subopulation. Integrating transcription factor regulatory modules with Alzheimer's disease risk loci revealed drivers of cell-type-specific state transitions towards Alzheimer's disease. For example, transcription factor EB, a master regulator of lysosomal function, regulates multiple disease genes in a specific Alzheimer's disease astrocyte subpopulation. These results provide insights into the coordinated control of Alzheimer's disease risk genes and their cell-type-specific contribution to disease susceptibility. These results are available at http://adsn.ddnetbio.com.   \n",
       "..                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     ...   \n",
       "452                                                                                                                                                                                                                                                                                                                                                                                                                 Background: Despite in-depth knowledge of the molecular mechanisms controlling embryonic heart development, little is known about the signals governing postnatal maturation of the human heart. Methods: Single nucleus RNA-sequencing (snRNA-seq) of 54,140 nuclei from 9 human donors was used to profile transcriptional changes in diverse cardiac cell types during maturation from fetal stages to adulthood. Bulk RNA-sequencing and the assay for transposase-accessible chromatin using sequencing (ATAC-seq) were used to further validate transcriptional changes and to profile alterations in the chromatin accessibility landscape in purified cardiomyocyte nuclei from 21 human donors. Functional validation studies of sex steroids implicated in cardiac maturation were performed in human pluripotent stem cell-derived cardiac organoids and mice. Results: Our data identify the progesterone receptor as a key mediator of sex-dependent transcriptional programs during cardiomyocyte maturation. Functional validation studies in human cardiac organoids and mice demonstrate the progesterone receptor drives sex-specific metabolic programs and maturation of cardiac contractile properties. Conclusions: These data provide a blueprint for understanding human heart maturation in both sexes and reveal an important role for the progesterone receptor in human heart development.   \n",
       "453  BackgroundAlthough ample knowledge exists about phenotype and function of cutaneous T lymphocytes, much less is known about the lymphocytic components of the skin's innate immune system.ObjectiveTo better understand the biologic role of cutaneous innate lymphoid cells (ILCs), we investigated their phenotypic and molecular features under physiologic (normal human skin [NHS]) and pathologic (lesional skin of patients with atopic dermatitis [AD]) conditions.MethodsSkin punch biopsies and reduction sheets as well as blood specimens were obtained from either patients with AD or healthy individuals. Cell and/or tissue samples were analyzed by flow cytometry, immunohistochemistry, and single-cell RNA sequencing or subjected to in vitro/ex vivo culture.ResultsNotwithstanding substantial quantitative differences between NHS and AD skin, we found that the vast majority of cutaneous ILCs belong to the CRTH2+ subset and reside in the upper skin layers. Single-cell RNA sequencing of cutaneous ILC-enriched cell samples confirmed the predominance of biologically heterogeneous group 2 ILCs and, for the first time, demonstrated considerable ILC lineage infidelity (coexpression of genes typical of either type 2 [GATA3 and IL13] or type 3/17 [RORC, IL22, and IL26] immunity within individual cells) in lesional AD skin, and to a much lesser extent, in NHS. Similar events were demonstrated in ILCs from skin explant cultures and in vitro expanded ILCs from the peripheral blood.ConclusionThese findings support the concept that instead of being a stable entity with well-defined components, the skin immune system consists of a network of highly flexible cellular players that are capable of adjusting their function to the needs and challenges of the environment.   \n",
       "454                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Our aims are to generate a full representation of human hematopoiesis in blood and bone marrow of humans using a multi-tier and iterative collection and analysis of 200,000 cells from ten healthy human bone   \n",
       "455                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     Cerebral organoids – three-dimensional cultures of human cerebral tissue derived from pluripotent stem cells – have emerged as models of human cortical development. However, the extent to which in vitro organoid systems recapitulate neural progenitor cell proliferation and neuronal differentiation programs observed in vivo remains unclear. Here we use single-cell RNA sequencing (scRNA-seq) to dissect and compare cell composition and progenitor-to-neuron lineage relationships in human cerebral organoids and fetal neocortex. Covariation network analysis using the fetal neocortex data reveals known and novel interactions among genes central to neural progenitor proliferation and neuronal differentiation. In the organoid, we detect diverse progenitors and differentiated cell types of neuronal and mesenchymal lineages, and identify cells that derived from regions resembling the fetal neocortex. We find that these organoid cortical cells use gene expression programs remarkably similar to those of the fetal tissue in order to organize into cerebral cortex-like regions. Our comparison of in vivo and in vitro cortical single cell transcriptomes illuminates the genetic features underlying human cortical development that can be studied in organoid cultures.   \n",
       "456                                                                                                                                                                          Some patients infected with severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) develop severe pneumonia and acute respiratory distress syndrome1 (ARDS). Distinct clinical features in these patients have led to speculation that the immune response to virus in the SARS-CoV-2-infected alveolus differs from that in other types of pneumonia2. Here we investigate SARS-CoV-2 pathobiology by characterizing the immune response in the alveoli of patients infected with the virus. We collected bronchoalveolar lavage fluid samples from 88 patients with SARS-CoV-2-induced respiratory failure and 211 patients with known or suspected pneumonia from other pathogens, and analysed them using flow cytometry and bulk transcriptomic profiling. We performed single-cell RNA sequencing on 10 bronchoalveolar lavage fluid samples collected from patients with severe coronavirus disease 2019 (COVID-19) within 48 h of intubation. In the majority of patients with SARS-CoV-2 infection, the alveolar space was persistently enriched in T cells and monocytes. Bulk and single-cell transcriptomic profiling suggested that SARS-CoV-2 infects alveolar macrophages, which in turn respond by producing T cell chemoattractants. These T cells produce interferon-γ to induce inflammatory cytokine release from alveolar macrophages and further promote T cell activation. Collectively, our results suggest that SARS-CoV-2 causes a slowly unfolding, spatially limited alveolitis in which alveolar macrophages containing SARS-CoV-2 and T cells form a positive feedback loop that drives persistent alveolar inflammation.   \n",
       "\n",
       "             Model                   Predictions  \\\n",
       "0    gpt-3.5-turbo                {RNA, Mapping}   \n",
       "1    gpt-3.5-turbo        {RNA, Gene expression}   \n",
       "2    gpt-3.5-turbo  {RNA, Developmental biology}   \n",
       "3    gpt-3.5-turbo                         {RNA}   \n",
       "4    gpt-3.5-turbo        {RNA, Transcriptomics}   \n",
       "..             ...                           ...   \n",
       "452  gpt-3.5-turbo                         {RNA}   \n",
       "453  gpt-3.5-turbo                         {RNA}   \n",
       "454  gpt-3.5-turbo                         {RNA}   \n",
       "455  gpt-3.5-turbo                         {RNA}   \n",
       "456  gpt-3.5-turbo                         {RNA}   \n",
       "\n",
       "                                                                                                                                            Hallucinations  \n",
       "0              {, sequencing, cell RNA, Comparative analysis with mouse progenitor populations, Single, Spatial mapping of developing human brain regions}  \n",
       "1    {cell RNA sequencing, 2, Genetic activation, Transcription factor inhibition, 5, 3, Transcriptomic analysis, Pharmacological treatment, 1, 4, Single}  \n",
       "2                                                              {, RNA sequencing, Development of light, Histochemistry, sensitive human retinal organoids}  \n",
       "3                                                                                                       {, cell RNA sequencing, Immunophenotyping, Single}  \n",
       "4                                   {Integrative analysis, 2, nucleus RNA sequencing, 3, Regulatory module analysis, 1, 4, Single, Transcriptome analysis}  \n",
       "..                                                                                                                                                     ...  \n",
       "452       {sequencing, 2, 3, Bulk RNA, 1, seq), sequencing (snRNA, accessible chromatin using sequencing (ATAC, Single nucleus RNA, Assay for transposase}  \n",
       "453                                                       {, cell RNA sequencing, Flow cytometry, In vitro, Immunohistochemistry, Single, ex vivo culture}  \n",
       "454                             {Bone marrow aspiration, cell RNA sequencing, 2, 3, tier collection and analysis, 1, 4, Single, Iterative analysis, Multi}  \n",
       "455                                                                             {, Covariation network analysis, cell RNA sequencing (scRNA, seq), Single}  \n",
       "456                                                                         {, cell RNA sequencing, Bulk transcriptomic profiling, Flow cytometry, Single}  \n",
       "\n",
       "[454 rows x 6 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['Hallucinations'].apply(len) > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add quotes back in for the predictions with commas (i.e, \"Data submission, annotation, and curation\")\n",
    "data['Predictions'] = data['Predictions'].apply(lambda preds: [f'\"{pred}\"' if f'\"{pred}\"' in quoted_topics else pred for pred in preds])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Predictions</th>\n",
       "      <th>Hallucinations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[RNA, Mapping]</td>\n",
       "      <td>{, sequencing, cell RNA, Comparative analysis with mouse progenitor populations, Single, Spatial mapping of developing human brain regions}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[RNA, Gene expression]</td>\n",
       "      <td>{cell RNA sequencing, 2, Genetic activation, Transcription factor inhibition, 5, 3, Transcriptomic analysis, Pharmacological treatment, 1, 4, Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[RNA, Developmental biology]</td>\n",
       "      <td>{, RNA sequencing, Development of light, Histochemistry, sensitive human retinal organoids}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{, cell RNA sequencing, Immunophenotyping, Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[RNA, Transcriptomics]</td>\n",
       "      <td>{Integrative analysis, 2, nucleus RNA sequencing, 3, Regulatory module analysis, 1, 4, Single, Transcriptome analysis}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>452</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{sequencing, 2, 3, Bulk RNA, 1, seq), sequencing (snRNA, accessible chromatin using sequencing (ATAC, Single nucleus RNA, Assay for transposase}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{, cell RNA sequencing, Flow cytometry, In vitro, Immunohistochemistry, Single, ex vivo culture}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>454</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{Bone marrow aspiration, cell RNA sequencing, 2, 3, tier collection and analysis, 1, 4, Single, Iterative analysis, Multi}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>455</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{, Covariation network analysis, cell RNA sequencing (scRNA, seq), Single}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>[RNA]</td>\n",
       "      <td>{, cell RNA sequencing, Bulk transcriptomic profiling, Flow cytometry, Single}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>457 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      Predictions  \\\n",
       "0                  [RNA, Mapping]   \n",
       "1          [RNA, Gene expression]   \n",
       "2    [RNA, Developmental biology]   \n",
       "3                           [RNA]   \n",
       "4          [RNA, Transcriptomics]   \n",
       "..                            ...   \n",
       "452                         [RNA]   \n",
       "453                         [RNA]   \n",
       "454                         [RNA]   \n",
       "455                         [RNA]   \n",
       "456                         [RNA]   \n",
       "\n",
       "                                                                                                                                            Hallucinations  \n",
       "0              {, sequencing, cell RNA, Comparative analysis with mouse progenitor populations, Single, Spatial mapping of developing human brain regions}  \n",
       "1    {cell RNA sequencing, 2, Genetic activation, Transcription factor inhibition, 5, 3, Transcriptomic analysis, Pharmacological treatment, 1, 4, Single}  \n",
       "2                                                              {, RNA sequencing, Development of light, Histochemistry, sensitive human retinal organoids}  \n",
       "3                                                                                                       {, cell RNA sequencing, Immunophenotyping, Single}  \n",
       "4                                   {Integrative analysis, 2, nucleus RNA sequencing, 3, Regulatory module analysis, 1, 4, Single, Transcriptome analysis}  \n",
       "..                                                                                                                                                     ...  \n",
       "452       {sequencing, 2, 3, Bulk RNA, 1, seq), sequencing (snRNA, accessible chromatin using sequencing (ATAC, Single nucleus RNA, Assay for transposase}  \n",
       "453                                                       {, cell RNA sequencing, Flow cytometry, In vitro, Immunohistochemistry, Single, ex vivo culture}  \n",
       "454                             {Bone marrow aspiration, cell RNA sequencing, 2, 3, tier collection and analysis, 1, 4, Single, Iterative analysis, Multi}  \n",
       "455                                                                             {, Covariation network analysis, cell RNA sequencing (scRNA, seq), Single}  \n",
       "456                                                                         {, cell RNA sequencing, Bulk transcriptomic profiling, Flow cytometry, Single}  \n",
       "\n",
       "[457 rows x 2 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['Hallucinations'].apply(len) > 0][['Predictions', 'Hallucinations']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of rows with 0 predictions: 38/457\n"
     ]
    }
   ],
   "source": [
    "print('Number of rows with 0 predictions:', f'{len(data[data[\"Predictions\"].apply(len) == 0])}/{len(data)}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].apply(lambda lst: set(lst))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_name = os.path.basename(DATA_PATH).replace('.', '_processed.')\n",
    "\n",
    "data.to_csv(f'outputs/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nde_gpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
