{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "\n",
    "import re\n",
    "import os\n",
    "import csv\n",
    "\n",
    "from tqdm.notebook import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded dataset with 65 rows\n"
     ]
    }
   ],
   "source": [
    "DATA_PATH = 'outputs/vdjserver.csv'\n",
    "\n",
    "data = pd.read_csv(DATA_PATH, lineterminator=\"\\n\")\n",
    "\n",
    "print(f'Loaded dataset with {len(data)} rows')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('EDAM/edam_topics.txt', 'r') as f:\n",
    "    edam_topics = [topic.strip() for topic in f.readlines()]\n",
    "\n",
    "quoted_topics = [topic for topic in edam_topics if topic.startswith('\"') and topic.endswith('\"')]\n",
    "\n",
    "# Remove quotes\n",
    "edam_topics = [topic[1:-1] if topic.startswith('\"') and topic.endswith('\"') else topic for topic in edam_topics]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Format Outputs\n",
    "\n",
    "Split outputs on tab, and check for other separators that GPT may have used in error."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].str.replace('\\\\t', '\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "def split_topics(topics):\n",
    "    cleaned_topics = [topic.strip() for topic in topics.split('\\t')]\n",
    "    for i in range(len(cleaned_topics)):\n",
    "        for quoted_topic in quoted_topics:\n",
    "            if quoted_topic.replace('\\\"', '').lower() in cleaned_topics[i].lower():\n",
    "                cleaned_topics[i] = cleaned_topics[i].replace(quoted_topic.replace('\\\"', ''), quoted_topic)\n",
    "                break\n",
    "            else:\n",
    "                cleaned_topics[i] = cleaned_topics[i].replace('\\\"', '')\n",
    "    return cleaned_topics\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(split_topics)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "83150f506ca046ac983477d91c680653",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/65 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "separators = ['    ', '   ', '  ', '\\n', '<TAB>', 'TAB', '<tab>', '(tab)', '<Tab>', '[tab]', '▪️', '(Tab)', '\\xa0\\xa0\\xa0\\xa0', '\\xa0', '\\u2003', '、', '\\x0b', '\\x0c', ';', '.', '--', '-', '–', '_', '\\\\', '\\\\n', '/', '@', '|', '\\r', '+', '<', '>', '·']\n",
    "\n",
    "# Join the separators with the regex OR operator |\n",
    "sep_pattern = '|'.join(map(re.escape, separators))\n",
    "\n",
    "for i in tqdm(range(len(data))):\n",
    "    # Replace commas not enclosed in double quotes with |\n",
    "    data['Predictions'][i] = [re.sub(r',(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)', '|', pred) for pred in data['Predictions'][i]]\n",
    "    # Split on the separators\n",
    "    split_list = [re.split(sep_pattern, pred) for pred in data['Predictions'][i]]\n",
    "    # Flatten the list\n",
    "    data['Predictions'][i] = [item for sublist in split_list for item in sublist]\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(lambda x: [item.strip() for item in x])\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(lambda x: [re.sub(r'Category \\d+:', '', pred) for pred in x])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "is_one_dimensional = all(isinstance(pred, str) for preds in data['Predictions'] for pred in preds)\n",
    "print(is_one_dimensional)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ## Capture any weirdly formatted outputs (using the wrong separators)\n",
    "\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('    ') if 0 < len(x) <= 1 and '    ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('   ') if 0 < len(x) <= 1 and '   ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('  ') if 0 < len(x) <= 1 and '  ' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\n') if 0 < len(x) <= 1 and '\\n' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<TAB>') if 0 < len(x) <= 1 and '<TAB>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('TAB') if 0 < len(x) <= 1 and 'TAB' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<tab>') if 0 < len(x) <= 1 and '<tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('(tab)') if 0 < len(x) <= 1 and '(tab)' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<Tab>') if 0 < len(x) <= 1 and '<Tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('[tab]') if 0 < len(x) <= 1 and '[tab]' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('▪️') if 0 < len(x) <= 1 and '▪️' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('(Tab)') if 0 < len(x) <= 1 and '<Tab>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\xa0\\xa0\\xa0\\xa0') if 0 < len(x) <= 1 and '\\xa0\\xa0\\xa0\\xa0' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\xa0') if 0 < len(x) <= 1 and '\\xa0' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\u2003') if 0 < len(x) <= 1 and '\\u2003' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('、') if 0 < len(x) <= 1 and '、' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\x0b') if 0 < len(x) <= 1 and '\\x0b' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\x0c') if 0 < len(x) <= 1 and '\\x0c' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split(';') if 0 < len(x) <= 1 and ';' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('.') if 0 < len(x) <= 1 and '.' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('--') if 0 < len(x) <= 1 and '--' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('-') if 0 < len(x) <= 1 and '-' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('–') if 0 < len(x) <= 1 and '–' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('_') if 0 < len(x) <= 1 and '_' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\\\') if 0 < len(x) <= 1 and '\\\\' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\\\n') if 0 < len(x) <= 1 and '\\\\n' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('/') if 0 < len(x) <= 1 and '/' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('@') if 0 < len(x) <= 1 and '@' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('|') if 0 < len(x) <= 1 and '|' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('\\r') if 0 < len(x) <= 1 and '\\r' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('+') if 0 < len(x) <= 1 and '+' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('<') if 0 < len(x) <= 1 and '<' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split('>') if 0 < len(x) <= 1 and '>' in list(x)[0] else x)\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: [pred.strip() for pred in csv.reader([list(x)[0]],skipinitialspace=True, delimiter=',', quotechar='\"').__next__()])\n",
    "# data['Predictions'] = data['Predictions'].apply(lambda x: [re.sub(r'Category \\d+:', '', pred) for pred in x])\n",
    "# # data['Predictions'] = data['Predictions'].apply(lambda x: list(x)[0].split(', ') if len(x) <= 1 and ', ' in list(x)[0] else x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [],
   "source": [
    "# check if any of the quoted topics, or their equivalent without quotes, \n",
    "# are in any of the prediction sets with length less than or equal to 1. \"\"\n",
    "# If there is, then add the quotes back in if they don't have them,\n",
    "#  and then split on commas while avoiding anything inside quotes\n",
    "def process_predictions(predictions):\n",
    "    processed_predictions = []\n",
    "    for prediction in predictions:\n",
    "        formatted = False\n",
    "        for topic in quoted_topics:\n",
    "            formatted_topic = topic.replace('\\\"', '')\n",
    "            if formatted_topic in prediction:\n",
    "                processed_prediction = prediction.replace(formatted_topic, f'{topic}')\n",
    "                processed_predictions.append(processed_prediction)\n",
    "                formatted = True\n",
    "                break\n",
    "        if not formatted:\n",
    "            processed_predictions.append(prediction)\n",
    "        \n",
    "    final_predictions = []\n",
    "    for prediction in processed_predictions:\n",
    "        if '\\\"' in prediction:\n",
    "            parts = re.findall(r'[^\"]+|\"[^\"]+\"', prediction)\n",
    "            final_predictions.extend(parts)\n",
    "        else:\n",
    "            final_predictions.extend([pred.strip() for pred in prediction.split(',')])\n",
    "    return set(final_predictions)\n",
    "\n",
    "data['Predictions'] = data['Predictions'].apply(process_predictions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of unexpected predictions: 0\n"
     ]
    }
   ],
   "source": [
    "filtered_predictions = data[data['Predictions'].apply(len) <= 1]['Predictions']\n",
    "\n",
    "# Filter out any expected predictions. May or may not be hallucinations\n",
    "unexpected_predictions = []\n",
    "for original_index, pred_set in filtered_predictions.items():\n",
    "    lst = list(pred_set)\n",
    "    if len(lst) == 0:\n",
    "        continue\n",
    "    prediction = lst[0]\n",
    "    if '\\\"' not in prediction and ' ' in prediction and prediction not in edam_topics:\n",
    "        unexpected_predictions.append((original_index, prediction))\n",
    "\n",
    "# Print the unexpected predictions and their corresponding original indices\n",
    "count = len(unexpected_predictions)\n",
    "print(f\"Number of unexpected predictions: {count}\")\n",
    "for original_index, prediction in unexpected_predictions:\n",
    "    print(f\"Original Index: {original_index}, Prediction: {prediction}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Hallucinations\n",
    "\n",
    "Filter out topics not in the EDAM topics list. The filtered topics may be matched to a topic or synonym->topic in the next section."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Hallucinations'] = data['Predictions'].apply(lambda preds: set([pred.replace('.', '').replace('\\\"', '') for pred in preds if pred.replace('.', '').replace('\\\"', '') not in edam_topics]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].apply(lambda preds: set([pred.replace('.', '').replace('\\\"', '') for pred in preds if pred.replace('.', '').replace('\\\"', '') in edam_topics]))\n",
    "data['Predictions'] = data.apply(lambda row: set([topic for topic in row['Predictions'] if topic not in row['Hallucinations']]), axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Synonym matching\n",
    "\n",
    "Check for mispelled/misformatted topics or synonyms using levenshtein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [],
   "source": [
    "edam = pd.read_csv('EDAM/EDAM.csv')\n",
    "\n",
    "edam = edam[edam['Class ID'].str.contains('topic')].reset_index(drop=True)\n",
    "# edam['Preferred Label'].apply(lambda topic: topic.replace('\\\"', ''))\n",
    "edam = edam[edam['Preferred Label'].isin([topic.replace('\\\"', '') for topic in edam_topics])].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "edam['Synonyms'] = edam['Synonyms'].fillna('').apply(lambda x: x.split('|') if x != '' else [])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 205,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_topics = set([topic.replace('\\\"', '') for topic in edam_topics]) - set(edam['Preferred Label'])\n",
    "missing_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "synonym_dict = {}\n",
    "\n",
    "for index, row in edam.iterrows():\n",
    "    for synonym in row['Synonyms']:\n",
    "        synonym_dict[synonym] = row['Preferred Label']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Molecular docking': 'Molecular modelling',\n",
       " 'Homology modeling': 'Molecular modelling',\n",
       " 'Docking': 'Molecular modelling',\n",
       " 'Comparative modelling': 'Molecular modelling',\n",
       " 'Homology modelling': 'Molecular modelling',\n",
       " 'Evolution': 'Evolutionary biology',\n",
       " 'Freshwater science': 'Freshwater biology',\n",
       " 'Nutrition': 'Nutritional science',\n",
       " 'Nutrition science': 'Nutritional science',\n",
       " 'Dietetics': 'Nutritional science',\n",
       " 'Cardiovascular medicine': 'Cardiology',\n",
       " 'Heart disease': 'Cardiology',\n",
       " 'Cardiovascular disease': 'Cardiology',\n",
       " 'Gene features': 'Gene structure',\n",
       " 'Fusion genes': 'Gene structure',\n",
       " 'Transcriptome': 'Transcriptomics',\n",
       " 'Comparative transcriptomics': 'Transcriptomics',\n",
       " 'Ancestral genomes': 'Paleogenomics',\n",
       " 'Paleogenetics': 'Paleogenomics',\n",
       " 'Panomics': 'Multiomics',\n",
       " 'Pan-omics': 'Multiomics',\n",
       " 'Integrative omics': 'Multiomics',\n",
       " 'Multi-omics': 'Multiomics',\n",
       " 'RNA metabarcoding': 'Metabarcoding',\n",
       " 'DNA metabarcoding': 'Metabarcoding',\n",
       " 'eDNA metabarcoding': 'Metabarcoding',\n",
       " 'Environmental metabarcoding': 'Metabarcoding',\n",
       " 'eRNA metabarcoding': 'Metabarcoding',\n",
       " 'Environmental RNA (eRNA)': 'Metabarcoding',\n",
       " 'Environmental sequencing': 'Metagenomics',\n",
       " 'Taxonomic profiling': 'Metabarcoding',\n",
       " 'Environmental DNA (eDNA)': 'Metagenomics',\n",
       " 'Expression': 'Gene expression',\n",
       " 'Codon usage': 'Gene expression',\n",
       " 'Gene transcription': 'Gene expression',\n",
       " 'Transcription': 'Gene expression',\n",
       " 'Gene expression profiling': 'Gene expression',\n",
       " 'Gene translation': 'Gene expression',\n",
       " 'DNA microarrays': 'Gene expression',\n",
       " 'DNA chips': 'Gene expression',\n",
       " 'Small RNA': 'RNA',\n",
       " 'Electron diffraction experiment': 'Electron microscopy',\n",
       " 'Transmission electron microscopy': 'Electron microscopy',\n",
       " 'SEM': 'Electron microscopy',\n",
       " 'Scanning electron microscopy': 'Electron microscopy',\n",
       " 'TEM': 'Electron microscopy',\n",
       " 'Electron crystallography': 'Electron microscopy',\n",
       " 'Single particle electron microscopy': 'Electron microscopy',\n",
       " 'Immunoproteins': 'Immunoproteins and antigens',\n",
       " 'Antigens': 'Immunoproteins and antigens',\n",
       " 'Immunopeptides': 'Immunoproteins and antigens',\n",
       " 'Therapeutic antibodies': 'Immunoproteins and antigens',\n",
       " 'Translation': 'Protein expression',\n",
       " 'Neuroscience': 'Neurobiology',\n",
       " 'Molecular neuroscience': 'Neurobiology',\n",
       " 'Systemetic neuroscience': 'Neurobiology',\n",
       " 'Neurophysiology': 'Neurobiology',\n",
       " 'Reproductive disorders': 'Reproductive health',\n",
       " 'Andrology': 'Reproductive health',\n",
       " 'Fertility medicine': 'Reproductive health',\n",
       " 'Family planning': 'Reproductive health',\n",
       " 'Findable, accessible, interoperable, reusable data': 'FAIR data',\n",
       " 'Open data': 'FAIR data',\n",
       " 'FAIRification': 'FAIR data',\n",
       " 'FAIR data principles': 'FAIR data',\n",
       " 'Genetic modification': 'Genetic engineering',\n",
       " 'Genetic manipulation': 'Genetic engineering',\n",
       " 'Genome engineering': 'Genetic engineering',\n",
       " 'Genome editing': 'Genetic engineering',\n",
       " 'Audiovestibular medicine': 'Ear, nose and throat medicine',\n",
       " 'Otorhinolaryngology': 'Ear, nose and throat medicine',\n",
       " 'Otolaryngology': 'Ear, nose and throat medicine',\n",
       " 'Head and neck disorders': 'Ear, nose and throat medicine',\n",
       " 'Data integration': 'Data integration and warehousing',\n",
       " 'Data warehousing': 'Data integration and warehousing',\n",
       " 'Neurological disorders': 'Neurology',\n",
       " 'Life sciences': 'Biosciences',\n",
       " 'Primers': 'Probes and primers',\n",
       " 'Primer quality': 'Probes and primers',\n",
       " 'Probes': 'Probes and primers',\n",
       " 'Epistatic interactions': 'Epistasis',\n",
       " 'Epistatic genetic interaction': 'Epistasis',\n",
       " 'Ecological informatics': 'Ecology',\n",
       " 'Computational ecology': 'Ecology',\n",
       " 'Ecosystem science': 'Ecology',\n",
       " 'Ecoinformatics': 'Ecology',\n",
       " 'Pipelines': 'Workflows',\n",
       " 'Tool interoperability': 'Workflows',\n",
       " 'Software integration': 'Workflows',\n",
       " 'Tool integration': 'Workflows',\n",
       " 'Membrane proteins': 'Membrane and lipoproteins',\n",
       " 'Transmembrane proteins': 'Membrane and lipoproteins',\n",
       " 'Lipoproteins': 'Membrane and lipoproteins',\n",
       " 'Biomedical research': 'Medicine',\n",
       " 'Experimental medicine': 'Medicine',\n",
       " 'Clinical medicine': 'Medicine',\n",
       " 'Internal medicine': 'Medicine',\n",
       " 'General medicine': 'Medicine',\n",
       " 'Environmental microbiology': 'Microbial ecology',\n",
       " 'Microbiome': 'Microbial ecology',\n",
       " 'Molecular community analysis': 'Microbial ecology',\n",
       " 'Community analysis': 'Microbial ecology',\n",
       " 'Pharmacodynamics': 'Drug metabolism',\n",
       " 'ADME': 'Drug metabolism',\n",
       " 'Pharmacokinetics and pharmacodynamics': 'Drug metabolism',\n",
       " 'Drug distribution': 'Drug metabolism',\n",
       " 'Pharmacokinetics': 'Drug metabolism',\n",
       " 'Drug absorption': 'Drug metabolism',\n",
       " 'Drug excretion': 'Drug metabolism',\n",
       " 'RNA structure prediction': 'Structure prediction',\n",
       " 'Nucleic acid folding': 'Structure prediction',\n",
       " 'Nucleic acid design': 'Structure prediction',\n",
       " 'DNA structure prediction': 'Structure prediction',\n",
       " 'Protein fold recognition': 'Structure prediction',\n",
       " 'Protein structure prediction': 'Structure prediction',\n",
       " 'Nucleic acid structure prediction': 'Structure prediction',\n",
       " 'Protein 3D motifs': 'Protein structural motifs and surfaces',\n",
       " 'Protein structural motifs': 'Protein structural motifs and surfaces',\n",
       " 'Protein structural features': 'Protein structural motifs and surfaces',\n",
       " 'Protein surfaces': 'Protein structural motifs and surfaces',\n",
       " 'Structural motifs': 'Protein structural motifs and surfaces',\n",
       " 'Data design': 'Data architecture, analysis and design',\n",
       " 'Data architecture': 'Data architecture, analysis and design',\n",
       " 'Data analysis': 'Data architecture, analysis and design',\n",
       " 'Applied ontology': 'Ontology and terminology',\n",
       " 'Ontologies': 'Ontology and terminology',\n",
       " 'Ontology relations': 'Ontology and terminology',\n",
       " 'Upper ontology': 'Ontology and terminology',\n",
       " 'Terminology': 'Ontology and terminology',\n",
       " 'Ontology': 'Ontology and terminology',\n",
       " 'Protein stability': 'Protein folding, stability and design',\n",
       " 'Protein design': 'Protein folding, stability and design',\n",
       " 'Protein residue interactions': 'Protein folding, stability and design',\n",
       " 'Protein folding': 'Protein folding, stability and design',\n",
       " 'Rational protein design': 'Protein folding, stability and design',\n",
       " 'Metadata management': 'Data management',\n",
       " 'Data stewardship': 'Data governance',\n",
       " 'Data privacy': 'Data security',\n",
       " 'Genes, gene family or system': 'Gene and protein families',\n",
       " 'Gene system': 'Gene and protein families',\n",
       " 'Gene family': 'Gene and protein families',\n",
       " 'Protein sequence classification': 'Gene and protein families',\n",
       " 'Protein families': 'Gene and protein families',\n",
       " 'Gene families': 'Gene and protein families',\n",
       " 'Protein post-translational modification': 'Protein modifications',\n",
       " 'Post-translational modifications': 'Protein modifications',\n",
       " 'PTMs': 'Protein modifications',\n",
       " 'Post-translation modifications': 'Protein modifications',\n",
       " 'Protein post-translational modifications': 'Protein modifications',\n",
       " 'Protein chemical modifications': 'Protein modifications',\n",
       " 'GO:0006464': 'Protein modifications',\n",
       " 'MOD:00000': 'Protein modifications',\n",
       " 'Pharmaceutical biotechnology': 'Medical biotechnology',\n",
       " 'Lipidomics': 'Lipids',\n",
       " 'Pain management': 'Pain medicine',\n",
       " 'Algiatry': 'Pain medicine',\n",
       " 'Protein functional sites': 'Protein binding sites',\n",
       " 'Protein cleavage sites': 'Protein binding sites',\n",
       " 'Enzyme active site': 'Protein binding sites',\n",
       " 'Protein-nucleic acid binding sites': 'Protein binding sites',\n",
       " 'Protein key folding sites': 'Protein binding sites',\n",
       " 'EEG': 'Electroencephalography',\n",
       " 'Communicable disease': 'Infectious disease',\n",
       " 'Transmissible disease': 'Infectious disease',\n",
       " 'Exomes': 'Genomics',\n",
       " 'Genomes': 'Genomics',\n",
       " 'Personal genomics': 'Genomics',\n",
       " 'Synthetic genomics': 'Genomics',\n",
       " 'Genome annotation': 'Genomics',\n",
       " 'Viral genomics': 'Genomics',\n",
       " 'Whole genomes': 'Genomics',\n",
       " 'Alternative splicing': 'RNA splicing',\n",
       " 'Splice sites': 'RNA splicing',\n",
       " 'Humans': 'Human biology',\n",
       " 'Biological processes': 'Molecular biology',\n",
       " 'Assembly': 'Sequence assembly',\n",
       " 'Diagnostic markers': 'Biomarkers',\n",
       " 'Clinical immunology': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Allergy': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immune disorders': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immunomodulators': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Immunotherapeutics': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Urology': 'Urology and nephrology',\n",
       " 'Nephrology': 'Urology and nephrology',\n",
       " 'Urological disorders': 'Urology and nephrology',\n",
       " 'Kidney disease': 'Urology and nephrology',\n",
       " 'Phylogeny reconstruction': 'Phylogeny',\n",
       " 'Phylogenetic dating': 'Phylogeny',\n",
       " 'Phylogenetic simulation': 'Phylogeny',\n",
       " 'Phylogenetic clocks': 'Phylogeny',\n",
       " 'Phylogenetic stratigraphy': 'Phylogeny',\n",
       " 'Optometry': 'Ophthalmology',\n",
       " 'Eye disoders': 'Ophthalmology',\n",
       " 'Chemoinformatics': 'Cheminformatics',\n",
       " 'Chemical informatics': 'Cheminformatics',\n",
       " 'AMR': 'Antimicrobial Resistance',\n",
       " 'Multidrug resistance': 'Antimicrobial Resistance',\n",
       " 'Multiple drug resistance (MDR)': 'Antimicrobial Resistance',\n",
       " 'Multiresistance': 'Antimicrobial Resistance',\n",
       " 'Antiprotozoal resistance': 'Antimicrobial Resistance',\n",
       " 'Antifungal resistance': 'Antimicrobial Resistance',\n",
       " 'Extensive drug resistance (XDR)': 'Antimicrobial Resistance',\n",
       " 'Pandrug resistance (PDR)': 'Antimicrobial Resistance',\n",
       " 'Total drug resistance (TDR)': 'Antimicrobial Resistance',\n",
       " 'Antiviral resistance': 'Antimicrobial Resistance',\n",
       " 'Biomimeic chemistry': 'Synthetic biology',\n",
       " 'Patient safety': 'Safety sciences',\n",
       " 'Drug safety': 'Safety sciences',\n",
       " 'Functional sites': 'Sequence sites, features and motifs',\n",
       " 'Sequence features': 'Sequence sites, features and motifs',\n",
       " 'Sequence sites': 'Sequence sites, features and motifs',\n",
       " 'Sequence profiles': 'Sequence sites, features and motifs',\n",
       " 'Sequence motifs': 'Sequence sites, features and motifs',\n",
       " 'HMMs': 'Sequence sites, features and motifs',\n",
       " 'Transplantation': 'Surgery',\n",
       " 'Environment': 'Environmental sciences',\n",
       " 'Biological sequences': 'Sequence analysis',\n",
       " 'Sequence databases': 'Sequence analysis',\n",
       " 'Animal experimentation': 'Laboratory animal science',\n",
       " 'Animal research': 'Laboratory animal science',\n",
       " 'Animal testing': 'Laboratory animal science',\n",
       " 'In vivo testing': 'Laboratory animal science',\n",
       " 'Medical physics': 'Biophysics',\n",
       " 'Neutron diffraction experiment': 'Neutron diffraction',\n",
       " 'Elastic neutron scattering': 'Neutron diffraction',\n",
       " 'Neutron microscopy': 'Neutron diffraction',\n",
       " 'NMR-based metabolomics': 'Metabolomics',\n",
       " 'LC-MS-based metabolomics': 'Metabolomics',\n",
       " 'MS-based metabolomics': 'Metabolomics',\n",
       " 'Exometabolomics': 'Metabolomics',\n",
       " 'Metabonomics': 'Metabolomics',\n",
       " 'MS-based targeted metabolomics': 'Metabolomics',\n",
       " 'Metabolome': 'Metabolomics',\n",
       " 'MS-based untargeted metabolomics': 'Metabolomics',\n",
       " 'Metabolites': 'Metabolomics',\n",
       " 'Mass spectrometry-based metabolomics': 'Metabolomics',\n",
       " 'Biological engineering': 'Bioengineering',\n",
       " 'Structure determination': 'Structural biology',\n",
       " 'Structural assignment': 'Structural biology',\n",
       " 'Structural determination': 'Structural biology',\n",
       " 'Rheumatology': 'Musculoskeletal medicine',\n",
       " 'Musculoskeletal disorders': 'Musculoskeletal medicine',\n",
       " 'Orthopaedics': 'Musculoskeletal medicine',\n",
       " 'Health care science': 'Medicines research and development',\n",
       " 'Health care research': 'Medicines research and development',\n",
       " 'Drug discovery and development': 'Medicines research and development',\n",
       " 'Hepatology': 'Hepatic and biliary medicine',\n",
       " 'Liver disorders': 'Hepatic and biliary medicine',\n",
       " 'Transmembrane regions': 'Protein folds and structural domains',\n",
       " 'Protein structural domains': 'Protein folds and structural domains',\n",
       " 'Protein membrane regions': 'Protein folds and structural domains',\n",
       " 'Intramembrane regions': 'Protein folds and structural domains',\n",
       " 'Protein folds': 'Protein folds and structural domains',\n",
       " 'Protein topological domains': 'Protein folds and structural domains',\n",
       " 'Protein domains': 'Protein folds and structural domains',\n",
       " 'Protein transmembrane regions': 'Protein folds and structural domains',\n",
       " 'Pharmacogenetics': 'Pharmacogenomics',\n",
       " 'Disease': 'Pathology',\n",
       " 'Data collection': 'Data acquisition',\n",
       " 'Public health': 'Public health and epidemiology',\n",
       " 'Epidemiology': 'Public health and epidemiology',\n",
       " 'Molecular motions': 'Molecular dynamics',\n",
       " 'Molecular flexibility': 'Molecular dynamics',\n",
       " 'Protein dynamics': 'Molecular dynamics',\n",
       " 'Fluidics': 'Microfluidics',\n",
       " 'Precision medicine': 'Personalised medicine',\n",
       " 'Molecular diagnostics': 'Personalised medicine',\n",
       " 'Computer programming': 'Software engineering',\n",
       " 'Software development': 'Software engineering',\n",
       " 'Data structures': 'Software engineering',\n",
       " 'Algorithms': 'Software engineering',\n",
       " 'Programming languages': 'Software engineering',\n",
       " 'Immunology and genetics': 'Immunogenetics',\n",
       " 'Immungenetics': 'Immunogenetics',\n",
       " 'Immune system genetics': 'Immunogenetics',\n",
       " 'Immunogenes': 'Immunogenetics',\n",
       " 'Methylation profiles': 'Epigenetics',\n",
       " 'DNA methylation': 'Epigenetics',\n",
       " 'Histone modification': 'Epigenetics',\n",
       " 'Clinical trial': 'Preclinical and clinical studies',\n",
       " 'Preclinical studies': 'Preclinical and clinical studies',\n",
       " 'Clinical studies': 'Preclinical and clinical studies',\n",
       " 'Drug trials': 'Preclinical and clinical studies',\n",
       " 'Clinical study': 'Preclinical and clinical studies',\n",
       " 'Preclinical study': 'Preclinical and clinical studies',\n",
       " 'Challenge study': 'Animal study',\n",
       " 'Functional RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Regulatory RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small nucleolar RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'miRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'microRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small interfering RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'siRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'lncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'snoRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'snRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Long non-coding RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'piwi-interacting RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'piRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small nuclear RNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'Small and long non-coding RNAs': 'Functional, regulatory and non-coding RNA',\n",
       " 'Long ncRNA': 'Functional, regulatory and non-coding RNA',\n",
       " 'DNA structural variation': 'Structural variation',\n",
       " 'Genomic structural variation': 'Structural variation',\n",
       " 'Insertion': 'Structural variation',\n",
       " 'Translocation': 'Structural variation',\n",
       " 'Deletion': 'Structural variation',\n",
       " 'Inversion': 'Structural variation',\n",
       " 'Duplication': 'Structural variation',\n",
       " 'Metabolic disorders': 'Endocrinology and metabolism',\n",
       " 'Endocrinology': 'Endocrinology and metabolism',\n",
       " 'Metabolism': 'Endocrinology and metabolism',\n",
       " 'Endocrine disorders': 'Endocrinology and metabolism',\n",
       " 'Primer binding sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Sequence tagged sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Nucleic acid sequence features': 'Nucleic acid sites, features and motifs',\n",
       " 'Nucleic acid functional sites': 'Nucleic acid sites, features and motifs',\n",
       " 'Spectroscopy': 'NMR',\n",
       " 'NMR spectroscopy': 'NMR',\n",
       " 'Nuclear magnetic resonance spectroscopy': 'NMR',\n",
       " 'Nuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'Heteronuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'Rotational Frame Nuclear Overhauser Effect Spectroscopy': 'NMR',\n",
       " 'NOESY': 'NMR',\n",
       " 'ROESY': 'NMR',\n",
       " 'HOESY': 'NMR',\n",
       " 'Molecular infection biology': 'Microbiology',\n",
       " 'Microbial physiology': 'Microbiology',\n",
       " 'Microbial genetics': 'Microbiology',\n",
       " 'Antimicrobial stewardship': 'Microbiology',\n",
       " 'Microbiological surveillance': 'Microbiology',\n",
       " 'Microbial surveillance': 'Microbiology',\n",
       " 'Molecular microbiology': 'Microbiology',\n",
       " 'Medical microbiology': 'Microbiology',\n",
       " 'Enzymology': 'Enzymes',\n",
       " 'Quality assurance': 'Quality affairs',\n",
       " 'Good laboratory practice': 'Quality affairs',\n",
       " 'Good clinical practice': 'Quality affairs',\n",
       " 'Good manufacturing practice': 'Quality affairs',\n",
       " 'Synthetic organic chemistry': 'Synthetic chemistry',\n",
       " 'Obstetrics': 'Gynaecology and obstetrics',\n",
       " 'Gynaecology': 'Gynaecology and obstetrics',\n",
       " 'Gynaecological disorders': 'Gynaecology and obstetrics',\n",
       " 'Protein repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Nucleic acid repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Repeat sequences': 'Sequence composition, complexity and repeats',\n",
       " 'Protein sequence repeats': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence complexity': 'Sequence composition, complexity and repeats',\n",
       " 'Sequence composition': 'Sequence composition, complexity and repeats',\n",
       " 'Low complexity sequences': 'Sequence composition, complexity and repeats',\n",
       " 'Community genomics': 'Metagenomics',\n",
       " 'Ecogenomics': 'Metagenomics',\n",
       " 'Biome sequencing': 'Metagenomics',\n",
       " 'Environmental genomics': 'Metagenomics',\n",
       " 'Environmental omics': 'Metagenomics',\n",
       " 'Shotgun metagenomics': 'Metagenomics',\n",
       " 'Computational toxicology': 'Toxicology',\n",
       " 'Toxicoinformatics': 'Toxicology',\n",
       " 'DNA analysis': 'DNA',\n",
       " 'Ancient DNA': 'DNA',\n",
       " 'Chromosomes': 'DNA',\n",
       " 'Protein sequence features': 'Protein sites, features and motifs',\n",
       " 'Signal peptide cleavage sites': 'Protein sites, features and motifs',\n",
       " 'Cellular processes': 'Cell biology',\n",
       " 'Cells': 'Cell biology',\n",
       " 'Protein subcellular localization': 'Cell biology',\n",
       " 'EKG': 'Electrocardiography',\n",
       " 'ECG': 'Electrocardiography',\n",
       " 'Electrophysiology': 'Physiology',\n",
       " 'Artificial Intelligence': 'Machine learning',\n",
       " 'Active learning': 'Machine learning',\n",
       " 'Knowledge representation': 'Machine learning',\n",
       " 'Unsupervised learning': 'Machine learning',\n",
       " 'Ensembl learning': 'Machine learning',\n",
       " 'Neural networks': 'Machine learning',\n",
       " 'Recommender system': 'Machine learning',\n",
       " 'Kernel methods': 'Machine learning',\n",
       " 'Reinforcement learning': 'Machine learning',\n",
       " 'Supervised learning': 'Machine learning',\n",
       " 'Reverse vaccinology': 'Vaccinology',\n",
       " 'Structural vaccinology': 'Vaccinology',\n",
       " 'Rational vaccine design': 'Vaccinology',\n",
       " 'Vaccine design': 'Vaccinology',\n",
       " 'Structure-based immunogen design': 'Vaccinology',\n",
       " 'Pulmonary medicine': 'Respiratory medicine',\n",
       " 'Pulmonology': 'Respiratory medicine',\n",
       " 'Pulmonary disorders': 'Respiratory medicine',\n",
       " 'Respiratory disease': 'Respiratory medicine',\n",
       " 'Biological models': 'Molecular interactions, pathways and networks',\n",
       " 'Genetic information processing pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Interactions': 'Molecular interactions, pathways and networks',\n",
       " 'Networks': 'Molecular interactions, pathways and networks',\n",
       " 'Signal transduction pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Biological pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Signaling pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Gene regulatory networks': 'Molecular interactions, pathways and networks',\n",
       " 'Disease pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Metabolic pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Interactome': 'Molecular interactions, pathways and networks',\n",
       " 'Environmental information processing pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Molecular interactions': 'Molecular interactions, pathways and networks',\n",
       " 'Biological networks': 'Molecular interactions, pathways and networks',\n",
       " 'Cellular process pathways': 'Molecular interactions, pathways and networks',\n",
       " 'Data clean-up': 'Data quality management',\n",
       " 'Data integrity': 'Data quality management',\n",
       " 'Data cleaning': 'Data quality management',\n",
       " 'Data quality': 'Data quality management',\n",
       " 'Biological system modelling': 'Systems biology',\n",
       " 'Systems modelling': 'Systems biology',\n",
       " 'Biological modelling': 'Systems biology',\n",
       " 'Cell culture': 'Cell culture collection',\n",
       " 'Biogeochemical cycle': 'Carbon cycle',\n",
       " 'Anaesthetics': 'Anaesthesiology',\n",
       " 'Gene imprinting': 'Genomic imprinting',\n",
       " 'Computational pharmacology': 'Pharmacology',\n",
       " 'Pharmacoinformatics': 'Pharmacology',\n",
       " 'Mental health': 'Psychiatry',\n",
       " 'Psychiatric disorders': 'Psychiatry',\n",
       " 'Plants': 'Plant biology',\n",
       " 'Plant science': 'Plant biology',\n",
       " 'Botany': 'Plant biology',\n",
       " 'Plant': 'Plant biology',\n",
       " 'Plant genetics': 'Plant biology',\n",
       " 'Plant ecology': 'Plant biology',\n",
       " 'Plant anatomy': 'Plant biology',\n",
       " 'Plant cell biology': 'Plant biology',\n",
       " 'Plant physiology': 'Plant biology',\n",
       " 'Clinical veterinary medicine': 'Veterinary medicine',\n",
       " 'Matrix/scaffold attachment region': 'DNA binding sites',\n",
       " 'Restriction sites': 'DNA binding sites',\n",
       " 'Matrix-attachment region': 'DNA binding sites',\n",
       " 'Scaffold-attachment region': 'DNA binding sites',\n",
       " 'Nucleosome exclusion sequences': 'DNA binding sites',\n",
       " 'Ribosome binding sites': 'DNA binding sites',\n",
       " 'Specimen collections': 'Sample collections',\n",
       " 'biosamples': 'Sample collections',\n",
       " 'samples': 'Sample collections',\n",
       " 'Biomedical informatics': 'Medical informatics',\n",
       " 'Clinical informatics': 'Medical informatics',\n",
       " 'Health informatics': 'Medical informatics',\n",
       " 'Healthcare informatics': 'Medical informatics',\n",
       " 'Health and disease': 'Medical informatics',\n",
       " 'Crystallography': 'X-ray diffraction',\n",
       " 'X-ray microscopy': 'X-ray diffraction',\n",
       " 'X-ray crystallography': 'X-ray diffraction',\n",
       " 'Mathematical biology': 'Computational biology',\n",
       " 'Biomathematics': 'Computational biology',\n",
       " 'Theoretical biology': 'Computational biology',\n",
       " 'Heredity': 'Genetics',\n",
       " 'Genes': 'Genetics',\n",
       " 'Information science': 'Informatics',\n",
       " 'Knowledge management': 'Informatics',\n",
       " 'Information management': 'Informatics',\n",
       " 'Alternative medicine': 'Complementary medicine',\n",
       " 'Integrative medicine': 'Complementary medicine',\n",
       " 'Holistic medicine': 'Complementary medicine',\n",
       " 'Protein sorting': 'Protein targeting and localisation',\n",
       " 'Protein targeting': 'Protein targeting and localisation',\n",
       " 'Protein localisation': 'Protein targeting and localisation',\n",
       " 'Child health': 'Paediatrics',\n",
       " 'KDD': 'Data mining',\n",
       " 'Knowledge discovery in databases': 'Data mining',\n",
       " 'Pattern recognition': 'Data mining',\n",
       " 'Chemical screening': 'Compound libraries and screening',\n",
       " 'Compound library': 'Compound libraries and screening',\n",
       " 'Small chemical compounds libraries': 'Compound libraries and screening',\n",
       " 'Chemical library': 'Compound libraries and screening',\n",
       " 'Small compounds libraries': 'Compound libraries and screening',\n",
       " 'Target identification and validation': 'Compound libraries and screening',\n",
       " 'Database curation': 'Data submission, annotation, and curation',\n",
       " 'Data provenance': 'Data submission, annotation, and curation',\n",
       " 'Data curation': 'Data submission, annotation, and curation',\n",
       " 'Bottom-up proteomics': 'Proteomics',\n",
       " 'Peptide identification': 'Proteomics',\n",
       " 'Quantitative proteomics': 'Proteomics',\n",
       " 'Discovery proteomics': 'Proteomics',\n",
       " 'Targeted proteomics': 'Proteomics',\n",
       " 'MS-based targeted proteomics': 'Proteomics',\n",
       " 'MS-based untargeted proteomics': 'Proteomics',\n",
       " 'Protein and peptide identification': 'Proteomics',\n",
       " 'Metaproteomics': 'Proteomics',\n",
       " 'Top-down proteomics': 'Proteomics',\n",
       " 'Laboratory mouse': 'Mouse clinic',\n",
       " 'Nucleic acid informatics': 'Nucleic acids',\n",
       " 'Nucleic acid bioinformatics': 'Nucleic acids',\n",
       " 'Nucleic acid physicochemistry': 'Nucleic acids',\n",
       " 'Nucleic acid properties': 'Nucleic acids',\n",
       " 'Cloud computing': 'Computer science',\n",
       " 'High performance computing': 'Computer science',\n",
       " 'High-performance computing': 'Computer science',\n",
       " 'HPC': 'Computer science',\n",
       " 'CNV deletion': 'Copy number variation',\n",
       " 'Copy number variant': 'Copy number variation',\n",
       " 'CNV duplication': 'Copy number variation',\n",
       " 'Complex CNV': 'Copy number variation',\n",
       " 'CNV insertion / amplification': 'Copy number variation',\n",
       " 'Animal biology': 'Zoology',\n",
       " 'Metazoa': 'Zoology',\n",
       " 'Animal': 'Zoology',\n",
       " 'Animals': 'Zoology',\n",
       " 'Animal physiology': 'Zoology',\n",
       " 'Animal genetics': 'Zoology',\n",
       " 'Entomology': 'Zoology',\n",
       " 'Drug formulation and delivery': 'Biotherapeutics',\n",
       " 'Drug delivery': 'Biotherapeutics',\n",
       " 'Drug formulation': 'Biotherapeutics',\n",
       " 'Traumatology': 'Trauma medicine',\n",
       " 'Structural bioinformatics': 'Structure analysis',\n",
       " 'Biomolecular structure': 'Structure analysis',\n",
       " 'Structure databases': 'Structure analysis',\n",
       " 'Structures': 'Structure analysis',\n",
       " 'Molecular structure': 'Structure analysis',\n",
       " 'Computational structural biology': 'Structure analysis',\n",
       " 'Structure data resources': 'Structure analysis',\n",
       " 'NLP': 'Natural language processing',\n",
       " 'Literature mining': 'Natural language processing',\n",
       " 'BioNLP': 'Natural language processing',\n",
       " 'Text analytics': 'Natural language processing',\n",
       " 'Text data mining': 'Natural language processing',\n",
       " 'Text mining': 'Natural language processing',\n",
       " 'Gastrointestinal disorders': 'Gastroenterology',\n",
       " 'biobanking': 'Biobank',\n",
       " 'Tissue collection': 'Biobank',\n",
       " 'Databases': 'Database management',\n",
       " 'Content management': 'Database management',\n",
       " 'Record management': 'Database management',\n",
       " 'Document management': 'Database management',\n",
       " 'File management': 'Database management',\n",
       " 'Database administration': 'Database management',\n",
       " 'Information systems': 'Database management',\n",
       " 'Biological science': 'Biology',\n",
       " 'Cryobiology': 'Biology',\n",
       " 'Aerobiology': 'Biology',\n",
       " 'Chronobiology': 'Biology',\n",
       " 'Biological rhythms': 'Biology',\n",
       " 'Reproductive biology': 'Biology',\n",
       " 'Behavioural biology': 'Biology',\n",
       " 'Protein-nucleic acid interactions': 'Protein interactions',\n",
       " 'Protein-DNA interactions': 'Protein interactions',\n",
       " 'Protein-RNA interactions': 'Protein interactions',\n",
       " 'Protein-RNA interaction': 'Protein interactions',\n",
       " 'Protein-ligand interactions': 'Protein interactions',\n",
       " 'Protein-protein interactions': 'Protein interactions',\n",
       " 'Protein interaction map': 'Protein interactions',\n",
       " 'Protein interactome': 'Protein interactions',\n",
       " 'Protein interaction networks': 'Protein interactions',\n",
       " 'Protein-DNA interaction': 'Protein interactions',\n",
       " 'Genetic linkage': 'Mapping',\n",
       " 'Synteny': 'Mapping',\n",
       " 'Linkage': 'Mapping',\n",
       " 'Linkage mapping': 'Mapping',\n",
       " 'Gerontology': 'Geriatric medicine',\n",
       " 'Ageing': 'Geriatric medicine',\n",
       " 'Aging': 'Geriatric medicine',\n",
       " 'Geriatrics': 'Geriatric medicine',\n",
       " 'Applied microbiology': 'Biotechnology',\n",
       " 'RiboSeq': 'Ribosome Profiling',\n",
       " 'ribosomal footprinting': 'Ribosome Profiling',\n",
       " 'ribo-seq': 'Ribosome Profiling',\n",
       " 'RIBO-seq': 'Ribosome Profiling',\n",
       " 'Ribo-Seq': 'Ribosome Profiling',\n",
       " 'translation footprinting': 'Ribosome Profiling',\n",
       " 'Blood disorders': 'Haematology',\n",
       " 'Haematological disorders': 'Haematology',\n",
       " 'Protein features (disordered structure)': 'Protein disordered structure',\n",
       " 'Drug design': 'Medicinal chemistry',\n",
       " 'Nucleic acid structure': 'Nucleic acid structure analysis',\n",
       " 'Nucleic acid denaturation': 'Nucleic acid structure analysis',\n",
       " 'Nucleic acid thermodynamics': 'Nucleic acid structure analysis',\n",
       " 'DNA melting': 'Nucleic acid structure analysis',\n",
       " 'DNA structure': 'Nucleic acid structure analysis',\n",
       " 'RNA structure': 'Nucleic acid structure analysis',\n",
       " 'RNA alignment': 'Nucleic acid structure analysis',\n",
       " 'RNA structure alignment': 'Nucleic acid structure analysis',\n",
       " 'Cancer biology': 'Oncology',\n",
       " 'Cancer': 'Oncology',\n",
       " 'Neoplasm': 'Oncology',\n",
       " 'Neoplasms': 'Oncology',\n",
       " 'Genotype-phenotype analysis': 'Genotype and phenotype',\n",
       " 'Genotype-phenotype': 'Genotype and phenotype',\n",
       " 'Genotype and phenotype resources': 'Genotype and phenotype',\n",
       " 'Genotype': 'Genotype and phenotype',\n",
       " 'Genotyping': 'Genotype and phenotype',\n",
       " 'Phenotype': 'Genotype and phenotype',\n",
       " 'Phenotyping': 'Genotype and phenotype',\n",
       " 'Drug development science': 'Drug development',\n",
       " 'Medicines development': 'Drug development',\n",
       " 'Medicine development': 'Drug development',\n",
       " 'DNA variation': 'Genetic variation',\n",
       " 'Polymorphism': 'Genetic variation',\n",
       " 'Mutation': 'Genetic variation',\n",
       " 'Genomic variation': 'Genetic variation',\n",
       " 'Somatic mutations': 'Genetic variation',\n",
       " 'Protein structure': 'Protein structure analysis',\n",
       " 'Protein tertiary structure': 'Protein structure analysis',\n",
       " 'Organisms': 'Model organisms',\n",
       " 'Stem cell research': 'Regenerative medicine',\n",
       " 'Nucleosome positioning': 'DNA packaging',\n",
       " 'Computational immunology': 'Immunoinformatics',\n",
       " 'Dermatological disorders': 'Dermatology',\n",
       " 'Biological chemistry': 'Biochemistry',\n",
       " 'Glycomics': 'Biochemistry',\n",
       " 'Pathobiochemistry': 'Biochemistry',\n",
       " 'Phytochemistry': 'Biochemistry',\n",
       " 'PolyA signal': 'Gene transcripts',\n",
       " 'PolyA site': 'Gene transcripts',\n",
       " 'Fusion transcripts': 'Gene transcripts',\n",
       " 'Gene transcript features': 'Gene transcripts',\n",
       " 'Coding RNA': 'Gene transcripts',\n",
       " 'Transit peptide coding sequence': 'Gene transcripts',\n",
       " 'Signal peptide coding sequence': 'Gene transcripts',\n",
       " 'mRNA features': 'Gene transcripts',\n",
       " 'Introns': 'Gene transcripts',\n",
       " 'Exons': 'Gene transcripts',\n",
       " 'cDNA': 'Gene transcripts',\n",
       " 'mRNA': 'Gene transcripts',\n",
       " 'EST': 'Gene transcripts',\n",
       " 'Development': 'Developmental biology',\n",
       " 'Transposons': 'Mobile genetic elements',\n",
       " 'Biomedical sciences': 'Biomedical science',\n",
       " 'Health science': 'Biomedical science',\n",
       " 'Protein physicochemistry': 'Protein properties',\n",
       " 'Protein hydropathy': 'Protein properties',\n",
       " 'Functional analysis': 'Function analysis',\n",
       " 'Protein function prediction': 'Function analysis',\n",
       " 'Protein function analysis': 'Function analysis',\n",
       " 'VT 1.7.10 Polymer science': 'Chemistry',\n",
       " 'Polymer science': 'Chemistry',\n",
       " 'Chemical science': 'Chemistry',\n",
       " 'Nuclear chemistry': 'Chemistry',\n",
       " 'Physical chemistry': 'Chemistry',\n",
       " 'Inorganic chemistry': 'Chemistry',\n",
       " 'Organic chemistry': 'Chemistry',\n",
       " 'Mathematical chemistry': 'Chemistry',\n",
       " 'Protein features (secondary structure)': 'Protein secondary structure',\n",
       " 'Protein super-secondary structure': 'Protein secondary structure',\n",
       " 'Agriculture': 'Agricultural science',\n",
       " 'Agronomy': 'Agricultural science',\n",
       " 'Agroecology': 'Agricultural science',\n",
       " 'Animal breeding': 'Agricultural science',\n",
       " 'Food security': 'Agricultural science',\n",
       " 'Phytomedicine': 'Agricultural science',\n",
       " 'Soil science': 'Agricultural science',\n",
       " 'Farming systems research': 'Agricultural science',\n",
       " 'Plant nutrition': 'Agricultural science',\n",
       " 'Animal husbandry': 'Agricultural science',\n",
       " 'Food process engineering': 'Agricultural science',\n",
       " 'Plant cultivation': 'Agricultural science',\n",
       " 'Plant breeding': 'Agricultural science',\n",
       " 'Animal nutrition': 'Agricultural science',\n",
       " 'Horticulture': 'Agricultural science',\n",
       " 'Plant pathology': 'Agricultural science',\n",
       " 'Agricultural economics': 'Agricultural science',\n",
       " 'Agricultural biotechnology': 'Agricultural science',\n",
       " 'Protein informatics': 'Proteins',\n",
       " 'Protein bioinformatics': 'Proteins',\n",
       " 'Protein databases': 'Proteins',\n",
       " 'Healthcare RA': 'Regulatory affairs',\n",
       " 'Tree of life': 'Cladistics',\n",
       " 'Microsatellites': 'DNA polymorphism',\n",
       " 'Variable number of tandem repeat polymorphism': 'DNA polymorphism',\n",
       " 'snps': 'DNA polymorphism',\n",
       " 'RFLP': 'DNA polymorphism',\n",
       " 'VNTR': 'DNA polymorphism',\n",
       " 'SNP': 'DNA polymorphism',\n",
       " 'Single nucleotide polymorphism': 'DNA polymorphism',\n",
       " 'Acute medicine': 'Critical care medicine',\n",
       " 'Emergency medicine': 'Critical care medicine',\n",
       " 'Intensive care medicine': 'Critical care medicine',\n",
       " 'Chemical structures': 'Small molecules',\n",
       " 'Toxins and targets': 'Small molecules',\n",
       " 'Amino acids': 'Small molecules',\n",
       " 'Targets': 'Small molecules',\n",
       " 'Drugs and target structures': 'Small molecules',\n",
       " 'Peptides': 'Small molecules',\n",
       " 'Toxins': 'Small molecules',\n",
       " 'Peptides and amino acids': 'Small molecules',\n",
       " 'Metabolite structures': 'Small molecules',\n",
       " 'Target structures': 'Small molecules',\n",
       " 'Drug structures': 'Small molecules',\n",
       " 'Drug targets': 'Small molecules',\n",
       " 'CHEBI:23367': 'Small molecules',\n",
       " 'Laboratory resources': 'Laboratory information management',\n",
       " 'Regulatory genomics': 'Gene regulation',\n",
       " 'Transcription factors': 'Transcription factors and regulatory sites',\n",
       " '-35 signals': 'Transcription factors and regulatory sites',\n",
       " 'CpG islands': 'Transcription factors and regulatory sites',\n",
       " '-10 signals': 'Transcription factors and regulatory sites',\n",
       " 'Enhancers': 'Transcription factors and regulatory sites',\n",
       " 'TFBS': 'Transcription factors and regulatory sites',\n",
       " 'Transcriptional regulatory sites': 'Transcription factors and regulatory sites',\n",
       " 'GC signals': 'Transcription factors and regulatory sites',\n",
       " 'Attenuators': 'Transcription factors and regulatory sites',\n",
       " 'CAAT signals': 'Transcription factors and regulatory sites',\n",
       " 'TATA signals': 'Transcription factors and regulatory sites',\n",
       " 'Terminators': 'Transcription factors and regulatory sites',\n",
       " 'Isochores': 'Transcription factors and regulatory sites',\n",
       " 'CAT box': 'Transcription factors and regulatory sites',\n",
       " 'Promoters': 'Transcription factors and regulatory sites',\n",
       " 'Transcription factor binding sites': 'Transcription factors and regulatory sites',\n",
       " 'CCAAT box': 'Transcription factors and regulatory sites',\n",
       " 'cryo-EM': 'Cryogenic electron microscopy',\n",
       " 'Computer graphics': 'Data visualisation',\n",
       " 'Data rendering': 'Data visualisation'}"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "synonym_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "49d9b6d3341a48f68bb0daad7b3baa20",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/65 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "{'Disease': 'Pathology',\n",
       " 'Iinfectious disease': 'Infectious disease',\n",
       " 'Infectious Disease': 'Infectious disease',\n",
       " 'T cell biology': 'Biology',\n",
       " 'Cancer Study': 'Oncology',\n",
       " 'Cancer': 'Oncology',\n",
       " 'Clinical Allergy': 'Allergy, clinical immunology and immunotherapeutics',\n",
       " 'Lung cancer': 'Oncology',\n",
       " 'Clinical medicine': 'Medicine'}"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import Levenshtein\n",
    "\n",
    "hallucinations = data['Hallucinations']\n",
    "\n",
    "matched_topics = {}\n",
    "\n",
    "for hallucination_set in tqdm(hallucinations):\n",
    "    for hallucination in hallucination_set:\n",
    "        if hallucination in matched_topics:\n",
    "            continue\n",
    "        matched = False\n",
    "        # First check for a match in the topics list\n",
    "        sorted_topics = sorted(edam_topics, key=lambda topic: Levenshtein.distance(hallucination, topic))\n",
    "        for topic in sorted_topics:\n",
    "            distance = Levenshtein.distance(hallucination, topic)\n",
    "            if  0 < distance <= 2:\n",
    "                matched_topics[hallucination] = topic\n",
    "                matched = True\n",
    "                break\n",
    "        \n",
    "        # If the hallucination has already been matched, skip to the next hallucination\n",
    "        if matched:\n",
    "            continue\n",
    "        \n",
    "        # If no match in the topics list, look through the available synonyms\n",
    "        sorted_synonyms = sorted(synonym_dict.keys(), key=lambda topic: Levenshtein.distance(hallucination, topic))\n",
    "        for topic in sorted_synonyms:\n",
    "            distance = Levenshtein.distance(hallucination, topic)\n",
    "            if 0 <= distance <= 1:\n",
    "                matched_topics[hallucination] = synonym_dict[topic]\n",
    "                matched = True\n",
    "                break     \n",
    "\n",
    "        if matched:\n",
    "            continue\n",
    "\n",
    "        for topic in sorted_topics:\n",
    "            if topic.lower() in hallucination.lower().split():\n",
    "                matched_topics[hallucination] = topic\n",
    "                break\n",
    "        # No break reached\n",
    "        else:\n",
    "            for topic in sorted_synonyms:\n",
    "                if topic.lower() in hallucination.lower().split():\n",
    "                    matched_topics[hallucination] = synonym_dict[topic]\n",
    "                    break\n",
    "\n",
    "matched_topics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'Disease' in row 7 matches topic 'Pathology'\n",
      "'Iinfectious disease' in row 8 matches topic 'Infectious disease'\n",
      "'Infectious Disease' in row 20 matches topic 'Infectious disease'\n",
      "'T cell biology' in row 24 matches topic 'Biology'\n",
      "'Cancer Study' in row 35 matches topic 'Oncology'\n",
      "'Cancer Study' in row 36 matches topic 'Oncology'\n",
      "'Cancer' in row 37 matches topic 'Oncology'\n",
      "'Cancer Study' in row 41 matches topic 'Oncology'\n",
      "'Clinical Allergy' in row 43 matches topic 'Allergy, clinical immunology and immunotherapeutics'\n",
      "'Cancer Study' in row 46 matches topic 'Oncology'\n",
      "'Lung cancer' in row 46 matches topic 'Oncology'\n",
      "'Clinical medicine' in row 53 matches topic 'Medicine'\n",
      "'Cancer' in row 58 matches topic 'Oncology'\n"
     ]
    }
   ],
   "source": [
    "for index, row in data.iterrows():\n",
    "    if len(row['Hallucinations']) > 0:\n",
    "        for hallucination in list(row['Hallucinations']):\n",
    "            if hallucination in matched_topics:\n",
    "                print(f\"'{hallucination}' in row {index} matches topic '{matched_topics[hallucination]}'\")\n",
    "                data.at[index, 'Predictions'].add(matched_topics[hallucination])\n",
    "                continue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>_id</th>\n",
       "      <th>Name</th>\n",
       "      <th>Description</th>\n",
       "      <th>Model</th>\n",
       "      <th>Predictions</th>\n",
       "      <th>Hallucinations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>VDJ_PRJNA315543</td>\n",
       "      <td>Sims et al., Diversity and divergence of the glioma-infiltrating T-cell receptor repertoire</td>\n",
       "      <td>Cancer Study\\n nonneoplastic, low-grade and high-grade glioma\\n NIH/National Institute of Biomedical Imaging and Bioengineering Grant K01EB016071, NIH/National Institute of Neurological Disorders and Stroke Grant R01NS066955</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{Genomics, Oncology}</td>\n",
       "      <td>{Immuno…strain, Cancer Study}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>VDJ_PRJNA506151</td>\n",
       "      <td>Jia et al., Local mutational diversity drives intratumoral immune heterogeneity in non-small cell lung cancer.</td>\n",
       "      <td>Cancer Study\\n Lung cancer\\n National Nature Science Foundation of China (81222031) to B.Z., and Beau Biden Cancer Moonshot Initiative from the NIH (R33CA225328)</td>\n",
       "      <td>gpt-3.5-turbo</td>\n",
       "      <td>{Immunology, Oncology}</td>\n",
       "      <td>{Cancer Study, Lung cancer}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                _id  \\\n",
       "35  VDJ_PRJNA315543   \n",
       "46  VDJ_PRJNA506151   \n",
       "\n",
       "                                                                                                              Name  \\\n",
       "35                     Sims et al., Diversity and divergence of the glioma-infiltrating T-cell receptor repertoire   \n",
       "46  Jia et al., Local mutational diversity drives intratumoral immune heterogeneity in non-small cell lung cancer.   \n",
       "\n",
       "                                                                                                                                                                                                                         Description  \\\n",
       "35  Cancer Study\\n nonneoplastic, low-grade and high-grade glioma\\n NIH/National Institute of Biomedical Imaging and Bioengineering Grant K01EB016071, NIH/National Institute of Neurological Disorders and Stroke Grant R01NS066955   \n",
       "46                                                                 Cancer Study\\n Lung cancer\\n National Nature Science Foundation of China (81222031) to B.Z., and Beau Biden Cancer Moonshot Initiative from the NIH (R33CA225328)   \n",
       "\n",
       "            Model             Predictions                 Hallucinations  \n",
       "35  gpt-3.5-turbo    {Genomics, Oncology}  {Immuno…strain, Cancer Study}  \n",
       "46  gpt-3.5-turbo  {Immunology, Oncology}    {Cancer Study, Lung cancer}  "
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['Hallucinations'].apply(len) > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add quotes back in for the predictions with commas (i.e, \"Data submission, annotation, and curation\")\n",
    "data['Predictions'] = data['Predictions'].apply(lambda preds: [f'\"{pred}\"' if f'\"{pred}\"' in quoted_topics else pred for pred in preds])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Predictions</th>\n",
       "      <th>Hallucinations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>[Drug discovery, Bioengineering, Pathology]</td>\n",
       "      <td>{Disease}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>[Immunology, Infectious disease, Transcriptomics]</td>\n",
       "      <td>{Iinfectious disease}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>[Vaccinology, Immunology, Infectious disease]</td>\n",
       "      <td>{Infectious Disease}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>[Biology, Immunology, Biomedical science]</td>\n",
       "      <td>{T cell biology}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>[Immunology, Neurology]</td>\n",
       "      <td>{Multiple Sclerosis Study}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>[Immunogenetics, Genetics]</td>\n",
       "      <td>{Immune system}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>[Genomics, Oncology]</td>\n",
       "      <td>{Immuno…strain, Cancer Study}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>[Immunology, Oncology, Data acquisition]</td>\n",
       "      <td>{Cancer Study}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>[Immunology, Oncology, Bioinformatics]</td>\n",
       "      <td>{Cancer}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>[Multiomics, Oncology]</td>\n",
       "      <td>{Cancer Study}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>[Immunology, Molecular biology, \"Allergy, clinical immunology and immunotherapeutics\"]</td>\n",
       "      <td>{Clinical Allergy}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>[Immunology, Oncology]</td>\n",
       "      <td>{Cancer Study, Lung cancer}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>[Immunology, Infectious disease, Medicine]</td>\n",
       "      <td>{Clinical medicine}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>[Immunology, Infectious disease]</td>\n",
       "      <td>{Women's health}</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>[Immunology, Oncology, Genetics]</td>\n",
       "      <td>{Cancer}</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                               Predictions  \\\n",
       "7                                              [Drug discovery, Bioengineering, Pathology]   \n",
       "8                                        [Immunology, Infectious disease, Transcriptomics]   \n",
       "20                                           [Vaccinology, Immunology, Infectious disease]   \n",
       "24                                               [Biology, Immunology, Biomedical science]   \n",
       "25                                                                 [Immunology, Neurology]   \n",
       "32                                                              [Immunogenetics, Genetics]   \n",
       "35                                                                    [Genomics, Oncology]   \n",
       "36                                                [Immunology, Oncology, Data acquisition]   \n",
       "37                                                  [Immunology, Oncology, Bioinformatics]   \n",
       "41                                                                  [Multiomics, Oncology]   \n",
       "43  [Immunology, Molecular biology, \"Allergy, clinical immunology and immunotherapeutics\"]   \n",
       "46                                                                  [Immunology, Oncology]   \n",
       "53                                              [Immunology, Infectious disease, Medicine]   \n",
       "56                                                        [Immunology, Infectious disease]   \n",
       "58                                                        [Immunology, Oncology, Genetics]   \n",
       "\n",
       "                   Hallucinations  \n",
       "7                       {Disease}  \n",
       "8           {Iinfectious disease}  \n",
       "20           {Infectious Disease}  \n",
       "24               {T cell biology}  \n",
       "25     {Multiple Sclerosis Study}  \n",
       "32                {Immune system}  \n",
       "35  {Immuno…strain, Cancer Study}  \n",
       "36                 {Cancer Study}  \n",
       "37                       {Cancer}  \n",
       "41                 {Cancer Study}  \n",
       "43             {Clinical Allergy}  \n",
       "46    {Cancer Study, Lung cancer}  \n",
       "53            {Clinical medicine}  \n",
       "56               {Women's health}  \n",
       "58                       {Cancer}  "
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[data['Hallucinations'].apply(len) > 0][['Predictions', 'Hallucinations']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "data['Predictions'] = data['Predictions'].apply(lambda lst: set(lst))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_name = os.path.basename(DATA_PATH).replace('.', '_processed.')\n",
    "\n",
    "data.to_csv(f'outputs/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nde_gpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
