{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from tqdm.notebook import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "response = requests.get('http://api.data.niaid.nih.gov/v1/query?q=lincs&fetch_all=true').json()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "482a8935e6534788bf31425ec911026a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/500 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "(['LINCS - preprocessed',\n",
       "  'Columbia LINCS Trans-Center project data',\n",
       "  'LINCS Canvas Browser: interactive web app to query, browse and interrogate LINCS L1000 gene expression signatures.',\n",
       "  'LINCS MCF10A GRNs calculated with ANANSE',\n",
       "  'Systematic Quality Control Analysis of LINCS Data.',\n",
       "  'Compound signature detection on LINCS L1000 big data.',\n",
       "  'L1000 Dataset -small molecule perturbagens- LINCS Phase 1',\n",
       "  'L1000 Dataset -small molecule perturbagens- LINCS Joint Project',\n",
       "  'L1000 Dataset -small molecule perturbagens- LINCS Trans-Center Project',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics LINCS PHASE *II* (n=354,123; updated March 30, 2017)',\n",
       "  'Drug-induced adverse events prediction with the LINCS L1000 data.',\n",
       "  'broadinstitute/lincs-cell-painting: Full release of LINCS Cell Painting dataset',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics LINCS Pilot PHASE I (n=1,319,138; updated March 03, 2017)',\n",
       "  'L1000 Dataset -small molecule perturbagens- LINCS Phase 2 (June 2015)',\n",
       "  'L1000 Dataset -small molecule, CRISPR perturbagens- LINCS Phase 2 (March 2017)',\n",
       "  'L1000 Dataset -small molecule, CRISPR perturbagens- LINCS Phase 2 (December 2015)',\n",
       "  'Harnessing the biological complexity of Big Data from LINCS gene expression signatures.',\n",
       "  'Perturbational proteomic and transcriptional profiles of drug perturbations from Broad Institute LINCS Proteomic Characterization Center for Signaling and Epigenetics and Broad Institute LINCS Center for Transcriptomics (U54HG008097) (n=5,063)',\n",
       "  'LINCS L1000 dataset-based repositioning of CGP-60474 as a highly potent anti-endotoxemic agent.',\n",
       "  'l1kdeconv: an R package for peak calling analysis with LINCS L1000 data.',\n",
       "  'Coarse-grained simulations of a phase-separated membrane with different LINCS settings (part 1/3)',\n",
       "  'Coarse-grained simulations of a phase-separated membrane with different LINCS settings (part 2/3)',\n",
       "  'Coarse-grained simulations of a phase-separated membrane with different LINCS settings (part 3/3)',\n",
       "  'Ontological representation, integration, and analysis of LINCS cell line cells and their cellular responses.',\n",
       "  'L1000 Dataset - small molecule, nucleic acid perturbagens - LINCS Phase 2 (December 2021)',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 12]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 11]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 3]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 9]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 4]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 5]',\n",
       "  'Metrics for growth responses of MDA-MB-231 and HME-1 cell lines to LINCS Pilot Phase TransCenter Project kinase inhibitors',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 9 of 9: Calculated dose-response metrics for biological replicate 3',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 7 of 15: End-point dose-response metrics for biological replicate 1.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 15 of 15: Calculated dose-response metrics.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 9 of 15: End-point dose-response metrics for biological replicate 3.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 8 of 9: Calculated dose-response metrics for biological replicate 2',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 8 of 15: End-point dose-response metrics for biological replicate 2.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 14 of 15: Time-dependent dose-response metrics for biological replicate 2.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 13 of 15: Time-dependent dose-response metrics for biological replicate 1.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 15 of 15: Time-dependent dose-response metrics for biological replicate 3.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 7 of 9: Calculated dose-response metrics for biological replicate 1',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 11 of 15: Time-dependent mean normalized growth rate inhibition values for biological replicate 2.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 1 of 9: Cell count and normalized growth rate inhibition values for biological replicate 1',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 10 of 15: Time-dependent mean normalized growth rate inhibition values for biological replicate 1.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 1 of 15: Cell count and relative growth within biological replicate 1.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 3 of 9: Cell count and normalized growth rate inhibition values for biological replicate 3',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 2 of 9: Cell count and normalized growth rate inhibition values for biological replicate 2',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 6 of 9: Mean cell count and mean normalized growth rate inhibition values for biological replicate 3',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist B in 2019 to assess reproducibility. Dataset 2 of 2: Calculated dose response metrics.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist C to assess reproducibility. Dataset 1 of 2: Normalized growth rate inhibition values. ',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist A in 2019 to assess reproducibility. Dataset 1 of 2: Normalized growth rate inhibition values. ',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist C in 2019 to assess reproducibility. Dataset 2 of 2: Calculated dose response metrics.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 5 of 9: Mean cell count and mean normalized growth rate inhibition values for biological replicate 2',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 4 of 9: Mean cell count and mean normalized growth rate inhibition values for biological replicate 1',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist A in 2019 to assess reproducibility. Dataset 2 of 2: Calculated dose response metrics.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 7 of 15: Mean cell count and mean growth rate across all replicates.',\n",
       "  'LINCS MCF 10A Common Project: Fixed-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Repeat performed by Scientist B in 2019 to assess reproducibility. Dataset 1 of 2: Normalized growth rate inhibition values. ',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 6 of 15: Mean cell count and mean growth rate across biological replicate 3.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 5 of 15: End-point mean cell counts and mean normalized growth rate inhibition values for biological replicate 2.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 2 of 15: End-point cell counts and normalized growth rate inhibition values for all technical replicates of biological replicate 2.',\n",
       "  'LINCS MCF 10A Common Project: Rolling-time-point sensitivity measures of the MCF 10A breast cell line to 8 small molecule perturbagens. Dataset 3 of 15: End-point cell counts and normalized growth rate inhibition values for all technical replicates of biological replicate 3.',\n",
       "  'LINCS MCF 10A Common Project: Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with 8 small molecule perturbagens. Dataset 4 of 6: Nuclear fluorescent intensity measurements at 48 hr.',\n",
       "  'LINCS MCF 10A Common Project: Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with 8 small molecule perturbagens. Dataset 1 of 6: Cytosolic fluorescent intensity measurements at 24 hr.',\n",
       "  'LINCS MCF 10A Common Project: Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with 8 small molecule perturbagens. Dataset 3 of 6: Cytosolic fluorescent intensity measurements at 48 hr.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 12 of 15: Mean cell count and mean normalized growth rate inhibition values across biological replicate 2.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 11 of 15: Mean cell count and mean normalized growth rate inhibition values across biological replicate 1.',\n",
       "  'LINCS MCF 10A Common Project: Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with 8 small molecule perturbagens. Dataset 2 of 6: Nuclear fluorescent intensity measurements at 24 hr.',\n",
       "  'LINCS MCF 10A Common Project: Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with 8 small molecule perturbagens. Dataset 5 of 6: Cytosolic fluorescent intensity measurements at 72 hr.',\n",
       "  'LINCS Pilot Phase Joint Project: Sensitivity measures of six breast cancer cell lines to a library of small molecule kinase inhibitors (single drug treatments). Dataset 13 of 15: Mean cell count and mean normalized growth rate inhibition values across biological replicate 3.',\n",
       "  'Ribociclib KINOMEscan',\n",
       "  'Torkinib KINOMEscan',\n",
       "  'BMX-IN-1 KINOMEscan',\n",
       "  'JNK-IN-11 KINOMEscan',\n",
       "  'BI-D1870 KINOMEscan',\n",
       "  'Tang Proliferation/Mitosis',\n",
       "  'Moerke 3 Color Apoptosis',\n",
       "  'BX795 KiNativ -- single dose experiment',\n",
       "  'On the impact of different templates on creating and understanding user stories - Supplemental Material',\n",
       "  'L1000 MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'ReNcell differentiation: Phosphoproteomics during differentiation under basal conditions. Dataset 2 of 3',\n",
       "  'Ceritinib KINOMEscan',\n",
       "  'THZ-2-98-01 KINOMEscan',\n",
       "  'RNA-seq MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'NG25 KINOMEscan',\n",
       "  'WZ4003 KINOMEscan',\n",
       "  'SCH772984 KINOMEscan',\n",
       "  'Bosutinib KINOMEscan',\n",
       "  'AS601245 KINOMEscan',\n",
       "  'Binding affinity of 6 small molecule JAK inhibitors for 5 kinases',\n",
       "  'Moerke 2 Color Apoptosis',\n",
       "  'ReNcell differentiation: Total proteomics during differentiation under basal conditions. Dataset 1 of 3',\n",
       "  'A549 cells -- High-throughput Secretomic Analysis of Single Cells',\n",
       "  'U87 cells -- High-throughput Secretomic Analysis of Single Cells',\n",
       "  'U937 cells -- High-throughput Secretomic Analysis of Single Cells',\n",
       "  'High-content imaging of six breast cancer cell lines treated with a library of small molecule kinase inhibitors.',\n",
       "  'RPPA MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'Targeted metabolomics profiling data (i.e., peak intensity measurements) after acute exposure of BT-549 cells to DMSO or rapamycin, AZD8055 or Torin2 at two doses (100nM, 1000nM) for 6 hours. ',\n",
       "  'CyC IF MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'GCP MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'ReNcell differentiation: Total proteomics during differentiation with small molecule perturbagens. Dataset 3 of 3',\n",
       "  'MEMA HMEC240L Staining Set C',\n",
       "  'MEMA HCC1954 Staining Set 100 (DMSO control)',\n",
       "  'MEMA AU565 Staining Set 100 (Lapatinib treatment)',\n",
       "  'MEMA HMEC240L Staining Set 4',\n",
       "  'MEMA MCF10A Staining Set 3',\n",
       "  'MEMA HMEC122L Staining Set 1',\n",
       "  'MEMA AU565 Staining Set 100 (DMSO control)',\n",
       "  'Synovial Fibroblast 1: Secretion response of primary human synovial fibroblast samples from one healthy and one rheumatoid arthritis donor to a panel of 10 stimuli and 10 small molecule inhibitors',\n",
       "  'Breast Cancer Cell Density: Imaging assay of the density- and context-dependence of small molecule perturbagen response in breast cancer cell lines. Dataset 1 of 3: Cell count and normalized growth rate inhibition values.',\n",
       "  'Synovial Fibroblast 3.1: Secretion response of seven primary human synovial fibroblast samples from healthy and rheumatoid arthritis donors to a panel of 3 stimuli and 5 small molecule inhibitors (replicate 1 of 2)',\n",
       "  'PLSR model loadings (pMEK and pERK included) from analysis of the covariation of molecular signals with cell viability and apoptosis fraction in BRAF(V600E/D) melanoma cell lines',\n",
       "  'MEMA MCF10A Staining Set 1',\n",
       "  'MEMA HMEC122L Staining Set 4',\n",
       "  'MEMA HMEC122L Staining Set C',\n",
       "  'MEMA MCF10A Staining Set 2 no H3',\n",
       "  'MEMA HMEC240L Staining Set 1',\n",
       "  'MEMA MCF10A Staining Set I',\n",
       "  'MEMA MCF10A Staining Set C',\n",
       "  'MEMA HCC1954 Staining Set 100 (Lapatinib treatment)',\n",
       "  'Breast Cancer Deubiquitinating Enzyme (DUB) Inhibitor Profiling: Fixed-cell GR measures of 21 breast cell lines to 32 small molecule perturbagens. Dataset 1 of 2: Normalized growth rate inhibition values.',\n",
       "  'Growth factor-induced ERK activity pulsing dynamics in a mammary epithelial cell line. Dataset 3 of 3: mean reporter pulsing metrics.',\n",
       "  'Evaluation of the sensitivity of two triple-negative breast cancer cell lines (HCC1806, HCC70) to a collection of dual mTOR and PI3K-like kinase (PIKK) inhibitors. Dataset 2 of 2: GR metrics.',\n",
       "  'Breast Cancer Profiling Project, Gene Expression 1: Baseline mRNA sequencing on 35 breast cell lines',\n",
       "  'Evaluation of the sensitivities of six triple-negative breast cancer (TNBC) cell lines to 23 different PI3K/AKT/mTOR inhibitors or to a MEK inhibitor (trametinib). Dataset 2 of 2: GR metrics.',\n",
       "  'Growth factor-induced ERK activity and FoxO3a nuclear-cytoplasmic pulsing dynamics in a mammary epithelial cell line. Dataset 1 of 1: single-cell reporter measurements.',\n",
       "  'PLSR model loadings (pMEK and pERK excluded) from analysis of the covariation of molecular signals with cell viability and apoptosis fraction in BRAF(V600E/D) melanoma cell lines',\n",
       "  'Growth rate-corrected (GR) dose-response metric signatures across a panel of 71 breast cancer cell lines treated with a library of small molecule and antibody perturbagens. Dataset 3 of 4: Median, upper quartile, and lower quartile GR metrics per perturbagen. ',\n",
       "  'Evaluation of the sensitivity of two triple-negative breast cancer cell lines (HCC1806, HCC70) to a collection of dual mTOR and PI3K-like kinase (PIKK) inhibitors. Dataset 1 of 2: GR values.',\n",
       "  'Breast Cancer Profiling Project - Proteomics 2: 1 phosphoproteome dataset (including phosphotyrosine enrichment) for a 35-cell line breast cancer panel under basal conditions ',\n",
       "  'MEMA MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'Growth factor-induced ERK activity pulsing dynamics in a mammary epithelial cell line. Dataset 1 of 3: single-cell reporter measurements.',\n",
       "  'Growth factor-induced ERK activity pulsing dynamics in a mammary epithelial cell line. Dataset 2 of 3: single-cell reporter pulsing metrics.',\n",
       "  'IGF-I-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of an AKT inhibitor. Dataset 1 of 3: single-cell reporter measurements.',\n",
       "  'EGF-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of a MEK inhibitor. Dataset 3 of 3: mean reporter pulsing metrics.',\n",
       "  'diMN (Exp 3) - ALS and Control (unaffected) diMN cell lines differentiated from iPS cell lines using a short and direct differentiation protocol - Proteomics',\n",
       "  'IGF-I-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of an AKT inhibitor. Dataset 3 of 3: mean reporter pulsing metrics.',\n",
       "  'Evaluation of the sensitivities of six triple-negative breast cancer (TNBC) cell lines to 23 different PI3K/AKT/mTOR inhibitors or to a MEK inhibitor (trametinib). Dataset 1 of 2: GR values.',\n",
       "  'EGF-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of a MEK inhibitor. Dataset 2 of 3: single-cell reporter pulsing metrics.',\n",
       "  'Growth rate-corrected (GR) dose-response metric signatures across a panel of 71 breast cancer cell lines treated with a library of small molecule and antibody perturbagens. Dataset 4 of 4: Median, upper quartile, and lower quartile GR metrics per perturbagen class.',\n",
       "  'Immunofluorescence MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'FoxO3a nuclear-cytoplasmic pulsing dynamics in a mammary epithelial cell line following growth-factor treatment with varying ERK/AKT activation loads. Dataset 2 of 3: single-cell reporter pulsing metrics.',\n",
       "  'Growth factor-induced FoxO3a nuclear-cytoplasmic pulsing dynamics of non-phosphorylatable reporters in mammary epithelial cell lines. Dataset 1 of 3: single-cell reporter measurements.',\n",
       "  'Growth factor-induced FoxO3a nuclear-cytoplasmic pulsing dynamics of non-phosphorylatable reporters in mammary epithelial cell lines. Dataset 2 of 3: single-cell reporter pulsing metrics.',\n",
       "  'Phosphorylation state and protein levels measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with combinations of two compounds (immunofluorescence dataset 2 of 4)',\n",
       "  'Protein levels measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with Vemurafenib in combination with chromatin-targeting compounds (immunofluorescence dataset 3 of 4)',\n",
       "  'Tyrosine Kinase Inhibitor-induced cardiotoxicity in hiPSC derived cardiomyocytes - pilot dataset',\n",
       "  'Growth factor-induced FoxO3a nuclear-cytoplasmic pulsing dynamics of non-phosphorylatable reporters in mammary epithelial cell lines. Dataset 3 of 3: mean reporter pulsing metrics.',\n",
       "  'FoxO3a nuclear-cytoplasmic pulsing dynamics in a mammary epithelial cell line following growth-factor treatment with varying ERK/AKT activation loads. Dataset 3 of 3: mean reporter pulsing metrics.',\n",
       "  'MEMA MCF7 Staining Set 1',\n",
       "  'MEMA YAPC Staining Set 2',\n",
       "  'Breast Cancer Profiling Project - Proteomics',\n",
       "  'GCP aggregated data - Chemical perturbations',\n",
       "  'Multiplexed imaging of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with nine kinase inhibitors',\n",
       "  'GCP - Drug treatment of vascular cell model systems',\n",
       "  'Breast Cancer Deubiquitinating Enzyme (DUB) Inhibitor Profiling: Fixed-cell GR measures of 21 breast cell lines to 32 small molecule perturbagens. Dataset 2 of 2: Calculated dose response metrics.',\n",
       "  'GCP data - neurodev compounds',\n",
       "  'LINCS-data',\n",
       "  'A review of connectivity map and computational approaches in pharmacogenomics.',\n",
       "  'GCP data - Selected Cell Lines treated with an HSP90 inhibitor',\n",
       "  'GCP - Selected chemical agents in schizophrenia patient-derived neural progenitor cells',\n",
       "  'MEMA MCF7 Staining Set 2',\n",
       "  'MEMA MCF7 Staining Set 3',\n",
       "  'MEMA YAPC Staining Set 3',\n",
       "  'GCP data - epigenetic compounds',\n",
       "  'MEMA PC3 Staining Set 3',\n",
       "  'MEMA PC3 Staining Set 1',\n",
       "  'MEMA PC3 Staining Set 2',\n",
       "  'MEMA YAPC Staining Set 1',\n",
       "  'Breast Cancer Profiling Project, Drug Sensitivity phase II: Fixed-cell GR measures of 33 breast cell lines and PDX lines to 16 small molecule perturbagens from library plate II. Dataset 2 of 2: Calculated dose response metrics',\n",
       "  'Breast Cancer Profiling Project, Drug Sensitivity 1: Fixed-cell GR measures of 35 breast cell lines to 34 small molecule perturbagens. Dataset 2 of 2: Calculated dose response metrics.',\n",
       "  'Breast Cancer Profiling Project, Drug Sensitivity phase II: Fixed-cell GR measures of 33 breast cell lines and PDX lines to 16 small molecule perturbagens from library plate II. Dataset 1 of 2: Normalized growth rate inhibition values',\n",
       "  'Breast Cancer Profiling Project - Proteomics 3: 1 total proteome dataset for a 27-cell line breast cancer panel under basal conditions ',\n",
       "  'Highly multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the COLO 858 melanoma cell line',\n",
       "  'ATAC-Seq MCF10A Deep Dive Dataset: treatment with ligands',\n",
       "  'Breast Cancer Profiling Project, Drug Sensitivity 1: Fixed-cell GR measures of 35 breast cell lines to 34 small molecule perturbagens. Dataset 1 of 2: Normalized growth rate inhibition values.',\n",
       "  'Cellarium single cell metabolic profiling assay',\n",
       "  'Transcriptomics dataset (March 2016)',\n",
       "  'Tyrosine Kinase Inhibitor-induced cardiotoxicity in hiPSC derived cardiomyocytes - pilot dataset (qPCR validation)',\n",
       "  'Breast cell line dose response to target inhibition measured by high throughput microscopy',\n",
       "  'P100 data - neurodev compounds',\n",
       "  'Homo sapiens',\n",
       "  'L1000 gene expression profiling assay - DOS small molecule perturbagens',\n",
       "  'P100 data - epigenetic compounds',\n",
       "  'MEMA PC3 Staining Set 2 no H3',\n",
       "  'GCP data - kinase inhibitors',\n",
       "  'Transcriptomics dataset (June 2015)',\n",
       "  'Cell signaling response to growth factors measured by high throughput microscopy',\n",
       "  'Breast Cancer Profiling Project - Proteomics 1: 1 total proteome dataset for a 35-cell line breast cancer panel under basal conditions',\n",
       "  'P100 aggregated data - Chemical perturbations',\n",
       "  'FoxO3a nuclear-cytoplasmic pulsing dynamics in a mammary epithelial cell line following growth-factor treatment with varying ERK/AKT activation loads. Dataset 1 of 3: single-cell reporter measurements.',\n",
       "  'IGF-I-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of an AKT inhibitor. Dataset 2 of 3: single-cell reporter pulsing metrics.',\n",
       "  'EGF-induced FoxO3a nuclear-cytoplasmic pulsing dynamics in mammary epithelial cell lines following treatment with increasing doses of a MEK inhibitor. Dataset 1 of 3: single-cell reporter measurements.',\n",
       "  'Metrics other than potency reveal systematic variation in responses to cancer drugs',\n",
       "  'Cell signaling response to cytokines measured by high throughput microscopy',\n",
       "  'Basal profile of receptor tyrosine kinase signaling network measured by ELISA',\n",
       "  'Growth rate-corrected (GR) dose-response metrics across a panel of 71 breast cancer cell lines treated with a library of small molecule and antibody perturbagens. Dataset 2 of 4: Calculated dose-response metrics. ',\n",
       "  'MEMA PC3 cell line dataset',\n",
       "  'Domain-Specific Language domain analysis and evaluation: a systematic literature review',\n",
       "  'diMN (Exp 3) - ALS and Control (unaffected) diMN cell lines differentiated from iPS cell lines using a short and direct differeintiation protocol - RNA-seq',\n",
       "  'Drug-drug synergy results from titration experiments',\n",
       "  'cyc D activation dynamics in MCF10A cells treated with the four core compounds.',\n",
       "  'iMN (Exp 2) - ALS, SMA and Control (unaffected) iMN cell lines differentiated from iPS cell lines using a long differentiation protocol - RNA-seq',\n",
       "  'Tyrosine Kinase Inhibitor-induced cardiotoxicity in hiPSC derived cardiomyocytes - Proteomics',\n",
       "  'Growth rate-corrected (GR) dose-response metrics across a panel of 71 breast cancer cell lines treated with a library of small molecule and antibody perturbagens. Dataset 1 of 4: Relative cell counts and normalized growth rate inhibition values across technical replicates.',\n",
       "  'Phosphorylation state and protein levels measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with combinations of two compounds (immunofluorescence dataset 1 of 4)',\n",
       "  'Phosphorylation state and protein levels measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with combinations of two compounds (immunofluorescence dataset 4 of 4)',\n",
       "  'Data for \"A Random Walk\" manuscript',\n",
       "  'DCC-2036 KiNativ -- single dose experiment',\n",
       "  'GCP aggregated data - CRISPR perturbations',\n",
       "  'GCP data - Chemical perturbations (MCF10A Common Project)',\n",
       "  'Live-cell imaging of cell-to-cell variability in caspase-8 FRET reporter activity and cell fate trajectory modeling',\n",
       "  'Sustainable data and metadata management at the BD2K-LINCS Data Coordination and Integration Center.',\n",
       "  'Multiplexed imaging of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with EGF',\n",
       "  'WZ3105 KiNativ -- single dose experiment',\n",
       "  'PCI-32765 KiNativ -- single dose experiment',\n",
       "  'Assessing user stories: the influence of template differences and gender-related problem-solving styles - Supplemental material',\n",
       "  'Deep Learning Benchmarks on L1000 Gene Expression Data.',\n",
       "  'Viability and apoptosis measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with combinations of three compounds (viability/apoptosis dataset 2 of 2)',\n",
       "  'Simulations of DPPC/DOPC/CHOL 2/1/1 mixture using the model by Carpenter et al., Part 3',\n",
       "  'Simulations of DPPC/DOPC/CHOL 3/3/2 mixture using the model by Carpenter et al., Part 3',\n",
       "  'Viability and apoptosis measured by imaging in BRAF(V600E/D) melanoma cell lines following treatment with combinations of two compounds (viability/apoptosis dataset 1 of 2)',\n",
       "  'Aggregated ChEMBL bioactivity data (v20)',\n",
       "  'Highly-multiplexed imaging by Cyclic Immunofluorescence (CycIF) of protein levels and protein phosphorylation states in the MCF 10A breast cell line treated with EGF or four kinase inhibitors',\n",
       "  'PD173074 KINOMEscan',\n",
       "  'XMD8-92 KiNativ -- single dose experiment',\n",
       "  'QL-X-138 KiNativ -- single dose experiment',\n",
       "  'XMD-12 KiNativ -- single dose experiment',\n",
       "  'XMD16-144 KiNativ -- single dose experiment',\n",
       "  'JWE-035 KiNativ -- single dose experiment',\n",
       "  'XMD11-50 KiNativ -- single dose experiment',\n",
       "  'XMD8-85 KiNativ -- single dose experiment',\n",
       "  'NPK76-II-72-1 KiNativ -- single dose experiment',\n",
       "  'P100 aggregated data - CRISPR perturbations',\n",
       "  'PLX-4720 KiNativ -- dose response experiment',\n",
       "  'JNK-IN-8 KiNativ -- single dose experiment',\n",
       "  'P100 data - Chemical perturbations (MCF10A Common Project)',\n",
       "  'CI-1033 KINOMEscan',\n",
       "  'WZ-1-84 KINOMEscan',\n",
       "  'Cell Painting morphological profiling assay',\n",
       "  'L1000 Consensus Signatures - shRNA perturbagens',\n",
       "  'WZ-4-145 KINOMEscan',\n",
       "  'WZ-7043 KINOMEscan',\n",
       "  'R406 KINOMEscan',\n",
       "  'A443654 KINOMEscan',\n",
       "  'AZ-628 KiNativ -- dose response experiment',\n",
       "  'Torin1 KiNativ -- single dose experiment',\n",
       "  'Sorafenib KiNativ -- dose response experiment',\n",
       "  'QL-XII-47 KiNativ -- single dose experiment',\n",
       "  'Torin2 KiNativ -- single dose experiment',\n",
       "  'NVP-BHG712 KiNativ -- multiple dose experiment',\n",
       "  'CH5424802 KiNativ -- single dose experiment',\n",
       "  '5z-7-oxozeaenol KiNativ -- single dose experiment',\n",
       "  'L1000 Consensus Signatures - CRISPR perturbagens',\n",
       "  'XMD11-85h KINOMEscan',\n",
       "  'XMD8-85 KINOMEscan',\n",
       "  'NPK76-II-72-1 KINOMEscan',\n",
       "  'JW-7-24-1 KINOMEscan',\n",
       "  'BMS-536924 KINOMEscan',\n",
       "  'BI-2536 KINOMEscan',\n",
       "  'ALW-II-38-3 KINOMEscan',\n",
       "  'ALW-II-49-7 KINOMEscan',\n",
       "  'A770041 KINOMEscan',\n",
       "  'PLX4032 KiNativ -- dose response experiment',\n",
       "  'Random-primed mRNA-sequencing transcriptomic dataset for 70 primary human cardiomyocyte cell samples',\n",
       "  \"3'-DGE mRNA-sequencing transcriptomic dataset for 914 primary human cardiomyocyte cell samples\",\n",
       "  'iMN (Exp 2) - iMN long Differentiation protocol - Imaging',\n",
       "  'PI3K inhibitors protect against glucocorticoid-induced skin atrophy.',\n",
       "  'FOXM1 functions collaboratively with PLAU to promote gastric cancer progression.',\n",
       "  'RO-3306 KINOMEscan',\n",
       "  'GSK429286A KINOMEscan',\n",
       "  'HKI-272 KINOMEscan',\n",
       "  'PLX4032 KINOMEscan',\n",
       "  'SB590885 KINOMEscan',\n",
       "  'CP724714 KINOMEscan',\n",
       "  'AZD-6482 KINOMEscan',\n",
       "  'XMD11-50 KINOMEscan',\n",
       "  'PLX-4720 KINOMEscan',\n",
       "  'PF02341066 KINOMEscan',\n",
       "  'Viability and apoptosis in BRAF(V600E/D) melanoma cell lines monitored by imaging',\n",
       "  'PF-3758309 KiNativ -- single dose experiment',\n",
       "  'BS-181 KINOMEscan',\n",
       "  'QL-XI-92 KINOMEscan',\n",
       "  'ZG-10 KINOMEscan',\n",
       "  'KU63794 KINOMEscan',\n",
       "  'Torin1 KINOMEscan',\n",
       "  'BMS345541 KINOMEscan',\n",
       "  'JWE-035 KINOMEscan',\n",
       "  'WZ3105 KINOMEscan',\n",
       "  '(R)-Roscovitine KINOMEscan',\n",
       "  'HG-6-64-1 KINOMEscan',\n",
       "  'ZM-447439 KINOMEscan',\n",
       "  'Datasets Used in Evaluation Of RNAi And CRISPR Technologies By Large Scale Gene Expression Profiling In The Connectivity Map',\n",
       "  'JNK-IN-5A KINOMEscan',\n",
       "  'CYT387 KINOMEscan',\n",
       "  'PHA-767491 KINOMEscan',\n",
       "  'GW843682 KINOMEscan',\n",
       "  'AZD4547 KINOMEscan',\n",
       "  'JNK-IN-8 KINOMEscan',\n",
       "  'OTSSP167 KINOMEscan',\n",
       "  'BX-912 KINOMEscan',\n",
       "  'AZ-628 KINOMEscan',\n",
       "  'XMD13-2 KINOMEscan',\n",
       "  'GSK2126458 KINOMEscan',\n",
       "  'AZD 5438 KINOMEscan',\n",
       "  'HG-14-8-02 KINOMEscan',\n",
       "  'KIN001-220 KINOMEscan',\n",
       "  'CP466722 KINOMEscan',\n",
       "  'PD0332991 KINOMEscan',\n",
       "  'XMD15-27 KINOMEscan',\n",
       "  'JNK-9L KINOMEscan',\n",
       "  'WYE-125132 KINOMEscan',\n",
       "  'NU7441 KINOMEscan',\n",
       "  'XMD14-99 KINOMEscan',\n",
       "  'Dinaciclib KINOMEscan',\n",
       "  'PD0325901 KINOMEscan',\n",
       "  'iMN (Exp 2) - ALS, SMA and Control (unaffected) iMN cell lines differentiated from iPS cell lines using a long differentiation protocol - Proteomics',\n",
       "  'Phosphorylation state and protein levels measured in BRAF(V600E/D) melanoma cell lines monitored by imaging',\n",
       "  'Average variable importance in the projection (VIP) scores from the PLSR models analyzing the covariation of molecular signals with cell viability and apoptosis fraction in BRAF(V600E/D) melanoma cell lines',\n",
       "  'AS-252424 KINOMEscan',\n",
       "  'ZSTK474 KINOMEscan',\n",
       "  'TGX221 KINOMEscan',\n",
       "  'Aggregated Tox21 bioactivity data',\n",
       "  'MPS-1-IN-1 KINOMEscan',\n",
       "  'WH-4-025 KINOMEscan',\n",
       "  'Proteomic cellular signatures of kinase inhibitor-induced cardiotoxicity',\n",
       "  'ERK and Akt activation dynamics in MCF10A cells treated with the four core compounds.',\n",
       "  'Discovering the anti-cancer potential of non-oncology drugs by systematic PRISM profiling',\n",
       "  'iPSC (Exp 1) - ALS, SMA and Control (unaffected) subject-derived iPSC lines - RNA-seq',\n",
       "  'BEZ235 KINOMEscan',\n",
       "  'A Next Generation Connectivity Map: L1000 Platform and the First 1,000,000 Profiles.',\n",
       "  'Measuring Cancer Drug Sensitivity and Resistance in Cultured Cells.',\n",
       "  'On transformative adaptive activation functions in neural networks for gene expression inference.',\n",
       "  'iPSC (Exp 1) - ALS, SMA and Control (unaffected) subject-derived iPSC lines- ATAC-seq',\n",
       "  'diMN (Exp 3) - ALS and Control (unaffected) diMN cell lines differentiated from iPS cell lines using a short and direct differeintiation protocol - Proteomics',\n",
       "  'iPSC (Exp 1) - ALS, SMA and Control (unaffected) subject-derived iPSC lines - SWATH-MS',\n",
       "  'GW-5074 KiNativ -- dose response experiment',\n",
       "  'SB590885 KiNativ -- dose response experiment',\n",
       "  'OTSSP167 KiNativ -- single dose experiment',\n",
       "  'EGF and Insulin combination dose and dynamic response of ERK, AKT, and mTOR signaling and the cell cycle (Cyclin D1) in serum- and growth-factor starved MCF10A cells.',\n",
       "  'XMD11-85h KiNativ -- single dose experiment',\n",
       "  '(s)-CR8 KINOMEscan',\n",
       "  'Broad Institute L1000 Connectivity Map and RNAseq gene expression profiles on tissue samples from the GTEx project',\n",
       "  'SP600125 KINOMEscan',\n",
       "  'HG-5-88-01 KINOMEscan',\n",
       "  'CGP60474 KINOMEscan',\n",
       "  'KIN001-127 KINOMEscan',\n",
       "  'GDC-0941 KINOMEscan',\n",
       "  'XMD8-92 KINOMEscan',\n",
       "  'BGJ398 KINOMEscan',\n",
       "  'Ibrutinib KINOMEscan',\n",
       "  'Baricitinib KINOMEscan',\n",
       "  'Abemaciclib KINOMEscan',\n",
       "  'NVP-TAE684 KINOMEscan',\n",
       "  'HG-5-113-01 KINOMEscan',\n",
       "  'WZ4002 KINOMEscan',\n",
       "  'Improving the Developer Experience with a Low-Code ProcessModelling Language: Companion site',\n",
       "  'Ibrutinib KiNativ -- single dose experiment',\n",
       "  'diMN (Exp 3) - ALS and Control (unaffected) diMN cell lines differentiated from iPS cell lines using a short and direct differeintiation protocol - ATAC-seq',\n",
       "  'iMN (Exp 2) - ALS, SMA and Control (unaffected) iMN cell lines differentiated from iPS cell lines using a long differentiation protocol - ATAC-seq',\n",
       "  'Imatinib KINOMEscan',\n",
       "  'R406 KINOMEscan-2',\n",
       "  'Pazopanib KINOMEscan',\n",
       "  'TG-101348 KINOMEscan',\n",
       "  'RAF-265 KINOMEscan',\n",
       "  'BIRB-796 KINOMEscan',\n",
       "  'Dasatinib KINOMEscan',\n",
       "  'HKI-272 KINOMEscan-2',\n",
       "  'LY-317615 KINOMEscan',\n",
       "  'Erlotinib KINOMEscan',\n",
       "  'GSK-461364A KINOMEscan',\n",
       "  'Ki-20227 KINOMEscan',\n",
       "  'Sorafenib KINOMEscan-2',\n",
       "  'Vandetanib KINOMEscan',\n",
       "  'GDC-0941 KINOMEscan-2',\n",
       "  'CEP-701 KINOMEscan',\n",
       "  'New ways to boost molecular dynamics simulations.',\n",
       "  'HG-6-64-01 KiNativ -- single dose experiment',\n",
       "  'PLX-4720 KINOMEscan-2',\n",
       "  'TAE-684 KINOMEscan',\n",
       "  'AZD-1152HQPA KINOMEscan-2',\n",
       "  'GSK-690693 KINOMEscan',\n",
       "  'Sunitinib KINOMEscan',\n",
       "  'GSK-1363089 KINOMEscan',\n",
       "  'SKI-606 KINOMEscan',\n",
       "  'MLN-8054 KINOMEscan',\n",
       "  'XMD-12 KINOMEscan',\n",
       "  'PHA-793887 KINOMEscan',\n",
       "  'HG-9-91-01 KINOMEscan',\n",
       "  'CGP74514A KINOMEscan',\n",
       "  'Sorafenib KINOMEscan',\n",
       "  'AZD7762 KINOMEscan',\n",
       "  'WH-4-023 KINOMEscan',\n",
       "  'TAK-715 KINOMEscan',\n",
       "  'Sigma A6730 KINOMEscan',\n",
       "  'AG1478 KINOMEscan',\n",
       "  'PHA-665752 KINOMEscan',\n",
       "  'AB-1010 KINOMEscan',\n",
       "  'VX-745 KINOMEscan',\n",
       "  'PKC-412 KINOMEscan',\n",
       "  'TKI-258 KINOMEscan',\n",
       "  'BYL719 KINOMEscan',\n",
       "  'BMS-345541 KINOMEscan',\n",
       "  'KW-2449 KINOMEscan',\n",
       "  'AC220 KINOMEscan',\n",
       "  'BIBW-2992 KINOMEscan',\n",
       "  'VX-680 KINOMEscan',\n",
       "  'Nilotinib KINOMEscan',\n",
       "  'SGX-523 KINOMEscan',\n",
       "  'Buparlisib KINOMEscan',\n",
       "  'PI-103 KINOMEscan',\n",
       "  'BI-2536 KINOMEscan-2',\n",
       "  'BMS-540215 KINOMEscan',\n",
       "  'AZD-6244 KINOMEscan',\n",
       "  'SB-203580 KINOMEscan',\n",
       "  'Gefitinib KINOMEscan',\n",
       "  'CG-930 KINOMEscan',\n",
       "  'Dabrafenib KINOMEscan',\n",
       "  'HG-14-10-04 KINOMEscan',\n",
       "  'GW-5074 KINOMEscan',\n",
       "  'GSK461364 KINOMEscan',\n",
       "  'MLN8054 KINOMEscan',\n",
       "  '5z-7-oxozeaenol KINOMEscan',\n",
       "  'AZD1152-HQPA KINOMEscan',\n",
       "  'QL-X-138 KINOMEscan',\n",
       "  'Torin2 KINOMEscan',\n",
       "  'XMD16-144 KINOMEscan',\n",
       "  'GSK1059615 KINOMEscan',\n",
       "  'Lapatinib KINOMEscan',\n",
       "  'Drug/Cell-line Browser: interactive canvas visualization of cancer drug/cell-line viability assay datasets.',\n",
       "  'Holistic optimization of an RNA-seq workflow for multi-threaded environments.',\n",
       "  'Connectivity Analyses of Bioenergetic Changes in Schizophrenia: Identification of Novel Treatments.',\n",
       "  'Transcriptomic profiling of human hippocampal progenitor cells treated with antidepressants and its application in drug repositioning.',\n",
       "  'Computational Drug Repositioning for Gastric Cancer using Reversal Gene Expression Profiles.',\n",
       "  'Centralized scientific communities are less likely to generate replicable results.',\n",
       "  'In Vitro and In Vivo Pipeline for Validation of Disease-Modifying Effects of Systems Biology-Derived Network Treatments for Traumatic Brain Injury-Lessons Learned.',\n",
       "  'Deep Learning Applications for Predicting Pharmacological Properties of Drugs and Drug Repurposing Using Transcriptomic Data.',\n",
       "  'Transcriptional profiling of LS1034 cells treated with tepoxalin',\n",
       "  'CGP082996 KINOMEscan',\n",
       "  'Molecular dynamics simulation trajectory of a fully hydrated DMPC lipid bilayer',\n",
       "  'Analysis of Post-TBI Gene Expression Signature Reveals Tubulins, NFE2L2, NFkB, CD44, and S100A4 as Treatment Targets for Traumatic Brain Injury',\n",
       "  'Genetic and transcriptional variation alters cancer cell line drug response',\n",
       "  'Genetic and transcriptional variation alters cancer cell line drug response [MCF7 strain L]',\n",
       "  'MGH (CMT) Growth Inhibition Assay Protocol (9 compound doses: 10.24 uM to 0.04 uM) -- DNA Staining',\n",
       "  'Inhibition of the neuromuscular acetylcholine receptor with atracurium activates FOXO/DAF-16-induced longevity (RNA-seq)',\n",
       "  'Systematically studying kinase inhibitor induced signaling network signatures by integrating both therapeutic and side effects.',\n",
       "  'BMS-387032 KINOMEscan',\n",
       "  'BIBF-1120 KINOMEscan',\n",
       "  'GSK-1838705A KINOMEscan',\n",
       "  'CI-1040 KINOMEscan',\n",
       "  'CI-1033 KINOMEscan-2',\n",
       "  'Transcriptomics-Based Drug Repurposing Approach Identifies Novel Drugs against Sorafenib-Resistant Hepatocellular Carcinoma.',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 6/94 mol% DMTAP/DMPC in 1.0 M NaCl',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 50/50 mol% DMTAP/DMPC in 1.0 M NaCl',\n",
       "  'Transcriptomics-based drug repurposing approach identifies novel drugs against sorafenib-resistant hepatocellular carcinoma',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 75/25 mol% DMTAP/DMPC in 1.0 M NaCl',\n",
       "  'INCB018424 KINOMEscan',\n",
       "  'GDC-0879 KINOMEscan',\n",
       "  'ABT-869 KINOMEscan',\n",
       "  'CP-690550 KINOMEscan',\n",
       "  'AT-7519 KINOMEscan',\n",
       "  'Flavopiridol KINOMEscan',\n",
       "  'Crizotinib KINOMEscan-2',\n",
       "  'MRT67307 KINOMEscan',\n",
       "  'Palbociclib KINOMEscan',\n",
       "  ' QL-XII-47 KINOMEscan',\n",
       "  'Transcriptional profiling of SF295 cells following MTF1 knockout by CRISPR/Cas9',\n",
       "  'AMG-706 KINOMEscan',\n",
       "  'Lapatinib KINOMEscan-2',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 75/25 mol% DMTAP/DMPC',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 50/50 mol% DMTAP/DMPC in 0.1 M NaCl',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 75/25 mol% DMTAP/DMPC in 0.1 M NaCl',\n",
       "  'Gene expression data of four cancer cell lines (A375, A549, MCF7, and PC3) treated with KR-37524',\n",
       "  'STL427944 suppresses FOXM1 via nuclear export and subsequent autophagic degradation',\n",
       "  'Analysis of Post-TBI Gene Expression Signature Reveals Tubulins, NFE2L2, NFkB, CD44, and S100A4 as Treatment Targets for Traumatic Brain Injury',\n",
       "  'Modulating membrane fluidity corrects Batten disease phenotypes in vitro and in vivo.',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 50/50 mol% DMTAP/DMPC in 0.5 M NaCl',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 6/94 mol% DMTAP/DMPC',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 75/25 mol% DMTAP/DMPC in 0.5 M NaCl',\n",
       "  'Molecular dynamics simulation trajectory of a cationic lipid bilayer: 50/50 mol% DMTAP/DMPC',\n",
       "  'SigMat: a classification scheme for gene signature matching.',\n",
       "  'GDC-0941 KINOMEscan-3',\n",
       "  'Taselisib KINOMEscan',\n",
       "  'Effects of mannose-binding lectin on pulmonary gene expression and innate immune inflammatory response to ozone.',\n",
       "  'Drug Gene Budger (DGB): an application for ranking drugs to modulate a specific gene based on transcriptomic signatures.',\n",
       "  'Screening of Drug Repositioning Candidates for Castration Resistant Prostate Cancer.',\n",
       "  'Translating in vitro CFTR rescue into small molecule correctors for cystic fibrosis using the Library of Integrated Network-based Cellular Signatures drug discovery platform',\n",
       "  'Transcriptomic metaanalyses of autistic brains reveals shared gene expression and biological pathway abnormalities with cancer.',\n",
       "  'LINCS-NeuroLINCS',\n",
       "  'Genetic and transcriptional variation alters cancer cell line drug response',\n",
       "  'Inhibition of the neuromuscular acetylcholine receptor with atracurium activates FOXO/DAF-16-induced longevity (RNA-seq)',\n",
       "  'Regulation of TROP2 expression in breast cancer',\n",
       "  'Multi-omics Analysis reveals novel therapeutic vulnerabilities in lung cancer',\n",
       "  'Encoding growth factor identity in the temporal dynamics of FoxO3 under the combinatorial control of ERK and AKT Kinases',\n",
       "  'Improving the Usability of a MAS DSML',\n",
       "  'Homo sapiens',\n",
       "  'TRANSCRIPT drug repurposing dataset',\n",
       "  'Homo sapiens',\n",
       "  'STL427944 suppresses FOXM1 via nuclear export and subsequent autophagic degradation',\n",
       "  'Data from: Gut microbiome critically impacts PCB-induced changes in metabolic fingerprints and the hepatic transcriptome in mice'],\n",
       " ['<p>LINCS - PDRGC - preprocessed</p>',\n",
       "  'The Library of Integrated Network-based Cellular Signatures (LINCS) is a multi-institutional, collaborative project initiated by the National Institutes of Health whose goal is to identify and categorize molecular signatures that occur when cells are exposed to agents that perturb their normal function. Understanding how perturbations affect the behavior of cells is important because it can provide insights into the causes of disease and help researchers develop new therapeutic strategies.',\n",
       "  'For the Library of Integrated Network-based Cellular Signatures (LINCS) project many gene expression signatures using the L1000 technology have been produced. The L1000 technology is a cost-effective method to profile gene expression in large scale. LINCS Canvas Browser (LCB) is an interactive HTML5 web-based software application that facilitates querying, browsing and interrogating many of the currently available LINCS L1000 data. LCB implements two compacted layered canvases, one to visualize clustered L1000 expression data, and the other to display enrichment analysis results using 30 different gene set libraries. Clicking on an experimental condition highlights gene-sets enriched for the differentially expressed genes from the selected experiment. A search interface allows users to input gene lists and query them against over 100 000 conditions to find the top matching experiments. The tool integrates many resources for an unprecedented potential for new discoveries in systems biology and systems pharmacology. The LCB application is available at http://www.maayanlab.net/LINCS/LCB. Customized versions will be made part of the http://lincscloud.org and http://lincs.hms.harvard.edu websites.',\n",
       "  '<p>GRNs calculated using ANANSE presumably active in LINCS MCF10A cell cultures perturbated with various TFs.</p>',\n",
       "  'The Library of Integrated Cellular Signatures (LINCS) project provides comprehensive transcriptome profiling of human cell lines before and after chemical and genetic perturbations. Its L1000 platform utilizes 978 landmark genes to infer the transcript levels of 14,292 genes computationally. Here we conducted the L1000 data quality control analysis by using MCF7, PC3, and A375 cell lines as representative examples. Before perturbations, a promising 80% correlation in transcriptome was observed between L1000- and Affymetrix HU133A-platforms. After library-based shRNA perturbations, a moderate 30% of differentially expressed genes overlapped between any two selected controls viral vectors using the L1000 platform. The mitogen-activated protein kinase, vascular endothelial growth factor, and T-cell receptor pathways were identified as the most significantly shared pathways between chemical and genetic perturbations in cancer cells. In conclusion, L1000 platform is reliable in assessing transcriptome before perturbation. Its response to perturbagens needs to be interpreted with caution. A quality control analysis pipeline of L1000 is recommended before addressing biological questions.',\n",
       "  'The Library of Integrated Network-based Cellular Signatures (LINCS) L1000 big data provide gene expression profiles induced by over 10?000 compounds, shRNAs, and kinase inhibitors using the L1000 platform. We developed csNMF, a systematic compound signature discovery pipeline covering from raw L1000 data processing to drug screening and mechanism generation. The csNMF pipeline demonstrated better performance than the original L1000 pipeline. The discovered compound signatures of breast cancer were consistent with the LINCS KINOMEscan data and were clinically relevant. The csNMF pipeline provided a novel and complete tool to expedite signature-based drug discovery leveraging the LINCS L1000 resources.',\n",
       "  'Transcriptional profiles of cultured human cancer cell lines treated with small molecules: Expression of 978 representative genes is measured in 39 cell lines perturbed by 13 thousands of small molecules.',\n",
       "  'Transcriptional profiles of cultured human breast cancer cell lines treated with small molecules: 5 cell lines are treated with 109 small molecules and the expression of 978 representative genes is measured, as a part of the LINCS Joint Project.',\n",
       "  'Transcriptional profiles of cultured human breast and prostate cancer cell lines treated with small molecules: 6 cancer cell lines are treated with 6 small molecules, as a part of the LINCS Trans-Center Project. The expression level for 978 representative genes is measured.',\n",
       "  'The Library of Integrated Cellular Signatures (LINCS) is an NIH program which funds the generation of perturbational profiles across multiple cell and perturbation types, as well as read-outs, at a massive scale. The LINCS Center for Transcriptomics at the Broad Institute uses the L1000 high-throughput gene-expression assay to build a Connectivity Map which seeks to enable the discovery of functional connections between drugs, genes and diseases through analysis of patterns induced by common gene-expression changes.The platform is GPL20573: Broad Institute Human L1000 epsilonhttp://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL20573For questions or assistance with this dataset, please email the CMap support team at: clue@broadinstitute.org\\nLINCS aims to enable a functional understanding of biology by cataloging changes in gene expression and other cellular processes that occur when cells are exposed to a variety of perturbing agents. The Broad Institute LINCS Center for Transcriptomics contributes to this collaborative effort by application of the Connectivity Map concept. In brief, the study design involves the generation of a compendium of transcriptional expression data from cultured human cells treated with small-molecule and genetic loss/gain of function perturbagens. These measurements are made using the L1000 high-throughput gene-expression assay that enables data generation at an unprecedented scale.  The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control)',\n",
       "  'MOTIVATION:Adverse drug reactions (ADRs) are a central consideration during drug development. Here we present a machine learning classifier to prioritize ADRs for approved drugs and pre-clinical small-molecule compounds by combining chemical structure (CS) and gene expression (GE) features. The GE data is from the Library of Integrated Network-based Cellular Signatures (LINCS) L1000 dataset that measured changes in GE before and after treatment of human cells with over 20 000 small-molecule compounds including most of the FDA-approved drugs. Using various benchmarking methods, we show that the integration of GE data with the CS of the drugs can significantly improve the predictability of ADRs. Moreover, transforming GE features to enrichment vectors of biological terms further improves the predictive capability of the classifiers. The most predictive biological-term features can assist in understanding the drug mechanisms of action. Finally, we applied the classifier to all ?>20 000 small-molecules profiled, and developed a web portal for browsing and searching predictive small-molecule/ADR connections. AVAILABILITY AND IMPLEMENTATION:The interface for the adverse event predictions for the ?>20 000 LINCS compounds is available at http://maayanlab.net/SEP-L1000/ CONTACT: avi.maayan@mssm.edu SUPPLEMENTARY INFORMATION:Supplementary data are available at Bioinformatics online.',\n",
       "  '<p>Summary</p>\\n\\n<p>This release contains finalized profile and metadata information for the CellProfiler-derived features from the LINCS Cell Painting dataset.</p>\\n\\n<p>We assayed two batches of data and applied an image analysis pipeline using CellProfiler to segment cells and extract morphology features. Using the CellProfiler output, we applied an image-based profiling pipeline to process the morphology profiles into an analysis-ready form.</p>\\n\\n<p>In this repository, we provide the full image-based profiling pipeline we used to process the data, as well as most of the intermediate data.</p>\\n\\n<p>Data counts</p>\\n\\n<ul>\\n\\t<li>Two batches (&quot;2016_04_01_a549_48hr_batch1&quot; and &quot;2017_12_05_Batch2&quot;)</li>\\n\\t<li>159,717,488 unique cells total</li>\\n</ul>\\n\\n<p>2016_04_01_a549_48hr_batch1</p>\\n\\n<ul>\\n\\t<li>52,223 profiles (batch corrected)</li>\\n\\t<li>10,752 unique perturbations</li>\\n\\t<li>1,571 unique compounds</li>\\n\\t<li>1 time point (48 hour)</li>\\n\\t<li>7 doses</li>\\n\\t<li>1 cell line (A549)</li>\\n\\t<li>110,012,425 unique cells</li>\\n</ul>\\n\\n<p>2017_12_05_Batch2</p>\\n\\n<ul>\\n\\t<li>51,447 profiles (batch corrected)</li>\\n\\t<li>10,368 unique perturbations</li>\\n\\t<li>349 unique compounds</li>\\n\\t<li>3 time points (6 hour, 24 hour, 48 hour)</li>\\n\\t<li>6 doses</li>\\n\\t<li>3 cell lines (A549, MCF7, U2OS)</li>\\n\\t<li>49,705,063 unique cells</li>\\n</ul>',\n",
       "  \"The Library of Integrated Cellular Signatures (LINCS) is an NIH program which funds the generation of perturbational profiles across multiple cell and perturbation types, as well as read-outs, at a massive scale. The LINCS Center for Transcriptomics at the Broad Institute uses the L1000 high-throughput gene-expression assay to build a Connectivity Map which seeks to enable the discovery of functional connections between drugs, genes and diseases through analysis of patterns induced by common gene-expression changes. These files represent L1000 data generated during the LINCS Pilot Phase (2012-2015), as well as profiles generated for more specific purposes, such as assay development and validation projects or testing custom compounds or non-standard cell lines (not part of the core LINCS cell lines). Note: Related GEO projects include (a) Additional L1000 and RNA-Seq data used to validate the assay and improve the inference model, available at GSE92743 (b)  The LINCS “production phase” (also termed Phase II, 2015-2020) which is generating an additional cohort of L1000 data, available at GSE70138.The Platform is GPL20573: Broad Institute Human L1000 epsilonhttps://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL20573For questions or assistance with this dataset, please email the CMap support team at: clue@broadinstitute.org\\nLINCS aims to enable a functional understanding of biology by cataloging changes in gene expression and other cellular processes that occur when cells are exposed to a variety of perturbing agents. The Broad Institute LINCS Center for Transcriptomics contributes to this collaborative effort by application of the Connectivity Map concept. In brief, the study design involves the generation of a compendium of transcriptional expression data from cultured human cells treated with small-molecule and genetic loss/gain of function perturbagens. These measurements are made using the L1000 high-throughput gene-expression assay that enables data generation at an unprecedented scale.  The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control). Level 5 (SIG) consists of the replicates, usually 3 per treatment, aggregated into a single differential expression vector derived from the weighted averages of the individual replicates.Please note that a description of the latest file, and a table listing the contents of the 'Broad_LINCS_auxiliary_datasets.tar.gz' file are updated in the following document;https://docs.google.com/document/d/1q2gciWRhVCAAnlvF2iRLuJ7whrGP6QjpsCMq1yWz7dU/edit#heading=h.l6bq0r1aih50\",\n",
       "  'L1000 is a bead-based, high-throughput gene expression assay in which cultured cells are treated with various chemical and genetic perturbations and the corresponding transcriptional responses are measured.The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control)',\n",
       "  'L1000 is a bead-based, high-throughput gene expression assay in which cultured cells are treated with various chemical and genetic perturbations and the corresponding transcriptional responses are measured.The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control)',\n",
       "  'L1000 is a bead-based, high-throughput gene expression assay in which cultured cells are treated with various chemical and genetic perturbations and the corresponding transcriptional responses are measured.The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control)',\n",
       "  'Gene expression profiling using transcriptional drug perturbations are useful for many biomedical discovery studies including drug repurposing and elucidation of drug mechanisms (MoA) and many other pharmacogenomic applications. However, limited data availability across cell types has severely hindered our capacity to progress in these areas. To fill this gap, recently, the LINCS program generated almost 1.3 million profiles for over 40,000 drug and genetic perturbations for over 70 different human cell types, including meta information about the experimental conditions and cell lines. Unfortunately, Big Data like the ones generated from the ongoing LINCS program do not enable easy insights from the data but possess considerable challenges toward their analysis. In this paper, we address some of these challenges. Specifically, first, we study the gene expression signature profiles from all cell lines and their perturbagents in order to obtain insights in the distributional characteristics of available conditions. Second, we investigate the differential expression of genes for all cell lines obtaining an understanding of condition dependent differential expression manifesting the biological complexity of perturbagents. As a result, our analysis helps the experimental design of follow-up studies, e.g., by selecting appropriate cell lines.',\n",
       "  'The Library of Integrated Cellular Signatures (LINCS) is an NIH program which funds the generation of perturbational profiles across multiple cell and perturbation types, as well as read-outs, at a massive scale. The LINCS PCCSE uses two high-throughput liquid chromatography-mass spectrometry (LCMS) assays to study the proteomic changes induced by drug and genetic perturbations. P100 monitors ~100 phosphorylated peptides from cellular proteins that serve as a reduced representation of the phosphoproteome. GCP monitors ~60 modified peptides from histones (e.g., methylated, acetylated, and combinations thereof) encompassing nearly every well-studied post-translational modification on the core nucleosomal histone proteins. Closely complementing these assays is the L1000 high-throughput gene-expression assay. The files available here contain P100, GCP, and L1000 data for 90 small-molecule perturbations in six cell lines (five cancer cell lines and a neurodevelopmental cell model). These data were generated under the auspices of the NIH LINCS Program (www.lincsproject.org). Note: Related GEO projects include a large corpus of additional L1000 data, available at GSE92742.The Platform for the L1000 data is GPL20573: Broad Institute Human L1000 epsilonhttp://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL20573For questions or assistance with this dataset, please email the Connectivity Map support team at: clue@broadinstitute.org\\nThese data represent three different read-outs (phosphosignaling, chromatin modifications, and transcriptional changes) of the same 90 small-molecule perturbations in six cell lines. The duration of treatment was 3 hours for P100, 24 hours for GCP, and 6 hours for L1000. These data are available at multiple levels of processing: level 1 is fluorescence intensity (for L1000) or mass spectrometry extracted ion chromatogram traces (for P100, GCP); level 2 is gene expression or proteomic values without normalization; level 3 is normalized; and level 4 is differential (i.e., each sample is compared to all other samples on a plate).',\n",
       "  'Sepsis is one of the most common clinical syndromes that causes death and disability. Although many studies have developed drugs for sepsis treatment, none have decreased the mortality rate. The aim of this study was to identify a novel treatment option for sepsis using the library of integrated network-based cellular signatures (LINCS) L1000 perturbation dataset based on an in vitro and in vivo sepsis model. Sepsis-related microarray studies of early-stage inflammatory processes in patients and innate immune cells were collected from the Gene Expression Omnibus (GEO) data repository and used for candidate drug selection based on the LINCS L1000 perturbation dataset. The anti-inflammatory effects of the selected candidate drugs were analyzed using activated macrophage cell lines. CGP-60474, an inhibitor of cyclin-dependent kinase, was the most potent drug. It alleviated tumor necrosis factor-? (TNF-?) and interleukin-6 (IL-6) in activated macrophages by downregulating the NF-?B activity, and it reduced the mortality rate in LPS induced endotoxemia mice. This study shows that CGP-60474 could be a potential therapeutic candidate to attenuate the endotoxemic process. Additionally, the virtual screening strategy using the LINCS L1000 perturbation dataset could be a cost and time effective tool in the early stages of drug development.',\n",
       "  'BACKGROUND:LINCS L1000 is a high-throughput technology that allows gene expression measurement in a large number of assays. However, to fit the measurements of ~1000 genes in the ~500 color channels of LINCS L1000, every two landmark genes are designed to share a single channel. Thus, a deconvolution step is required to infer the expression values of each gene. Any errors in this step can be propagated adversely to the downstream analyses. RESULTS:We presented a LINCS L1000 data peak calling R package l1kdeconv based on a new outlier detection method and an aggregate Gaussian mixture model (AGMM). Upon the remove of outliers and the borrowing information among similar samples, l1kdeconv showed more stable and better performance than methods commonly used in LINCS L1000 data deconvolution. CONCLUSIONS:Based on the benchmark using both simulated data and real data, the l1kdeconv package achieved more stable results than the commonly used LINCS L1000 data deconvolution methods.',\n",
       "  '<p>systems with&nbsp;lincs_iter = 1. To be written.</p>',\n",
       "  '<p>systems with&nbsp;lincs_iter = 2. To be written.</p>',\n",
       "  '<p>Additional simulations with lincs_order 10 &amp; 12</p>',\n",
       "  'BACKGROUND:Aiming to understand cellular responses to different perturbations, the NIH Common Fund Library of Integrated Network-based Cellular Signatures (LINCS) program involves many institutes and laboratories working on over a thousand cell lines. The community-based Cell Line Ontology (CLO) is selected as the default ontology for LINCS cell line representation and integration. RESULTS:CLO has consistently represented all 1097 LINCS cell lines and included information extracted from the LINCS Data Portal and ChEMBL. Using MCF 10A cell line cells as an example, we demonstrated how to ontologically model LINCS cellular signatures such as their non-tumorigenic epithelial cell type, three-dimensional growth, latrunculin-A-induced actin depolymerization and apoptosis, and cell line transfection. A CLO subset view of LINCS cell lines, named LINCS-CLOview, was generated to support systematic LINCS cell line analysis and queries. In summary, LINCS cell lines are currently associated with 43 cell types, 131 tissues and organs, and 121 cancer types. The LINCS-CLO view information can be queried using SPARQL scripts. CONCLUSIONS:CLO was used to support ontological representation, integration, and analysis of over a thousand LINCS cell line cells and their cellular responses.',\n",
       "  'L1000 is a bead-based, high-throughput gene expression assay in which cultured cells are treated with various chemical and genetic perturbations and the corresponding transcriptional responses are measured.The data are processed through a computational system, that converts raw fluorescence intensities into differential gene expression signatures. The data at each stage of the pre-processing are available: Level 1 (LXB) - raw, unprocessed flow cytometry data from Luminex scanners. One LXB file is generated for each well of a 384-well plate, and each file contains a fluorescence intensity value for every observed analyte in the well. Level 2 (GEX) - gene expression values per 1,000 genes after deconvolution from Luminex beads. Level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalized using invariant set scaling followed by quantile normalization. Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control)',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 12]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 11]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 3]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 9]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 4]',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366) [part 5]',\n",
       "  '-',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'No description',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'No description',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'No description',\n",
       "  '-',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, and by other HMS scientists in 2017 (datasets #20278-20286 ) and 2019 (datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, and by other HMS scientists in 2017 (datasets #20278-20286 ) and 2019 (datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, by the same HMS scientist in 2017 (datasets #20278-20286 ) and by other HMS scientists in 2019 (see datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, and by other HMS scientists in 2017 (datasets #20278-20286 ) and 2019 (datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'To generate measures of the sensitivities of three independent batches and one derivative of the MCF 10A cell line to 8 small molecule perturbagens, we treated cells with single drugs over a minimum 9-point dilution series using SQRT(10) dilutions centered around the GR<sub>50</sub> but not exceeding 10 µM and then measured cell number after three days of drug exposure.',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, by the same HMS scientist in 2017 (datasets #20278-20286 ) and by other HMS scientists in 2019 (see datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  '-',\n",
       "  'As part of the multi-center study on factors influencing the reproducibility of in vitro drug-response studies, we measured the sensitivities of the MCF 10A cell line to 8 small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure. Replicate experiments were performed at other centers, and by other HMS scientists in 2017 (datasets #20278-20286 ) and 2019 (datasets #20358-20363). These datasets as well as replicate datasets from other LINCS centers can also be accessed at <a href=\"https://www.synapse.org/#!Synapse:syn18456348/wiki/\" target = \"_blank\">https://www.synapse.org/#!Synapse:syn18456348/wiki/</a>.',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  '-',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  '-',\n",
       "  '-',\n",
       "  '-',\n",
       "  '<p>Here we include&nbsp;the supplementary material that may be used as a replication package for the quasi-experiment reported in the paper &quot;On the impact of different templates on creating and understanding user stories&quot;, submitted to RE 2021 (International Requirements Engineering Conference).</p>',\n",
       "  'No description',\n",
       "  'ReNcell-VM neuroprogenitors were differentiated by growth factor (EGF and FGF) removal for 15 days in culture, giving rise to dopaminergic neurons, astrocytes, and oligodendrocytes. Cells were collected at 10 time points and processed for phosphoproteomics by liquid chromatography mass spectrometry (LC/MS) using 10-plex tandem mass tag (TMT) labelling combined with phosphopeptide enrichment by titanium dioxide (TiO2), allowing deep quantification of proteome dynamics across ~3,500 phospho-proteins (~11,000 sites of phosphorylation).',\n",
       "  'No description',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  '-',\n",
       "  '-',\n",
       "  'We tested the binding affinity of 6 small molecule JAK inhibitors (Alvocidib, Nintedanib, Bosutinib, Tofacitinib, Baricitinib, and S-Ruxolitinib)  for JAK and other kinases (JAK1, JAK2, JAK3, TYK2 and DCLK3).',\n",
       "  '-',\n",
       "  'ReNcell-VM neuroprogenitors were differentiated by growth factor (EGF and FGF) removal for 15 days in culture, giving rise to dopaminergic neurons, astrocytes, and oligodendrocytes. Cells were collected at 10 time points and processed for total proteomics by liquid chromatography mass spectrometry (LC/MS) using 10-plex tandem mass tag (TMT) labelling to achieve deep quantification of proteome dynamics across ~8,900 proteins.',\n",
       "  'This micro-fluidics assay is performed using a microchip developed by the Yale LINCS U01 team. This microchip consists of a PDMS microwell array containing >5000 single-cell trapping chambers and a high-density antibody barcode array fabricated by Yale LINCS flow patterning method. The assay does not require external fluid handling equipment and can be performed anywhere in a typical cell biology lab.',\n",
       "  'This micro-fluidics assay is performed using a microchip developed by the Yale LINCS U01 team. This microchip consists of a PDMS microwell array containing >5000 single-cell trapping chambers and a high-density antibody barcode array fabricated by Yale LINCS flow patterning method. The assay does not require external fluid handling equipment and can be performed anywhere in a typical cell biology lab.',\n",
       "  'This micro-fluidics assay is performed using a microchip developed by the Yale LINCS U01 team. This microchip consists of a PDMS microwell array containing >5000 single-cell trapping chambers and a high-density antibody barcode array fabricated by Yale LINCS flow patterning method. The assay does not require external fluid handling equipment and can be performed anywhere in a typical cell biology lab.',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'Recovery and measurement of intracellular polar metabolites from drug-treated breast cancer cells.',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'ReNcell-VM neuroprogenitors were differentiated by growth factor (EGF and FGF) removal for 15 days in culture, giving rise to dopaminergic neurons, astrocytes, and oligodendrocytes. Drugs (Mevastatin and Kenpaullone) previously screened for their promoting effects on neural differentiation or DMSO control were added to the growth factor free maintenance media during the course of differentiation. Cells were collected at 10 time points and processed for total proteomics by liquid chromatography mass spectrometry (LC/MS) using 10-plex tandem mass tag (TMT) labelling to achieve deep quantification of proteome dynamics.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  '-',\n",
       "  '-',\n",
       "  '-',\n",
       "  'This analysis dataset presents partial-least-squares regression (PLSR) model loadings arising from analysis of HMS LINCS datasets #20217 and 20218 for covariation of molecular signals (measured by RPPA; dataset #20218) with cellular responses (relative viability and apoptotic fractions; datasets #20217) when pMEK and pERK signals are included.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'We measured the sensitivities of one non-malignant breast cell line and 20 breast cancer cell lines (six triple negative, six hormone receptor positive, five Her2 amplified, and three established from triple negative patient-derived xenografts) to 32 pre-clinical small molecule deubiquitinating enzyme (DUB) inhibitors. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series and then measured cell number and viability after three days of drug exposure.',\n",
       "  'ERK activity as measured by the EKAREV fluorescent reporter was assessed in the MCF 10A mammary epithelial cell line at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors.',\n",
       "  'We measured the sensitivities of two triple-negative breast cancer cell lines to nine uncharacterized dual mTOR-PIKK inhibitors, Torin1, Torin2, and the conventional mTOR kinase inhibitor AZD8055 (N=12 drugs). Cells were treated with each drug using a 9-point ½ log dilution series. Live cell counts were measured at the start and end of the 72 hour treatment period using fixed cell microscopy. Drug potency and efficacy were calculated using GR metrics. ',\n",
       "  'We measured the baseline mRNAseq profiles of two non-malignant breast cell lines and 33 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, four were Her2 amplified, and three were established from triple negative patient-derived xenografts. ',\n",
       "  'We measured the sensitivities of six TNBC cell lines to 23 clinical and pre-clinical small molecule inhibitors of kinases in the PI3K/AKT/mTOR signaling pathway and to one inhibitor of MEK. Cells were treated with each drug using a 9-point ½ log dilution series. Live cell counts were measured at the start and end of the 72 hour treatment period using fixed cell microscopy. Drug potency and efficacy were calculated using GR metrics.',\n",
       "  'ERK activity and the nuclear-cytoplasmic pulsing behavior of FoxO3a as measured by a dual fluorescent reporter (EKAREV-FoxO3aN400-mCherry) were assessed in the MCF 10A mammary epithelial cell line at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors.',\n",
       "  'This analysis dataset presents partial-least-squares regression (PLSR) model loadings arising from analysis of HMS LINCS datasets #20217 and 20218 for covariation of molecular signals (measured by RPPA; dataset #20218) with cellular responses (relative viability and apoptotic fractions; datasets #20217) when pMEK and pERK signals are excluded.',\n",
       "  'Dose-response data were collected for a panel of 73 breast cancer cell lines grown under standard or modified culture conditions and treated with 139 small molecule and antibody perturbagens. A subset of these data was previously described in Heiser et al. (2012) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/22003129\" target = \"_blank\">22003129</a>) and Daemen et al. (2013) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/24176112\" target = \"_blank\">24176112</a>). Here, the complete dataset was analyzed using the growth rate inhibition (GR) metrics described in Hafner et al. (2016) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/27135972\" target = \"_blank\">27135972</a>) to account for differences in growth rates across the cell lines and treatment conditions.',\n",
       "  'We measured the sensitivities of two triple-negative breast cancer cell lines to nine uncharacterized mTOR-PIKK inhibitors, Torin1, Torin2, and the conventional mTOR kinase inhibitor AZD8055 (N=12 drugs). Cells were treated with each drug using a 9-point ½ log dilution series. Live cell counts were measured at the start and end of the 72 hour treatment period using fixed cell microscopy. Drug potency and efficacy were calculated using GR metrics.',\n",
       "  'We measured the baseline phosphoproteomic profiles of two non-malignant breast cell lines and 33 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, four were Her2 amplified, and three were established from triple negative patient-derived xenografts. ',\n",
       "  'No description',\n",
       "  'ERK activity as measured by the EKAREV fluorescent reporter was assessed in the MCF 10A mammary epithelial cell line at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors.',\n",
       "  'ERK activity as measured by the EKAREV fluorescent reporter was assessed in the MCF 10A mammary epithelial cell line at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with IGF-I in the absence or presence of increasing doses of an AKT inhibitor.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with EGF in the absence or presence of increasing doses of a MEK inhibitor.',\n",
       "  'DIA proteomics was carried out on total protein obtained from motor neuron lines generated from subject induced pluripotent stem cell (iPSC) lines. The iPSC lines were derived from 11 ALS and 4 Control (unaffected) individuals, in duplicate.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with IGF-I in the absence or presence of increasing doses of an AKT inhibitor.',\n",
       "  'We measured the sensitivities of six TNBC cell lines to 23 clinical and pre-clinical small molecule inhibitors of kinases in the PI3K/AKT/mTOR signaling pathway and to one inhibitor of MEK. Cells were treated with each drug using a 9-point ½ log dilution series. Live cell counts were measured at the start and end of the 72 hour treatment period using fixed cell microscopy. Drug potency and efficacy were calculated using GR metrics.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with EGF in the absence or presence of increasing doses of a MEK inhibitor.',\n",
       "  'Dose-response data were collected for a panel of 73 breast cancer cell lines grown under standard or modified culture conditions and treated with 139 small molecule and antibody perturbagens. A subset of these data was previously described in Heiser et al. (2012) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/22003129\" target = \"_blank\">22003129</a>) and Daemen et al. (2013) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/24176112\" target = \"_blank\">24176112</a>). Here, the complete dataset was analyzed using the growth rate inhibition (GR) metrics described in Hafner et al. (2016) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/27135972\" target = \"_blank\">27135972</a>) to account for differences in growth rates across the cell lines and treatment conditions.',\n",
       "  'No description',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-Venus) expressed in the 184A1 mammary epithelial cell line was assessed at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors in the absence or presence of an AKT or MEK inhibitor.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of FoxO3a fluorescent reporters (FoxO3aN400-Venus) with or without AKT- and ERK-dependent phosphorylation site mutations was assessed in two mammary epithelial cell lines at the single-cell level using live imaging of cells that were untreated or treated with one of five growth factors.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of FoxO3a fluorescent reporters (FoxO3aN400-Venus) with or without AKT- and ERK-dependent phosphorylation site mutations was assessed in two mammary epithelial cell lines at the single-cell level using live imaging of cells that were untreated or treated with one of five growth factors.',\n",
       "  'Three human melanoma cell lines were treated in 2 replicates with Vemurafenib at 3 doses in combination with DMSO or 1 dose of 9 other compounds. This dataset presents the mean intensity measurements for 3 protein or phosphoprotein readouts (KI-67, Rb (pS807; pS811), and ERK-1 (pT202; pY204) / ERK-2 (pT185; pY187) extracted from microscopy images across each condition. ',\n",
       "  'Three human melanoma cell lines were treated in 2 replicates with 1 dose of Vemurafenib in combination with DMSO or 3 doses of each of 41 compounds in a chromatin-targeting library (LINCS 3 Chromatin Targeting Library). This dataset presents the number of nuclei assayed and the mean intensity measurements for 2 protein readouts (KI-67 and NGFR) extracted from microscopy images across each condition.',\n",
       "  'To define molecular network markers of tyrosine kinase inhibitor-induced cardiotoxicity, we treated human induced pluripotent stem cell-derived cardiomyocytes (hiPSC-CMs) with one of four tyrosine kinase inhibitors displaying a range of mild to severe effects on heart function (Erlotinib, Lapatinib, Sorafenib, or Sunitinib) or vehicle control (DMSO). We performed 6 different treatments for each drug, including three doses (1, 3.16 and 10 µM) at 24 hours and the intermediate dose (3.16 µM) for an additional three time points (6h, 72h and 168h). Gene expression changes were assessed at the cell population level using total RNA-Seq, which measured levels of both mRNAs and long non-coding RNAs. Validation of RNA-Seq results was performed by qPCR (HMS-LINCS Dataset <a href = \"http://lincs.hms.harvard.edu/db/datasets/20325/\">#20325</a>).',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of FoxO3a fluorescent reporters (FoxO3aN400-Venus) with or without AKT- and ERK-dependent phosphorylation site mutations was assessed in two mammary epithelial cell lines at the single-cell level using live imaging of cells that were untreated or treated with one of five growth factors.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-Venus) expressed in the 184A1 mammary epithelial cell line was assessed at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors in the absence or presence of an AKT or MEK inhibitor.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'We measured the baseline proteomic profiles of non-malignant breast cell lines and breast cancer cell lines. ',\n",
       "  'Global chromatin profiling technology is a mass spectrometry-based assay to identify and quantify post-translational modifications on histone proteins from bulk chromatin. The assay quantifies around 60 different combinations of histone modifications on H3, with even more possible combinations on H4, H2A (including isoforms), and H2B. GCP can be considered as orthogonal to ChIP-Seq. Instead of characterizing one histone mark at a time with locus resolution across the genome, one can profile almost all combinations of histone marks together in bulk chromatin with no genomic resolution. The technique is particularly useful in assessing the functional consequences of genetic mutations to the epigenetic machinery and can be employed to make connections among biological conditions and perturbations.',\n",
       "  'The MCF 10A breast cell line was plated in 96-well plates and treated with different doses of one of 9 kinase inhibitors (Getifinib, Erlotinib, Lapatinib, PD0325901, Selumetinib, MK2206, Triciribine, Dactolisib, and Torkinib) for 24 or 48 hr before being fixed and analyzed by immunofluorescence microscopy according to the protocol described below.',\n",
       "  'No description',\n",
       "  'We measured the sensitivities of one non-malignant breast cell line and 20 breast cancer cell lines (six triple negative, six hormone receptor positive, five Her2 amplified, and three established from triple negative patient-derived xenografts) to 32 pre-clinical small molecule deubiquitinating enzyme (DUB) inhibitors. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series and then measured cell number and viability after three days of drug exposure.',\n",
       "  'Global chromatin epigentic profiles of NPC cell lines treated with neurodevelopmental compounds.',\n",
       "  '<p>LINCS data</p>',\n",
       "  'Large-scale perturbation databases, such as Connectivity Map (CMap) or Library of Integrated Network-based Cellular Signatures (LINCS), provide enormous opportunities for computational pharmacogenomics and drug design. A reason for this is that in contrast to classical pharmacology focusing at one target at a time, the transcriptomics profiles provided by CMap and LINCS open the door for systems biology approaches on the pathway and network level. In this article, we provide a review of recent developments in computational pharmacogenomics with respect to CMap and LINCS and related applications.',\n",
       "  'No description',\n",
       "  'No description',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'Global chromatin epigentic profiles of human cancer and NPC cell lines treated with epigenetic compounds.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'We measured the sensitivities of two non-malignant breast cell lines and 31 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, five were Her2 amplified, to 16 clinically-relevant small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure.',\n",
       "  'We measured the sensitivities of two non-malignant breast cell lines and 33 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, four were Her2 amplified, and three were established from triple negative patient-derived xenografts to 34 clinically-relevant small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure.',\n",
       "  'We measured the sensitivities of two non-malignant breast cell lines and 31 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, five were Her2 amplified, to 16 clinically-relevant small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure.',\n",
       "  'We measured the baseline proteomic profiles of four non-malignant breast cell lines and 23 breast cancer cell lines of which three were triple negative, ten were hormone receptor positive, nine were Her2 amplified, and one was established from a triple negative patient-derived xenograft. ',\n",
       "  '-',\n",
       "  'No description',\n",
       "  'We measured the sensitivities of two non-malignant breast cell lines and 33 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, four were Her2 amplified, and three were established from triple negative patient-derived xenografts to 34 clinically-relevant small molecule perturbagens. A microscopy-based dose response assay was used to measure drug potency, and to quantify drug efficacy in terms of growth inhibition (GR metrics) and cell death. We treated cells with single drugs over a 9-point ½ log dilution series from a maximum dose not exceeding 10 µM and then measured cell number and viability after three days of drug exposure.',\n",
       "  'Metabolic parameters (pH, oxygen, and glucose) are measured for 2 cell lines perturbed by 3 small molecules.',\n",
       "  'Transcriptional profiles of four PromoCell cardiomyocytes treated with known toxic compounds alone and in combination with \"mitigating\" compounds and an antibody.',\n",
       "  'To define molecular network markers of tyrosine kinase inhibitor-induced cardiotoxicity, we treated human induced pluripotent stem cell-derived cardiomyocytes (hiPSC-CMs) with one of four tyrosine kinase inhibitors displaying a range of mild to severe effects on heart function (Erlotinib, Lapatinib, Sorafenib, or Sunitinib) or vehicle control (DMSO). We performed 3 different treatments for each drug, including two doses (3.16 and 10  µM) at 24 hours and the lower dose (3.16  µM) at 72h. Gene expression changes were assessed at the cell population level using qPCR. This is a validation dataset for RNA-seq HMS-LINCS Dataset <a href = \"http://lincs.hms.harvard.edu/db/datasets/20324/\">#20324</a>. ',\n",
       "  '-',\n",
       "  'Protein expression profiles of human cancer and NPC cell lines treated with neurodevelopmental compounds.',\n",
       "  'L1000 Connectivity Map perturbational profiles from Broad Institute LINCS Center for Transcriptomics (NIH U54HL127366)',\n",
       "  'Transcriptional profiles of U2 OS cell line treated with ~21 thousands of compounds from diversity-oriented-synthesis (DOS) library.',\n",
       "  'Protein expression profiles of human cancer and NPC cell lines treated with epigenetic compounds',\n",
       "  'The MEP-LINCS uses a novel microenvironment microarray (MEMA) based platform to assess the impacts of different pairwise combinations of ME perturbagens (MEPs) on specific biological and molecular endpoints.',\n",
       "  'Global chromatin epigentic profiles of human cancer and NPC cell lines treated with kinase inhibitors.',\n",
       "  'Transcriptional profiles of three PromoCell cardiomyocytes treated with known toxic compounds alone and in combination with \"mitigating\" compounds.',\n",
       "  '-',\n",
       "  'We measured the baseline proteomic profiles of two non-malignant breast cell lines and 33 breast cancer cell lines of which twenty were triple negative, six were hormone receptor positive, four were Her2 amplified, and three were established from triple negative patient-derived xenografts. ',\n",
       "  'The P100 assay is a mass spectrometry-based targeted proteomics assay that detects and quantifies a representative set of ~100 phosphopeptide probes that are present in a wide range of cell types and have been demonstrated to be modulated via perturbations.  Briefly, phosphopeptides are enriched via an automated protocol from proteins derived from cell pellets that have been perturbed by various reagents.  These peptides are mixed with a set of isotopically labeled internal standards that correspond to the analytes in the P100 assay.  This mixture is introduced into a mass spectrometer using ultra-high pressure liquid chromatography (LCMS) and subjected to an analysis method specific for quantification of the P100 analytes.  The behavior of more than 1000 unique phosphosites may be imputed from the profiles of the P100 analytes.  The resulting profiles are suitable for molecular signature generation, querying of signatures across cell perturbations and types, and modeling of response networks in cellular systems.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-Venus) expressed in the 184A1 mammary epithelial cell line was assessed at the single-cell level using live imaging of cells that were untreated or treated with one of six growth factors in the absence or presence of an AKT or MEK inhibitor.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with IGF-I in the absence or presence of increasing doses of an AKT inhibitor.',\n",
       "  'The nuclear-cytoplasmic pulsing behavior of a FoxO3a fluorescent reporter (FoxO3aN400-HA-Venus) expressed in two mammary epithelial cell lines was assessed at the single-cell level using live imaging of cells that were untreated or treated with EGF in the absence or presence of increasing doses of a MEK inhibitor.',\n",
       "  '-',\n",
       "  '-',\n",
       "  '-',\n",
       "  'Dose-response data were collected for a panel of 73 breast cancer cell lines grown under standard or modified culture conditions and treated with 139 small molecule and antibody perturbagens. A subset of these data was previously described in Heiser et al. (2012) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/22003129\" target = \"_blank\">22003129</a>) and Daemen et al. (2013) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/24176112\" target = \"_blank\">24176112</a>). Here, the complete dataset was analyzed using the growth rate inhibition (GR) metrics described in Hafner et al. (2016) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/27135972\" target = \"_blank\">27135972</a>) to account for differences in growth rates across the cell lines and treatment conditions.',\n",
       "  'Cell cycle and nuclear state imaging readouts for PC3 cells treated with ligand and ECMp combinations to explore the role of microenvironmental signals in cancer growth.',\n",
       "  '<p>In order to successfully implement Domain-Specific Languages (DSLs), it is needed to systematically define and to support its development process; namely its Evaluation and the Domain Analysis phase. For that purpose, the studies were systematically selected from the most relevant venues that focus on the implementation of DSLs, in order to get insight if and how these development phases were performed. The special focus was given to the human-machine DSLs (excluding the machine-machine languages), the involvement of its end-users in the development process and the evaluation of DSLs usability, i.e. quality in use of DSLs.&nbsp;</p>\\n\\n<p>Preliminary results give us a notion that there is increased the state of practice of performing the evaluation of the DSLs, mostly including usability concerns, at least after its implementation. Generally, the quality of the reviewed studies was high. On another hand, rarely the assessments are done during domain analysis, which in general is not reporting inclusion of end-users or consideration of different use-cases, although the majority of studies refer to target non-programmers and to contribute easy in use.&nbsp;</p>\\n\\n<p>We did collect the valuable body of primary studies that are giving us answers to our research questions, however, to raise the credibility of the conclusions we should extend the analysis to other venues. Also, as we get insights into the categorization of practices we could specify more concretely answers that would give us means to perform more detailed meta-analysis.&nbsp;</p>',\n",
       "  'Transcriptomic analysis using total RNA isolated from iPSC-derived diMNs (short protocol with direct differentiation to motor neurons) cell lines from amyotrophic lateral sclerosis (ALS) and control (unaffected) subjects.',\n",
       "  'The Library of Integrated Network-based Cellular Signatures (LINCS) is a multi-institutional, collaborative project initiated by the National Institutes of Health whose goal is to identify and categorize molecular signatures that occur when cells are exposed to agents that perturb their normal function. Understanding how perturbations affect the behavior of cells is important because it can provide insights into the causes of disease and help researchers develop new therapeutic strategies.',\n",
       "  'The microwestern is a meso-scale quantitative western blotting assay using infrared fluorescence. Level 1 data are raw signal (background subtracted) from a single \"spot\" (band). Level 2 data are Level 1 data divided by a loading control signal (e.g. tubulin). Level 3 data are Level 2 data divided by a biological internal control (e.g. Level 2 data corresponding to serum-starved conditions). Level 4 data are averages across Level 3 replicates.',\n",
       "  'Transcriptomic analysis using total RNA isolated from iPSC-derived iMNs cell lines from amyotrophic lateral sclerosis (ALS), spinal muscular atrophy (SMA), and control (unaffected) subjects.',\n",
       "  'To define protein markers of tyrosine kinase inhibitor-induced cardiotoxicity, we treated human induced pluripotent stem cell-derived cardiomyocytes (hiPSC-CMs) with tyrosine kinase inhibitors (TKIs) displaying a range of mild to severe effects on heart function. Two independent cell experiments were performed. In the first experiment, hiPSC-CMs were treated with one of four TKIs (Erlotinib, Lapatinib, Sorafenib and Sunitinib) at 3 &mu;M or a DMSO vehicle-only control in biological replicates (10 samples in total) and harvested at 72 hours for tandem mass tag mass spectrometry. In the second experiment, a different lot of hiPSC-CMs were treated the same way, but samples were collected at both 24 and 72 hours (20 samples in total).<br />\\nCorresponding gene expression data can be found in datasets <a href=\"http://lincs.hms.harvard.edu/db/datasets/20324/\" target=\"_blank\">20324</a> and <a href=\"http://lincs.hms.harvard.edu/db/datasets/20325/\" target=\"_blank\">20325</a>.',\n",
       "  'Dose-response data were collected for a panel of 73 breast cancer cell lines grown under standard or modified culture conditions and treated with 139 small molecule and antibody perturbagens. A subset of these data was previously described in Heiser et al. (2012) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/22003129\" target = \"_blank\">22003129</a>) and Daemen et al. (2013) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/24176112\" target = \"_blank\">24176112</a>). Here, the complete dataset was analyzed using the growth rate inhibition (GR) metrics described in Hafner et al. (2016) (PMID: <a href=\"http://www.ncbi.nlm.nih.gov/pubmed/27135972\" target = \"_blank\">27135972</a>) to account for differences in growth rates across the cell lines and treatment conditions.',\n",
       "  'Three human melanoma cell lines were treated in 2 replicates with Vemurafenib at 3 doses in combination with DMSO or 2 doses of 9 other compounds. This dataset presents the mean intensity measurements for 6 protein or phosphoprotein readouts (KI-67, Rb (pS807; pS811), NGFR, S6 (pS240; pS244), c-Jun, and c-Jun (pS73)) extracted from microscopy images across each condition.',\n",
       "  'Three human melanoma cell lines were treated in 2 replicates with 5 doses of Vemurafenib in combination with DMSO or 1 dose of 5 other compounds. This dataset presents the mean intensity measurements for 5 protein and phosphoprotein readouts (KI-67, NGFR, S6 (pS240; pS244), ERK-1 (pT202; pY204) / ERK-2 (pT185; pY187), and c-Jun) extracted from microscopy images across each condition.',\n",
       "  '<p>&quot;GO D-G/&quot; includes gene-annotation bipartite graph</p>\\n\\n<p>&quot;LINCS D-G/&quot; includes gene-drug bipartite graph</p>\\n\\n<p>&quot;LINCS G-G/&quot; includes gene-gene co-expression graph for LINCS cell line data</p>\\n\\n<p>&quot;scRNA G-G/&quot; includes gene-gene co-expression graph for single-cell data</p>',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'Global chromatin profiling technology is a mass spectrometry-based assay to identify and quantify post-translational modifications on histone proteins from bulk chromatin. The assay quantifies around 60 different combinations of histone modifications on H3, with even more possible combinations on H4, H2A (including isoforms), and H2B. GCP can be considered as orthogonal to ChIP-Seq. Instead of characterizing one histone mark at a time with locus resolution across the genome, one can profile almost all combinations of histone marks together in bulk chromatin with no genomic resolution. The technique is particularly useful in assessing the functional consequences of genetic mutations to the epigenetic machinery and can be employed to make connections among biological conditions and perturbations.',\n",
       "  'Global chromatin profiling technology is a mass spectrometry-based assay to identify and quantify post-translational modifications on histone proteins from bulk chromatin. The assay quantifies around 60 different combinations of histone modifications on H3, with even more possible combinations on H4, H2A (including isoforms), and H2B. GCP can be considered as orthogonal to ChIP-Seq. Instead of characterizing one histone mark at a time with locus resolution across the genome, one can profile almost all combinations of histone marks together in bulk chromatin with no genomic resolution. The technique is particularly useful in assessing the functional consequences of genetic mutations to the epigenetic machinery and can be employed to make connections among biological conditions and perturbations.',\n",
       "  '-',\n",
       "  'The NIH-funded LINCS Consortium is creating an extensive reference library of cell-based perturbation response signatures and sophisticated informatics tools incorporating a large number of perturbagens, model systems, and assays. To date, more than 350 datasets have been generated including transcriptomics, proteomics, epigenomics, cell phenotype and competitive binding profiling assays. The large volume and variety of data necessitate rigorous data standards and effective data management including modular data processing pipelines and end-user interfaces to facilitate accurate and reliable data exchange, curation, validation, standardization, aggregation, integration, and end user access. Deep metadata annotations and the use of qualified data standards enable integration with many external resources. Here we describe the end-to-end data processing and management at the DCIC to generate a high-quality and persistent product. Our data management and stewardship solutions enable a functioning Consortium and make LINCS a valuable scientific resource that aligns with big data initiatives such as the BD2K NIH Program and concords with emerging data science best practices including the findable, accessible, interoperable, and reusable (FAIR) principles.',\n",
       "  'The MCF 10A breast cell line was plated in 96-well plates and treated with different doses of epidermal growth factor (EGF) for 24 hr before being fixed and analyzed by immunofluorescence microscopy according to the protocol described below.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  '<p>Here we include the supplementary material that may be used as a replication package for the quasi-experiment reported in the paper &quot;Assessing user stories: the influence of template differences and gender-related problem-solving styles&quot;, submitted to REJ Special Issue - RE 2021.</p>',\n",
       "  'Gene expression data can offer deep, physiological insights beyond the static coding of the genome alone. We believe that realizing this potential requires specialized, high-capacity machine learning methods capable of using underlying biological structure, but the development of such models is hampered by the lack of published benchmark tasks and well characterized baselines. In this work, we establish such benchmarks and baselines by profiling many classifiers against biologically motivated tasks on two curated views of a large, public gene expression dataset (the LINCS corpus) and one privately produced dataset. We provide these two curated views of the public LINCS dataset and our benchmark tasks to enable direct comparisons to future methodological work and help spur deep learning method development on this modality. In addition to profiling a battery of traditional classifiers, including linear models, random forests, decision trees, K nearest neighbor (KNN) classifiers, and feed-forward artificial neural networks (FF-ANNs), we also test a method novel to this data modality: graph convolugtional neural networks (GCNNs), which allow us to incorporate prior biological domain knowledge. We find that GCNNs can be highly performant, with large datasets, whereas FF-ANNs consistently perform well. Non-neural classifiers are dominated by linear models and KNN classifiers.',\n",
       "  'The dose response of human melanoma cell lines to Vemurafenib in combination with Trametinib and a third compound was measured at 72 hr to determine their effects on apoptosis and viability. Cells were treated in 3 replicates using a Hewlett-Packard (HP) D300 Digital Dispenser with 4 combined doses of Vemurafenib/Trametinib (1:3.16 dilution ratios) and 1 or 3 doses of a third compound for 72 hr. To score viability and apoptosis, a dye-based imaging assay was used. The cell-permeable DNA dye Hoechst 33342 was used to mark all nuclei, and DEVD-NucView488 Caspase-3 substrate was used to mark the nuclei of cells undergoing apoptosis (i.e., in which caspase 3 is active).',\n",
       "  '<p>Simulations of 2/1/1 mixture of DPPC/DOPC/CHOL with the refined Martini model by Carpenter et al.&nbsp;</p>\\n\\n<p>Simulations are performed using a time step&nbsp;of 35&nbsp;fs for&nbsp;15 &micro;s. The files are named 211_XX.YYY, where XX is either &quot;sep&quot; for simulations in which all lipid types are coupled separately to the thermostat, or &quot;lincs&quot;, where different values for lincs_order (8 instead of standard 4) and lincs_iter (2 intead of standard 1) are used.&nbsp;YYY are GROMACS-compatible file formats: xtc for trajectory, edr for energy outputs, cpt for checkpoint file, tpr for run input file, and gro for final structure at the end of simulation. Additionally, files with _trr are included, which are 100 ns-long extensions to the aforementioned files. For them, a trr trajectory with velocities and coordinates is provided.</p>\\n\\n<p>The topology (topol.top) and index file (index.ndx) are common for all simulations, and the TOP.tar contains all the force field parameters (itp). The run input parameters (mdp) are provided for each system.</p>',\n",
       "  '<p>Simulations of 3/3/2 mixture of DPPC/DOPC/CHOL with the refined Martini model by Carpenter et al.&nbsp;</p>\\n\\n<p>Simulations are performed using a time step&nbsp;of 35&nbsp;fs for&nbsp;15 &micro;s. The files are named 332_XX.YYY, where XX is either &quot;sep&quot; for simulations in which all lipid types are coupled separately to the thermostat, or &quot;lincs&quot;, where different values for lincs_order (8 instead of standard 4) and lincs_iter (2 intead of standard 1) are used.&nbsp;YYY are GROMACS-compatible file formats: xtc for trajectory, edr for energy outputs, cpt for checkpoint file, tpr for run input file, and gro for final structure at the end of simulation. Additionally, files with _trr are included, which are 100 ns-long extensions to the aforementioned files. For them, a trr trajectory with velocities and coordinates is provided.</p>\\n\\n<p>The topology (topol.top) and index file (index.ndx) are common for all simulations, and the TOP.tar contains all the force field parameters (itp). The run input parameters (mdp) are provided for each system.</p>',\n",
       "  'The dose response of human melanoma cell lines in response to Vemurafenib in combination with a second compound was measured at 72 hr to determine their effects on apoptosis and viability. Cells were treated in 3 replicates using a Hewlett-Packard (HP) D300 Digital Dispenser with 4 doses of Vemurafenib (1:3.16 dilution ratios) and 1 or 3 doses of a second compound for 72 hr. To score viability and apoptosis, a dye-based imaging assay was used. The cell-permeable DNA dye Hoechst 33342 was used to mark all nuclei, and DEVD-NucView488 Caspase-3 substrate was used to mark the nuclei of cells undergoing apoptosis (i.e., in which caspase 3 is active).',\n",
       "  'No description',\n",
       "  'Cyclic Immunofluorescence (CycIF) is a fluorescence imaging assay used for multiplexed measurement of protein levels and protein phosphorylation states in single cells. The MCF 10A breast cell line was plated in 96-well plates and treated with different doses of one of five perturbagens (EGF and the kinase inhibitors Lapatinib, Selumetinib, Dactolisib, and Torkinib) for 24 hr before being fixed and subjected to nine cycles of CycIF staining according to the protocol described below.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The P100 assay is a mass spectrometry-based targeted proteomics assay that detects and quantifies a representative set of ~100 phosphopeptide probes that are present in a wide range of cell types and have been demonstrated to be modulated via perturbations.  Briefly, phosphopeptides are enriched via an automated protocol from proteins derived from cell pellets that have been perturbed by various reagents.  These peptides are mixed with a set of isotopically labeled internal standards that correspond to the analytes in the P100 assay.  This mixture is introduced into a mass spectrometer using ultra-high pressure liquid chromatography (LCMS) and subjected to an analysis method specific for quantification of the P100 analytes.  The behavior of more than 1000 unique phosphosites may be imputed from the profiles of the P100 analytes.  The resulting profiles are suitable for molecular signature generation, querying of signatures across cell perturbations and types, and modeling of response networks in cellular systems.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The P100 assay is a mass spectrometry-based targeted proteomics assay that detects and quantifies a representative set of ~100 phosphopeptide probes that are present in a wide range of cell types and have been demonstrated to be modulated via perturbations.  Briefly, phosphopeptides are enriched via an automated protocol from proteins derived from cell pellets that have been perturbed by various reagents.  These peptides are mixed with a set of isotopically labeled internal standards that correspond to the analytes in the P100 assay.  This mixture is introduced into a mass spectrometer using ultra-high pressure liquid chromatography (LCMS) and subjected to an analysis method specific for quantification of the P100 analytes.  The behavior of more than 1000 unique phosphosites may be imputed from the profiles of the P100 analytes.  The resulting profiles are suitable for molecular signature generation, querying of signatures across cell perturbations and types, and modeling of response networks in cellular systems.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'Morphology profiles of cultured human bone cancer cell lines treated with small molecules.',\n",
       "  'No description',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'No description',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  \"The RNA library of 70 PromoCell samples are prepared using llumina's TruSeq Stranded mRNA Library Prep Kit and sequenced by the Illumina NovaSeq 6000 sequencing system.\",\n",
       "  \"The RNA library of 914 PromoCell samples are prepared using llumina's TruSeq Stranded mRNA Library Prep Kit and sequenced by the Illumina HiSeq 2500 sequencing system.\",\n",
       "  'No description',\n",
       "  'BACKGROUND:Skin atrophy is a major adverse effect of topical glucocorticoids. We recently reported that REDD1 (regulated in development and DNA damage 1) and FKBP51 (FK506 binding protein 5), negative regulators of mTOR/Akt signaling, are induced by glucocorticoids in mouse and human skin and are central drivers of steroid skin atrophy. Thus, we hypothesized that REDD1/FKBP51 inhibitors could protect skin against catabolic effects of glucocorticoids. METHODS:Using drug repurposing approach, we screened LINCS library (http://lincsproject.org/LINCS/) to identify repressors of REDD1/FKBP51 expression. Candidate compounds were tested for their ability to inhibit glucocorticoid-induced REDD1/FKBP51 expression in human primary/immortalized keratinocytes and in mouse skin. Reporter gene expression, microarray, and chromatin immunoprecipitation were employed to evaluate effect of these inhibitors on the glucocorticoid receptor (GR) signaling. FINDINGS:Bioinformatics analysis unexpectedly identified phosphoinositide-3-kinase (PI3K)/mTOR/Akt inhibitors as a pharmacological class of REDD1/FKBP51 repressors. Selected PI3K/mTOR/Akt inhibitors-Wortmannin (WM), LY294002, AZD8055, and two others indeed blocked REDD1/FKBP51expression in human keratinocytes. PI3K/mTOR/Akt inhibitors also modified global effect of glucocorticoids on trascriptome, shifting it towards therapeutically important transrepression; negatively impacted GR phosphorylation; nuclear translocation; and GR loading on REDD1/FKBP51 gene promoters. Further, topical application of LY294002 together with glucocorticoid fluocinolone acetonide (FA) protected mice against FA-induced proliferative block and skin atrophy but did not alter the anti-inflammatory activity of FA in ear edema test. INTERPRETATION:Our results built a strong foundation for development of safer GR-targeted therapies for inflammatory skin diseases using combination of glucocorticoids with PI3K/mTOR/Akt inhibitors. FUND: Work is supported by NIH grants R01GM112945, R01AI125366, and HESI-THRIVE foundation.',\n",
       "  'Background: Gastric cancer (GC) is one of the main mortality cause worldwide. Previously, we found Forkhead box protein (FOXM1) or Urokinase-type plasminogen activator (PLAU) are independent prognostic markers of GC. This study aims to explore the combining prognostic efficacy and the potential insights underlying additive effect of FOXM1 to PLAU in GC progression through in-silico analyses. Method: The expression of FOXM1 and PLAU were profiled in 33 cancer types using public data. A merged GC expression dataset containing 598 samples was used for evaluating prognostic significance of FOXM1/PLAU. Gene Set Enrichment Analysis (GSEA) was performed to elucidate the mechanisms underlying FOXM1/PLAU promoted GC progression. The Cancer Genome Atlas (TCGA) was used for analyzing the association between FOXM1/PLAU and tumor immune infiltration. Genomic and proteomic differences between FOXM1+PLAU+ and FOXM1-PLAU- groups were also computed using TCGA GC data. Drugs targeting FOXM1/PLAU associated gene expression pattern was analyzed using LINCs database. Results: FOXM1 and PLAU are overexpressed in 17/33 cancer types including GC. Kaplan-Meier analyses indicate that the FOXM1+PLAU+ subgroup have the worst prognosis, while FOXM1-PLAU- subgroup have the best survival. Bioinformatics analysis indicated that FOXM1+PLAU+ associated genes are enriched in TGF-beta, DNA repair and drug resistance signaling pathways; FOXM1 and PLAU expression are negatively correlated with tumor immune infiltration. Genomic and proteomic differences between FOXM1+PLAU+ and FOXM1-PLAU- groups were presented. Data mining from LINCs suggested several chemicals or drugs that could target the gene expression pattern of FOXM1+PLAU+ patients. Conclusion: FOXM1+PLAU+ can serve as effective prognostic biomarkers and potential therapeutic targets for GC. Due to the additive effect of these two genes, screening for drugs or chemicals that targeting the expression patterns PLAU+FOXM1+ subgroup may exert important clinical impact on GC management.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  '-',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'This GEO Series contains datasets used in the paper \"Evaluation Of RNAi And CRISPR Technologies By Large Scale Gene Expression Profiling In The Connectivity Map\".  The application of RNA interference (RNAi) to mammalian cells has provided the means to perform phenotypic screens to determine the functions of genes. Although RNAi has revolutionized loss of function genetic experiments, it has been difficult to systematically assess the prevalence and consequences of off-target effects. The Connectivity Map (CMAP) represents an unprecedented resource to study the gene expression consequences of expressing short hairpin RNAs (shRNAs). Analysis of signatures for over 13,000 shRNAs applied in 9 cell lines revealed that miRNA-like off-target effects of RNAi are far stronger and more pervasive than generally appreciated. We show that mitigating off-target effects is feasible in these datasets via computational methodologies to produce a Consensus Gene Signature (CGS). In addition, we compared RNAi technology to clustered regularly interspaced short palindromic repeat (CRISPR)-based knockout by analysis of 373 sgRNAs in 6 cells lines, and show that the on-target efficacies are comparable, but CRISPR technology is far less susceptible to systematic off-target effects. These results will help guide the proper use and analysis of loss-of-function reagents for the determination of gene function.  Note: Related GEO projects include a large corpus of additional L1000 data, available at GSE92742.The Platform is GPL20573: Broad Institute Human L1000 epsilonhttps://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL20573For questions or assistance with this dataset, please email the CMap support team at: clue@broadinstitute.org\\nThe datasets in this Series were generated as part of the LINCS project, which aims to enable a functional understanding of biology by cataloging changes in gene expression and other cellular processes that occur when cells are exposed to a variety of perturbing agents. The Broad Institute LINCS Center for Transcriptomics contributes to this collaborative effort by application of the Connectivity Map concept. In brief, the study design involves the generation of a compendium of transcriptional expression data from cultured human cells treated with small-molecule and genetic loss/gain of function perturbagens. These measurements are made using the L1000 high-throughput gene-expression assay that enables data generation at an unprecedented scale.  The data provided here include Levels 4 and 5, as well as additional data described in the README:  Level 4 (Z-SCORES) - signatures with differentially expressed genes computed by robust z-scores for each profile relative to control (PC relative to plate population as control; VC relative to vehicle control). Level 5 (SIG) consists of the replicates, usually 3 per treatment, aggregated into a single differential expression vector derived from the weighted averages of the individual replicates.  For more data levels and their descriptions, please consult GSE92742.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'No description',\n",
       "  '-',\n",
       "  '-',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'Confirmatory and associated toxicity counter assays from the Tox21 project were filtered by compound purity and aggregated by CID.  Data are presented by assay pairings and include relevant dataset annotations.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'High throughput shot-gun proteomics experiments have been conducted for 317 cell line/drug combinations (from four primary human cardiomyocyte cell lines purchased from PromoCell) and 64 HeLa control lysates. The proteins contained in cell samples were isolated and digested into peptides, following by the LC-MS/MS assay of resulted peptide compositions. The RAW data files generated from the MS assay were analyzed by MaxQuant software to map detected peptides to corresponding proteins.',\n",
       "  'The microwestern is a meso-scale quantitative western blotting assay using infrared fluorescence. Level 1 data are raw signal (background subtracted) from a single \"spot\" (band). Level 2 data are Level 1 data divided by a loading control signal (e.g. tubulin). Level 3 data are Level 2 data divided by a biological internal control (e.g. Level 2 data corresponding to serum-starved conditions). Level 4 data are averages across Level 3 replicates.',\n",
       "  'This SuperSeries is composed of the SubSeries listed below.\\nRefer to individual Series',\n",
       "  'mRNA transcriptomic profiles of polyA+ enriched RNA isolated from frozen pellets of iPSCs derived from amyotrophic lateral sclerosis (ALS), spinal muscular atrophy (SMA), and control (unaffected) subjects.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'We previously piloted the concept of a Connectivity Map (CMap), whereby genes, drugs, and disease states are connected by virtue of common gene-expression signatures. Here, we report more than a 1,000-fold scale-up of the CMap as part of the NIH LINCS Consortium, made possible by a new, low-cost, high-throughput reduced representation expression profiling method that we term L1000. We show that L1000 is highly reproducible, comparable to RNA sequencing, and suitable for computational inference of the expression levels of 81% of non-measured transcripts. We further show that the expanded CMap can be used to discover mechanism of action of small molecules, functionally annotate genetic variants of disease genes, and inform clinical trials. The 1.3 million L1000 profiles described here, as well as tools for their analysis, are available at https://clue.io.',\n",
       "  'Measuring the potencies of small-molecule drugs in cell lines is a critical aspect of preclinical pharmacology. Such experiments are also prototypical of high-throughput experiments in multi-well plates. The procedure is simple in principle, but many unrecognized factors can affect the results, potentially making data unreliable. The procedures for measuring drug response described here were developed by the NIH LINCS program to improve reproducibility. Key features include maximizing uniform cell growth during the assay period, accounting for the effects of cell density on response, and correcting sensitivity measures for differences in proliferation rates. Two related protocols are described: one involves an endpoint measure well-suited to large-scale studies and the second is a time-dependent measurement that reveals changes in response over time. The methods can be adapted to other types of plate-based experiments. © 2017 by John Wiley & Sons, Inc.',\n",
       "  'Gene expression profiling was made more cost-effective by the NIH LINCS program that profiles only ?1, 000 selected landmark genes and uses them to reconstruct the whole profile. The D-GEX method employs neural networks to infer the entire profile. However, the original D-GEX can be significantly improved. We propose a novel transformative adaptive activation function that improves the gene expression inference even further and which generalizes several existing adaptive activation functions. Our improved neural network achieves an average mean absolute error of 0.1340, which is a significant improvement over our reimplementation of the original D-GEX, which achieves an average mean absolute error of 0.1637. The proposed transformative adaptive function enables a significantly more accurate reconstruction of the full gene expression profiles with only a small increase in the complexity of the model and its training procedure compared to other methods.',\n",
       "  'Epigenomic profiles using nuclei isolated from cryopreserved cultures of iPCS cells derived from amyotrophic lateral sclerosis (ALS), spinal muscular atrophy (SMA), and control (unaffected) subjects.The ATAC-seq technique detects open-chromatin regions and maps transcription factor binding events genome-wide by means of direct\\xa0in vitro\\xa0transposition of native chromatin.',\n",
       "  'Proteomic analysis using total protein isolated from iPSC-derived diMNs (short protocol with direct differentiation to motor neurons) cell lines from amyotrophic lateral sclerosis (ALS) and control (unaffected) subjects.',\n",
       "  'Protein isolated from induced Pluripotnent Stem Cells (iPSC) cell lines was used to generate high-quality libraries for SWATH-MS, a data-independent acquisition method.  These lines were derived from ALS, SMA and Control (unaffected) individuals (three of each). Output files contain peptide lists and protein names.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The microwestern is a meso-scale quantitative western blotting assay using infrared fluorescence. Level 1 data are raw signal (background subtracted) from a single \"spot\" (band). Level 2 data are Level 1 data divided by a loading control signal (e.g. tubulin). Level 3 data are Level 2 data divided by a biological internal control (e.g. Level 2 data corresponding to serum-starved conditions). Level 4 data are averages across Level 3 replicates.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The Library of Integrated Cellular Signatures (LINCS) is an NIH program which funds the generation of perturbational profiles across multiple cell and perturbation types, as well as read-outs, at a massive scale. The LINCS Center for Transcriptomics at the Broad Institute uses the L1000 high-throughput gene-expression assay to profile a range of cellular models and perturbations of cellular state. These data relate to using  RNA-Seq datasets from the GTEx consortium (http://www.gtexportal.org/) to validate the L1000 assay and to improve the statistical model used in imputing the transcriptome.The Platform is GPL20573: Broad Institute Human L1000 epsilonhttps://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL20573For questions or assistance with this dataset, please email the CMap support team at: clue@broadinstitute.org\\nThe Genotype Tissue Expression (GTEx) project aims to find associations between genomic features and gene expression levels. As part of this project, aliquots of mRNA obtained from 3,176 patient-derived samples from 30 tissue types were profiled on two contemporary gene expression platforms; RNAseq and L1000, a high-throughput gene-expression assay that enables data generation at an unprecedented scale. In L1000, the data are processed through a computational system that converts raw fluorescence intensities into normalized expression profiles generating level 3 (Q2NORM) - gene expression profiles of both directly measured landmark transcripts plus inferred genes. Normalization is done using invariant set scaling followed by quantile normalization. The RNAseq data were collected using Illumina TrueSeq RNA sequencing and converted into reads per kilobase-million (RPKM) units. These RPKM data and underlying read count files are available via the GTEx portal (http://www.gtexportal.org/home/datasets). In this series, the RPKM data were also log2-scaled and quantile normalized as a cohort.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  '<p>This companion site contains additional data to complement the paper:</p>\\n\\n<p><em><strong>Henriques, H., Louren&ccedil;o, H., Amaral, V., and Goul&atilde;o, M. (2018). Improving the developer experience with a low-code process </strong></em><em><strong>modelling</strong></em><em><strong> language. In ACM/IEEE 21st International Conference on Model Driven Engineering Languages and Systems (MODELS 2018), Copenhagen, Denmark. ACM. https://doi.org/10.1145/3239372.3239387</strong></em></p>\\n\\n<p><strong>Abstract</strong></p>\\n\\n<p><strong>Context</strong><strong>:&nbsp;</strong>The OutSystems Platform is a development environment composed of several DSLs, used to specify, quickly build and validate web and mobile applications. The DSLs allow users to model different perspectives such as interfaces and data models, define custom business logic and construct process models.</p>\\n\\n<p><strong>Problem</strong><strong>:&nbsp;</strong>TheDSL for process modelling (Business Process Technology (BPT)), has a low adoption rate and is perceived as having usability problems hampering its adoption. This is problematic given the language maintenance costs.</p>\\n\\n<p><strong>Method:</strong> We used a combination of interviews, a critical review of BPT using the &ldquo;Physics of Notation&rdquo; and empirical evaluations of BPT using the System Usability Scale (SUS)and the NASA Task Load indeX (TLX), to develop a new version ofBPT, taking these inputs and Outsystems&rsquo; engineers culture into account.</p>\\n\\n<p><strong>Results:&nbsp;</strong>Evaluations conducted with 25 professional soft-ware engineers showed an increase of the semantic transparency on the new version, from 31% to 69%, an increase in the correctness of responses, from 51% to 89%, an increase in the SUS score, from 42.25 to 64.78, and a decrease of the TLX score, from 36.50 to 20.78. These differences were statistically significant.</p>\\n\\n<p><strong>Conclusions:</strong> These results suggest the new version of BPT significantly improved the developer experience of the previous version. The end users background with OutSystems had a relevant impact on the final concrete syntax choices and achieved usability indicators.</p>\\n\\n<p>&nbsp;</p>\\n\\n<p><strong>Contents</strong></p>\\n\\n<p>This companion site provides a permanent link for additional data to the supported paper.</p>\\n\\n<p>This repository includes:</p>\\n\\n<ul>\\n\\t<li>Surveys and Questionnaires used in the evaluation reported in the paper\\n\\t<ul>\\n\\t\\t<li>Survey on OutSystems BPT notations (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/survey.pdf\">survey.pdf</a>)</li>\\n\\t\\t<li>Prototype Symbol Set Questionnaire (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/PrototypeSymbolSetQuestionnaire.pdf\">PrototypeSymbolSetQuestionnaire.pdf</a>)</li>\\n\\t\\t<li>Original BPT Evaluation (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/languages.png\">languages.png</a>)</li>\\n\\t\\t<li>Usability Evaluation (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/sus.png\">sus.png</a>)</li>\\n\\t\\t<li>Cognitive Effort Evaluation (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/tlx.png\">tlx.png</a>)</li>\\n\\t\\t<li>Testing environment screenshot (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/Testing%20Environment%20Screenshot.png\">Testing Environment Screenshot</a>)</li>\\n\\t</ul>\\n\\t</li>\\n\\t<li>Statistics\\n\\t<ul>\\n\\t\\t<li>SUS and NASA TLX\\n\\t\\t<ul>\\n\\t\\t\\t<li>Descriptive statistics (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/SUSTLXDescriptiveStats.pdf\">SUSTLXDescriptiveStats.pdf</a>)</li>\\n\\t\\t\\t<li>Normality tests (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/SUSTLXNormalityTests.pdf\">SUSTLXNormality.pdf</a>)</li>\\n\\t\\t\\t<li>Correlation test (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/SUSTLXCorrelation.pdf\">SUSTLXCorrelation.pdf</a>)</li>\\n\\t\\t\\t<li>Scatterplot (<a href=\"https://zenodo.org/api/files/68bdc7fa-684a-496d-ab63-d956271f1f7d/SUSTLXScatterPlot.pdf\">SUSTLXScatterplot.pdf</a>)</li>\\n\\t\\t</ul>\\n\\t\\t</li>\\n\\t</ul>\\n\\t</li>\\n</ul>\\n\\n<p>&nbsp;</p>',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP and ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'Epigenomic profiles using nuclei isolated from cryopreserved cultures of iPCS cells derived from amyotrophic lateral sclerosis (ALS), spinal muscular atrophy (SMA), and control (unaffected) subjects.The ATAC-seq technique detects open-chromatin regions and maps transcription factor binding events genome-wide by means of direct\\xa0in vitro\\xa0transposition of native chromatin.',\n",
       "  'Epigenomic profiles using nuclei isolated from cryopreserved cultures of iPCS cells derived from amyotrophic lateral sclerosis (ALS), spinal muscular atrophy (SMA), and control (unaffected) subjects.The ATAC-seq technique detects open-chromatin regions and maps transcription factor binding events genome-wide by means of direct\\xa0in vitro\\xa0transposition of native chromatin.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'We describe a set of algorithms that allow to simulate dihydrofolate reductase (DHFR, a common benchmark) with the AMBER all-atom force field at 160 nanoseconds/day on a single Intel Core i7 5960X CPU (no graphics processing unit (GPU), 23,786 atoms, particle mesh Ewald (PME), 8.0 Å cutoff, correct atom masses, reproducible trajectory, CPU with 3.6 GHz, no turbo boost, 8 AVX registers). The new features include a mixed multiple time-step algorithm (reaching 5 fs), a tuned version of LINCS to constrain bond angles, the fusion of pair list creation and force calculation, pressure coupling with a \"densostat,\" and exploitation of new CPU instruction sets like AVX2. The impact of Intel\\'s new transactional memory, atomic instructions, and sloppy pair lists is also analyzed. The algorithms map well to GPUs and can automatically handle most Protein Data Bank (PDB) files including ligands. An implementation is available as part of the YASARA molecular modeling and simulation program from www.YASARA.org.',\n",
       "  'The KiNativ assay platform is based on the use of biotinylated acyl phosphates of ATP or ADP that act as probes by reacting with protein kinases on conserved lysine residues in the ATP binding pocket to covalently attach a biotin moiety.  Using these probes, cell lysates treated with a kinase inhibitor can be labeled with BHAcATP or BHAcADP, with kinases inhibited by the compound of interest exhibiting reduced or no labeling.  This is followed by digestion with trypsin, isolation of biotinylated peptides, and analysis by mass spectrometry to determine the extent of labeling of peptides from each kinase.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\".',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition. The KINOMEscan data are presented graphically on TREEspot Kinase Dendrograms (http://www.kinomescan.com/Tools---Resources/Study-Reports---Data-Analysis).  For this study, HMS LINCS investigators have graphed results for kinases classified as 35 \"percent of control\" (in the presence of the compound, the kinase is 35% as active for binding ligand in the presence of DMSO), 5 \"percent of control\" and 1 \"percent of control\". ',\n",
       "  'Recently, several high profile studies collected cell viability data from panels of cancer cell lines treated with many drugs applied at different concentrations. Such drug sensitivity data for cancer cell lines provide suggestive treatments for different types and subtypes of cancer. Visualization of these datasets can reveal patterns that may not be obvious by examining the data without such efforts. Here we introduce Drug/Cell-line Browser (DCB), an online interactive HTML5 data visualization tool for interacting with three of the recently published datasets of cancer cell lines/drug-viability studies. DCB uses clustering and canvas visualization of the drugs and the cell lines, as well as a bar graph that summarizes drug effectiveness for the tissue of origin or the cancer subtypes for single or multiple drugs. DCB can help in understanding drug response patterns and prioritizing drug/cancer cell line interactions by tissue of origin or cancer subtype.DCB is an open source Web-based tool that is freely available at: http://www.maayanlab.net/LINCS/DCB CONTACT: avi.maayan@mssm.eduSupplementary data are available at Bioinformatics online.',\n",
       "  'SUMMARY:For many next generation-sequencing pipelines, the most computationally intensive step is the alignment of reads to a reference sequence. As a result, alignment software such as the Burrows-Wheeler Aligner is optimized for speed and is often executed in parallel on the cloud. However, there are other less demanding steps that can also be optimized to significantly increase the speed especially when using many threads. We demonstrate this using a unique molecular identifier RNA-sequencing pipeline consisting of 3 steps: split, align, and merge. Optimization of all three steps yields a 40% increase in speed when executed using a single thread. However, when executed using 16 threads, we observe a 4-fold improvement over the original parallel implementation and more than an 8-fold improvement over the original single-threaded implementation. In contrast, optimizing only the alignment step results in just a 13% improvement over the original parallel workflow using 16 threads. AVAILABILITY AND IMPLEMENTATION:Code (M.I.T. license), supporting scripts and Dockerfiles are available at https://github.com/BioDepot/LINCS_RNAseq_cpp and Docker images at https://hub.docker.com/r/biodepot/rnaseq-umi-cpp/. SUPPLEMENTARY INFORMATION:Supplementary data are available at Bioinformatics online.',\n",
       "  'We utilized a cell-level approach to examine glycolytic pathways in the DLPFC of subjects with schizophrenia (n =?16) and control (n =?16) and found decreased mRNA expression of glycolytic enzymes in pyramidal neurons, but not astrocytes. To replicate these novel bioenergetic findings, we probed independent datasets for bioenergetic targets and found similar abnormalities. Next, we used a novel strategy to build a schizophrenia bioenergetic profile by a tailored application of the Library of Integrated Network-Based Cellular Signatures data portal (iLINCS) and investigated connected cellular pathways, kinases, and transcription factors using Enrichr. Finally, with the goal of identifying drugs capable of \"reversing\" the bioenergetic schizophrenia signature, we performed a connectivity analysis with iLINCS and identified peroxisome proliferator-activated receptor (PPAR) agonists as promising therapeutic targets. We administered a PPAR agonist to the GluN1 knockdown model of schizophrenia and found it improved long-term memory. Taken together, our findings suggest that tailored bioinformatics approaches, coupled with the LINCS library of transcriptional signatures of chemical and genetic perturbagens, may be employed to identify novel treatment strategies for schizophrenia and related diseases.',\n",
       "  \"Current pharmacological treatments for major depressive disorder (MDD) are ineffective in a significant proportion of patients, and the identification of new antidepressant compounds has been difficult. 'Connectivity mapping' is a method that can be used to identify drugs that elicit similar downstream effects on mRNA levels when compared to current treatments, and thus may point towards possible repositioning opportunities. We investigated genome-wide transcriptomic changes to human hippocampal progenitor cells treated with therapeutically relevant concentrations of a tricyclic antidepressant (nortriptyline) and a selective serotonin reuptake inhibitor (escitalopram). We identified mRNA changes common to both drugs to create an 'antidepressant mRNA signature'. We used this signature to probe the Library of Integrated Network-based Cellular Signatures (LINCS) and to identify other compounds that elicit similar changes to mRNA in neural progenitor cells. Results from LINCS revealed that the tricyclic antidepressant clomipramine elicited mRNA changes most similar to our mRNA signature, and we identified W-7 and vorinostat as functionally relevant drug candidates, which may have repositioning potential. Our results are encouraging and represent the first attempt to use connectivity mapping for drug repositioning in MDD.\",\n",
       "  'Treatment of gastric cancer (GC) often produces poor outcomes. Moreover, predicting which GC treatments will be effective remains challenging. Computational drug repositioning using public databases is a promising and efficient tool for discovering new uses for existing drugs. Here we used a computational reversal of gene expression approach based on effects on gene expression signatures by GC disease and drugs to explore new GC drug candidates. Gene expression profiles for individual GC tumoral and normal gastric tissue samples were downloaded from the Gene Expression Omnibus (GEO) and differentially expressed genes (DEGs) in GC were determined with a meta-signature analysis. Profiles drug activity and drug-induced gene expression were downloaded from the ChEMBL and the LINCS databases, respectively. Candidate drugs to treat GC were predicted using reversal gene expression score (RGES). Drug candidates including sorafenib, olaparib, elesclomol, tanespimycin, selumetinib, and ponatinib were predicted to be active for treatment of GC. Meanwhile, GC-related genes such as PLOD3, COL4A1, UBE2C, MIF, and PRPF5 were identified as having gene expression profiles that can be reversed by drugs. These findings support the use of a computational reversal gene expression approach to identify new drug candidates that can be used to treat GC.',\n",
       "  'Concerns have been expressed about the robustness of experimental findings in several areas of science, but these matters have not been evaluated at scale. Here we identify a large sample of published drug-gene interaction claims curated in the Comparative Toxicogenomics Database (for example, benzo(a)pyrene decreases expression of SLC22A3) and evaluate these claims by connecting them with high-throughput experiments from the LINCS L1000 program. Our sample included 60,159 supporting findings and 4253 opposing findings about 51,292 drug-gene interaction claims in 3363 scientific articles. We show that claims reported in a single paper replicate 19.0% (95% confidence interval [CI], 16.9-21.2%) more frequently than expected, while claims reported in multiple papers replicate 45.5% (95% CI, 21.8-74.2%) more frequently than expected. We also analyze the subsample of interactions with two or more published findings (2493 claims; 6272 supporting findings; 339 opposing findings; 1282 research articles), and show that centralized scientific communities, which use similar methods and involve shared authors who contribute to many articles, propagate less replicable claims than decentralized communities, which use more diverse methods and contain more independent teams. Our findings suggest how policies that foster decentralized collaboration will increase the robustness of scientific findings in biomedical research.',\n",
       "  'We developed a pipeline for the discovery of transcriptomics-derived disease-modifying therapies and used it to validate treatments in vitro and in vivo that could be repurposed for TBI treatment. Desmethylclomipramine, ionomycin, sirolimus and trimipramine, identified by in silico LINCS analysis as candidate treatments modulating the TBI-induced transcriptomics networks, were tested in neuron-BV2 microglial co-cultures, using tumour necrosis factor ? as a monitoring biomarker for neuroinflammation, nitrite for nitric oxide-mediated neurotoxicity and microtubule associated protein 2-based immunostaining for neuronal survival. Based on (a) therapeutic time window in silico, (b) blood-brain barrier penetration and water solubility, (c) anti-inflammatory and neuroprotective effects in vitro (p < 0.05) and (d) target engagement of Nrf2 target genes (p < 0.05), desmethylclomipramine was validated in a lateral fluid-percussion model of TBI in rats. Despite the favourable in silico and in vitro outcomes, in vivo assessment of clomipramine, which metabolizes to desmethylclomipramine, failed to demonstrate favourable effects on motor and memory tests. In fact, clomipramine treatment worsened the composite neuroscore (p < 0.05). Weight loss (p < 0.05) and prolonged upregulation of plasma cytokines (p < 0.05) may have contributed to the worsened somatomotor outcome. Our pipeline provides a rational stepwise procedure for evaluating favourable and unfavourable effects of systems-biology discovered compounds that modulate post-TBI transcriptomics.',\n",
       "  'Deep learning is rapidly advancing many areas of science and technology with multiple success stories in image, text, voice and video recognition, robotics, and autonomous driving. In this paper we demonstrate how deep neural networks (DNN) trained on large transcriptional response data sets can classify various drugs to therapeutic categories solely based on their transcriptional profiles. We used the perturbation samples of 678 drugs across A549, MCF-7, and PC-3 cell lines from the LINCS Project and linked those to 12 therapeutic use categories derived from MeSH. To train the DNN, we utilized both gene level transcriptomic data and transcriptomic data processed using a pathway activation scoring algorithm, for a pooled data set of samples perturbed with different concentrations of the drug for 6 and 24 hours. In both pathway and gene level classification, DNN achieved high classification accuracy and convincingly outperformed the support vector machine (SVM) model on every multiclass classification problem, however, models based on pathway level data performed significantly better. For the first time we demonstrate a deep learning neural net trained on transcriptomic data to recognize pharmacological properties of multiple drugs across different biological systems and conditions. We also propose using deep neural net confusion matrices for drug repositioning. This work is a proof of principle for applying deep learning to drug discovery and development.',\n",
       "  'LS1034 colorectal cancer cells were treated with DMSO vehicle control or 12 uM tepoxalin for 6 hours.Anti-cancer uses of non-oncology drugs have been found on occasion, but such discoveries have been serendipitous and rare. To fully discover activity against multiple tumor types, resource-intensive screening across hundreds of cancer cell lines is needed. We sought to create a public resource containing the growth inhibitory activity of 4,518 drugs tested across 578 human cancer cell lines. To accomplish this, we used PRISM, which involves the molecular barcoding of each cell line, followed by pooling of the barcoded lines. Relative barcode abundance following drug treatment versus controls thus reflects cell viability following drug treatment. We found that an unexpectedly large number of non-oncology drugs selectively inhibited subsets of cancer cell lines. Moreover, the killing activity of the majority of these drugs was predictable based on the molecular features of the cell lines. Mechanistic follow-up of several of these compounds revealed novel mechanisms. These results illustrate the potential of the PRISM drug repurposing resource as a starting point for new oncology therapeutic development.\\n6 samples total:  LS1034 cells treated with tepoxalin for 6 hours (triplicate) or DMSO vehicle control (triplicate)',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  '<p><strong>System:&nbsp;</strong>DMPC (dimyristoylphosphatidylcholine)&nbsp;bilayer in water.</p>\\n\\n<p><strong>Number of lipids:&nbsp;</strong>128.<br>\\n<strong>Number of waters:</strong>&nbsp;5097.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;110 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;20 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  'TBI was induced with lateral fluid-percussion injury in adult male rats. Genome-wide RNA-seq of the perilesional cortex, ipsilateral thalamus and dorsal hippocampus was performed at 3 months post-TBI. The data highlighted chronic transcriptional changes, particularly, in the perilesional cortex and thalamus. Genes showing a significantly altered expression both in the cortex and thalamus were submitted to the LINCS web query to identify novel pharmacotherapies to improve post-TBI outcome.\\nTBI was induced to 5 rats, 5 sham operated served as a controls.',\n",
       "  'Genetic and transcriptional variation alters cancer cell line drug response',\n",
       "  'Genetic and transcriptional variation alters cancer cell line drug response [MCF7 strain L]',\n",
       "  'No description',\n",
       "  'Transcriptome-based drug screening is emerging as a powerful tool to identify geroprotective compounds to intervene in age-related disease. We hypothesized that, by mimicking the transcriptional signature of the highly conserved longevity intervention of FOXO3 (daf-16 in worms) overexpression, we could identify and repurpose compounds with similar downstream effects to increase longevity. Our in silico screen, utilizing the LINCS transcriptome database of genetic and compound interventions, identified several FDA-approved compounds that activate FOXO downstream targets in mammalian cells. These included the neuromuscular blocker atracurium, which also robustly extends both lifespan and healthspan in C. elegans. This longevity is dependent on both daf-16 signaling and inhibition of the neuromuscular acetylcholine receptor. Other neuromuscular blockers tubocurarine and pancuronium caused similar healthspan benefits. Together, these data demonstrate the capacity to mimic genetic lifespan interventions with drugs, and in doing so, reveal that the neuromuscular acetylcholine receptor regulates the highly conserved FOXO/DAF-16 longevity pathway. Overall design: 4 samples each of C. elegans treated for 24 hours from larval stage 4 to day 1 of adulthood with 50 µM atracurium besylate or water vehicle controls',\n",
       "  'Substantial effort in recent years has been devoted to analyzing data based large-scale biological networks, which provide valuable insight into the topologies of complex biological networks but are rarely context specific and cannot be used to predict the responses of cell signaling proteins to specific ligands or compounds. In this work, we proposed a novel strategy to investigate kinase inhibitor induced pathway signatures by integrating multiplex data in Library of Integrated Network-based Cellular Signatures (LINCS), e.g. KINOMEscan data and cell proliferation/mitosis imaging data. Using this strategy, we first established a PC9 cell line specific pathway model to investigate the pathway signatures in PC9 cell line when perturbed by a small molecule kinase inhibitor GW843682. This specific pathway revealed the role of PI3K/AKT in modulating the cell proliferation process and the absence of two anti-proliferation links, which indicated a potential mechanism of abnormal expansion in PC9 cell number. Incorporating the pathway model for side effects on primary human hepatocytes, it was used to screen 27 kinase inhibitors in LINCS database and PF02341066, known as Crizotinib, was finally suggested with an optimal concentration 4.6 uM to suppress PC9 cancer cell expansion while avoiding severe damage to primary human hepatocytes. Drug combination analysis revealed that the synergistic effect region can be predicted straightforwardly based on a threshold which is an inherent property of each kinase inhibitor. Furthermore, this integration strategy can be easily extended to other specific cell lines to be a powerful tool for drug screen before clinical trials.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'Objective: Hepatocellular carcinoma (HCC) is frequently diagnosed in patients with late-stage disease who are ineligible for curative surgical therapies. The majority of patients become resistant to sorafenib, the only approved first-line therapy for advanced cancer, underscoring the need for newer, more effective drugs. The purpose of this study is to expedite identification of novel drugs against sorafenib resistant (SR)-HCC. Methods: We employed a transcriptomics-based drug repurposing method termed connectivity mapping using gene signatures from in vitro-derived SR Huh7 HCC cells. For proof of concept validation, we focused on drugs that were FDA-approved or under clinical investigation and prioritized two anti-neoplastic agents (dasatinib and fostamatinib) with targets associated with HCC. We also prospectively validated predicted gene expression changes in drug-treated SR Huh7 cells as well as identified and validated the targets of Fostamatinib in HCC. Results: Dasatinib specifically reduced the viability of SR-HCC cells that correlated with up-regulated activity of SRC family kinases, its targets, in our SR-HCC model. However, fostamatinib was able to inhibit both parental and SR HCC cells in vitro and in xenograft models. Ingenuity pathway analysis of fostamatinib gene expression signature from LINCS predicted JAK/STAT, PI3K/AKT, ERK/MAPK pathways as potential targets of fostamatinib that were validated by Western blot analysis. Fostamatinib treatment reversed the expression of genes that were deregulated in SR HCC. Conclusion: We provide proof of concept evidence for the validity of this drug repurposing approach for SR-HCC with implications for personalized medicine.',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 6&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 94&nbsp;mol-%) lipids&nbsp;in 1.0 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;120.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;8.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;89.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;97.<br>\\n<strong>Number of waters:</strong>&nbsp;4921.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;230 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;12&nbsp;ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 50&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 50 mol-%) lipids&nbsp;in 1.0 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;64.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;64.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;94.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;158.<br>\\n<strong>Number of waters:</strong>&nbsp;5148.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;170 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;21 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  'Hepatocellular carcinoma (HCC) is frequently diagnosed in patients with late-stage disease who are ineligible for curative surgical therapies. Furthermore, the majority of patients become resistant to sorafenib. Recently, computational methods for drug repurposing have shown great promise to accelerate the discovery of new uses for existing drugs. In order to identify novel drugs for use against sorafenib resistant (SR)-HCC, we employed a transcriptomics-based drug repurposing method termed connectivity mapping. We conducted a comprehensive analysis of available in vitro and in vivo gene signatures of (SR)-HCC, and generated our own in vitro model using the Huh7 HCC cell line. We compared coverage of SR-HCC gene signatures across seven patient-derived HCC gene expression datasets, and observed that patients harboring the Huh7 SR-HCC gene signature had significantly reduced survival. Utilizing the Huh7 SR-HCC gene signature, we applied connectivity mapping to drug-induced gene expression profiles (n= 3,740 drugs) in the HepG2 HCC cell line from the LINCS database in order to find drugs that could oppose sorafenib resistance. We validated the use of two non-receptor tyrosine kinase inhibitors, dasatinib and fostamatinib, to reduce viability of sorafenib-resistant HCC cells and confirmed up-regulated activity of Src family kinases, the targets of dasatinib, in our SR-HCC models. We prospectively validated predicted gene expression changes in fostamatinib treated Huh7-SR via RNA-seq analysis. Overall design: Gene expression profiles of Huh7-SR hepatocellular carcinoma cells resistant to sorafenib were generated by RNAseq (Illumina HiSeq 4000) following treatment with fostamatinib and vehicle control.',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 75&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 25&nbsp;mol-%) lipids&nbsp;in 1.0 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;32.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;96.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;96.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;192.<br>\\n<strong>Number of waters:</strong>&nbsp;5304.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;110 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;11 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'SF295 glioblastoma cells were subjected to CRISPR/Cas9-mediated knockout of the MTF1 (metal regulatory transcription factor 1) gene or non-targeted control (GFP). Two guides targeting MTF1 and one guide targeting GFP were cloned into the pXPR_023 all-in-one CRISPR/Cas9 vector. SF295 cells were transduced by lentiviral infection and selected using puromycin. Following selection, RNA was harvested from cell lines growing in culture. Wild-type SF295 cells were also included as a control.Anti-cancer uses of non-oncology drugs have been found on occasion, but such discoveries have been serendipitous and rare. To fully discover activity against multiple tumor types, resource-intensive screening across hundreds of cancer cell lines is needed. We sought to create a public resource containing the growth inhibitory activity of 4,518 drugs tested across 578 human cancer cell lines. To accomplish this, we used PRISM, which involves the molecular barcoding of each cell line, followed by pooling of the barcoded lines. Relative barcode abundance following drug treatment versus controls thus reflects cell viability following drug treatment. We found that an unexpectedly large number of non-oncology drugs selectively inhibited subsets of cancer cell lines. Moreover, the killing activity of the majority of these drugs was predictable based on the molecular features of the cell lines. Mechanistic follow-up of several of these compounds revealed novel mechanisms. These results illustrate the potential of the PRISM drug repurposing resource as a starting point for new oncology therapeutic development.\\nEleven biological samples are included in the study. SF295 wild-type cells (control, duplicate), SF295 GFP sg1 (non-targeting guide control, triplicate), SF295 MTF1 sg1 (triplicate), and SF295 MTF1 sg2 (triplicate).',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 442 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag. Kd values (nM) reported for each compound were determined using 11 serial threefold dilutions of test compound and a DMSO control. A null result means no inhibition of kinase binding to the ligand in the presence of the compound and low Kd means strong inhibition.',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 75&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 25&nbsp;mol-%) lipids&nbsp;in water.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;32.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;96.<br>\\n<strong>Number of Cl--ions:</strong>&nbsp;96.<br>\\n<strong>Number of waters:</strong>&nbsp;5496.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;109&nbsp;ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;31&nbsp;ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 50 mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 50 mol-%) lipids&nbsp;in 0.1 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;64.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;64.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;10.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;74.<br>\\n<strong>Number of waters:</strong>&nbsp;5316.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;190 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;21 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 75&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 25&nbsp;mol-%) lipids&nbsp;in 0.1 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;32.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;96.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;10.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;106.<br>\\n<strong>Number of waters:</strong>&nbsp;5476.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;110 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;11 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  'KR-37524, known as a Urotensin-II receptor antagonist, has low usability due to its low affinity. For this reason, gene expression information was obtained by treating KR-37524 in four cell cancer cell lines to find a new target. The change in gene expression profile was measured and compared with the LINCS L1000 database and used to find a new target.\\nSix sheets of each cell line were seeded in a 60mm dish and incubated for 24 hours. After 24 hours, DMSO and KR-37524 were diluted 1000-fold in medium and the cells were treated. After 6 hours, the medium was removed and cells were obtained by treatment with 1ml of TRIzol. Cells were disrupted by pipetting and stored in a deep freezer.',\n",
       "  'FOXM1 transcription factor is an oncogene and a crucial regulator of cancer chemoresistance. Existing pharmacological FOXM1 inhibitors attenuate resistance of cancer cells to treatment but exhibit insufficient specificity. We performed in silico screening of small molecules from NIH LINCS database to identify STL427944 compound that suppresses FOXM1 through a novel two-step mechanism. STL427944 treatment induces relocalization of nuclear FOXM1 protein to cytoplasm, where it is subsequently degraded in an autophagosome-dependent manner. Human cancer cells treated with STL427944 exhibit increased sensitivity to cytotoxic effects of conventional chemotherapeutic agents. Transcriptome-based analysis of STL427944-induced gene expression changes revealed prominent suppression of gene signatures characteristic for FOXM1 and its downstream targets but no significant changes in other important regulatory pathways, thereby suggesting high selectivity of STL427944 towards FOXM1. We expect that STL427944 or its derivatives may be used in combination with different chemotherapeutical drugs to overcome tumor chemoresistance and improve anti-cancer treatment efficacy. Overall design: For each experimental model and condition, two biological replicates were analyzed. Control samples were treated with vehicle alone (DMSO), with vehicle percentage not exceeding 0.3%.',\n",
       "  'TBI was induced with lateral fluid-percussion injury in adult male rats. Genome-wide RNA-seq of the perilesional cortex, ipsilateral thalamus and dorsal hippocampus was performed at 3 months post-TBI. The data highlighted chronic transcriptional changes, particularly, in the perilesional cortex and thalamus. Genes showing a significantly altered expression both in the cortex and thalamus were submitted to the LINCS web query to identify novel pharmacotherapies to improve post-TBI outcome. Overall design: TBI was induced to 5 rats, 5 sham operated served as a controls.',\n",
       "  'The neuronal ceroid lipofuscinoses are a class of inherited neurodegenerative diseases characterized by the accumulation of autofluorescent storage material. The most common neuronal ceroid lipofuscinosis has juvenile onset with rapid onset blindness and progressive degeneration of cognitive processes. The juvenile form is caused by mutations in the CLN3 gene, which encodes the protein CLN3. While mouse models of Cln3 deficiency show mild disease phenotypes, it is apparent from patient tissue- and cell-based studies that its loss impacts many cellular processes. Using Cln3 deficient mice, we previously described defects in mouse brain endothelial cells and blood-brain barrier (BBB) permeability. Here we expand on this to other components of the BBB and show that Cln3 deficient mice have increased astrocyte endfeet area. Interestingly, this phenotype is corrected by treatment with a commonly used GAP junction inhibitor, carbenoxolone (CBX). In addition to its action on GAP junctions, CBX has also been proposed to alter lipid microdomains. In this work, we show that CBX modifies lipid microdomains and corrects membrane fluidity alterations in Cln3 deficient endothelial cells, which in turn improves defects in endocytosis, caveolin-1 distribution at the plasma membrane, and Cdc42 activity. In further work using the NIH Library of Integrated Network-based Cellular Signatures (LINCS), we discovered other small molecules whose impact was similar to CBX in that they improved Cln3-deficient cell phenotypes. Moreover, Cln3 deficient mice treated orally with CBX exhibited recovery of impaired BBB responses and reduced autofluorescence. CBX and the compounds identified by LINCS, many of which have been used in humans or approved for other indications, may find therapeutic benefit in children suffering from CLN3 deficiency through mechanisms independent of their original intended use.',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 50&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 50 mol-%) lipids&nbsp;in 0.5&nbsp;M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;64.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;64.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;48.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;112.<br>\\n<strong>Number of waters:</strong>&nbsp;5240.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;179&nbsp;ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;21 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 6 mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 94 mol-%) lipids&nbsp;in water.</p>\\n\\n<p><strong>Number of DMPC:&nbsp;</strong>120.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;8.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;8.<br>\\n<strong>Number of waters:</strong>&nbsp;5099.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J. </em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;109&nbsp;ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;32&nbsp;ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 75&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 25&nbsp;mol-%) lipids&nbsp;in 0.5 M NaCl solution.</p>\\n\\n<p><strong>Number of DMPC:</strong>&nbsp;32.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;96.<br>\\n<strong>Number of Na<sup>+</sup>-ions:</strong>&nbsp;49.<br>\\n<strong>Number of Cl<sup>-</sup>-ions:</strong>&nbsp;145.<br>\\n<strong>Number of waters:</strong>&nbsp;5398.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;110 ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;11 ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  '<p><strong>System:&nbsp;</strong>Symmetric bilayer of cationic&nbsp;DMTAP (dimyristoyltrimethylammoniumpropane, 50&nbsp;mol-%) and&nbsp;zwitterionic DMPC (dimyristoylphosphatidylcholine, 50 mol-%) lipids&nbsp;in water.</p>\\n\\n<p><strong>Number of DMPC:&nbsp;</strong>64.<br>\\n<strong>Number of DMTAP:</strong>&nbsp;64.<br>\\n<strong>Number of Cl--ions:</strong>&nbsp;64.<br>\\n<strong>Number of waters:</strong>&nbsp;5336.</p>\\n\\n<p><strong>Lipid model:</strong>&nbsp;&#39;Berger&#39; united-atom [<em>Biophys. J.</em>&nbsp;<strong>72</strong>&nbsp;2002 (1997)] with&nbsp;DMTAP&nbsp;modification&nbsp;by&nbsp;Gurtovenko et al. [<em>Biophys. J.&nbsp;</em><strong>86</strong>&nbsp;3461 (2004)].<br>\\n<strong>Water model:</strong>&nbsp;SPC [In&nbsp;<em>Intermolecular Forces,</em>&nbsp;ed. Pullman. 331 (1981)].</p>\\n\\n<p><strong>Simulation engine:</strong>&nbsp;Gromacs 3.x [www.gromacs.org]</p>\\n\\n<p><strong>Trajectory length:</strong>&nbsp;149&nbsp;ns.<br>\\n<strong>Previously equilibrated for:</strong>&nbsp;41&nbsp;ns.<br>\\n<strong>Sampling rate:</strong>&nbsp;every 10 ps.</p>\\n\\n<p><strong>Time integration step:</strong>&nbsp;2 fs.</p>\\n\\n<p><strong>Thermodynamic ensemble:</strong>&nbsp;NpT.&nbsp;<br>\\n<strong>Temperature coupling:</strong>&nbsp;&#39;Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with lipids and water coupled separately at T = 323 K.<br>\\n<strong>Pressure coupling: &#39;</strong>Berendsen&#39; [<em>J. Chem. Phys.</em>&nbsp;<strong>81</strong>&nbsp;3684 (1984)] with xy and z coupled separately at p = 1.0 bar.</p>\\n\\n<p><strong>Electrostatics:&nbsp;</strong>PME [<em>J. Chem. Phys.</em>&nbsp;<strong>98</strong>&nbsp;10089 (1993);&nbsp;<em>J. Chem. Phys.</em>&nbsp;<strong>103</strong>&nbsp;8577 (1995)], real-space cutoff at 1.0 nm.<br>\\n<strong>Van der Waals:</strong>&nbsp;Truncated at 1.0 nm.</p>\\n\\n<p><strong>Constraints:&nbsp;</strong>Covalent bond lengths in lipids using LINCS [<em>J. Comput. Chem.</em>&nbsp;<strong>18</strong>&nbsp;1463 (1997)], in water using SETTLE [J. Comput. Chem.&nbsp;<strong>13</strong>&nbsp;952 (1992)].</p>\\n\\n<p><strong>Used in publications:&nbsp;</strong>[1]&nbsp;Markus S. Miettinen, Andrey A. Gurtovenko, Ilpo Vattulainen, and Mikko Karttunen: &quot;Ion Dynamics in Cationic Lipid Bilayer Systems in Saline Solutions&quot;.&nbsp;<em>J. Phys. Chem. B</em>&nbsp;<strong>113</strong>&nbsp;9226 (2009). DOI:&nbsp;10.1021/jp810233q. [2] Markus S. Miettinen: &quot;Computational Modeling of Cationic Lipid Bilayers in Saline Solutions&quot;. PhD Thesis.&nbsp;Aalto University School of Science and Technology, Helsinki, Finland. (2010). ISBN&nbsp;978-952-60-3194-1.</p>',\n",
       "  \"Motivation:Several large-scale efforts have been made to collect gene expression signatures from a variety of biological conditions, such as response of cell lines to treatment with drugs, or tumor samples with different characteristics. These gene signature collections are utilized through bioinformatics tools for 'signature matching', whereby a researcher studying an expression profile can identify previously cataloged biological conditions most related to their profile. Signature matching tools typically retrieve from the collection the signature that has highest similarity to the user-provided profile. Alternatively, classification models may be applied where each biological condition in the signature collection is a class label; however, such models are trained on the collection of available signatures and may not generalize to the novel cellular context or cell line of the researcher's expression profile. Results:We present an advanced multi-way classification algorithm for signature matching, called SigMat, that is trained on a large signature collection from a well-studied cellular context, but can also classify signatures from other cell types by relying on an additional, small collection of signatures representing the target cell type. It uses these 'tuning data' to learn two additional parameters that help adapt its predictions for other cellular contexts. SigMat outperforms other similarity scores and classification methods in identifying the correct label of a query expression profile from as many as 244 or 500 candidate classes (drug treatments) cataloged by the LINCS L1000 project. SigMat retains its high accuracy in cross-cell line applications even when the amount of tuning data is severely limited. Availability and implementation:SigMat is available on GitHub at https://github.com/JinfengXiao/SigMat. Supplementary information:Supplementary data are available at Bioinformatics online.\",\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'The KINOMEscan assay platform is based on a competition binding assay that is run for a compound of interest against each of a panel of 317 to 456 kinases. The assay has three components: a kinase-tagged phage, a test compound, and an immobilized ligand that the compound competes with to displace the kinase. The amount of kinase bound to the immobilized ligand is determined using quantitative PCR of the DNA tag.  Results for each kinase are reported as \"Percent of control\", where the control is DMSO and where a 100% result means no inhibition of kinase binding to the ligand in the presence of the compound, and where low percent results mean strong inhibition.',\n",
       "  'Ozone is a common, potent oxidant pollutant in industrialized nations. Ozone exposure causes airway hyperreactivity, lung hyperpermeability, inflammation, and cell damage in humans and laboratory animals, and exposure to ozone has been associated with exacerbation of asthma, altered lung function, and mortality. The mechanisms of ozone-induced lung injury and differential susceptibility are not fully understood. Ozone-induced lung inflammation is mediated, in part, by the innate immune system. We hypothesized that mannose-binding lectin (MBL), an innate immunity serum protein, contributes to the proinflammatory events caused by ozone-mediated activation of the innate immune system. Wild-type (Mbl(+/+)) and MBL-deficient (Mbl(-/-)) mice were exposed to ozone (0.3 ppm) for up to 72 h, and bronchoalveolar lavage fluid was examined for inflammatory markers. Mean numbers of eosinophils and neutrophils and levels of the neutrophil attractants C-X-C motif chemokines 2 [Cxcl2 (major intrinsic protein 2)] and 5 [Cxcl5 (limb expression, LIX)] in the bronchoalveolar lavage fluid were significantly lower in Mbl(-/-) than Mbl(+/+) mice exposed to ozone. Using genome-wide mRNA microarray analyses, we identified significant differences in transcript response profiles and networks at baseline [e.g., nuclear factor erythroid-related factor 2 (NRF2)-mediated oxidative stress response] and after exposure (e.g., humoral immune response) between Mbl(+/+) and Mbl(-/-) mice. The microarray data were further analyzed to discover several informative differential response patterns and subsequent gene sets, including the antimicrobial response and the inflammatory response. We also used the lists of gene transcripts to search the LINCS L1000CDS(2) data sets to identify agents that are predicted to perturb ozone-induced changes in gene transcripts and inflammation. These novel findings demonstrate that targeted deletion of Mbl caused differential levels of inflammation-related gene sets at baseline and after exposure to ozone and significantly reduced pulmonary inflammation, thus indicating an important innate immunomodulatory role of the gene in this model.',\n",
       "  'SUMMARY:Mechanistic molecular studies in biomedical research often discover important genes that are aberrantly over- or under-expressed in disease. However, manipulating these genes in an attempt to improve the disease state is challenging. Herein, we reveal Drug Gene Budger (DGB), a web-based and mobile application developed to assist investigators in order to prioritize small molecules that are predicted to maximally influence the expression of their target gene of interest. With DGB, users can enter a gene symbol along with the wish to up-regulate or down-regulate its expression. The output of the application is a ranked list of small molecules that have been experimentally determined to produce the desired expression effect. The table includes log-transformed fold change, P-value and q-value for each small molecule, reporting the significance of differential expression as determined by the limma method. Relevant links are provided to further explore knowledge about the target gene, the small molecule and the source of evidence from which the relationship between the small molecule and the target gene was derived. The experimental data contained within DGB is compiled from signatures extracted from the LINCS L1000 dataset, the original Connectivity Map (CMap) dataset and the Gene Expression Omnibus (GEO). DGB also presents a specificity measure for a drug-gene connection based on the number of genes a drug modulates. DGB provides a useful preliminary technique for identifying small molecules that can target the expression of a single gene in human cells and tissues. AVAILABILITY AND IMPLEMENTATION:The application is freely available on the web at http://DGB.cloud and as a mobile phone application on iTunes https://itunes.apple.com/us/app/drug-gene-budger/id1243580241? mt=8 and Google Play https://play.google.com/store/apps/details? id=com.drgenebudger. SUPPLEMENTARY INFORMATION:Supplementary data are available at Bioinformatics online.',\n",
       "  'Purpose: Most prostate cancers (PCs) initially respond to androgen deprivation therapy (ADT), but eventually many PC patients develop castration resistant PC (CRPC). Currently, available drugs that have been approved for the treatment of CRPC patients are limited. Computational drug repositioning methods using public databases represent a promising and efficient tool for discovering new uses for existing drugs. The purpose of the present study is to predict drug candidates that can treat CRPC using a computational method that integrates publicly available gene expression data of tumors from CRPC patients, drug-induced gene expression data and drug response activity data. Methods: Gene expression data from tumoral and normal or benign prostate tissue samples in CRPC patients were downloaded from the Gene Expression Omnibus (GEO) and differentially expressed genes (DEGs) in CRPC were determined with a meta-signature analysis by a metaDE R package. Additionally, drug activity data were downloaded from the ChEMBL database. Furthermore, the drug-induced gene expression data were downloaded from the LINCS database. The reversal relationship between the CRPC and drug gene expression signatures as the Reverse Gene Expression Scores (RGES) were computed. Drug candidates to treat CRPC were predicted using summarized scores (sRGES). Additionally, synergic effects of drug combinations were predicted with a Target Inhibition interaction using the Minimization and Maximization Averaging (TIMMA) algorithm. Results: The drug candidates of sorafenib, olaparib, elesclomol, tanespimycin, and ponatinib were predicted to be active for the treatment of CRPC. Meanwhile, CRPC-related genes, in this case MYL9, E2F2, APOE, and ZFP36, were identified as having gene expression data that can be reversed by these drugs. Additionally, lenalidomide in combination with pazopanib was predicted to be most potent for CRPC. Conclusion: These findings support the use of a computational reversal gene expression approach to identify new drug and drug combination candidates that can be used to treat CRPC.',\n",
       "  'Cystic fibrosis is a lethal autosomal recessive disease caused by mutations in the cystic fibrosis transmembrane conductance regulator (CFTR) gene. The common delF508-CFTR mutation results in protein misfolding and proteasomal degradation. If delF508-CFTR trafficks to the cell surface, its anion channel function may be partially restored. Several in vitro strategies can partially correct delF508-CFTR trafficking and function, including low-temperature, small molecules, overexpression of miR-138, or knockdown of SIN3A. The challenge remains to translate such interventions into therapies and to understand their mechanisms. One approach for connecting such interventions to small molecule therapies that has previously succeeded for cystic fibrosis and other diseases is via mRNA expression profiling and iterative searches of small molecules with similar expression signatures. Here, we query the Library of Integrated Network-based Cellular Signatures (LINCS) using transcriptomic signatures from previously generated CF expression data, including RNAi- and low temperature-based rescue signatures. This LINCS in silico screen prioritized 135 small molecules that mimicked our rescue interventions based on their genome-wide transcriptional perturbations. Functional screens of these small molecules identified nine compounds that partially restored delF508-CFTR function, as assessed by cAMP-activated chloride conductance. Of these, XL147 rescued delF508-CFTR function in primary CF airway epithelia, while also showing cooperativity when administered with C18. Improved CF corrector therapies are greatly needed and this integrative drug prioritization approach offers a novel method to both identify small molecules that may rescue delF508-CFTR function and identify gene networks underlying such rescue.',\n",
       "  'Background:Epidemiological and clinical evidence points to cancer as a comorbidity in people with autism spectrum disorders (ASD). A significant overlap of genes and biological processes between both diseases has also been reported. Methods:Here, for the first time, we compared the gene expression profiles of ASD frontal cortex tissues and 22 cancer types obtained by differential expression meta-analysis and report gene, pathway, and drug set-based overlaps between them. Results:Four cancer types (brain, thyroid, kidney, and pancreatic cancers) presented a significant overlap in gene expression deregulations in the same direction as ASD whereas two cancer types (lung and prostate cancers) showed differential expression profiles significantly deregulated in the opposite direction from ASD. Functional enrichment and LINCS L1000 based drug set enrichment analyses revealed the implication of several biological processes and pathways that were affected jointly in both diseases, including impairments of the immune system, and impairments in oxidative phosphorylation and ATP synthesis among others. Our data also suggest that brain and kidney cancer have patterns of transcriptomic dysregulation in the PI3K/AKT/MTOR axis that are similar to those found in ASD. Conclusions:Comparisons of ASD and cancer differential gene expression meta-analysis results suggest that brain, kidney, thyroid, and pancreatic cancers are candidates for direct comorbid associations with ASD. On the other hand, lung and prostate cancers are candidates for inverse comorbid associations with ASD. Joint perturbations in a set of specific biological processes underlie these associations which include several pathways previously implicated in both cancer and ASD encompassing immune system alterations, impairments of energy metabolism, cell cycle, and signaling through PI3K and G protein-coupled receptors among others. These findings could help to explain epidemiological observations pointing towards direct and inverse comorbid associations between ASD and specific cancer types and depict a complex scenario regarding the molecular patterns of association between ASD and cancer.',\n",
       "  '\"The NeuroLINCS Center is part of the NIH Common Fund\\'\\'s Library of Integrated Network-based Cellular Signatures (LINCS) program, which aims to characterize how a variety of human cells, tissues and the entire organism respond to perturbations by drugs and other molecular factors.  As Part of the LINCS program, the NeuroLINCS study concentrates on human brain cells, which are far less understood than other cells in the body. Our initial focus is to produce diseased motor neurons from patients by utilizing high-quality induced pluripotent stem cell (iPSC) lines from Amyotrophic Lateral Sclerosis (ALS) and Spinal Muscular Atrophy (SMA) patients in addition to unaffected normal healthy controls. Using state-of-the-art OMICS methods (genomics, epigenomics, transcriptomics, and proteomics), we intend to create a wealth of cellular data that is patient-specific in the context of their baseline genetic perturbations and in the presence of other genetic and environmental perturbagens... (for more see dbGaP study page.)\"',\n",
       "  'This SuperSeries is composed of the SubSeries listed below.\\nRefer to individual Series',\n",
       "  'Transcriptome-based drug screening is emerging as a powerful tool to identify geroprotective compounds to intervene in age-related disease. We hypothesized that, by mimicking the transcriptional signature of the highly conserved longevity intervention of FOXO3 (daf-16 in worms) overexpression, we could identify and repurpose compounds with similar downstream effects to increase longevity. Our in silico screen, utilizing the LINCS transcriptome database of genetic and compound interventions, identified several FDA-approved compounds that activate FOXO downstream targets in mammalian cells. These included the neuromuscular blocker atracurium, which also robustly extends both lifespan and healthspan in C. elegans. This longevity is dependent on both daf-16 signaling and inhibition of the neuromuscular acetylcholine receptor. Other neuromuscular blockers tubocurarine and pancuronium caused similar healthspan benefits. Together, these data demonstrate the capacity to mimic genetic lifespan interventions with drugs, and in doing so, reveal that the neuromuscular acetylcholine receptor regulates the highly conserved FOXO/DAF-16 longevity pathway.\\n4 samples each of C. elegans treated for 24 hours from larval stage 4 to day 1 of adulthood with 50 μM atracurium besylate or water vehicle controls',\n",
       "  'Expression of genes highly correlated with TACSTD2 expression in breast tumors and corresponding patient-derived xenografts (PDXs) and found that 29 genes were overlapping between in 18 breast cancer samples and matching PDXs.  Among these genes, TACSTD2 expression strongly correlated with expression of epithelial marker CDH1 (E-Cadherin) (r2=0.823; p<0.001).  Expression of TACSTD2 was then compared to expression of CDH1 in breast cancer cell lines from The Cancer Cell Line Encyclopedia (CCLE) (N = 51) and The Library of Integrated Network-Based Cellular Signatures (LINCS) (N = 35).  There was also a significant correlation between TACSTD2 and CDH1 in both CCLE (r2=0.564, p < 0.001) and LINCS (r2= 0.755; p < 0.001).\\nComparision of gene expression between PDX and tumors.',\n",
       "  'Non-small cell lung cancer (NSCLC) comprises the majority (~85%) of all lung tumors, with lung adenocarcinoma (LUAD) and squamous cell carcinoma (LUSC) being the most frequently diagnosed histological subtypes. Currently, multi-modal omics profiles had been carried out in NSCLC, but no studies reported yet a systems biology approach to provide a complete picture of molecular perturbations specifically for LUAD and LUSC.\\nIn this study, we integrated transcriptomics, metabolomics and adducteomics for the first time to analyze 30 paired human lung tumors and adjacent noncancerous tissues. We portrayed unique metabolite related gene signatures to distinguish LUAD from LUSC and several altered pathways aberrantly expressed at both transcriptional and metabolic levels. Additionally, the integration studies revealed a 28-metabolite gene signature in TCGA-LUAD cohort and demonstrated a significant association with survival, especially for 6 metabolite related genes as therapeutic targets. To further explore, we identified AZD-6482, a clinical trial drug from LINCS/connectivity map and found that it significantly inhibited 3 genes from the 28-gene signature. By integrating adducteomics, we found a differential expression of 307 genes correlated with 6 DNA adducts in LUSC although LUAD did not show much correlation. We found that γ-hydroxy-propanodeoxyguanosine (γ-OH-PdG) is the major DNA adduct formed in LUSC, but not in adjacent noncancerous tissue and significantly correlated with 295 modulated genes as well. We also have identified aberrant signaling pathways for each specific DNA adducts in LUSC compared to adjacent normal. Overall, we have provided a broad picture of integration of omics that can distinguish LUAD and LUSC, prediction of patient survival and identification of therapeutic targets and anticancer drugs. Ultimately, genes modulated by DNA adducts in LUSC could serve as biomarkers or therapeutic targets for cigarette smoke induced lung cancer.',\n",
       "  'Sampattavanich et al., Cell Systems (2018)\\n\\nThis data repository provides raw datasets for accompanying the paper, \"Encoding growth factor identity in the temporal dynamics of FoxO3 under combinatorial regulation by ERK and Akt\" by Sampattavanich et al., Cell Systems (2018)\\n\\nSynopsis:\\nForkhead box O3 transcription factor (FoxO3) functions as an integrative node for diverse upstream signaling networks, and has been implicated in a number of biological processes including cycle arrest, apoptosis, oxidative stress, cell migration and cell metabolism. Regulators of FoxO3, including phosphorylation, energy stress, genotoxic stress, and cytokines, control its abundance, subcellular localization and DNA-binding capacity. In this paper we study how the identities and concentrations of growth factors are encoded in the dynamics of FoxO3 activation. Transcription factors like FoxO3 often switch between on and off states repeatedly over the course of a 12-24 hours period. Immunoblotting and similar methods can obscure this switching behavior due to asynchronicity between cells. In contrast, our use of time-lapse microscopy of fluorescent reporter proteins reveals that FoxO3 exhibits complex low wavelength and pulsatile translocation dynamics responsive to combinatorial control by ERK and Akt with the potential to encode the identities and concentrations of multiple extracellular growth factors.\\n\\nDataset Content:\\n\\nThis dataset includes raw datasets either in .MAT, or .CSV for regenerating all figures in the publication.  These folders should be placed inside the /rawdata folder in the git repository folder so that the reader can generate these figures on their own.  \\n/parentalVSreporter - This dataset is used to regenerate fig.1B, S1B and S1C. \\n/Workspace - This dataset is important for all fPCA and pulse score analysis in Fig.1-6\\n/dualsensors - This dataset is used to regenerate Fig.6B\\n/western - This dataset is used to regenerate Fig.1C and S1G\\n/fixedcell - This dataset is used to regenerate Fig.7\\n/Time-lapse video - This folder gives video files of the main dataset for Fig.1-2, varying different growth factors at various dosing, with different drug pre-treatment. \\n\\nTo learn more about this data, please also visit the accompanying HMS LINCS website shown below.',\n",
       "  '<p><strong>Improving the Usability of a MAS DSML</strong></p>\\n\\n<p>Tom&aacute;s Miranda, Moharram Challenger, Baris Tezel, Omer Faruk Alaca, Vasco Amaral, Miguel Goul&atilde;o, and Geylani Kardas</p>\\n\\n<p>Paper accepted for publication in the <a href=\"http://emas2018.dibris.unige.it/\">6th International Workshop on Engineering Multi-Agent Systems (EMAS 2018)</a></p>\\n\\n<p>&nbsp;</p>\\n\\n<p><em><strong>Abstract&nbsp;</strong></em></p>\\n\\n<p><strong>&nbsp;Context:&nbsp;</strong>A significant effort has been devoted to the design and implementation of various domain-specific modeling languages (DSMLs) for the software agents domain.</p>\\n\\n<p><strong>Problem:&nbsp;</strong>Language usability is often tackled in an ad-hoc way, with the collection of anecdotal evidence supporting the process. However, usability plays an important role in the productivity, learnability and, ultimately, in the adoption of a MAS DSML by agent developers.</p>\\n\\n<p><strong>Method:&nbsp;</strong>In this paper, we apply the principles of The &ldquo;Physics&rdquo; of Notations (PoN) to improve the visual notation of a MAS DSML, called SEA_ML and evaluate the result in terms of usability.</p>\\n\\n<p><strong>Results:&nbsp;</strong>The evolved version of the language, SEA_ML++, was perceived as significantly improved in terms of icons comprehensibility, adequacy and usability, as a direct result of employing the principles of PoN. However, users were not significantly more efficient and effective with SEA_ML++, suggesting these two properties were not chiefly constrained by the identified shortcomings of the SEA_ML concrete syntax.</p>\\n\\n<p><em><strong>Study Participants</strong></em></p>\\n\\n<p><strong>Summary.&nbsp;</strong>This evaluation is comprised of a set of studies, each with its own subset of participants. No participants were involved in more than one study.</p>\\n\\n<ul>\\n\\t<li>Symbol selection study (25 participants)</li>\\n\\t<li>Concrete syntax evaluation experiment (24 + 12 participants)</li>\\n</ul>\\n\\n<p><strong>Participants in the symbol selection study.&nbsp;</strong>The symbol selection experiment had 25 participants, 20 males, 5 females, all trained in Computer Science. Of these, 5 had completed their MSc, 16 their BSc and 4 were undergraduate Computer Science students at UNL. 11 had learned about MAS in the context of a course (although not using any of the particular languages in this experiment). The remaining 14 were not familiar with MAS. 9 participants had learned about the semantic web in the context of a course, 9 had only informal knowledge about it and the remaining 7 were not familiar with the semantic web.</p>\\n\\n<p>More details concerning the development of the SEA_ML++ version of SEA_ML can be found in: <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/SEA_MLPoN.pdf\">SEA_MLPoN.pdf</a> .</p>\\n\\n<p><strong>Participants in the concrete syntax evaluation experiment.&nbsp;</strong>This evaluation experiment was run as a pair of replicated studies. The first replica was run in Universidade Nova de Lisboa, in Portugal. The second replica was run in the International Computer Institute, Ege University, Izmir, in Turkey.</p>\\n\\n<ul>\\n\\t<li><strong>UNL evaluation.&nbsp;</strong>At UNL we had 24 participants, including 19 males and 5 females, all trained in Computer Science. Of these, 8 have completed their MSc, 15 their BSc and 1 is an undergraduate Computer Science student. 12 of them had learned about MAS in the context of a course (although not using any of the particular languages in this experiment), and 1 had only informal knowledge about MAS. The remaining 11 were not familiar with MAS. 14 participants had learned about semantic web in the context of a course, 5 had informal knowledge about it, while 4 were not familiar with the semantic web.&nbsp;</li>\\n\\t<li><strong>EGE evaluation.&nbsp;</strong>At Ege University, we had 12 participants, all Computer Science graduate students who had learned about MAS in the context of a course.</li>\\n</ul>\\n\\n<p>&nbsp;</p>\\n\\n<p><em><strong>Case studies</strong></em></p>\\n\\n<p><strong>Music Trader.&nbsp;</strong>In this case study, participants are requested to develop a system that allows agents to trade their music albums without using any currency. Agents want to trade their music albums for other albums, with this trade being made on an N to N basis (Agent A wants to trade the album A1 for the album B1 from Agent B and vice versa). Agents are not able to trade more than one album for only one album.)</p>\\n\\n<p><strong>Expert Finding.&nbsp;</strong>In this case study, participants are requested to develop a system that allows agents to find information about other agents that they are searching for in order to communicate with them. Agents have some information about the other agent they are looking for (they are family related or were friends at the past), which is crucial in order to find the correct SemanticWeb Service to search the right person. The communication between agents can be made through Social Networks, E-Mail, VoIP or Phone Call. This case study is an adaptation of the case study &quot;Expert Finding&quot;&nbsp;for an evaluation of SEA_ML.</p>\\n\\n<p><em><strong>Exercises</strong></em></p>\\n\\n<p>Each case study presents the participants with 2 different exercises (4 exercises in total). Each exercise has a description that defines all variables of the system to be modeled. For each exercise, an incomplete version of this system is presented. Participants should read the description that is provided to them and compare to the model they have in hands. When the participant thinks the model is according to the description, the exercise is complete, passing to an inquiry about the system they have modeled and afterward to the next exercise. Each exercise should take around 10 minutes to be completed. The total experiment should take around 40 minutes.</p>\\n\\n<p>A short description of each exercise is as follows:</p>\\n\\n<p>1. Music Trader: Exercise 1 &mdash; In this exercise, the participants will be modeling the M.A.S viewpoint. An environment and one customer are missing from the original model. A model is&nbsp;presented to the participant for this exercise (either in SEA_ML or in SEA_ML++).</p>\\n\\n<p>2. Music Trader: Exercise 2 &mdash; In this exercise, the participants will be modeling the Agent-SWS viewpoint. A SS_RegisterPlan is missing from the original model. A model is&nbsp;presented to the participant for this exercise (either in SEA_ML or in SEA_ML++).</p>\\n\\n<p>3. Expert Finding: Exercise 1 &mdash; In this exercise, the participants will be modeling the Agent Internal viewpoint. A goal, a belief and a behavior (and its respective connections) are missing from the original model. A model is&nbsp;presented to the participant for this exercise (either in SEA_ML or in SEA_ML++).</p>\\n\\n<p>4. Expert Finding: Exercise 2&mdash;In this exercise, the participants will be modeling the Ontology viewpoint. A fact and a semantic web organization are missing from the original model. A model is&nbsp;presented to the participant for this exercise (either in SEA_ML or in SEA_ML++).</p>\\n\\n<p>&nbsp;</p>\\n\\n<p><em><strong>How does the usage of SEA_ML vs SEA_ML++ affect the effectiveness and efficiency of developers?</strong></em></p>\\n\\n<p><strong>Summary.&nbsp;</strong>We briefly present the data analysis for comparing the effectiveness and efficiency of developers with the case studies, using SEA_ML and SEA_ML++. The data collection was performed in two replicate experiments: one conducted in Universidade Nova de Lisboa, the other in EGE University. The tasks and experimental design were identical and planned so that we are able to aggregate the results. In the end, we outline how the data collected in each of these replications compares with each other.</p>\\n\\n<p><strong>Descriptive statistics.&nbsp;</strong>We start by presenting the descriptive statistics for the correctness and duration of the task. The correction is measured as the percentage of elements correctly identified by participants in our case studies. The duration measures how long it took them to perform the&nbsp;task (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/CorrectnessEfficiencyDescriptives.pdf?versionId=9e46c43e-349f-4ec6-a97c-fa358cee72bf\">CorrectnessEfficiencyDescriptives.pdf</a>).&nbsp;</p>\\n\\n<p><strong>Normality tests.</strong> We conducted a Kolmogorov-Smirnov and a Shapiro-Wilk tests to check for normality. Neither the correctness nor the duration have a normal distribution, p &lt; 0.001 (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/CorrectnessEfficiencyNormality.pdf?versionId=f27fe63c-45c0-4e41-869d-f91af58bc97e\">CorrectnessEfficiencyNormality.pdf</a>).</p>\\n\\n<p><strong>Hypothesis testing.&nbsp;</strong>Our null hypotheses are that:</p>\\n\\n<ul>\\n\\t<li>Using SEA_ML and SEA_ML++ has no impact in the correctness of the answers provided by our participants.</li>\\n\\t<li>Using SEA_ML and SEA_ML++ has no impact in the duration of the tasks performed by our participants.</li>\\n</ul>\\n\\n<p>We summarize the main descriptive statistics here. We note a slight increase from .8021 to .8368 in the achieved correctness and a slight decrease in the mean time to complete the tasks, from 14m48s to 13m20s (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/CorrectnessEfficiencySummary.pdf?versionId=b1d2d603-74fe-4149-89e2-484efcdd6998\">CorrectnessEfficiencySummary.pdf</a>).</p>\\n\\n<p>We can visually compare the distributions of correctness and duration, depending on the particular version of the language. Concerning correctness, as we can observe, the mode is 1.00 for both languages, although the distribution seems to be more skewed towards the highest rank. The values below 1.00 are considered extreme values, for SEA_ML++, meaning they are relatively scarce. It was relatively infrequent not to obtain maximum correctness with SEA_ML++ (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/CorrectnessBoxplot.pdf?versionId=2c15f46f-c02d-4a0a-8005-09734fe2f2af\">CorrectnessBoxplot.pdf</a>).</p>\\n\\n<p>Concerning duration, the distributions are even more similar (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/EfficiencyBoxplot.pdf?versionId=f875ffcc-b6dc-4116-9d3e-839b3f1b8b09\">EfficiencyBoxplot.pdf</a>).&nbsp;</p>\\n\\n<p>In order to assess whether there is any statistically significant difference between correctness using SEA_ML vs SEA_ML++ and between duration, again using SEA_ML vs SEA_ML++, we conducted a Welch T test, which is robust to deviations from normality, unequal sample sizes and does not assume equal variances of the compared variables (see <a href=\"https://zenodo.org/api/files/b8f7c0ab-6121-479c-b356-440ff1cde8e1/WelchTTestCorrectnessEfficiency.pdf?versionId=3caff774-4f6f-4395-81dc-37cd7b6f3be1\">WelchTTestCorrectnessEfficiency.pdf</a>).</p>\\n\\n<p>The <em>correctness does not differ significantly</em>, according to Welch&rsquo;s&nbsp;t-test,&nbsp;t(141.968) = .417,&nbsp;p=.519 from the SEA_ML (M = .80,&nbsp; SD&nbsp;=&nbsp;.32)&nbsp;to the SEA_ML++ (M&nbsp;=&nbsp;.84,&nbsp;SD&nbsp;= .32) concrete syntax. These results suggest that there was <em>no difference between the two concrete syntaxes, in terms of&nbsp;complexity</em>.</p>\\n\\n<p>The <em>duration does not differ&nbsp;significantly</em>,&nbsp;according to Welch&rsquo;s&nbsp;t-test, t(122.030) = 1.180,&nbsp;p&nbsp;=&nbsp;.280 from the SEA_ML (M = 14 : 48,&nbsp;SD = 09 : 32) to SEA_ML++ (M = 13 : 20,&nbsp;SD = 06 : 12) concrete syntax.&nbsp;These results suggest that there was <em>no difference between the two concrete&nbsp;syntaxes, in terms of duration</em>.</p>\\n\\n<p><strong>Conclusions</strong></p>\\n\\n<p>The slight differences are not statistically significant. We found no evidence supporting the hypothesis that using SEA_ML++ instead of SEA_ML leads to more correct models. Likewise, we also did not find evidence supporting the hypothesis that using SEA_ML++ instead of SEA_ML&nbsp;will make practitioners reduce the duration of this task.</p>\\n\\n<p><strong>Acknowledgments</strong></p>\\n\\n<p>The authors would like to thank the followings:</p>\\n\\n<ul>\\n\\t<li>the Scientific and Technological Research Council of Turkey (TUBITAK) under grant 115E591</li>\\n\\t<li>Portuguese grants NOVA LINCS Research Laboratory (Grant: FCT/MCTES&nbsp;PEst UID/ CEC/04516/2013) and DSML4MA Project (Grant: FCT/MCTES&nbsp;TUBITAK/0008/2014)</li>\\n</ul>',\n",
       "  'Experimental validation of candidate regulators of KRAS transcriptional components.',\n",
       "  '<p>Version 2.0.0 (05/29/2023)</p>\\n\\n<p>This is a drug repurposing dataset under MIT licence, compiled by Dr. Cl&eacute;mence R&eacute;da &lt;clemence.reda@uni-rostock.de&gt; at Universit&auml;t Rostock, comprising a drug-disease association matrix, and several drug-drug and disease-disease similarity matrices. It only uses transcriptomic data (i.e., gene activity/expression). The sparsity number is the percentage of nonzero values in the association matrix.</p>\\n\\n<p># drugs | # diseases | Sparsity number | # positive associations | # negative associations | # genes<br>\\n------- | ---------- | --------------- | ----------------------- | ----------------------- | -------<br>\\n204&nbsp;&nbsp;&nbsp;&nbsp; | 116&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | 0.44%&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | 401&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | 11&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | 12,096</p>\\n\\n<p>All drugs (resp., diseases) are associated with a gene expression feature vector of length 12,096 (that is, all drugs and diseases in the feature matrices appear in the association matrix, and vice versa).</p>\\n\\n<p>----------</p>\\n\\n<p>This dataset consists of three .CSV files:</p>\\n\\n<p>* Drug-Disease Association Matrix</p>\\n\\n<p>1. &quot;ratings_mat.csv&quot;</p>\\n\\n<p>This matrix contains values in {-1,0,1} where -1 stands for a negative association (i.e., the drug failed for some reason to treat the considered disease: e.g., lack of accrual in the associated clinical trial, or proven toxicity), 1 for a positive association (i.e., the drug was shown to treat the disease), and 0 for unknown associated status. The columns are diseases, identified by their MedGen Concept ID, whereas rows are drugs, identified by their DrugBank IDs or PubChem CIDs.</p>\\n\\n<p>* Drug Feature Matrix</p>\\n\\n<p>1. &quot;items.csv&quot;</p>\\n\\n<p>This matrix has drugs in its columns, identified by their DrugBank IDs or PubChem CIDs, and genes in its rows, identified by their HUGO Gene Symbol. Genewise transcriptomic variation induced by drug treatment, from the CREEDS or the LINCS L1000 databases.</p>\\n\\n<p>* Disease Feature Matrix</p>\\n\\n<p>1. &quot;users.csv&quot;</p>\\n\\n<p>This matrix has diseases in its columns, identified by their MedGen Concept IDs, and genes in its rows, identified by their HUGO Gene Symbol. Genewise transcriptomic variation induced by the disease, from the CREEDS database.</p>\\n\\n<p>----------</p>\\n\\n<p>Further information about the generation of those matrices is available by running the Jupyter notebook TRANSCRIPT_dataset.ipynb on the following GitHub repository: https://github.com/RECeSS-EU-Project/drug-repurposing-datasets. For any questions, please contact the author at &lt;clemence.reda@uni-rostock.de&gt; or the RECeSS project contributors at &lt;recess-project@proton.me&gt;.</p>',\n",
       "  'Generation of KRAS signatures using immortalized isogenic lung cells.',\n",
       "  'FOXM1 transcription factor is an oncogene and a crucial regulator of cancer chemoresistance. Existing pharmacological FOXM1 inhibitors attenuate resistance of cancer cells to treatment but exhibit insufficient specificity. We performed in silico screening of small molecules from NIH LINCS database to identify STL427944 compound that suppresses FOXM1 through a novel two-step mechanism. STL427944 treatment induces relocalization of nuclear FOXM1 protein to cytoplasm, where it is subsequently degraded in an autophagosome-dependent manner. Human cancer cells treated with STL427944 exhibit increased sensitivity to cytotoxic effects of conventional chemotherapeutic agents. Transcriptome-based analysis of STL427944-induced gene expression changes revealed prominent suppression of gene signatures characteristic for FOXM1 and its downstream targets but no significant changes in other important regulatory pathways, thereby suggesting high selectivity of STL427944 towards FOXM1. We expect that STL427944 or its derivatives may be used in combination with different chemotherapeutical drugs to overcome tumor chemoresistance and improve anti-cancer treatment efficacy.\\nFor each experimental model and condition, two biological replicates were analyzed. Control samples were treated with vehicle alone (DMSO), with vehicle percentage not exceeding 0.3%.',\n",
       "  '<p class=\"western\"><span><span><span>Polychlorinated biphenyls (PCBs) are ubiquitously detected in the environment and have been linked to metabolic diseases. The liver serves as a central hub for the metabolism of xenobiotics and endogenous metabolites. Gut dysbiosis is recognized as a critical regulator of disease susceptibility, however, little is known regarding how PCBs and gut microbiome interact to modulate the interface between xenobiotic and intermediary metabolism. We hypothesized that the gut microbiome regulates PCBs-mediated changes in the metabolic fingerprints and hepatic transcriptome. Ninety-day-old female conventional (CV) and germ-free (GF) C57BL/6 mice were orally exposed to the PCB Fox River Mixture (synthetic PCB mixture, 6 or 30 mg/kg) or corn oil (vehicle control, 10 ml/kg), once daily for 3 consecutive days. Organs were collected 24 hours after the final dose. RNA-Seq was conducted on liver, and endogenous aqueous metabolites (amino acids, carbohydrates, and nucleotides) were measured in liver and serum by LC-MS. The primary factor in clustering the transcriptomic and metabolomic signatures within the same exposure was by enterotype. The numbers of PCB-regulated genes were higher in CV than in GF conditions. The prototypical target genes of the major xenobiotic-sensing transcription factors AhR, PXR, and CAR were more readily up-regulated by PCBs in CV than in GF conditions, indicating the effect of PCBs on the hepatic transcriptome act partly through the gut microbiome. Xenobiotic and steroid metabolism pathways were up-regulated, whereas response to incorrect proteins pathway was down-regulated by PCBs in a gut microbiome-dependent manner. At the high PCB dose, NADP and arginine appear to interact with drug-metabolizing enzymes (Cyp1-3 family, DhcR7, and Nqo1), which are highly correlated with <i>Anaerotruncus</i> and <i>Roseburia</i> in CV mice, providing a novel explanation of gut-liver interaction in toxicant exposures. In GF exposure groups, hepatic glucose was down-regulated, whereas fructose 6-phosphate and glucose 6-phosphate were up-regulated, indicating increased glucose utilization potentiated by lack of gut microbiota. Through querying the LINCS L1000 chemical database, Enrichr predicted that therapeutic drugs targeting the anti-inflammatory and ER stress pathways are potential remedies to mitigate PCB toxicity. In conclusion, our findings demonstrate that habitation of the gut microbiota drives PCBs-mediated hepatic responses, possibly due to crosstalk between gut and liver. </span></span></span></p>'])"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names = []\n",
    "descriptions = []\n",
    "\n",
    "for record in tqdm(response['hits']):\n",
    "    if 'name' not in record:\n",
    "        names.append('No name')\n",
    "    else:\n",
    "        names.append(record['name'])\n",
    "    if 'description' not in record:\n",
    "        descriptions.append('No description')\n",
    "    else:\n",
    "        descriptions.append(record['description'])\n",
    "\n",
    "names, descriptions\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINCS - preprocessed</td>\n",
       "      <td>&lt;p&gt;LINCS - PDRGC - preprocessed&lt;/p&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Columbia LINCS Trans-Center project data</td>\n",
       "      <td>The Library of Integrated Network-based Cellul...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LINCS Canvas Browser: interactive web app to q...</td>\n",
       "      <td>For the Library of Integrated Network-based Ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LINCS MCF10A GRNs calculated with ANANSE</td>\n",
       "      <td>&lt;p&gt;GRNs calculated using ANANSE presumably act...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Systematic Quality Control Analysis of LINCS D...</td>\n",
       "      <td>The Library of Integrated Cellular Signatures ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Experimental validation of candidate regulator...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>TRANSCRIPT drug repurposing dataset</td>\n",
       "      <td>&lt;p&gt;Version 2.0.0 (05/29/2023)&lt;/p&gt;\\n\\n&lt;p&gt;This i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Generation of KRAS signatures using immortaliz...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>STL427944 suppresses FOXM1 via nuclear export ...</td>\n",
       "      <td>FOXM1 transcription factor is an oncogene and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>Data from: Gut microbiome critically impacts P...</td>\n",
       "      <td>&lt;p class=\"western\"&gt;&lt;span&gt;&lt;span&gt;&lt;span&gt;Polychlor...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  Name  \\\n",
       "0                                 LINCS - preprocessed   \n",
       "1             Columbia LINCS Trans-Center project data   \n",
       "2    LINCS Canvas Browser: interactive web app to q...   \n",
       "3             LINCS MCF10A GRNs calculated with ANANSE   \n",
       "4    Systematic Quality Control Analysis of LINCS D...   \n",
       "..                                                 ...   \n",
       "495                                       Homo sapiens   \n",
       "496                TRANSCRIPT drug repurposing dataset   \n",
       "497                                       Homo sapiens   \n",
       "498  STL427944 suppresses FOXM1 via nuclear export ...   \n",
       "499  Data from: Gut microbiome critically impacts P...   \n",
       "\n",
       "                                           Description  \n",
       "0                  <p>LINCS - PDRGC - preprocessed</p>  \n",
       "1    The Library of Integrated Network-based Cellul...  \n",
       "2    For the Library of Integrated Network-based Ce...  \n",
       "3    <p>GRNs calculated using ANANSE presumably act...  \n",
       "4    The Library of Integrated Cellular Signatures ...  \n",
       "..                                                 ...  \n",
       "495  Experimental validation of candidate regulator...  \n",
       "496  <p>Version 2.0.0 (05/29/2023)</p>\\n\\n<p>This i...  \n",
       "497  Generation of KRAS signatures using immortaliz...  \n",
       "498  FOXM1 transcription factor is an oncogene and ...  \n",
       "499  <p class=\"western\"><span><span><span>Polychlor...  \n",
       "\n",
       "[500 rows x 2 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.DataFrame({'Name': names, 'Description': descriptions})\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/2s/m_y5ncrj7wv6_6bqg4l26hq00000gn/T/ipykernel_40945/126205139.py:3: MarkupResemblesLocatorWarning: The input looks more like a filename than markup. You may want to open this file and pass the filehandle into Beautiful Soup.\n",
      "  df['Name'] = df['Name'].apply(lambda x: BeautifulSoup(x, 'html.parser').get_text())\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINCS - preprocessed</td>\n",
       "      <td>LINCS - PDRGC - preprocessed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Columbia LINCS Trans-Center project data</td>\n",
       "      <td>The Library of Integrated Network-based Cellul...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LINCS Canvas Browser: interactive web app to q...</td>\n",
       "      <td>For the Library of Integrated Network-based Ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LINCS MCF10A GRNs calculated with ANANSE</td>\n",
       "      <td>GRNs calculated using ANANSE presumably active...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Systematic Quality Control Analysis of LINCS D...</td>\n",
       "      <td>The Library of Integrated Cellular Signatures ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Experimental validation of candidate regulator...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>TRANSCRIPT drug repurposing dataset</td>\n",
       "      <td>Version 2.0.0 (05/29/2023)\\nThis is a drug rep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Generation of KRAS signatures using immortaliz...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>STL427944 suppresses FOXM1 via nuclear export ...</td>\n",
       "      <td>FOXM1 transcription factor is an oncogene and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>Data from: Gut microbiome critically impacts P...</td>\n",
       "      <td>Polychlorinated biphenyls (PCBs) are ubiquitou...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  Name  \\\n",
       "0                                 LINCS - preprocessed   \n",
       "1             Columbia LINCS Trans-Center project data   \n",
       "2    LINCS Canvas Browser: interactive web app to q...   \n",
       "3             LINCS MCF10A GRNs calculated with ANANSE   \n",
       "4    Systematic Quality Control Analysis of LINCS D...   \n",
       "..                                                 ...   \n",
       "495                                       Homo sapiens   \n",
       "496                TRANSCRIPT drug repurposing dataset   \n",
       "497                                       Homo sapiens   \n",
       "498  STL427944 suppresses FOXM1 via nuclear export ...   \n",
       "499  Data from: Gut microbiome critically impacts P...   \n",
       "\n",
       "                                           Description  \n",
       "0                         LINCS - PDRGC - preprocessed  \n",
       "1    The Library of Integrated Network-based Cellul...  \n",
       "2    For the Library of Integrated Network-based Ce...  \n",
       "3    GRNs calculated using ANANSE presumably active...  \n",
       "4    The Library of Integrated Cellular Signatures ...  \n",
       "..                                                 ...  \n",
       "495  Experimental validation of candidate regulator...  \n",
       "496  Version 2.0.0 (05/29/2023)\\nThis is a drug rep...  \n",
       "497  Generation of KRAS signatures using immortaliz...  \n",
       "498  FOXM1 transcription factor is an oncogene and ...  \n",
       "499  Polychlorinated biphenyls (PCBs) are ubiquitou...  \n",
       "\n",
       "[500 rows x 2 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from bs4 import BeautifulSoup\n",
    "\n",
    "df['Name'] = df['Name'].apply(lambda x: BeautifulSoup(x, 'html.parser').get_text())\n",
    "df['Description'] = df['Description'].apply(lambda x: BeautifulSoup(x, 'html.parser').get_text())\n",
    "\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('../datasets/LINCS.csv', index=False, lineterminator=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LINCS - preprocessed</td>\n",
       "      <td>LINCS - PDRGC - preprocessed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Columbia LINCS Trans-Center project data</td>\n",
       "      <td>The Library of Integrated Network-based Cellul...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LINCS Canvas Browser: interactive web app to q...</td>\n",
       "      <td>For the Library of Integrated Network-based Ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LINCS MCF10A GRNs calculated with ANANSE</td>\n",
       "      <td>GRNs calculated using ANANSE presumably active...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Systematic Quality Control Analysis of LINCS D...</td>\n",
       "      <td>The Library of Integrated Cellular Signatures ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Experimental validation of candidate regulator...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>TRANSCRIPT drug repurposing dataset</td>\n",
       "      <td>Version 2.0.0 (05/29/2023)\\nThis is a drug rep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Generation of KRAS signatures using immortaliz...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>STL427944 suppresses FOXM1 via nuclear export ...</td>\n",
       "      <td>FOXM1 transcription factor is an oncogene and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>Data from: Gut microbiome critically impacts P...</td>\n",
       "      <td>Polychlorinated biphenyls (PCBs) are ubiquitou...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>500 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                  Name  \\\n",
       "0                                 LINCS - preprocessed   \n",
       "1             Columbia LINCS Trans-Center project data   \n",
       "2    LINCS Canvas Browser: interactive web app to q...   \n",
       "3             LINCS MCF10A GRNs calculated with ANANSE   \n",
       "4    Systematic Quality Control Analysis of LINCS D...   \n",
       "..                                                 ...   \n",
       "495                                       Homo sapiens   \n",
       "496                TRANSCRIPT drug repurposing dataset   \n",
       "497                                       Homo sapiens   \n",
       "498  STL427944 suppresses FOXM1 via nuclear export ...   \n",
       "499  Data from: Gut microbiome critically impacts P...   \n",
       "\n",
       "                                           Description  \n",
       "0                         LINCS - PDRGC - preprocessed  \n",
       "1    The Library of Integrated Network-based Cellul...  \n",
       "2    For the Library of Integrated Network-based Ce...  \n",
       "3    GRNs calculated using ANANSE presumably active...  \n",
       "4    The Library of Integrated Cellular Signatures ...  \n",
       "..                                                 ...  \n",
       "495  Experimental validation of candidate regulator...  \n",
       "496  Version 2.0.0 (05/29/2023)\\nThis is a drug rep...  \n",
       "497  Generation of KRAS signatures using immortaliz...  \n",
       "498  FOXM1 transcription factor is an oncogene and ...  \n",
       "499  Polychlorinated biphenyls (PCBs) are ubiquitou...  \n",
       "\n",
       "[500 rows x 2 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv('../datasets/LINCS.csv', lineterminator=\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "outputs = pd.read_csv('../outputs/9sroovd8.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      Data integration and warehousing, Computationa...\n",
       "1                 Cell biology, Genetics, Bioinformatics\n",
       "2      Bioinformatics, Systems biology, Computational...\n",
       "3        Bioinformatics, Computational biology, Genomics\n",
       "4      Bioinformatics, Computational biology, Transcr...\n",
       "                             ...                        \n",
       "495    Genetics, Gene expression, Transcription facto...\n",
       "496      Bioinformatics, Drug discovery, Transcriptomics\n",
       "497            Cell biology, Genetics, Molecular biology\n",
       "498                 Biochemistry, Pharmacology, Oncology\n",
       "499               Animal study, Microbiology, Toxicology\n",
       "Name: Predictions, Length: 500, dtype: object"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "outputs['Predictions']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Data integration and warehousing, Computational biology, \"Data submission, annotation, and curation\"',\n",
       " 'Cell biology, Genetics, Bioinformatics',\n",
       " 'Bioinformatics, Systems biology, Computational biology',\n",
       " 'Bioinformatics, Computational biology, Genomics',\n",
       " 'Bioinformatics, Computational biology, Transcriptomics',\n",
       " 'Bioinformatics, Drug discovery, Oncology',\n",
       " 'Cell biology, Drug discovery, Gene expression',\n",
       " 'Transcriptomics, Small molecules, Cancer',\n",
       " 'Transcriptomics, Small molecules, Cancer',\n",
       " 'Transcriptomics, Drug development, Bioinformatics',\n",
       " 'Bioinformatics, Drug discovery, Machine learning',\n",
       " 'Cell biology, Bioinformatics, Data management',\n",
       " 'Bioinformatics, Genomics, Transcriptomics',\n",
       " 'Gene expression, Computational biology, Bioinformatics',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Bioinformatics, Genomics, Drug discovery',\n",
       " 'Bioinformatics, Computational biology, Gene expression',\n",
       " 'Bioinformatics, Proteomics, Transcriptomics',\n",
       " 'Bioinformatics, Drug discovery, Immunology',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Computational biology, Biophysics, Membrane and lipoproteins',\n",
       " 'Bioinformatics, Computational biology, Biophysics',\n",
       " 'Biomolecular simulation, Biophysics, Computational biology',\n",
       " 'Cell biology, Bioinformatics, Ontology and terminology',\n",
       " 'Bioinformatics, Computational biology, Transcriptomics',\n",
       " 'Bioinformatics, Transcriptomics, Pharmacology',\n",
       " 'Transcriptomics, Bioinformatics, Biomedical science',\n",
       " 'Bioinformatics, Transcriptomics, Biomedical science',\n",
       " 'Transcriptomics, Data integration and warehousing, Drug discovery',\n",
       " 'Transcriptomics, Bioinformatics, Biomedical science',\n",
       " 'Bioinformatics, Transcriptomics, Genetics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Data integration and warehousing',\n",
       " 'Cell biology, Drug discovery, Molecular biology',\n",
       " 'Pharmacology, Oncology, Cell biology',\n",
       " 'Cell biology, Drug development, Biotechnology',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Pharmacology',\n",
       " 'Cell biology, Drug discovery, Biomedical science',\n",
       " 'Cell biology, Drug discovery, Biomedical science',\n",
       " 'Cell biology, Drug discovery, Pharmacology',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Cancerology',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Biotechnology',\n",
       " 'Cell biology, Drug discovery, Molecular biology',\n",
       " 'Cell biology, Drug discovery, Genetics',\n",
       " 'Cell biology, Drug discovery, Genetics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, Reproductive health',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Data integration and warehousing',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Computational biology',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Drug discovery, High-throughput screening',\n",
       " 'Cell biology, Drug discovery, Genetics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Proteomics, Drug development',\n",
       " 'Cell biology, Proteomics, Small molecules',\n",
       " 'Cell biology, Bioinformatics, Proteomics',\n",
       " 'Cell biology, Drug discovery, Oncology',\n",
       " 'Cell biology, Drug discovery, Biotechnology',\n",
       " 'Bioinformatics, Cell biology, Proteomics',\n",
       " 'Cell biology, Protein expression, Proteomics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Drug discovery, Proteins',\n",
       " 'Chemical biology, Drug discovery, Computational chemistry',\n",
       " 'Biochemistry, Biotechnology, Drug discovery',\n",
       " 'Cell biology, Molecular biology, Genetics',\n",
       " 'Cell biology, Genetics, Immunology',\n",
       " 'Bioinformatics, Drug discovery, Pharmacology',\n",
       " 'Computer science, Software engineering, Data mining',\n",
       " 'Cell biology, Drug discovery, Genetics',\n",
       " 'Cell biology, Proteomics, Neuroscience',\n",
       " 'Bioinformatics, Biotechnology, Pharmacology',\n",
       " 'Biochemistry, Biology, Genetics',\n",
       " 'RNA, Gene expression, Treatment effects',\n",
       " 'Bioinformatics, Biotechnology, Proteomics',\n",
       " 'Bioinformatics, Biotechnology, Drug discovery',\n",
       " '\"Chemical biology\",\"Drug discovery\",\"Biochemistry\"',\n",
       " 'Biochemistry, Drug development, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Cell biology, Biochemistry, Molecular biology',\n",
       " 'Cell biology, Proteomics, Neuroscience',\n",
       " 'Cell biology, Biotechnology, Microfluidics',\n",
       " 'Cell biology, Bioengineering, Microfluidics',\n",
       " 'Cell biology, Microfluidics, Biochemistry',\n",
       " 'Cell biology, Drug discovery, Proteomics',\n",
       " 'Cell biology, Genetics, Molecular biology',\n",
       " 'Metabolomics, Cell biology, Drug discovery',\n",
       " 'Cell biology, Drug discovery, Genetics',\n",
       " 'Cell biology, Molecular biology, Drug discovery',\n",
       " 'Proteomics, Cell biology, Drug discovery',\n",
       " 'Bioinformatics, Biology, Biotechnology',\n",
       " 'Bioinformatics, Biology, Biomarkers',\n",
       " 'Bioinformatics, Computational biology, Molecular biology',\n",
       " 'Bioinformatics, Biomarkers, Molecular interactions, pathways, and networks',\n",
       " 'Bioinformatics, Molecular biology, Computational biology',\n",
       " 'Bioinformatics, Computational biology, Functional genomics',\n",
       " 'Bioinformatics, Computational biology, Systems biology',\n",
       " 'Cell biology, Immunology, Rheumatology',\n",
       " 'Cell biology, Drug discovery, Cancer oncology',\n",
       " 'Cell biology, Biomedical science, Immunology',\n",
       " 'Bioinformatics, Cell biology, Biomarkers',\n",
       " 'Bioinformatics, Biology, Computational biology',\n",
       " 'Bioinformatics, Biotechnology, Computational biology',\n",
       " 'Biochemistry, Biology, Genetics',\n",
       " 'Biochemistry, Biology, Biotechnology',\n",
       " 'Bioinformatics, Computational biology, Systems biology',\n",
       " 'Biochemistry, Biology, Biotechnology',\n",
       " 'Bioinformatics, Biology, Biotechnology',\n",
       " 'Genomics, Drug development, Oncology',\n",
       " 'Cell biology, Drug discovery, Cancer',\n",
       " 'Cell biology, Biology, Bioinformatics',\n",
       " 'Cell biology, Bioinformatics, Drug discovery',\n",
       " 'Gene expression, Oncology, Cell biology',\n",
       " 'Cell biology, Drug development, Oncology',\n",
       " 'Cell biology, Bioinformatics, Computational biology',\n",
       " 'Bioinformatics, Molecular biology, Oncology',\n",
       " 'Drug discovery, Cancer research, Computational biology',\n",
       " 'Cell biology, Drug discovery, Pharmacology',\n",
       " 'Proteomics, Breast cancer, Cell biology',\n",
       " 'Cell biology, Drug discovery, Molecular biology',\n",
       " 'Biochemistry, Cell biology, Biotechnology',\n",
       " 'Cell biology, Biochemistry, Computational biology',\n",
       " 'Cell biology, Biochemistry, Molecular biology',\n",
       " 'Cell biology, Biochemistry, Drug discovery',\n",
       " 'Proteomics, Biochemistry, Genetics',\n",
       " 'Cell biology, Biomedical science, Drug discovery',\n",
       " 'Cell biology, Drug discovery, Pharmacology',\n",
       " 'Cell biology, Molecular biology, Nuclear acid sites, features and motifs',\n",
       " 'Drug discovery, Pharmacology, Biotechnology',\n",
       " 'Cell biology, Immunology, Biomedical science',\n",
       " 'Cell biology, Biomarkers, Single-cell analysis.',\n",
       " 'Cell biology, Bioinformatics, Computational biology',\n",
       " 'Cell biology, Molecular biology, Bioinformatics',\n",
       " 'Cell biology, Bioinformatics, Drug discovery',\n",
       " 'Cell biology, Drug discovery, Proteomics',\n",
       " 'Cell biology, Drug development, Molecular biology',\n",
       " 'Cell biology, Genetics, Molecular biology',\n",
       " 'Cell biology, Bioinformatics, Genetics',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Microbiology, Biotechnology, Molecular biology',\n",
       " 'Proteomics, Cancer, Cell biology',\n",
       " 'Biochemistry, Epigenetics, Genetics',\n",
       " 'Cell biology, Drug discovery, Proteomics',\n",
       " 'Cell biology, Drug development, Drug discovery',\n",
       " 'Cell biology, Drug discovery, Oncology',\n",
       " 'Epigenetics, Neurobiology, Drug discovery',\n",
       " 'Bioinformatics, Genomics, Data integration and warehousing',\n",
       " 'Computational biology, Pharmacogenomics, Drug discovery',\n",
       " 'Cell biology, Drug discovery, Biochemistry.',\n",
       " 'Cell biology, Neuroscience, Psychiatry',\n",
       " 'Bioinformatics, Biology, Biotechnology',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Bioinformatics, Molecular biology, Biotechnology',\n",
       " 'Epigenetics, Biology, Biotechnology',\n",
       " 'Biology, Biotechnology, Bioengineering',\n",
       " 'Biology, Bioinformatics, Biomarkers',\n",
       " 'Biology, Biotechnology, Bioinformatics',\n",
       " 'Bioinformatics, Biotechnology, Biology',\n",
       " 'Cell culture collection, Drug discovery, Drug development',\n",
       " 'Drug discovery, Oncology, Cell biology',\n",
       " 'Cell biology, Drug development, Oncology',\n",
       " 'Proteomics, Oncology, Cell biology',\n",
       " 'Cell biology, Proteomics, Biochemistry',\n",
       " 'Bioinformatics,Biology,Drug discovery',\n",
       " 'Drug discovery, Oncology, Cell biology',\n",
       " 'Cell biology, Metabolomics, Small molecules',\n",
       " 'Cell biology, Pharmacology, Transcriptomics',\n",
       " 'Cell biology, Pharmacology, Transcriptomics',\n",
       " 'Cell biology, Drug discovery, High throughput microscopy',\n",
       " 'Proteomics, Drug development, Cancer',\n",
       " 'Transcriptomics, Biomarkers, Biotechnology',\n",
       " 'Bioinformatics, Drug discovery, Gene expression',\n",
       " 'Protein expression, Epigenetics, Cell biology',\n",
       " 'Bioinformatics, Biotechnology, Molecular biology',\n",
       " 'Bioinformatics, Genetics, Epigenomics',\n",
       " 'Cell biology, Drug discovery, Toxicology',\n",
       " 'Cell biology, Bioinformatics, High-throughput technologies',\n",
       " 'Proteomics, Breast Cancer, Cell biology',\n",
       " 'Bioinformatics, Proteomics, Data integration and warehousing',\n",
       " 'Cell biology, Genetics, Molecular biology',\n",
       " 'Cell biology, Molecular biology, Data integration and warehousing',\n",
       " 'Cell biology, Molecular biology, Bioengineering',\n",
       " 'Pharmacology, Cancer, Drug development',\n",
       " 'Cell biology, Immunology, High throughput microscopy',\n",
       " 'Biochemistry, Biology, Proteomics',\n",
       " 'Cell biology, Drug discovery, Bioinformatics',\n",
       " 'Cell biology, Oncology, Biotechnology',\n",
       " 'Software engineering, Computational biology, Data mining',\n",
       " 'Genomics, Transcriptomics, Neurology',\n",
       " 'Bioinformatics, Drug discovery, Pharmacology',\n",
       " 'Cell biology, Biochemistry, Biotechnology',\n",
       " 'Transcriptomics, Genetics, Neurology',\n",
       " 'Cell biology, Proteomics, Drug discovery',\n",
       " 'Cell biology, Oncology, Drug discovery',\n",
       " 'Cell biology, Molecular biology, Drug discovery',\n",
       " 'Cell biology, Biochemistry, Biomarkers',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Biochemistry, Proteomics, Mass Spectrometry',\n",
       " 'Epigenetics, Genetics, Bioinformatics',\n",
       " 'Epigenetics, Functional genomics, Computational biology',\n",
       " 'Cell biology, Computational biology, Systems biology',\n",
       " 'Data management, Data integration and warehousing, Computational biology',\n",
       " 'Cell biology, Protein expression, Proteomics',\n",
       " 'Biochemistry, Proteomics, Drug discovery',\n",
       " 'Biochemistry, Proteomics, Pharmacology',\n",
       " 'Computer science, Gender medicine, \"Data architecture, analysis and design\"',\n",
       " 'Bioinformatics, Computational biology, Machine learning',\n",
       " 'Cell biology, Oncology, Drug discovery',\n",
       " 'Biophysics, Computational biology, Lipids',\n",
       " 'Biomolecular simulation, Lipids, Computational biology',\n",
       " 'Cell biology, Drug discovery, Biochemistry',\n",
       " 'Chemical biology, Cheminformatics, Data integration and warehousing',\n",
       " 'Cell biology, Biochemistry, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Molecular biology',\n",
       " 'Biochemistry, Proteomics, Mass spectrometry',\n",
       " 'Biochemistry, Proteomics, Drug discovery',\n",
       " 'Biochemistry, Analytical chemistry, Proteomics',\n",
       " 'Analytical chemistry, Biochemistry, Proteomics',\n",
       " 'Biochemistry, Molecular biology, Proteomics',\n",
       " 'Biochemistry, Proteomics, Analytical chemistry',\n",
       " 'Biochemistry, Analytical chemistry, Cell biology',\n",
       " 'Biochemistry, Biology, Proteomics',\n",
       " 'Bioinformatics, Computational biology, Proteomics',\n",
       " 'Analytical chemistry, Biochemistry, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Proteomics',\n",
       " 'Biochemistry, Computational biology, Proteomics',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Cell biology, Drug discovery, Small molecules',\n",
       " 'Cell biology, Genetics, Molecular biology',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Data integration and warehousing',\n",
       " 'Bioinformatics, Chemical biology, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Computational biology, Proteomics',\n",
       " 'Biochemistry, Proteomics, Drug development',\n",
       " 'Biochemistry, Biomarkers, Proteomics',\n",
       " 'Analytical chemistry, Biochemistry, Proteomics',\n",
       " 'Biochemistry, Biology, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Proteomics',\n",
       " 'Bioinformatics, Genomics, Drug discovery.',\n",
       " 'Biochemistry, Drug discovery, Structural biology',\n",
       " 'Biochemistry, Drug discovery, Chemistry',\n",
       " 'Biochemistry, Proteomics, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Drug discovery, Molecular biology',\n",
       " 'Analytical chemistry, Bioinformatics, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Protein interactions',\n",
       " 'Bioinformatics, Computational biology, Drug discovery',\n",
       " 'Biochemistry, Proteomics, Data integration and warehousing',\n",
       " 'Transcriptomics, Cell biology, Cardiology',\n",
       " 'Transcriptomics, Cell biology, Cardiology',\n",
       " 'Cell biology, Imaging, Bioinformatics',\n",
       " 'Dermatology,Drug discovery,Translational medicine',\n",
       " 'Bioinformatics, Genetics, Oncology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology.',\n",
       " 'Bioinformatics, Chemical biology, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Drug discovery, Bioinformatics, Chemical biology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Genetics',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Bioinformatics, Compound libraries and screening, Genetics',\n",
       " 'Pharmacology, Drug discovery, Bioinformatics',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Cell biology, Molecular biology, Oncology',\n",
       " 'Biochemistry, Proteomics, Biotechnology',\n",
       " 'Bioinformatics, Drug discovery, Genetics',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Bioinformatics, Chemical biology, Data visualisation',\n",
       " 'Bioinformatics, Chemistry, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Biotechnology',\n",
       " 'Analytical chemistry, Pharmacology, Proteomics',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Pharmacology, Drug discovery',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Analytical chemistry, Bioinformatics, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Chemistry',\n",
       " 'Bioinformatics, Genetics, Transcriptomics',\n",
       " 'Bioengineering, Computational biology, Drug discovery',\n",
       " 'Biochemistry, Pharmacology, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Chemistry',\n",
       " 'Biochemistry, Drug discovery, Data mining',\n",
       " 'Biochemistry, Drug discovery, Data integration and warehousing',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Computational biology, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Chemical biology, Drug development',\n",
       " 'Biochemistry, Drug discovery, Data visualisation',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Cell biology, Proteomics, Biotechnology',\n",
       " 'Cell biology, Molecular biology, Proteomics',\n",
       " 'Bioinformatics, Cell biology, Genomics',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Enzymes',\n",
       " 'Biochemistry, Drug discovery, Genetics',\n",
       " 'Bioinformatics, Toxicology, \"Data submission, annotation, and curation\"',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Proteomics, Drug development, Cardiology',\n",
       " 'Cell biology, Biochemistry, Molecular biology',\n",
       " 'Drug discovery, Pharmacology, Oncology',\n",
       " 'Transcriptomics, Genetics, Neurology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Bioinformatics, Genomics, Drug discovery',\n",
       " 'Cell biology, Pharmacology, Drug discovery',\n",
       " 'Bioinformatics, Genetics, Computational biology',\n",
       " 'Epigenetics, Transcriptomics, Neurology',\n",
       " 'Neurobiology, Proteomics, Cell biology',\n",
       " 'Proteomics, Bioinformatics, Cell biology',\n",
       " 'Biochemistry, Molecular biology, Proteomics',\n",
       " 'Biochemistry, Chemistry, Proteomics',\n",
       " 'Biochemistry, Proteomics, Mass spectrometry',\n",
       " 'Bioinformatics, Cell biology, Proteomics',\n",
       " 'Biochemistry,Biology,Proteomics',\n",
       " 'Biochemistry, Biotechnology, Drug discovery',\n",
       " 'Bioinformatics, Transcriptomics, Genomics',\n",
       " 'Biochemistry, Pharmacology, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Drug discovery',\n",
       " 'Bioinformatics, Data mining, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Enzymes',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Compound libraries and screening, Bioengineering, Drug discovery',\n",
       " 'Software engineering, Data integration and warehousing, Data management',\n",
       " 'Biochemistry, Cell biology, Proteomics',\n",
       " 'Epigenetics, Cell biology, Transcriptomics',\n",
       " 'Epigenomics, Neurology, Biomarkers',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Chemistry',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Computational biology, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Molecular biology, Drug discovery',\n",
       " 'Bioinformatics, Biomolecular simulation, Computational biology',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Molecular biology',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Computational biology, Structural biology',\n",
       " 'Biochemistry, Proteomics, Data integration and warehousing',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Compound libraries and screening, Biochemistry, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Chemical biology, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug development, Compound libraries and screening',\n",
       " 'Biochemistry, Drug discovery, Protein interactions',\n",
       " 'Analytical chemistry, Biochemistry, Drug discovery',\n",
       " 'Biochemistry, Drug development, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Analytical chemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Cell biology, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Bioinformatics, Biotechnology, Drug discovery',\n",
       " 'Bioinformatics, Computational biology, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Biochemistry, Drug discovery, Biotechnology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Bioengineering, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Chemical biology, Drug discovery, Computational biology',\n",
       " 'Compound libraries and screening, Biochemistry, Biotechnology',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Analytical chemistry, Bioinformatics, Drug discovery',\n",
       " 'Biochemistry, Pharmacology, Data integration and warehousing',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Compound libraries and screening, Bioinformatics, Proteomics',\n",
       " 'Bioinformatics, Biotechnology, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteins',\n",
       " 'Compound libraries and screening, Drug discovery, Biochemistry',\n",
       " 'Bioinformatics, Computational biology, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Drug development, Computational biology',\n",
       " 'Bioinformatics, Computational biology, Genetics',\n",
       " 'Bioinformatics, Biology, Psychiatry',\n",
       " 'Transcriptomics, Drug discovery, Psychiatry',\n",
       " 'Bioinformatics, Computational biology, Drug discovery',\n",
       " 'Biology, Biomedical science, Genetics',\n",
       " 'Neurobiology, Systems biology, Trauma medicine',\n",
       " 'Bioinformatics, Drug development, Transcriptomics',\n",
       " 'Cell biology, Drug discovery, Oncology',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biomolecular simulation, Bioinformatics, Biophysics',\n",
       " 'Animal study, Genetics, Transcriptomics',\n",
       " 'Genetic variation, Transcriptomics, Drug response',\n",
       " 'Genetic variation, Transcription factors and regulatory sites, Oncology',\n",
       " 'Bioinformatics, Chemistry, Genetics',\n",
       " 'Animal study, Functional genomics, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Systems biology',\n",
       " 'Bioinformatics, Chemistry, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Chemical biology, Bioinformatics, Drug discovery',\n",
       " 'Analytical chemistry, Biochemistry, Drug discovery',\n",
       " 'Transcriptomics, Drug discovery, Oncology',\n",
       " 'Biomolecular simulation, Biophysics, Computational chemistry',\n",
       " 'Biomolecular simulation, Lipids, Biophysics',\n",
       " 'Transcriptomics, Drug discovery, Hepatic and biliary medicine',\n",
       " 'Biophysics, Computational biology, Molecular dynamics',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Biotechnology, Drug development',\n",
       " 'Chemical biology, Pharmacology, Drug discovery',\n",
       " 'Biochemistry, Pharmacology, Drug discovery',\n",
       " 'Biochemistry, Biotechnology, Drug discovery',\n",
       " 'Bioinformatics, Drug discovery, Chemistry',\n",
       " 'Biochemistry, Drug discovery, Pharmacology',\n",
       " 'Bioinformatics, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Computational biology',\n",
       " 'Bioinformatics, Drug discovery, Oncology',\n",
       " 'Compound libraries and screening, Cheminformatics, Drug discovery',\n",
       " 'Biochemistry, Drug discovery, Chemistry',\n",
       " 'Biophysics, Computational biology, Structural biology',\n",
       " 'Biomolecular simulation, Biophysics, Computational biology',\n",
       " 'Biophysics, Computational chemistry, Lipids',\n",
       " 'Gene expression, Drug discovery, Cancer',\n",
       " 'Biochemistry, Oncology, Drug discovery',\n",
       " 'Gene expression, Trauma medicine, Animal study',\n",
       " 'Cell biology, Drug development, Genetics',\n",
       " 'Biophysics, Computational biology, Membrane and lipoproteins',\n",
       " 'Biophysics, Computational biology, Bioinformatics',\n",
       " 'Biophysics, Computational biology, Structural biology',\n",
       " 'Biomolecular simulation, Biophysics, Molecular dynamics',\n",
       " 'Bioinformatics, Biology, Computational biology',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Biochemistry, Drug discovery, Proteomics',\n",
       " 'Immunology, Gene expression, Transcriptomics',\n",
       " 'Bioinformatics, Pharmacology, Drug development',\n",
       " 'Bioinformatics, Drug development, Oncology',\n",
       " 'Bioinformatics, Computational biology, Drug discovery',\n",
       " 'Transcriptomics, Genomics, Oncology',\n",
       " 'Cell biology, Genomics, Omics',\n",
       " 'Genetics, Transcriptomics, Drug development',\n",
       " 'Bioinformatics, Drug discovery, Transcriptomics',\n",
       " 'Gene expression, Oncology, Comparative genomics',\n",
       " 'Cancer, Omics, Therapeutics',\n",
       " 'Cell biology, Bioinformatics, Genetics',\n",
       " 'Computer science, Bioinformatics, Data integration and warehousing',\n",
       " 'Genetics, Gene expression, Transcription factors and regulatory sites',\n",
       " 'Bioinformatics, Drug discovery, Transcriptomics',\n",
       " 'Cell biology, Genetics, Molecular biology',\n",
       " 'Biochemistry, Pharmacology, Oncology',\n",
       " 'Animal study, Microbiology, Toxicology']"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open('../EDAM/edam_topics.txt', 'r') as f:\n",
    "    lines = f.readlines()\n",
    "\n",
    "# Filter the lines to only include those with quotes\n",
    "quoted_lines = [line.strip().replace('\"', '') for line in lines if '\"' in line]\n",
    "# quoted_subsets = []\n",
    "out = []\n",
    "for line in outputs['Predictions']:\n",
    "    ln = line\n",
    "    for topic in quoted_lines:\n",
    "        if topic in line:\n",
    "            ln = ln.replace(topic, f'\"{topic}\"')\n",
    "    out.append(ln)\n",
    "\n",
    "out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "outputs['Predictions'] = out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "outputs['Ground Truth'] = ['\\\"Molecular interactions, pathways and networks\\\", Metabolomics, Protein binding sites'] * len(outputs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "outputs.to_csv('../outputs/9sroovd8.csv', index=False, lineterminator='\\n')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nde_gpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
