{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm.notebook import tqdm\n",
    "import json\n",
    "from collections import Counter\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('prod_topic_categories.json', 'r') as file:\n",
    "    data = json.load(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "05b29477817d4fa688c8ad2c4dad2098",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/1726653 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Translational medicine: 16\n",
      "Immunology: 594\n",
      "Genomics: 43\n",
      "Biochemistry: 13\n",
      "Pharmacology: 11\n",
      "Medicine: 31\n",
      "Oncology: 35\n",
      "Bioinformatics: 33\n",
      "Biomarkers: 30\n",
      "Machine learning: 8\n",
      "Vaccinology: 193\n",
      "Biology: 27\n",
      "Genetics: 87\n",
      "Pathology: 4\n",
      "Drug discovery: 8\n",
      "Bioengineering: 8\n",
      "Microbiology: 38\n",
      "Infectious disease: 275\n",
      "Laboratory animal science: 3\n",
      "Computational biology: 24\n",
      "Molecular biology: 37\n",
      "Animal study: 48\n",
      "Virology: 45\n",
      "Biomedical science: 34\n",
      "Neurology: 6\n",
      "Medical biotechnology: 5\n",
      "Evolutionary biology: 4\n",
      "Immunogenetics: 9\n",
      "Public health and epidemiology: 58\n",
      "Zoology: 8\n",
      "Structural biology: 1\n",
      "Data acquisition: 2\n",
      "Multiomics: 6\n",
      "Ecology: 2\n",
      "Allergy, clinical immunology and immunotherapeutics: 37\n",
      "Biotechnology: 8\n",
      "Data mining: 2\n",
      "Respiratory medicine: 17\n",
      "Cell biology: 40\n",
      "Database management: 1\n",
      "Data integration and warehousing: 2\n",
      "Drug development: 26\n",
      "Dermatology: 12\n",
      "Proteomics: 10\n",
      "Parasitology: 4\n",
      "Microbial ecology: 4\n",
      "Paediatrics: 45\n",
      "Epigenetics: 6\n",
      "Preclinical and clinical studies: 18\n",
      "Musculoskeletal medicine: 2\n",
      "Transcriptomics: 34\n",
      "Surgery: 11\n",
      "Developmental biology: 3\n",
      "Human biology: 8\n",
      "Haematology: 2\n",
      "Geriatric medicine: 23\n",
      "Systems biology: 37\n",
      "Comparative genomics: 4\n",
      "Reproductive health: 21\n",
      "Gynaecology and obstetrics: 25\n",
      "Metagenomics: 2\n",
      "Genetic variation: 6\n",
      "Mouse clinic: 3\n",
      "DNA: 2\n",
      "Gene expression: 12\n",
      "Functional genomics: 3\n",
      "Pain medicine: 3\n",
      "Psychiatry: 3\n",
      "Personalised medicine: 6\n",
      "Sequence analysis: 3\n",
      "Environmental sciences: 2\n",
      "Gastroenterology: 2\n",
      "Tropical medicine: 3\n",
      "Antimicrobial Resistance: 1\n",
      "Hepatic and biliary medicine: 2\n",
      "Metabolomics: 7\n",
      "Agricultural science: 3\n",
      "Data submission, annotation, and curation: 3\n",
      "Cytogenetics: 2\n",
      "Electron microscopy: 1\n",
      "Biobank: 1\n",
      "Immunomics: 1\n",
      "Endocrinology and metabolism: 5\n",
      "Data quality management: 1\n",
      "Molecular genetics: 1\n",
      "Genetic engineering: 1\n",
      "Population genetics: 9\n",
      "Cardiology: 7\n",
      "Sample collections: 1\n",
      "Immunoinformatics: 5\n",
      "Lipids: 2\n",
      "Pharmacogenomics: 1\n",
      "Nutritional science: 5\n",
      "RNA: 1\n",
      "Human genetics: 7\n",
      "Embryology: 1\n",
      "Ear, nose and throat medicine: 1\n",
      "Microfluidics: 1\n",
      "Omics: 3\n",
      "Urology and nephrology: 4\n",
      "Phylogenetics: 1\n",
      "Trauma medicine: 2\n",
      "Proteins: 1\n",
      "Critical care medicine: 1\n",
      "Population genomics: 3\n",
      "Taxonomy: 1\n",
      "Veterinary medicine: 1\n",
      "Molecular interactions, pathways and networks: 1\n",
      "Genotype and phenotype: 1\n",
      "Neurobiology: 1\n",
      "Biomolecular simulation: 1\n",
      "Analytical chemistry: 1\n",
      "Protein expression: 1\n",
      "Medicines research and development: 1\n"
     ]
    }
   ],
   "source": [
    "# Initialize a list to store names of categories curated by gpt-3.5-turbo\n",
    "gpt_curated_categories = []\n",
    "\n",
    "# Iterate through each item and its topicCategory list\n",
    "for item in tqdm(data):\n",
    "    for category in item['topicCategory']:\n",
    "        if 'curatedBy' in category:\n",
    "            if category['curatedBy']['name'] == \"gpt-3.5-turbo\":\n",
    "                gpt_curated_categories.append(category['name'])\n",
    "\n",
    "# Use Counter to count occurrences of each category name\n",
    "category_counts = Counter(gpt_curated_categories)\n",
    "\n",
    "# Print the count of each category curated by gpt-3.5-turbo\n",
    "for category, count in category_counts.items():\n",
    "    print(f\"{category}: {count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('Immunology', 594),\n",
       " ('Infectious disease', 275),\n",
       " ('Vaccinology', 193),\n",
       " ('Genetics', 87),\n",
       " ('Public health and epidemiology', 58),\n",
       " ('Animal study', 48),\n",
       " ('Virology', 45),\n",
       " ('Paediatrics', 45),\n",
       " ('Genomics', 43),\n",
       " ('Cell biology', 40),\n",
       " ('Microbiology', 38),\n",
       " ('Molecular biology', 37),\n",
       " ('Allergy, clinical immunology and immunotherapeutics', 37),\n",
       " ('Systems biology', 37),\n",
       " ('Oncology', 35),\n",
       " ('Biomedical science', 34),\n",
       " ('Transcriptomics', 34),\n",
       " ('Bioinformatics', 33),\n",
       " ('Medicine', 31),\n",
       " ('Biomarkers', 30),\n",
       " ('Biology', 27),\n",
       " ('Drug development', 26),\n",
       " ('Gynaecology and obstetrics', 25),\n",
       " ('Computational biology', 24),\n",
       " ('Geriatric medicine', 23),\n",
       " ('Reproductive health', 21),\n",
       " ('Preclinical and clinical studies', 18),\n",
       " ('Respiratory medicine', 17),\n",
       " ('Translational medicine', 16),\n",
       " ('Biochemistry', 13),\n",
       " ('Dermatology', 12),\n",
       " ('Gene expression', 12),\n",
       " ('Pharmacology', 11),\n",
       " ('Surgery', 11),\n",
       " ('Proteomics', 10),\n",
       " ('Immunogenetics', 9),\n",
       " ('Population genetics', 9),\n",
       " ('Machine learning', 8),\n",
       " ('Drug discovery', 8),\n",
       " ('Bioengineering', 8),\n",
       " ('Zoology', 8),\n",
       " ('Biotechnology', 8),\n",
       " ('Human biology', 8),\n",
       " ('Metabolomics', 7),\n",
       " ('Cardiology', 7),\n",
       " ('Human genetics', 7),\n",
       " ('Neurology', 6),\n",
       " ('Multiomics', 6),\n",
       " ('Epigenetics', 6),\n",
       " ('Genetic variation', 6),\n",
       " ('Personalised medicine', 6),\n",
       " ('Medical biotechnology', 5),\n",
       " ('Endocrinology and metabolism', 5),\n",
       " ('Immunoinformatics', 5),\n",
       " ('Nutritional science', 5),\n",
       " ('Pathology', 4),\n",
       " ('Evolutionary biology', 4),\n",
       " ('Parasitology', 4),\n",
       " ('Microbial ecology', 4),\n",
       " ('Comparative genomics', 4),\n",
       " ('Urology and nephrology', 4),\n",
       " ('Laboratory animal science', 3),\n",
       " ('Developmental biology', 3),\n",
       " ('Mouse clinic', 3),\n",
       " ('Functional genomics', 3),\n",
       " ('Pain medicine', 3),\n",
       " ('Psychiatry', 3),\n",
       " ('Sequence analysis', 3),\n",
       " ('Tropical medicine', 3),\n",
       " ('Agricultural science', 3),\n",
       " ('Data submission, annotation, and curation', 3),\n",
       " ('Omics', 3),\n",
       " ('Population genomics', 3),\n",
       " ('Data acquisition', 2),\n",
       " ('Ecology', 2),\n",
       " ('Data mining', 2),\n",
       " ('Data integration and warehousing', 2),\n",
       " ('Musculoskeletal medicine', 2),\n",
       " ('Haematology', 2),\n",
       " ('Metagenomics', 2),\n",
       " ('DNA', 2),\n",
       " ('Environmental sciences', 2),\n",
       " ('Gastroenterology', 2),\n",
       " ('Hepatic and biliary medicine', 2),\n",
       " ('Cytogenetics', 2),\n",
       " ('Lipids', 2),\n",
       " ('Trauma medicine', 2),\n",
       " ('Structural biology', 1),\n",
       " ('Database management', 1),\n",
       " ('Antimicrobial Resistance', 1),\n",
       " ('Electron microscopy', 1),\n",
       " ('Biobank', 1),\n",
       " ('Immunomics', 1),\n",
       " ('Data quality management', 1),\n",
       " ('Molecular genetics', 1),\n",
       " ('Genetic engineering', 1),\n",
       " ('Sample collections', 1),\n",
       " ('Pharmacogenomics', 1),\n",
       " ('RNA', 1),\n",
       " ('Embryology', 1),\n",
       " ('Ear, nose and throat medicine', 1),\n",
       " ('Microfluidics', 1),\n",
       " ('Phylogenetics', 1),\n",
       " ('Proteins', 1),\n",
       " ('Critical care medicine', 1),\n",
       " ('Taxonomy', 1),\n",
       " ('Veterinary medicine', 1),\n",
       " ('Molecular interactions, pathways and networks', 1),\n",
       " ('Genotype and phenotype', 1),\n",
       " ('Neurobiology', 1),\n",
       " ('Biomolecular simulation', 1),\n",
       " ('Analytical chemistry', 1),\n",
       " ('Protein expression', 1),\n",
       " ('Medicines research and development', 1)]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(category_counts.items(), key=lambda x: x[1], reverse=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2292\n"
     ]
    }
   ],
   "source": [
    "total_counts = sum(category_counts.values())\n",
    "print(total_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Data Repository Frequency Table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from collections import Counter, defaultdict\n",
    "\n",
    "# Load the JSON data\n",
    "with open('gpt_curated_data_w_ids.json', 'r') as file:\n",
    "    data = json.load(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f29b712dbd704ebabfb03913a5a0585d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/1874465 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genetics: 427171\n",
      "Molecular biology: 283443\n",
      "Genomics: 273243\n",
      "Microbiology: 253419\n",
      "Cell biology: 206439\n",
      "Immunology: 185135\n",
      "Transcriptomics: 184228\n",
      "Bioinformatics: 161118\n",
      "Biochemistry: 157387\n",
      "Biology: 152482\n",
      "Oncology: 150547\n",
      "Environmental sciences: 126864\n",
      "Animal study: 123989\n",
      "Plant biology: 118565\n",
      "Infectious disease: 107063\n",
      "Public health and epidemiology: 101586\n",
      "Computational biology: 100855\n",
      "Gene expression: 89892\n",
      "Neurobiology: 69910\n",
      "Epigenetics: 60982\n",
      "Ecology: 59380\n",
      "Metagenomics: 59135\n",
      "Developmental biology: 50479\n",
      "Zoology: 49087\n",
      "RNA: 46466\n",
      "Medicine: 45990\n",
      "Structural biology: 44504\n",
      "Chemistry: 44206\n",
      "Biotechnology: 43150\n",
      "Biodiversity: 42339\n",
      "Cardiology: 41563\n",
      "Evolutionary biology: 40617\n",
      "Psychiatry: 39982\n",
      "Neurology: 38042\n",
      "Proteomics: 37962\n",
      "Pharmacology: 34654\n",
      "Data submission, annotation, and curation: 33908\n",
      "Biomedical science: 33195\n",
      "Agricultural science: 30686\n",
      "Database management: 29885\n",
      "Virology: 29507\n",
      "Data integration and warehousing: 28773\n",
      "Microbial ecology: 28730\n",
      "Antimicrobial Resistance: 28512\n",
      "Biomarkers: 28451\n",
      "Population genetics: 28055\n",
      "Physiology: 25590\n",
      "Drug development: 25194\n",
      "Proteins: 24358\n",
      "Drug discovery: 23370\n",
      "Sequence analysis: 23175\n",
      "Gastroenterology: 22509\n",
      "Taxonomy: 21988\n",
      "Marine biology: 21581\n",
      "DNA: 21526\n",
      "Preclinical and clinical studies: 20638\n",
      "Data mining: 19523\n",
      "Parasitology: 19019\n",
      "Comparative genomics: 18397\n",
      "Paediatrics: 18317\n",
      "Nutritional science: 18028\n",
      "Bioengineering: 17493\n",
      "Transcription factors and regulatory sites: 15701\n",
      "Metabolomics: 15221\n",
      "Functional genomics: 14472\n",
      "Physics: 13796\n",
      "Respiratory medicine: 13166\n",
      "Biophysics: 12951\n",
      "Analytical chemistry: 12676\n",
      "Enzymes: 12380\n",
      "X-ray diffraction: 12114\n",
      "Informatics: 11800\n",
      "Genetic variation: 11573\n",
      "Toxicology: 11454\n",
      "Gene regulation: 11417\n",
      "Reproductive health: 11344\n",
      "Haematology: 11319\n",
      "Pathology: 11170\n",
      "Anatomy: 11141\n",
      "Geriatric medicine: 10878\n",
      "Data management: 10640\n",
      "Musculoskeletal medicine: 10636\n",
      "Vaccinology: 10294\n",
      "Human genetics: 10267\n",
      "Personalised medicine: 10253\n",
      "Phylogenetics: 9764\n",
      "Surgery: 9193\n",
      "Data acquisition: 8653\n",
      "Sequence assembly: 8632\n",
      "Gynaecology and obstetrics: 8481\n",
      "Biomaterials: 8072\n",
      "Epigenomics: 8043\n",
      "Endocrinology and metabolism: 7933\n",
      "Population genomics: 7654\n",
      "Trauma medicine: 7424\n",
      "Veterinary medicine: 7325\n",
      "Mouse clinic: 7035\n",
      "Allergy, clinical immunology and immunotherapeutics: 7005\n",
      "Computer science: 6945\n",
      "Molecular interactions, pathways and networks: 6941\n",
      "Data architecture, analysis and design: 6915\n",
      "Critical care medicine: 6663\n",
      "Computational chemistry: 6640\n",
      "Hepatic and biliary medicine: 6571\n",
      "Pain medicine: 6486\n",
      "Regenerative medicine: 6313\n",
      "Tropical medicine: 6037\n",
      "Machine learning: 6003\n",
      "Medical informatics: 5913\n",
      "Dermatology: 5903\n",
      "Human biology: 5647\n",
      "Molecular genetics: 5523\n",
      "Rare diseases: 5516\n",
      "Laboratory animal science: 5370\n",
      "Embryology: 4909\n",
      "Freshwater biology: 4868\n",
      "Medicinal chemistry: 4627\n",
      "Data governance: 4590\n",
      "DNA replication and recombination: 4471\n",
      "Molecular dynamics: 4450\n",
      "NMR: 4440\n",
      "Urology and nephrology: 4135\n",
      "Acoustics: 4064\n",
      "Dentistry: 3978\n",
      "Lipids: 3978\n",
      "Protein interactions: 3845\n",
      "Translational medicine: 3753\n",
      "Omics: 3694\n",
      "Systems biology: 3457\n",
      "Cytogenetics: 3297\n",
      "Pharmacogenomics: 3287\n",
      "Gender medicine: 3104\n",
      "Open science: 3043\n",
      "Data visualisation: 3025\n",
      "Software engineering: 2985\n",
      "Mapping: 2978\n",
      "Paleogenomics: 2968\n",
      "Medical biotechnology: 2805\n",
      "DNA binding sites: 2783\n",
      "Biomolecular simulation: 2742\n",
      "Metatranscriptomics: 2544\n",
      "Protein folding, stability and design: 2513\n",
      "RNA splicing: 2462\n",
      "Quality affairs: 2262\n",
      "Metabolic engineering: 2235\n",
      "Sample collections: 2195\n",
      "Copy number variation: 2183\n",
      "Microfluidics: 2058\n",
      "Biobank: 2010\n",
      "Electroencephalography: 1959\n",
      "Chemical biology: 1953\n",
      "Laboratory information management: 1926\n",
      "Gene transcripts: 1854\n",
      "Electron microscopy: 1842\n",
      "Synthetic chemistry: 1832\n",
      "Genotype and phenotype: 1820\n",
      "Anaesthesiology: 1816\n",
      "Drug metabolism: 1805\n",
      "Carbon cycle: 1784\n",
      "Electrocardiography: 1759\n",
      "Membrane and lipoproteins: 1712\n",
      "Model organisms: 1682\n",
      "Ribosome Profiling: 1646\n",
      "Phylogenomics: 1601\n",
      "Immunogenetics: 1598\n",
      "Quantitative genetics: 1581\n",
      "Protein structure analysis: 1562\n",
      "Synthetic biology: 1533\n",
      "Structural genomics: 1505\n",
      "DNA mutation: 1480\n",
      "Carbohydrates: 1449\n",
      "Ear, nose and throat medicine: 1405\n",
      "Biosciences: 1345\n",
      "Genetic engineering: 1341\n",
      "Cell culture collection: 1320\n",
      "Pharmacovigilance: 1256\n",
      "Complementary medicine: 1251\n",
      "Protein expression: 1061\n",
      "Cryogenic electron microscopy: 1051\n",
      "Cheminformatics: 1046\n",
      "Safety sciences: 1046\n",
      "Multiomics: 1039\n",
      "Data security: 956\n",
      "Medicines research and development: 951\n",
      "Microbial collection: 913\n",
      "Small molecules: 890\n",
      "Structure analysis: 795\n",
      "Nucleic acids: 765\n",
      "Phenomics: 699\n",
      "Regulatory affairs: 671\n",
      "Biotherapeutics: 656\n",
      "Data quality management: 623\n",
      "Immunoinformatics: 619\n",
      "Compound libraries and screening: 594\n",
      "Natural language processing: 561\n",
      "Molecular evolution: 497\n",
      "Mobile genetic elements: 482\n",
      "Workflows: 479\n",
      "Molecular medicine: 464\n",
      "Molecular modelling: 412\n",
      "Protein binding sites: 407\n",
      "Protein modifications: 398\n",
      "Function analysis: 374\n",
      "Phylogeny: 365\n",
      "Chemometrics: 304\n",
      "Structural variation: 300\n",
      "Protein targeting and localisation: 300\n",
      "Medical toxicology: 297\n",
      "Clone library: 275\n",
      "Epistasis: 234\n",
      "Neutron diffraction: 224\n",
      "Functional, regulatory and non-coding RNA: 206\n",
      "Gene structure: 206\n",
      "Ontology and terminology: 204\n",
      "Sequence composition, complexity and repeats: 190\n",
      "Protein properties: 169\n",
      "Proteogenomics: 167\n",
      "Gene and protein families: 151\n",
      "DNA packaging: 128\n",
      "Protein disordered structure: 128\n",
      "Protein structural motifs and surfaces: 124\n",
      "Immunomics: 121\n",
      "Probes and primers: 119\n",
      "Nucleic acid structure analysis: 118\n",
      "DNA polymorphism: 101\n",
      "Cladistics: 101\n",
      "Protein secondary structure: 96\n",
      "Sequence sites, features and motifs: 79\n",
      "Nucleic acid sites, features and motifs: 75\n",
      "FAIR data: 70\n",
      "Fluxomics: 61\n",
      "Structure prediction: 56\n",
      "Immunoproteins and antigens: 53\n",
      "Data rescue: 43\n",
      "Systems medicine: 40\n",
      "Data identity and mapping: 39\n",
      "Protein folds and structural domains: 38\n",
      "Protein variants: 24\n",
      "Protein sites, features and motifs: 8\n"
     ]
    }
   ],
   "source": [
    "# Initialize a list to store names of categories curated by gpt-3.5-turbo\n",
    "gpt_curated_categories = []\n",
    "\n",
    "# Iterate through each item and its topicCategory list\n",
    "for item in tqdm(data):\n",
    "    for category in item['topicCategory']:\n",
    "        if 'curatedBy' in category:\n",
    "            if category['curatedBy']['name'] == \"gpt-3.5-turbo\":\n",
    "                gpt_curated_categories.append(category['name'])\n",
    "\n",
    "# Use Counter to count occurrences of each category name\n",
    "category_counts = Counter(gpt_curated_categories)\n",
    "sorted_category_counts = sorted(category_counts.items(), key=lambda x: x[1], reverse=True)\n",
    "for category, count in sorted_category_counts:\n",
    "    print(f\"{category}: {count}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "28a832537f79490a8092d8f633baa58b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/1874465 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Initialize a dictionary to hold the frequency table for each data catalog\n",
    "catalog_frequency_tables = {}\n",
    "\n",
    "# Initialize a dictionary to track total lengths and counts for each category\n",
    "category_lengths = defaultdict(lambda: {'total_length': [], 'count': 0})\n",
    "\n",
    "all_lengths = []\n",
    "\n",
    "# Iterate through each item in the JSON data\n",
    "for item in tqdm(data):\n",
    "    if 'includedInDataCatalog' not in item or 'topicCategory' not in item:\n",
    "        continue\n",
    "    \n",
    "    if 'name' not in item['includedInDataCatalog']:\n",
    "        continue\n",
    "    catalog_name = item['includedInDataCatalog']['name']\n",
    "    topic_categories = [category['name'] for category in item['topicCategory'] if category['fromGPT'] == True]\n",
    "    \n",
    "    # Update the category_lengths dictionary\n",
    "    list_length = len(topic_categories)\n",
    "\n",
    "    all_lengths.append(list_length)\n",
    "\n",
    "    for category in topic_categories:\n",
    "        category_lengths[category]['total_length'].append(list_length)\n",
    "        category_lengths[category]['count'] += 1\n",
    "    \n",
    "    # Update the frequency table for each data catalog\n",
    "    if catalog_name not in catalog_frequency_tables:\n",
    "        catalog_frequency_tables[catalog_name] = Counter(topic_categories)\n",
    "    else:\n",
    "        catalog_frequency_tables[catalog_name].update(topic_categories)\n",
    "\n",
    "# Calculate the average list size for each category\n",
    "average_list_sizes = {category: sum(lengths['total_length']) / len(lengths['total_length']) for category, lengths in category_lengths.items()}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data Catalog: NCBI SRA\n",
      "  Microbiology: 82057, Average List Size: 2.96\n",
      "  Genomics: 77173, Average List Size: 2.97\n",
      "  Genetics: 70695, Average List Size: 2.96\n",
      "  Transcriptomics: 59556, Average List Size: 2.95\n",
      "  Molecular biology: 37008, Average List Size: 2.98\n",
      "  Plant biology: 35105, Average List Size: 2.95\n",
      "  Environmental sciences: 34728, Average List Size: 2.85\n",
      "  Bioinformatics: 30430, Average List Size: 2.98\n",
      "  Biology: 29363, Average List Size: 2.95\n",
      "  Metagenomics: 27775, Average List Size: 2.96\n",
      "\n",
      "Data Catalog: Qiita\n",
      "  Microbiology: 472, Average List Size: 2.96\n",
      "  Environmental sciences: 178, Average List Size: 2.85\n",
      "  Metagenomics: 126, Average List Size: 2.96\n",
      "  Microbial ecology: 122, Average List Size: 2.97\n",
      "  Animal study: 109, Average List Size: 2.95\n",
      "  Immunology: 84, Average List Size: 2.97\n",
      "  Gastroenterology: 71, Average List Size: 2.93\n",
      "  Ecology: 70, Average List Size: 2.92\n",
      "  Nutritional science: 59, Average List Size: 2.86\n",
      "  Metabolomics: 57, Average List Size: 2.94\n",
      "\n",
      "Data Catalog: Mendeley\n",
      "  Environmental sciences: 7532, Average List Size: 2.85\n",
      "  Computational biology: 6286, Average List Size: 2.93\n",
      "  Data mining: 4739, Average List Size: 2.81\n",
      "  Molecular biology: 4280, Average List Size: 2.98\n",
      "  Biology: 4248, Average List Size: 2.95\n",
      "  Biochemistry: 4103, Average List Size: 2.96\n",
      "  Bioinformatics: 3655, Average List Size: 2.98\n",
      "  Genetics: 2941, Average List Size: 2.96\n",
      "  Public health and epidemiology: 2882, Average List Size: 2.68\n",
      "  Chemistry: 2849, Average List Size: 2.59\n",
      "\n",
      "Data Catalog: AccessClinicalData@NIAID\n",
      "  Infectious disease: 6, Average List Size: 2.96\n",
      "  Preclinical and clinical studies: 4, Average List Size: 2.76\n",
      "  Public health and epidemiology: 3, Average List Size: 2.68\n",
      "  Immunology: 2, Average List Size: 2.97\n",
      "  Medicine: 2, Average List Size: 2.86\n",
      "  Pharmacology: 1, Average List Size: 2.94\n",
      "\n",
      "Data Catalog: Dryad Digital Repository\n",
      "  Ecology: 23586, Average List Size: 2.92\n",
      "  Environmental sciences: 13438, Average List Size: 2.85\n",
      "  Evolutionary biology: 10676, Average List Size: 2.96\n",
      "  Animal study: 10619, Average List Size: 2.95\n",
      "  Genetics: 9804, Average List Size: 2.96\n",
      "  Biodiversity: 9408, Average List Size: 2.95\n",
      "  Zoology: 8153, Average List Size: 2.96\n",
      "  Biology: 6408, Average List Size: 2.95\n",
      "  Plant biology: 5553, Average List Size: 2.95\n",
      "  Population genetics: 4648, Average List Size: 2.94\n",
      "\n",
      "Data Catalog: VDJServer\n",
      "  Immunology: 44, Average List Size: 2.97\n",
      "  Oncology: 14, Average List Size: 2.86\n",
      "  Genetics: 14, Average List Size: 2.96\n",
      "  Genomics: 9, Average List Size: 2.97\n",
      "  Bioinformatics: 9, Average List Size: 2.98\n",
      "  Infectious disease: 8, Average List Size: 2.96\n",
      "  Biology: 5, Average List Size: 2.95\n",
      "  Vaccinology: 4, Average List Size: 2.99\n",
      "  Molecular biology: 4, Average List Size: 2.98\n",
      "  Virology: 4, Average List Size: 2.97\n",
      "\n",
      "Data Catalog: HuBMAP\n",
      "  Genetics: 701, Average List Size: 2.96\n",
      "  Human biology: 446, Average List Size: 2.86\n",
      "  Anatomy: 407, Average List Size: 2.88\n",
      "  Biology: 398, Average List Size: 2.95\n",
      "  Cell biology: 375, Average List Size: 2.96\n",
      "  Microbiology: 344, Average List Size: 2.96\n",
      "  Molecular biology: 336, Average List Size: 2.98\n",
      "  Genomics: 313, Average List Size: 2.97\n",
      "  Transcriptomics: 230, Average List Size: 2.95\n",
      "  Biomedical science: 202, Average List Size: 2.86\n",
      "\n",
      "Data Catalog: ClinEpiDB\n",
      "  Public health and epidemiology: 44, Average List Size: 2.68\n",
      "  Infectious disease: 28, Average List Size: 2.96\n",
      "  Parasitology: 14, Average List Size: 2.99\n",
      "  Population genetics: 12, Average List Size: 2.94\n",
      "  Medicine: 6, Average List Size: 2.86\n",
      "  Microbiology: 6, Average List Size: 2.96\n",
      "  Tropical medicine: 5, Average List Size: 2.91\n",
      "  Paediatrics: 3, Average List Size: 2.83\n",
      "  Nutritional science: 2, Average List Size: 2.86\n",
      "  Data acquisition: 2, Average List Size: 2.90\n",
      "\n",
      "Data Catalog: Harvard Dataverse\n",
      "  Database management: 28018, Average List Size: 2.92\n",
      "  Data submission, annotation, and curation: 22232, Average List Size: 3.00\n",
      "  Data integration and warehousing: 17375, Average List Size: 2.99\n",
      "  Informatics: 8968, Average List Size: 2.94\n",
      "  Public health and epidemiology: 7996, Average List Size: 2.68\n",
      "  Environmental sciences: 7662, Average List Size: 2.85\n",
      "  Data management: 6481, Average List Size: 2.71\n",
      "  Data mining: 5057, Average List Size: 2.81\n",
      "  Computational biology: 4807, Average List Size: 2.93\n",
      "  Data architecture, analysis and design: 3791, Average List Size: 2.92\n",
      "\n",
      "Data Catalog: ImmPort\n",
      "  Immunology: 550, Average List Size: 2.97\n",
      "  Infectious disease: 267, Average List Size: 2.96\n",
      "  Vaccinology: 189, Average List Size: 2.99\n",
      "  Genetics: 73, Average List Size: 2.96\n",
      "  Public health and epidemiology: 56, Average List Size: 2.68\n",
      "  Animal study: 46, Average List Size: 2.95\n",
      "  Paediatrics: 45, Average List Size: 2.83\n",
      "  Virology: 41, Average List Size: 2.97\n",
      "  Cell biology: 39, Average List Size: 2.96\n",
      "  Systems biology: 37, Average List Size: 3.05\n",
      "\n",
      "Data Catalog: ReframeDB\n",
      "  Drug discovery: 78, Average List Size: 2.96\n",
      "  Biochemistry: 37, Average List Size: 2.96\n",
      "  Microbiology: 35, Average List Size: 2.96\n",
      "  Antimicrobial Resistance: 35, Average List Size: 2.97\n",
      "  Molecular biology: 30, Average List Size: 2.98\n",
      "  Cell biology: 24, Average List Size: 2.96\n",
      "  Infectious disease: 23, Average List Size: 2.96\n",
      "  Compound libraries and screening: 16, Average List Size: 2.98\n",
      "  Parasitology: 16, Average List Size: 2.99\n",
      "  Virology: 16, Average List Size: 2.97\n",
      "\n",
      "Data Catalog: Vivli\n",
      "  Preclinical and clinical studies: 2999, Average List Size: 2.76\n",
      "  Pharmacology: 2246, Average List Size: 2.94\n",
      "  Immunology: 1611, Average List Size: 2.97\n",
      "  Drug development: 1574, Average List Size: 2.92\n",
      "  Medicine: 884, Average List Size: 2.86\n",
      "  Infectious disease: 874, Average List Size: 2.96\n",
      "  Vaccinology: 847, Average List Size: 2.99\n",
      "  Oncology: 817, Average List Size: 2.86\n",
      "  Public health and epidemiology: 765, Average List Size: 2.68\n",
      "  Paediatrics: 747, Average List Size: 2.83\n",
      "\n",
      "Data Catalog: Omics Discovery Index (OmicsDI)\n",
      "  Genetics: 261872, Average List Size: 2.96\n",
      "  Molecular biology: 192631, Average List Size: 2.98\n",
      "  Microbiology: 156804, Average List Size: 2.96\n",
      "  Genomics: 156204, Average List Size: 2.97\n",
      "  Biochemistry: 134726, Average List Size: 2.96\n",
      "  Cell biology: 133780, Average List Size: 2.96\n",
      "  Immunology: 129837, Average List Size: 2.97\n",
      "  Oncology: 110882, Average List Size: 2.86\n",
      "  Biology: 92570, Average List Size: 2.95\n",
      "  Bioinformatics: 88000, Average List Size: 2.98\n",
      "\n",
      "Data Catalog: LINCS\n",
      "  Drug discovery: 53, Average List Size: 2.96\n",
      "  Biochemistry: 49, Average List Size: 2.96\n",
      "  Molecular biology: 39, Average List Size: 2.98\n",
      "  Cell biology: 26, Average List Size: 2.96\n",
      "  Bioinformatics: 21, Average List Size: 2.98\n",
      "  Compound libraries and screening: 20, Average List Size: 2.98\n",
      "  Proteomics: 19, Average List Size: 2.97\n",
      "  Computational biology: 16, Average List Size: 2.93\n",
      "  Pharmacology: 9, Average List Size: 2.94\n",
      "  Chemical biology: 9, Average List Size: 2.84\n",
      "\n",
      "Data Catalog: NCBI GEO\n",
      "  Genetics: 25648, Average List Size: 2.96\n",
      "  Transcriptomics: 18633, Average List Size: 2.95\n",
      "  Cell biology: 15902, Average List Size: 2.96\n",
      "  Molecular biology: 14242, Average List Size: 2.98\n",
      "  Bioinformatics: 11018, Average List Size: 2.98\n",
      "  Genomics: 10917, Average List Size: 2.97\n",
      "  Immunology: 10575, Average List Size: 2.97\n",
      "  Gene expression: 10492, Average List Size: 2.98\n",
      "  Oncology: 8434, Average List Size: 2.86\n",
      "  Epigenetics: 8027, Average List Size: 2.96\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Sort and print the frequency table for each data catalog, including average list size\n",
    "for catalog, frequency_table in catalog_frequency_tables.items():\n",
    "    print(f\"Data Catalog: {catalog}\")\n",
    "    # Sort the frequency table by frequency (highest to lowest)\n",
    "    sorted_frequency_table = sorted(frequency_table.items(), key=lambda item: item[1], reverse=True)\n",
    "    for category, frequency in sorted_frequency_table[:10]:\n",
    "        avg_size = average_list_sizes[category]\n",
    "        print(f\"  {category}: {frequency}, Average List Size: {avg_size:.2f}\")\n",
    "    print()  # Print a newline for better readability between catalogs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data exported to data_catalogs.xlsx successfully.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Initialize a list to hold all rows for the DataFrame\n",
    "data_for_df = []\n",
    "\n",
    "# Iterate through each data catalog to populate the list\n",
    "for catalog, frequency_table in catalog_frequency_tables.items():\n",
    "    # Sort the frequency table by frequency (highest to lowest)\n",
    "    sorted_frequency_table = sorted(frequency_table.items(), key=lambda item: item[1], reverse=True)\n",
    "    for category, frequency in sorted_frequency_table[:10]:\n",
    "        avg_size = average_list_sizes[category]\n",
    "        # Append a dictionary for each row\n",
    "        data_for_df.append({\n",
    "            \"Data Catalog\": catalog,\n",
    "            \"Category\": category,\n",
    "            \"Frequency\": frequency,\n",
    "            \"Average List Size\": avg_size\n",
    "        })\n",
    "\n",
    "# Create a DataFrame\n",
    "df = pd.DataFrame(data_for_df)\n",
    "\n",
    "# Export the DataFrame to an Excel file\n",
    "excel_filename = \"data_catalogs.xlsx\"\n",
    "df.to_excel(excel_filename, index=False)\n",
    "\n",
    "print(f\"Data exported to {excel_filename} successfully.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Empty List Metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import json\n",
    "import os\n",
    "\n",
    "def process_json_file(file_path):\n",
    "    key_values = {}\n",
    "    with open(file_path, 'r') as file:\n",
    "        data = json.load(file)\n",
    "        for item in data:\n",
    "            for key, values in item.items():\n",
    "                if key not in key_values:\n",
    "                    key_values[key] = []\n",
    "                key_values[key].extend(values)\n",
    "    rows = [[key, value] for key, value in key_values.items()]\n",
    "    return rows\n",
    "\n",
    "def process_folder(folder_path):\n",
    "    all_rows = []\n",
    "    for file_name in tqdm(os.listdir(folder_path)):\n",
    "        if file_name.endswith('.json'):\n",
    "            file_path = os.path.join(folder_path, file_name)\n",
    "            all_rows.extend(process_json_file(file_path))\n",
    "    \n",
    "    df = pd.DataFrame(all_rows, columns=['Key', 'Values'])  # Note 'Values' is plural now\n",
    "    return df\n",
    "\n",
    "# Example usage\n",
    "folder_path = '../outputs/jsons'\n",
    "df = process_folder(folder_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.drop_duplicates(subset='Key', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "zero_predictions = df[df['Values'].apply(len) == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2770d40b61a545f5b7a1dd24842192bf",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/18 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "\n",
    "# Define the path to the folder containing the CSV files\n",
    "folder_path = 'nde_prod'\n",
    "\n",
    "# Use glob to find all CSV files in the folder\n",
    "csv_files = glob.glob(f\"{folder_path}/*.csv\")\n",
    "\n",
    "# Initialize an empty DataFrame\n",
    "big_dataframe = pd.DataFrame()\n",
    "\n",
    "# Loop through each CSV file\n",
    "for file in tqdm(csv_files):\n",
    "    # Read the CSV file into a DataFrame\n",
    "    df = pd.read_csv(file, lineterminator='\\n')\n",
    "    \n",
    "    # Select only the 'id', 'Name', and 'Description' columns\n",
    "    selected_data = df[['_id', 'Name', 'Description', 'includedInDataCatalog']]\n",
    "    \n",
    "    # Append this data to the big DataFrame\n",
    "    big_dataframe = pd.concat([big_dataframe, selected_data], ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_315154/403595838.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  zero_predictions.rename(columns={'Key': '_id'}, inplace=True)\n"
     ]
    }
   ],
   "source": [
    "zero_predictions.rename(columns={'Key': '_id'}, inplace=True)    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "matched_rows = big_dataframe[big_dataframe['_id'].isin(zero_predictions['_id'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_315154/1408403160.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  matched_rows['combined'] = matched_rows['Name'] + matched_rows['Description']\n",
      "/tmp/ipykernel_315154/1408403160.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  matched_rows['combined'] = matched_rows['combined'].fillna('')\n",
      "/tmp/ipykernel_315154/1408403160.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  matched_rows['length'] = matched_rows['combined'].apply(list).apply(len)\n"
     ]
    }
   ],
   "source": [
    "matched_rows['combined'] = matched_rows['Name'] + matched_rows['Description']\n",
    "matched_rows['combined'] = matched_rows['combined'].fillna('')\n",
    "\n",
    "matched_rows['length'] = matched_rows['combined'].apply(list).apply(len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                 <100 chars  100-200 chars  201-300 chars  \\\n",
      "includedInDataCatalog                                                       \n",
      "Dryad Digital Repository                  1              1              0   \n",
      "Harvard Dataverse                       637           1883           1771   \n",
      "HuBMAP                                    5              1              0   \n",
      "ImmPort                                   0              0              0   \n",
      "Mendeley                                139            510            404   \n",
      "NCBI GEO                                  0              2              2   \n",
      "NCBI SRA                                  3              5              3   \n",
      "Omics Discovery Index (OmicsDI)          29             26             13   \n",
      "VEuPathDB                                 0              4              1   \n",
      "Vivli                                     0              0              0   \n",
      "Zenodo                                  247            933            700   \n",
      "\n",
      "                                 301-400 chars  >400 chars  \n",
      "includedInDataCatalog                                       \n",
      "Dryad Digital Repository                     1         176  \n",
      "Harvard Dataverse                         1397        5128  \n",
      "HuBMAP                                       0           0  \n",
      "ImmPort                                      1           0  \n",
      "Mendeley                                   230        1428  \n",
      "NCBI GEO                                     2          15  \n",
      "NCBI SRA                                     5          13  \n",
      "Omics Discovery Index (OmicsDI)             13        7971  \n",
      "VEuPathDB                                    2           0  \n",
      "Vivli                                        1           2  \n",
      "Zenodo                                     398        3540  \n"
     ]
    }
   ],
   "source": [
    "# Step 3: Define the character length ranges\n",
    "ranges = [(0, 99), (100, 200), (201, 300), (301, 400), (401, float('inf'))]\n",
    "range_labels = ['<100 chars', '100-200 chars', '201-300 chars', '301-400 chars', '>400 chars']\n",
    "\n",
    "# Step 4: Group by 'repository' and count entries within each length range\n",
    "def count_in_ranges(group):\n",
    "    counts = {}\n",
    "    for range_label, (start, end) in zip(range_labels, ranges):\n",
    "        counts[range_label] = group[(group['length'] > start) & (group['length'] <= end)].shape[0]\n",
    "    return pd.Series(counts)\n",
    "\n",
    "# Apply the custom aggregation function\n",
    "result = matched_rows.groupby('includedInDataCatalog').apply(count_in_ranges)\n",
    "\n",
    "# Step 5: Display the result\n",
    "print(result)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "vscode": {
     "languageId": "markdown"
    }
   },
   "outputs": [],
   "source": [
    "result.to_excel(\"output.xlsx\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "markdown"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nde_gpt",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
